{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Environmental analysis\n",
    "\n",
    "![scheme](img01.jpg\n",
    ")\n",
    "\n",
    "RW: Raw Water before treatment\n",
    "\n",
    "BC: Before Chlorination\n",
    "\n",
    "FW: Finished Water \n",
    "\n",
    "DS1-DS3: tap water\n",
    "\n",
    "PB1-PB2: biofilm in pipes\n",
    "\n",
    "WM: biofilm in water meters\n",
    "\n",
    "We are going to study the variation of microbial populations before entering the circuit (RW), at the end of the treatment (FW) and tap water samples in three houses (DS1, DS2 and DS3). We will also study samples taken in June and July. \n",
    "\n",
    "I propose you to run the pipeline I showed you in Unit 2 whit this dataset and answer the following questions:\n",
    "\n",
    "1)\tHow many reads have we got for each sample?\n",
    "\n",
    "2)\tWhich is the trimming length you are using for the denoising step?\n",
    "\n",
    "3)\tHow many ASVs do you have before filtering deblur-table? And after filtering? \n",
    "\n",
    "4)\tWhich is the average frequency of sequences per sample before filtering deblur table?\n",
    "\n",
    "5)\tWhich is the sample with the lower number of sequences after filtering deblur table? And the one with the highest number?\n",
    "\n",
    "6)\tWhich is the most abundant phylum in each sample?\n",
    "\n",
    "7)\tHas the study enough coverage to allow us to make any statistical inference on communities’ diversity?\n",
    "\n",
    "8)\tStudying the Unifrac Weighted PCoA plot. Is there any effect of water treatment or sampling moment on the bacteria communities? \n",
    "\n",
    "9)\tIf we compare Untreated samples (RW) vs Treated samples (FW, DS1, DS2, DS3) which are the Phyla or Classes explaining the differences among both groups? (Hint: use LEfSe)\n",
    "\n",
    "**Dataset Contents**:\n",
    "\n",
    "1)\tfastq folder: raw sample sequences\n",
    "\n",
    "2)\t85_otus.fasta and 85_otus_taxonomy.txt: taxonomy database\n",
    "\n",
    "3)\tmetadata.txt: sample metadata. I have included some columns useful for diversity analyses.\n",
    "\n",
    "4)\tprimers.txt: information on primers used for 16S PCR amplification\n",
    "\n",
    "5)\tquiz.docx: this file\n",
    "\n",
    "6)\tsamplemanifest: manifest file with information of the ubication of  fastq files and their corresponding tags."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Responses"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1)\tHow many reads have we got for each sample?\n",
    "\n",
    "\n",
    ">From *grep* against fastq files and verified after import on *qiime*.\n",
    "\n",
    "| File              |#reads|\n",
    "| :------------------|--------:|\n",
    "|SRR3593621_1.fastq |52945| \n",
    "|SRR3593621_2.fastq |52945|\n",
    "|SRR3593622_1.fastq |62218|\n",
    "|SRR3593622_2.fastq |62218|\n",
    "|SRR3593623_1.fastq |92740|\n",
    "|SRR3593623_2.fastq |92740|\n",
    "|SRR3593625_1.fastq |70366|\n",
    "|SRR3593625_2.fastq |70366|\n",
    "|SRR3593627_1.fastq |100615|\n",
    "|SRR3593627_2.fastq |100615|\n",
    "|SRR3593628_1.fastq |78495|\n",
    "|SRR3593628_2.fastq |78495|\n",
    "|SRR3593631_1.fastq |97332|\n",
    "|SRR3593631_2.fastq |97332|\n",
    "|SRR3593632_1.fastq |84361|\n",
    "|SRR3593632_2.fastq |84361|\n",
    "|SRR3593664_1.fastq |101827|\n",
    "|SRR3593664_2.fastq |101827|\n",
    "|SRR3593665_1.fastq |84850|\n",
    "|SRR3593665_2.fastq |84850|\n",
    "\n",
    "\n",
    "2)\tWhich is the trimming length you are using for the denoising step?\n",
    "\n",
    ">$300$. It's the length at which the drop of sequence quality is produced.\n",
    "\n",
    "3)\tHow many ASVs do you have before filtering deblur-table? And after filtering? \n",
    "\n",
    ">Before filtering I've $813$ features after $468$.\n",
    "\n",
    "4)\tWhich is the average frequency of sequences per sample before filtering deblur table?\n",
    ">$35,290$\n",
    "\n",
    "5)\tWhich is the sample with the lower number of sequences after filtering deblur table? And the one with the highest number?\n",
    "\n",
    "6)\tWhich is the most abundant phylum in each sample?\n",
    "\n",
    "7)\tHas the study enough coverage to allow us to make any statistical inference on communities’ diversity?\n",
    "\n",
    "8)\tStudying the Unifrac Weighted PCoA plot. Is there any effect of water treatment or sampling moment on the bacteria communities? \n",
    "\n",
    "9)\tIf we compare Untreated samples (RW) vs Treated samples (FW, DS1, DS2, DS3) which are the Phyla or Classes explaining the differences among both groups? (Hint: use LEfSe)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "#Alignment and phylogeny\n",
    "mkdir tree_out\n",
    "\n",
    "qiime alignment mafft --i-sequences deblur_output/rep_seqs_filt.qza \\\n",
    "                      --p-n-threads 3 \\\n",
    "                      --o-alignment tree_out/rep_seqs_filt_aligned.qza\n",
    "\n",
    "qiime alignment mask --i-alignment tree_out/rep_seqs_filt_aligned.qza \\\n",
    "                     --o-masked-alignment tree_out/rep_seqs_filt_aligned_masked.qza\n",
    "\n",
    "qiime phylogeny fasttree --i-alignment tree_out/rep_seqs_filt_aligned_masked.qza \\\n",
    "                         --p-n-threads 2 \\\n",
    "                         --o-tree tree_out/rep_seqs_filt_aligned_masked_tree\n",
    "\n",
    "qiime phylogeny midpoint-root --i-tree tree_out/rep_seqs_filt_aligned_masked_tree.qza \\\n",
    "                              --o-rooted-tree tree_out/rep_seqs_filt_aligned_masked_tree_rooted.qza\n",
    "\n",
    "#Training database\n",
    "qiime tools import \\\n",
    "  --type 'FeatureData[Sequence]' \\\n",
    "  --input-path 85_otus.fasta \\\n",
    "  --output-path 85_otus.qza\n",
    "\n",
    "qiime tools import \\\n",
    "  --type 'FeatureData[Taxonomy]' \\\n",
    "  --input-format HeaderlessTSVTaxonomyFormat \\\n",
    "  --input-path 85_otu_taxonomy.txt \\\n",
    "  --output-path ref-taxonomy.qza\n",
    "\n",
    "qiime feature-classifier extract-reads \\\n",
    "  --i-sequences 85_otus.qza \\\n",
    "  --p-f-primer AACMGGATTAGATACCCKG \\\n",
    "  --p-r-primer ACGTCATCCCCACCTTCC \\\n",
    "  --p-min-length 100 \\\n",
    "  --p-max-length 400 \\\n",
    "  --o-reads ref-seqs.qza\n",
    "\n",
    "qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier classifier.qza\n",
    "\n",
    "#Taxonomic assignation\n",
    "\n",
    "qiime feature-classifier classify-sklearn --i-reads deblur_output/rep_seqs_filt.qza \\\n",
    "                                          --i-classifier classifier.qza \\\n",
    "                                          --p-n-jobs 2 \\\n",
    "                                          --output-dir taxa\n",
    "\n",
    "qiime tools export --input-path taxa/classification.qza --output-path taxa\n",
    "\n",
    "qiime taxa barplot --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                   --i-taxonomy taxa/classification.qza \\\n",
    "                   --m-metadata-file metadata.txt \\\n",
    "                   --o-visualization taxa/taxa_barplot.qzv\n",
    "\n",
    "qiime feature-table group --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                          --p-axis sample \\\n",
    "                          --p-mode sum \\\n",
    "                          --m-metadata-file metadata.txt \\\n",
    "                          --m-metadata-column Plant \\\n",
    "                          --o-grouped-table deblur_output/deblur_table_filt_Plant.qza\n",
    "\n",
    "qiime taxa barplot --i-table deblur_output/deblur_table_filt_Plant.qza \\\n",
    "                   --i-taxonomy taxa/classification.qza \\\n",
    "                   --m-metadata-file metadata.txt \\\n",
    "                   --o-visualization taxa/taxa_barplot_Plant.qzv\n",
    "\n",
    "#Diversity\n",
    "\n",
    "qiime diversity alpha-rarefaction --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                                  --p-max-depth 5496 \\\n",
    "                                  --p-steps 20 \\\n",
    "                                  --i-phylogeny tree_out/rep_seqs_filt_aligned_masked_tree_rooted.qza \\\n",
    "                                  --m-metadata-file metadata.txt \\\n",
    "                                  --o-visualization rarefaction_curves.qzv\n",
    "\n",
    "qiime diversity core-metrics-phylogenetic --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                                          --i-phylogeny tree_out/rep_seqs_filt_aligned_masked_tree_rooted.qza \\\n",
    "                                          --p-sampling-depth 1411 \\\n",
    "                                          --m-metadata-file metadata.txt \\\n",
    "                                          --p-n-jobs 2 \\\n",
    "                                          --output-dir diversity\n",
    "\n",
    "qiime diversity alpha-group-significance --i-alpha-diversity diversity/shannon_vector.qza \\\n",
    "                                         --m-metadata-file metadata.txt \\\n",
    "                                         --o-visualization diversity/shannon_compare_groups.qzv\n",
    "\n",
    "#Differential Abundance\n",
    "#Ancom\n",
    "\n",
    "qiime taxa collapse \\\n",
    "  --i-table deblur_output/deblur_table_filt.qza \\\n",
    "  --i-taxonomy taxa/classification.qza \\\n",
    "  --p-level 6 \\\n",
    "  --o-collapsed-table deblur_output/deblur_collapsed.qza\n",
    "\n",
    "qiime composition add-pseudocount \\\n",
    "  --i-table deblur_output/deblur_collapsed.qza \\\n",
    "  --o-composition-table deblur_output/comp-deblur-l6.qza\n",
    "\n",
    "qiime composition ancom \\\n",
    "  --i-table deblur_output/comp-deblur-l6.qza \\\n",
    "  --m-metadata-file metadata.txt \\\n",
    "  --m-metadata-column Plant \\\n",
    "  --o-visualization deblur_output/l6-ancom-Subject.qzv\n",
    "\n",
    "#LEFSe\n",
    "\n",
    "qiime taxa collapse --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                    --o-collapsed-table deblur_output/L3_collapse_table.qza \\\n",
    "                    --p-level 3 \\\n",
    "                    --i-taxonomy taxa/classification.qza\n",
    "\n",
    "qiime tools export --input-path deblur_output/L3_collapse_table.qza \\\n",
    "                   --output-path lefse_table/\n",
    "\n",
    "biom convert -i lefse_table/feature-table.biom \\\n",
    "             -o lefse_table/feature-table.txt \\\n",
    "             --header-key “taxonomy” --to-tsv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Preprocessing and quality check\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "FILE_ID = \"SRR\"\n",
    "FASTQ_STR = \"@SRR\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=SRR, FASTQ_STR=@SRR\n",
      "==> SRR3593621_1.fastq <==\n",
      "@SRR3593621.1 1 length=300\n",
      "TACGTAGGGTGCAAGCGTTATCCGGATTCACTGTTCTTCCCTATTCGTTTAGTTTTTTTTTTCCTTCTCACTTCACAGCCCTTTGCTTTACCTCGTCCTTCTTTTCTTCTTTACTATACTCGTTTTTTATATTTGTACGTGGTTCTCCTTTTTGTTCTGTGCCTTGCGTTGTGTTCTTGTTGTACCCCAATTGCCATTGCTCCTTTCTTCTTCATTCCTGTCACTCTTCCACGCAAGCTATCGTACTCCATCAGTTTAGTCCCCTCCTTTTTTCTAGCCCTCAACTCTTCCCTGCTAGTT\n",
      "+SRR3593621.1 1 length=300\n",
      "<6BCCGGGGGGGGGGGGGG7CEE6+,8@@,;C,,,<,<,,,,,,;6,;,6,,,;-,:+8+86,,,<9,:,6CE,,,,,,66,,,,,5:,,6696,,8,89,,,9,<,,,,9,,,:,<,,95+,+++9+,9,,,,,,,,+,,+++:>;,74,4,+,,5,,8,,,,8,,+66+++,66,7,,,,,,:,:5++,,+2,,,,+3+74<2=2@,,+22,5,5*4+,4*/5++++3*.)*)((//++(-(((*+*))+)))3.)))++,/;*/(/)//587))6((,(*),-.).)-44)43)).-\n",
      "\n",
      "==> SRR3593621_2.fastq <==\n",
      "@SRR3593621.1 1 length=300\n",
      "TTTCCTCTTTCTCTCTTCCTCCCCACCCTTCCTCCTTCCCTCTTTCTCTTCCTTCTCTCCTCACATTCTCCTCCCTCCCCTGCTTCCCTTTATTTCTCCCTCTACTCCTCTCTCTTCTCTGCTTCTCTCCTCTACCTTTCGTTTCTCAGTGTCATCTATTTCCCTGCCCGTTGCCTTCTCCTTTCTTCTTCCTCCTCATCTCCACTCATTTCACTTCTCCTCCATGCTTTCCCCTTTCCTCTATCACCCTCCACTCTCTTCCTCTCCTCCCCCACGCCCCCTTTACGCCCCGTCCTTTCC\n",
      "+SRR3593621.1 1 length=300\n",
      "88@--;<C@,C,-6;,6,<C@E@C,+,8,,,;;,B@C,,65;,4>,,@A?,5,6;,;,C6,,,,,,6,,,,6,,8;,,8,,,4,:,5,,94,,,,99,,,,,9,9,4,,,,,,,5+,3>9+934459526389+,923<9*0*09;@**3*26*++6*6++20+*))++**4)10008*03>*0*+0*3:;+:9):)**320**3*))*3;;*205:;=D*80)1*)**+*001010:8;87**3*;018>5))>@E):7A*19/))*)-/-)-5(-(((((()))((((.(((,//./)\n",
      "\n",
      "==> SRR3593622_1.fastq <==\n",
      "@SRR3593622.1 1 length=300\n",
      "TACGAAGGGGGCTAGCGTTGTTCGGATTTACTGGGCGTAAAGCGCACGTAGGCGGTTTTTTAAGTCAGGGGTGAAATCCCAAGGCTCAACCTTGGAACTGCCTTTGATACTGGAAGTCTCGAGTCCGGGAGAGGTGAGTGGAATTGCTAGTGTAGAGGTGAAATTCGTAGATATTCGCAAGAACACCAGTGGCGAAGGCGGCTCACTGGCCCGGTACTGACGCTGAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACTATGGATGCTAGCCGT\n",
      "+SRR3593622.1 1 length=300\n",
      "CCCC@FFGGGGGGGGGGGGGGGGGGGGGFFCFFFGECFFGGGFG:+8@EFFG@FF+>BDDEE<EFC,C,,@FEGFGEE?=,,:?DFAFFGEEE9<,,4A?=<EFD<,A<?F9,,,::AF;?<7DF?>F+8@==FF7,,9=+F,,:,@,,8@FDFE9=:D*>D,@3=CDGGDFFGFAD>>7CC,>,>FE,C<7<=E=@EDB75>5+2A8+*8*/*211:48CE1:*775C)4>?>>B73;@DB66>>*;D5(<<07@?F66<CF86*546<AAB6<BF?0>?::<<2<62).,3994(,,4\n",
      "\n",
      "==> SRR3593622_2.fastq <==\n",
      "@SRR3593622.1 1 length=300\n",
      "TTTAACCTTGCGACCGTACTCCCCAGGCGGGATGCTTAATGCGTTAGCTGCGCCACTGAACAGCTAGCTGCCCAACGGCTAGCATCCATAGTTTACGGCGTTGACTACCACGCTATCTAATCCTCTTTTCTCCCCACGCTTTCGCACCTCAGCGTCAGTACCTTGCCAGTGTGCCGCCTTCGCCACTGTTGTTCTCCCGACTCTCTACGCCTTTCACCTCTCCACTCCCAGTTCCACTCTCCCCTCCCGTACTCGAGACTTCCATTTTCAAAGGCAGTTCCAAGTTTTCGCCTTGGCATT\n",
      "+SRR3593622.1 1 length=300\n",
      "AC<8-;FFFFGC+7@:FFGGEGEGGF@C@+++8,BFG@,,:@>CE@6EE,@76,+CC6,,:,,4,,,:6,99C,,8=+84+,,,:94,9,,9C@,54+>68+++B;,4A,,++8+<>;,+53?;,5>@,24;+;@8>+3=0=0+0*@)1A)*389**)53+4*+/)0+***,*14)-55-7(-624.))/.;17/)),(,(((/6)676((/,=2)63,4<(,-343(.))5<6))66:).2(-.:(33,,(-.-1(((..4<A)))))))))((((,.4))),).))),-3,()((,).\n",
      "\n",
      "==> SRR3593623_1.fastq <==\n",
      "@SRR3593623.1 1 length=300\n",
      "TACGGAGGGAGCTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCACGTAGGCGGCTTTGTCAGTTAGAGGTGAAAGCCTTGAGCTCACCTCCAGAATTGCCTTTAAGACTGCATCTCTTGAATCCAGGAGAGGTGAGTGGAATTCCTAGTGTAGAGGTGAAATTCGTAGATATTCGGATGAACACCAGTGGCGAAGGCGGCTCACTGGACTGTTATTGACGCTGCGGTGCGAAAGCTTGGGGAGCAACCAGGATTAGATACCCTGGTAGTCCACGCCGTAACCGATGATAACTAGCTGT\n",
      "+SRR3593623.1 1 length=300\n",
      "C@CCCGGGGGGGGGGGGGGDFFG7@;,CE<;FC<8C@C@,CFC77+8@C:E@6@:+8=C,C,,<CC,,,,6C,,CE,,:C,,,:@E=C,CC==,9,,<<,<?EFA,,,<:F,,,:A,9E@,,5A@,,5,++,8:A+,,98+A,CB>=,,9CA@==,,8@+,,,@=D+@A?6@,ECC,,++53,:,@F*4@3,;55*>*9>):::,4B8,*22/+314<11<*(6*)/):)421*72:*15)3(413*0*)(0(,8=>?2.0<<<?*/...49@*8(4,(42:549(3,).)).6).(46:\n",
      "\n",
      "==> SRR3593623_2.fastq <==\n",
      "@SRR3593623.1 1 length=300\n",
      "TTTACTCTTGCGTCCGTACTCCCCAGGCGGTTCTCTTCTTGCGTTAGCTTCGCCCCCACACTTCTTTCCACCCTCTCCCCTCTTTTTCATCCTTTTCCCCCTTTACTTCCCCCTTACCTCATCCTCTTTTCTCCCCCCTCTTTCGCTCCTCCTCTTCATTTCCTCTCCCGTTATCCTCCTTCGCCTCTCTTCTTCTTCCGAATATCTACCTATTTCACCTCTACCCCCGTAATTCCACTCTCCTCTCCCGCCTTCAAGCTCTGCACTCTTACATGCACTTCTGCATTTCAGCCCCATCCT\n",
      "+SRR3593623.1 1 length=300\n",
      "8@6--6CFC;E++7@+C6BF@EEEB,,98+++,,;EE,,;,66>B,,EF<@+,6,8+++,,,5C<,,,,,69,,,,:,,,4,,99,99,4,,9?@,9,,,+4,,,4,,9A,,++,*,)9+*4<,4,9?;++0+368++*03=<+*0+380)*+)+42*+4+20+*5:7+*3,))10**359*15>*10)**+*1*2//*)))))*2/*/****10*/00075*9*/((,0((//)6)743)),.6631((((-()))))))-)))(.(-5-)))),)-4))-))))..)))-(-(((((,\n",
      "\n",
      "==> SRR3593625_1.fastq <==\n",
      "@SRR3593625.1 1 length=300\n",
      "TACGTAGGGTCCAAGCGTTAATCGGAATTACTGGGCGTACAGCGTTCGCAGGCGGTTTTTCAAGTCCGATGTGAAATCCCCTTGCTTAACTTTGGTATTGCTTTTGTTACTGCTCGTCTTGAGTTTGTCAGAGGTGGCTAGAATTCCTCGTGTAGCAGTGAAATGCGTAGTGATGTGGTGGAATACCGTTGGCGAAGGCAGCCCCCTGGGATAACACTGACGCTCATGCACGAAAGCTTGGGGAGCAACCAGGTTTAGTTACCCTGGTAGTCCTCGCCCTAAACGATGTCAACTAGTTGT\n",
      "+SRR3593625.1 1 length=300\n",
      "BBCCCGGGGGFGFGGGGGG>,<FDGC7DAC6ECEFC@F@+@FCC@,88+88@+86@@,9,,,69,:66+88E,,69<C666,,:@F<:,CCC,,,,,9<55,9A,,:,55C,,,9++6A+,,,C,:,B,4,,:+,++,8,,C,C?4@,7@F:@=,7,7C8,5,@,@+6=C+,,7,7,6,+3+,3,:>:+@*,9:***13,3*1*1;C***)0,,4,,868>*(67)C+1*7(,*):3*2):0((.4*2*0(04)5756)4+<<06**.46.6=*639((4.,+5((2(.))).62)64)6\n",
      "\n",
      "==> SRR3593625_2.fastq <==\n",
      "@SRR3593625.1 1 length=300\n",
      "TTTACTCTTGCGACCGTACTCCCCAGGCGGTCTACTTCTCGCGTTAGCTTCGTTACTACCCTCTTTCCTCCCCCACACCTACTTTTCATCCTTTTTGGCGTTGACTACCCCTCTCTCTATTCCTCTTTTCTCCCCACTCTTTCGTCCATGACCGTCACTTTTATCCCACCTCGCTCCCTTCTCCATCTGTATTCCTCCTCATCTCTACGCCTTTCTCTGCTACACCTCGTATTCTACCCCCCTCTGCCCCCCTCTACCCCTTCTGTCTCCAATTCAACTCCCTAGTTATCCTCCTCGCTT\n",
      "+SRR3593625.1 1 length=300\n",
      "8B<--6CDDCE@+7@+B6BF@E@C,,,68+86,,CFFF,6,84>C,4EF,C496,@6,,,,,,,6,,,,66?:++7,,89,,59,,5,4,,59C,,++,48+++:9,,9,+++9,543,+2,47,5>?,33;+9@,=+5=+?++3*1)2+**6):9+)6+62*+303++**,)0*)1))1)*:6*1:**3*2;****-)76/8AE76;)/(0(4**1**0088*5(-(()((./.)173(.3(63)))(,((,-,))6((,(()).))--)(-))))).)),()(.)))-.).,((((,(\n",
      "\n",
      "==> SRR3593627_1.fastq <==\n",
      "@SRR3593627.1 1 length=300\n",
      "TACGTAGGATGCAAGCGTTGTCCGGATTTATTGGGCGTAAAGAGTTCGTAGGTGGTTTGTTAAGTTTGGTGTTAAAGATTGGGGCTCAACCCTGATACTGCTCTTAATACTGTCAGACTCGAGTGTGGTAGAGGCTAGTGGAATTCCCAGTGTAGCGGTGAATTGCGTAGTTATTGGGATGATCACCGGTGGCGTAGGCGACTTCCTGGGCCATTACTGACACTGAGGAACGAAAGCCAGGGGAGCGAATGGGATTAGATACCCCTGTAGTCCTGGCCGTAAACGTTGTCTACTAGTCGT\n",
      "+SRR3593627.1 1 length=300\n",
      "CCCCCGGGDGGGFGFFGGGDCCFEG@@FE<CFECEC@F+,6C,;<CC@B,CCCCCBCC,:C9,EE9C<,:CFE,CC,,:9,,,@CG<CCFC==,,,,6<,6,,<,,,<<5C,,5,,,9A,4,+C+<57A;,,,=+4,,9,,C,CA>>5,,8,C@=;>:C775,@==7@6C+@,@=,,3,,6,,:,:>8+@<+;@3**5***4;*,4=;,*/4*4,,,9+14<0>**2**055@752(0);.((2(3=*/).92(4??2.62<<0)*1.).65=@).3<(4(-/66(41,).).6).))-4\n",
      "\n",
      "==> SRR3593627_2.fastq <==\n",
      "@SRR3593627.1 1 length=300\n",
      "TTCAACCTTGCGCCCGTACTACTCAGGCGGGATACTTATCGCGTTAGCTGCGTCACATCACTGTTCGATACCCCATACGCCTAGTATCCATCGTTTACCGCCAGCACTACTGCGGTATCTAATCCCTTTCGCTCCCCTCGCTTTCGTTCCTCCGTGTCAGTTATGCCCCAGCTCCTCGCCTACGCCACCTTTGTTCTTCCCCATATCTACGCATTTCACCGCTACACTGGGATTTCCACTAGCCTCTACCCCCCTCGATTCTGCCCGTTTCCAGTTCAGTTTCAGGTTTGACCCCCAATC\n",
      "+SRR3593627.1 1 length=300\n",
      "@C@8-;CFFFFC+6@:F@FGAF9E?66C@+++8,BFGC@698BEE7AEE,C+,6,8,9C,,,,,95,,6,86C,,4,:,,8?,,:,594,9,+:A<+A,++6+++,94,A,,,++3+3>3+,47=5,:=+0::DBDC8*+*=9900418ACB+0:*40++51*+**3;?**10)*00)-7>(7256(-50)+*109/96;3((6(1?557(,(247C37;22(49).)))(((..:<)44().44:72:)(((-,((((.64-5)((-(.)6)-.)6))..--))).,-(-).3,,((.(\n",
      "\n",
      "==> SRR3593628_1.fastq <==\n",
      "@SRR3593628.1 1 length=300\n",
      "TACGTAGGATGCAAGCGTTGTCCGGATTTATTGGGCGTAAAGCGTTCGTAGGTGGTTTTTTAAGTTTGGTGTTAAATCTTTTGGCTCAACCCTGAAACTGCTCTTAATACTGTCTGACTCGAGTGTGGTAGAGGCTATTGGATTTCCCAGTGTAGCGGTGAAATGCGTAGTTATTGGGAAGATCACCGGTGGCGTAGGCGACTTCCTGGGCCTTTACTGACACTGAGGAACGAAAGCCTGGGGAGCGAATCGGATTAGATACCCCTGTAGTCCTGGCCTTACACGTTGGCTACTAGTCGT\n",
      "+SRR3593628.1 1 length=300\n",
      "-CCCCGGGFGGGFGGGGGG>BCFCC,6CE;<FEFC@6C6,CC,888668,:CC,CBCC,:@,,CC9C,,:CFF66,,,<9,,,8@F5:,C9,6,,,,,9,5,,<,,,955C,,9,,,4:44,+:+<,+9,,,98+4,,9,,9,:A4=7,7A:C@,;++B++,,@9=36+@+@,@=,,,,,+@,,,:>++@339@3*5*9**11*,3=,,**/*4,519+4<,0>*9***)/6(1)438).)2(1(:**9.(0(-8@28)6+3<+.**.6).6=/..36)4),<.8(4(((341:)5))(4\n",
      "\n",
      "==> SRR3593628_2.fastq <==\n",
      "@SRR3593628.1 1 length=300\n",
      "TTCAACCTTGCGCCCGTACTACTCAGGCGGGATACTTATCGCGTTAGCTTCGCCACTTCACTTGTCTATACCCGATACCCCTAGTATCCATCGTTTACCGCCAGGACTACTGCGGTATCTACTCCCTTTCGCTCCCCTGTCTTTCGTTCCTCAGTGTCAGTTATCGCCCAGCTACTCGCCTTCCCCACCTTTGTTCTTCCCTATATCTACGCATTTCACCTCTACCCTGGGAATTCCACTTCCCTCTTCCACACTCGAGTCTGCCAGTTTCCCGTGCACTTTCAGGTTTCACCCCCACTC\n",
      "+SRR3593628.1 1 length=300\n",
      "AC---;CFFFFC+6@@FCEG,C,B9;66@:++8,BFFDA;9@CEG?6ED,C+,646,6:,,,,,:5,,9,<<6++68>,9:A,,:,5@=,9,+:A<+9,448++++99,85,,++34:><+,47@,,:@3*::FEFEC*+5=CD;5681AEA+0<+5;++53*;*)8;C**4)**00)/7<):)09)(/()4*17CC?5>=*.0*9@C6701(.,9C))7.2)76)/)()(((//7?)))))-44?2)65>4(4<?:((.4).53)).)-).)-((,().,-.)))).)).).((-((,,\n",
      "\n",
      "==> SRR3593631_1.fastq <==\n",
      "@SRR3593631.1 1 length=300\n",
      "TACGGAGGGAGCTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCCCGTAGGCGGCTTTTTAAGTCTGGTGTGACATCCCTGGGCTCAACCCCGGTACTGCCCTTGATACTGTCTATCTTGATTCTTGTAGTGGCGATTGGAATTCCTAGTGTAGAGGTGAAATTCGTAGTTATTCGGAAGATCACCAGTGGCGAAGGCGACTCCCTGGACCTGTCTTGCCGCTGAGGTTCGACAGCTTGGGGAGCCACCCGGTTTAGTTACCCTGTTAGTCCTCGCCGTACACGTTGTTACCTATCTTT\n",
      "+SRR3593631.1 1 length=300\n",
      "B@CCC@FGD7CFEFGFGGG>C@F@B@,CCC6EC@C@8C+,,CC@6+77@68@,+8+8=CCE,,:CC,,,,9C,9,<95,4,,,4@E=,:F44+++++4<,5,,:=,,,,5C,,,,,,8C,,,,<7@@,,,7,,,++++83,3,:>=@,,3C,@3,,,5@+,8,7>;35;@+7,@<=,,***,,3,8>,,67,34***/8*)*4*+2;:**.***5+49+/0*(4))/*2)4(.*.,1)1)04(-(3*+0.(04(45.:).+3<+1**.5).<?*.,22(-4(*64(4((.))-.).).)6\n",
      "\n",
      "==> SRR3593631_2.fastq <==\n",
      "@SRR3593631.1 1 length=300\n",
      "TTTACTCTTGCGACCGTACTCCCCCGGCGGTTCACTTATTGCGTTAGCTTCGCCACCCCCCTTCTTTCCACCCCGCCCCCTTTTTATCCTCTTTTTCCCCCTTTCCTTCCCCCGTTTCTTATCCTCTTTTCTCCCCACGCTTTCGCTCCCTCTCTTCCTTTCTTTTCCCGCTATTCTCCTTCGCCCCTCTTTTTCTTCCCACTTTCTCCCCCTTTCCCCTCTTCACTCGGACTTCCCCTCCCCCCCCCCTCATTCCCGCTTCCCCCTTTCCACCTCCCTTCCCGCTTTTCCCCCCTCTCT\n",
      "+SRR3593631.1 1 length=300\n",
      "@B@--6CEDCE@+6@+B,BF@F@E+++77++6,,:FF,,6966>C7,EE,9+,6,@6=+++449C,5,,,59=+++7+++4,,95,59,7,,78?,@,,,+4,,,37,9=,,+++*,),++3573,7::+2049>C,0*+033+*0*11*)**0*3+*+4+4/2*1:2**)*((+1*05.>*((5)3,(()+/.(...)()(/))).)/(((,0..))4((,).).).()((,.).(.(,(-,(-,((((((.))))-,-,,,(((-.)))))((.(,(..((((,-())))32(((((,\n",
      "\n",
      "==> SRR3593632_1.fastq <==\n",
      "@SRR3593632.1 1 length=300\n",
      "TACGTAGGATGCAAGCGTTGTCCGGATTTATTGGGCGTAAAGAGTTCGTAGGTGGTTTGTTAAGTTTGGTGTTAAAGATTGGGGCTCAACCCTGATACTGCACTGAATACTGGCAGACTCGAGTGTGGTAGAGGCTAGTGGAATTCCCAGTGTAGCGGTGAAATGCGTAGATATTGGGAAGAACACCGGTGGCGTAGGCGACTAGCTGGGCCATAACTGACACTGAGGAACGAAAGCCAGGGGAGCGAATGGGATTAGATACCCCAGTAGTCCTGGCCGTAAACGATGGATACTAGGCGT\n",
      "+SRR3593632.1 1 length=300\n",
      "-ACCCFGGFFFGGGFFEFGEFEGGGFEGEFCFGGGGGGGGFFCGEE@FCCFGEGGFED,CECCFG<CFCFGGGGGGF,6C6,CCEG<CFG<8C<?,,6C@9,,C,:,?C,FE,,4?,9A=D<FF+BFFFFGFGG:9=,CFFG9FE=>5,EFGEEEDFGGGFF,@,=E@CE+@=DA,BEE,AF,@@FGG6EEEDE:@FEG8C9;*,0BFF*04*=,>8C@BF60509FC*):??EFFF)08>?66(./:>5:FF??FFF2AGDC8585>44:<FFB:::2906CA:6<8?(.6:AA2029<\n",
      "\n",
      "==> SRR3593632_2.fastq <==\n",
      "@SRR3593632.1 1 length=300\n",
      "TTCAACCTTGCGGCCGTACTTCTCAGGCGGTATACTTATTGCGTTAGCTGCGCCACTCCACCTTTCGATCCCCCACACCCCTTGTTCCCATCTTTTACCGCCCGGACTACTGCGGTATCTAACCCCTTTCGCTCCCCTCGCTTTCGTTCCTCCCTGTCTTTTTTCCCCCAGCTCCCCCCTTTCGCCTCCTGTTTTCTTCCCTCTATCTCCGTTTTTCCCCTCTACCCGGTGATTTCCTCTCCCCTCTCCCACACTCGATTCTGCCATTTTACAGTCCTTTTTCGTTGTTCACCCCCCATC\n",
      "+SRR3593632.1 1 length=300\n",
      "AC6A-,=CDFFF+7F:CFGG,E,CFC<68@++6,CFF@6,;,>BE7@EF,67+,+8,,:,,,,,6,,,,,,96+++,6,4+,,,5,,4,,44,<A,9:,+++++++8+49,,,+83+8>2+,+38++8@=*1*;==A8*+*6*3304)1;C=+)5*3)++5+*3)*3*?45,))*)))-(-)1((/((,()2)1()).)/()(/)16.5)(()0((()/8(-().)((()(().--6).-((,((-3(,,41((81(.()))))).))--)..4.).))).).-.((-(--.-4(((4,.\n",
      "\n",
      "==> SRR3593664_1.fastq <==\n",
      "@SRR3593664.1 1 length=300\n",
      "TACGTAGGATGCAAGCGTTGTCCGGATTTATTGGGCGTAAAGCGTTCGTAGGTGGTTTTTTAAGTTTGGTGTTAAATCTTCCGGCTCCACCCTGATACTGCTCTTAATACTGTCAGACTCGAGTGTGGTAGAGGCTATTGGATTTCCCAGTGTAGCGGTGAATTGCGTAGTTTTTTGGATTATCCCCGTTGGCGTAGGCGACTTTCTGGTCCTTTACTGACACTGAGGAACGAAAGCCTGGGTAGCGAATCGGATTAGTTCCCCCTTTAGTCCTGGCCGTACACGATGGTTCCTAGTCGT\n",
      "+SRR3593664.1 1 length=300\n",
      "CCCCCGGG?FFGCGGGGGG7C@FCF86CC,<ECCEC@F@,,C,88686@,::;,98C@,9@,,:<6@,,::EF<@9,,99,,,8@F4,6C94=,,,,5<,9,,<,,,9<9C,,5,,,9A4+,+C+<,+995,,,+,,,9,,5,:A447,,8,8@,4++@++5,@,@3@+@+@,@=+++,,4>,7,:@**6*,3**1**3**14*,4=,,,10,4,419+4>908*20**)/61*573)).2((/,3/*)0(03(4;<<).+3<:)**.-).6@6)(36(-3(*.4(3-()-),.:))-,4\n",
      "\n",
      "==> SRR3593664_2.fastq <==\n",
      "@SRR3593664.1 1 length=300\n",
      "TTCAACCTTGCGCCCGTACTACTCAGGCGGTATACTTATCGCGTTAGCTTCGCCACTTCCCCCTTCTATACCCCATACCCCTTGTATCCTTCTTTTACCGCCTGCACTACTGCGGTATCTATTCCCTTTCGCTCCCCTCGCTTTCGTTCCTCAGTGTCTTTTCTTCCCCCGCTCCTCTCCTTCGCCTCCTGTTTTATTCCCAATATCTTCGCTTTTCCCCGCTACACTTGGTTTTCCTCTCGCCTCTACCACCCTCGATTCTCCCATTTTCCAGTGCAGTTTCAGTTTTTCTCCCCCCTT\n",
      "+SRR3593664.1 1 length=300\n",
      "@B6--6CFECEF+,@+@+BF9F,EF;6C8+++:5EFG<@,:6@CE76ED,C+,6,?,,:,,,++64,,6,669,,6,9,,:?,,:,,44,9,,<:9,A,,+4++,,94,9,,,++3+3>++,444,,9=338:?C=C8*+*60=0*151@DD+)5+60++21*2**5=D**+))*03*308*:)85))/))5*10/1/*/7*(7)98**1/((055@*.5(4..2)6.)+))(,/)7))1((,4,4()6)).(--((((.)))..()-).)))...6)))--))))-))(,)66,((-,(\n",
      "\n",
      "==> SRR3593665_1.fastq <==\n",
      "@SRR3593665.1 1 length=300\n",
      "TACGGAGGGAGCTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCACGTAGGCGGCTTTGTAAGTTAGAGGTGAAAGCCTGGAGCTCAACTCCAGAATTGCCTTTAAGACTGCATCGCTTGAATCCAGGAGAGGTGAGTGGAATTCCGAGTGTAGAGGTGAAATTCGTAGATATTCGGAAGAACACCAGTGGCGAAGGCGGCTCACTGGACTGGTATTGACGCTGAGGTGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGATAACTAGCTGT\n",
      "+SRR3593665.1 1 length=300\n",
      "CCCCCGGGGGGGGGGGGGGGGGGGGGDFFFEGGGGFGGFFFGEF@@@@FFGGCCG7CCC,C<<FFF,C9,CFFGEF9CAE,,=FFGCFFGDFD8=,,<?=AAEFE9,,C?F@@9EE+@FD@8=FD?EF,+?<FFF798CDFGFFGFF,+8@@EFC9EGFC:8,ECF@DFG>DFGGEDC85;D7@>FC>5DFGGGGDDGF5CCED+;CFC*2=55>;=CFCGE1?DBF:D3?B:5<9?)8>>B0<>;CC>;6F?<BF7>95+3<8F6D84<<>F4>?9:69??:6?9:365)6<A<7(64<\n",
      "\n",
      "==> SRR3593665_2.fastq <==\n",
      "@SRR3593665.1 1 length=300\n",
      "TTTAATCTTGCGACCGTACTCCCCAGGCGGATAACTTAATGCGTTAGCTTCGCCACCAAAGTTCTGTGAACCCTGACAGCTAGTTATCATCGTTTTCGGCGTGGACTACCAGGGTATCTTATCCTGTTTGCTCCCCACGCTTTCGCACCCCAGCGTCAATACCAGTCCAGTTATCCGCCTTCGCCACTCTTGTTCTTCCGAATATCTACGAATTTCACCTCTACACTCGGAATTCCACCCACCTCTCCTGGATTCATGCGATCCAGTCTTAAAGGCAATTCTGGATTTTAGCTCCATGCT\n",
      "+SRR3593665.1 1 length=300\n",
      "CCCC-CFFFFFGC@CCFFGGDFEEEFDCCC7@,5FFG<,<E<EEGFEGF,C@78,?8<,,,9AEE,<9,,:C?,4,:,,95,,CA,:@,@;,AED,9++4+8+++4,,AB,,+?439:C++7===,@EE?79??@EF8*60@F+*:+15*)*+05::)+;+3)+)3:?7*?,;*4;5):=<349A*:.)*/*18**976(---/*/*5@((((27796@3273>)67@3)(233)9)7(/(462,),46.(())<<)))((-(-))..642<))),((.)...)).)4))(.).4().()\n",
      "SRR3593621_1.fastq:52945\n",
      "SRR3593621_2.fastq:52945\n",
      "SRR3593622_1.fastq:62218\n",
      "SRR3593622_2.fastq:62218\n",
      "SRR3593623_1.fastq:92740\n",
      "SRR3593623_2.fastq:92740\n",
      "SRR3593625_1.fastq:70366\n",
      "SRR3593625_2.fastq:70366\n",
      "SRR3593627_1.fastq:100615\n",
      "SRR3593627_2.fastq:100615\n",
      "SRR3593628_1.fastq:78495\n",
      "SRR3593628_2.fastq:78495\n",
      "SRR3593631_1.fastq:97332\n",
      "SRR3593631_2.fastq:97332\n",
      "SRR3593632_1.fastq:84361\n",
      "SRR3593632_2.fastq:84361\n",
      "SRR3593664_1.fastq:101827\n",
      "SRR3593664_2.fastq:101827\n",
      "SRR3593665_1.fastq:84850\n",
      "SRR3593665_2.fastq:84850\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "cd Documentos/Tema_2/fastq\n",
    "head -4 ${FILE_ID}*fastq\n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq\n",
    "\n",
    "#echo \"#### Compute quality\"\n",
    "#mkdir ${FILE_ID}_Quality\n",
    "#fastqc ${FILE_ID}_R1.fastq -o ${FILE_ID}_Quality/\n",
    "#fastqc ${FILE_ID}_R2.fastq -o ${FILE_ID}_Quality/\n",
    "#\n",
    "#echo \"#### Replace ' ' by '_' in header\"\n",
    "#head -n 1 ${FILE_ID}*fastq\n",
    "#cat ${FILE_ID}_R1.fastq | sed 's/ /_/g' > ${FILE_ID}_R1_.fastq\n",
    "#cat ${FILE_ID}_R2.fastq | sed 's/ /_/g' > ${FILE_ID}_R2_.fastq\n",
    "#head -n 1 ${FILE_ID}*fastq\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=SRR, FASTQ_STR=@SRR\n",
      "Usage: qiime [OPTIONS] COMMAND [ARGS]...\n",
      "\n",
      "  QIIME 2 command-line interface (q2cli)\n",
      "  --------------------------------------\n",
      "\n",
      "  To get help with QIIME 2, visit https://qiime2.org.\n",
      "\n",
      "  To enable tab completion in Bash, run the following command or add it to\n",
      "  your .bashrc/.bash_profile:\n",
      "\n",
      "      source tab-qiime\n",
      "\n",
      "  To enable tab completion in ZSH, run the following commands or add them to\n",
      "  your .zshrc:\n",
      "\n",
      "      autoload bashcompinit && bashcompinit && source tab-qiime\n",
      "\n",
      "Options:\n",
      "  --version  Show the version and exit.\n",
      "  --help     Show this message and exit.\n",
      "\n",
      "Commands:\n",
      "  info                Display information about current deployment.\n",
      "  tools               Tools for working with QIIME 2 files.\n",
      "  dev                 Utilities for developers and advanced users.\n",
      "  alignment           Plugin for generating and manipulating alignments.\n",
      "  composition         Plugin for compositional data analysis.\n",
      "  cutadapt            Plugin for removing adapter sequences, primers, and\n",
      "                      other unwanted sequence from sequence data.\n",
      "  dada2               Plugin for sequence quality control with DADA2.\n",
      "  deblur              Plugin for sequence quality control with Deblur.\n",
      "  demux               Plugin for demultiplexing & viewing sequence quality.\n",
      "  diversity           Plugin for exploring community diversity.\n",
      "  emperor             Plugin for ordination plotting with Emperor.\n",
      "  feature-classifier  Plugin for taxonomic classification.\n",
      "  feature-table       Plugin for working with sample by feature tables.\n",
      "  fragment-insertion  Plugin for extending phylogenies.\n",
      "  gneiss              Plugin for building compositional models.\n",
      "  longitudinal        Plugin for paired sample and time series analyses.\n",
      "  metadata            Plugin for working with Metadata.\n",
      "  phylogeny           Plugin for generating and manipulating phylogenies.\n",
      "  quality-control     Plugin for quality control of feature and sequence data.\n",
      "  quality-filter      Plugin for PHRED-based filtering and trimming.\n",
      "  sample-classifier   Plugin for machine learning prediction of sample\n",
      "                      metadata.\n",
      "  taxa                Plugin for working with feature taxonomy annotations.\n",
      "  vsearch             Plugin for clustering and dereplicating with vsearch.\n",
      "Imported samplemanifest as PairedEndFastqManifestPhred33 to paired-end-demux.qza\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "source activate qiime2-2018.11\n",
    "cd Documentos/Tema_2\n",
    "qiime\n",
    "qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "                   --input-path samplemanifest \\\n",
    "                   --output-path paired-end-demux.qza \\\n",
    "                   --input-format PairedEndFastqManifestPhred33\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=SRR, FASTQ_STR=@SRR\n",
      "Saved Visualization to: paired-end-demux.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "source activate qiime2-2018.11\n",
    "cd Documentos/Tema_2\n",
    "qiime demux summarize --i-data paired-end-demux.qza --o-visualization paired-end-demux.qzv\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Determination of sOTUs using Deblur"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sample pre-processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=SRR, FASTQ_STR=@SRR\n",
      "Saved SampleData[JoinedSequencesWithQuality] to: joined-reads.qza\n",
      "Saved Visualization to: joined-reads.qzv\n",
      "Saved SampleData[JoinedSequencesWithQuality] to: joined-filt-reads.qza\n",
      "Saved QualityFilterStats to: filt_stats.qza\n",
      "Saved Visualization to: joined-filt-reads.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "source activate qiime2-2018.11\n",
    "cd Documentos/Tema_2\n",
    "\n",
    "# Deblur does not currently support unpaired paired-end readings, \n",
    "#so we have to use the VSEARCH algorithm to merge the readings:\n",
    "qiime vsearch join-pairs --i-demultiplexed-seqs paired-end-demux.qza \\\n",
    "                         --o-joined-sequences joined-reads.qza\n",
    "qiime demux summarize --i-data joined-reads.qza --o-visualization joined-reads.qzv\n",
    "\n",
    "# Filtering of the readings according to their quality.\n",
    "qiime quality-filter q-score-joined --i-demux joined-reads.qza \\\n",
    "                                    --o-filter-stats filt_stats.qza \\\n",
    "                                    --o-filtered-sequences joined-filt-reads.qza\n",
    "\n",
    "qiime demux summarize --i-data joined-filt-reads.qza --o-visualization joined-filt-reads.qzv\n",
    "\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Determination of sOTUs/features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved Visualization to: deblur_output/deblur_table_summary.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "source activate qiime2-2018.11\n",
    "cd Documentos/Tema_2\n",
    "\n",
    "qiime deblur denoise-16S --i-demultiplexed-seqs joined-filt-reads.qza \\\n",
    "                         --p-trim-length 300 \\\n",
    "                         --p-sample-stats \\\n",
    "                         --p-jobs-to-start 2 \\\n",
    "                         --p-min-reads 1 \\\n",
    "                         --output-dir deblur_output\n",
    "\n",
    "qiime feature-table summarize --i-table deblur_output/table.qza --o-visualization deblur_output/deblur_table_summary.qzv\n",
    "\n",
    "\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saved FeatureTable[Frequency] to: deblur_output/deblur_table_filt.qza\n",
      "Saved FeatureData[Sequence] to: deblur_output/rep_seqs_filt.qza\n",
      "Saved Visualization to: deblur_output/deblur_table_filt_summary.qzv\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<\"EOT\"\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "source activate qiime2-2018.11\n",
    "cd Documentos/Tema_2\n",
    "\n",
    "# Exclude sOTUS that have a frequency below 0.1% of the mean depth. \n",
    "# This threshold would exclude those sOTUs that are due to Illumina sequencing errors (0.1% of total reads).\n",
    "\n",
    "qiime feature-table filter-features --i-table deblur_output/table.qza \\\n",
    "                                    --p-min-frequency 3 \\\n",
    "                                    --p-min-samples 1 \\\n",
    "                                    --o-filtered-table deblur_output/deblur_table_filt.qza\n",
    "\n",
    "# Exclude low frequency sOTUS\n",
    "qiime feature-table filter-seqs --i-data deblur_output/representative_sequences.qza \\\n",
    "                                --i-table deblur_output/deblur_table_filt.qza \\\n",
    "                                --o-filtered-data deblur_output/rep_seqs_filt.qza\n",
    "\n",
    "\n",
    "# Summarize\n",
    "qiime feature-table summarize --i-table deblur_output/deblur_table_filt.qza --o-visualization deblur_output/deblur_table_filt_summary.qzv\n",
    "\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Phylogenetic distances determination using FastTree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "latex_metadata": {
     "hidden": "true",
     "lexer": "bash"
    }
   },
   "source": [
    "# Outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "latex_metadata": {
     "hidden": "true",
     "lexer": "bash"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "This is pdfTeX, Version 3.14159265-2.6-1.40.19 (TeX Live 2018) (preloaded format=pdflatex)\n",
      " \\write18 enabled.\n",
      "entering extended mode\n",
      "(./environmental_population_analysis.tex\n",
      "LaTeX2e <2018-04-01> patch level 2\n",
      "Babel <3.18> and hyphenation patterns for 84 language(s) loaded.\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/article.cls\n",
      "Document Class: article 2014/09/29 v1.4h Standard LaTeX document class\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/size11.clo))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/placeins/placeins.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsfonts/amssymb.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsfonts/amsfonts.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsmath/amsmath.sty\n",
      "For additional information on amsmath, use the `?' option.\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsmath/amstext.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsmath/amsgen.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsmath/amsbsy.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amsmath/amsopn.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/amscls/amsthm.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/mathtools/mathtools.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics/keyval.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tools/calc.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/mathtools/mhsetup.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/fontenc.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/t1enc.def))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/mathpazo.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics/graphicx.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics/graphics.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics/trig.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics-def/pdftex.def)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/caption/caption.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/caption/caption3.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/adjustbox/adjustbox.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/xkeyval/xkeyval.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/xkeyval/xkeyval.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/xkeyval/xkvutils.tex)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/adjustbox/adjcalc.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/adjustbox/trimclip.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/collectbox/collectbox.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/adjustbox/tc-pdftex.def))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ifoddpage/ifoddpage.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/varwidth/varwidth.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/xcolor/xcolor.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/graphics-cfg/color.cfg))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tools/enumerate.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/geometry/geometry.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/ifpdf.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/ifvtex.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/ifxetex/ifxetex.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/textcomp.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/ts1enc.def\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/ts1enc.dfu)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/upquote/upquote.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/eurosym/eurosym.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ucs/ucs.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ucs/data/uni-global.def))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/inputenc.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ucs/utf8x.def))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/fancyvrb/fancyvrb.sty\n",
      "Style option: `fancyvrb' v2.7a, with DG/SPQR fixes, and firstline=lastline fix \n",
      "<2008/02/07> (tvz))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/grffile.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/kvoptions.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/ltxcmds.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/kvsetkeys.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/infwarerr.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/etexcmds.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/ifluatex.sty))))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/pdftexcmds.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/hyperref/hyperref.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/hobsub-hyperref.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/hobsub-generic.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/auxhook.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/hyperref/pd1enc.def)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/latexconfig/hyperref.cfg)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/url/url.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/hyperref/hpdftex.def\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/rerunfilecheck.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tools/longtable.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/booktabs/booktabs.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/enumitem/enumitem.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/ulem/ulem.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/minted/minted.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/fvextra/fvextra.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/ifthen.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/etoolbox/etoolbox.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/lineno/lineno.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tools/shellesc.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ifplatform/ifplatform.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/catchfile.sty)\n",
      "(./environmental_population_analysis.w18))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/xstring/xstring.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/xstring/xstring.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/framed/framed.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/float/float.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcolorbox.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/basiclayer/pgf.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/utilities/pgfrcs.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfutil-common.te\n",
      "x\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfutil-common-li\n",
      "sts.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfutil-latex.def\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ms/everyshi.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfrcs.code.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/basiclayer/pgfcore.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/systemlayer/pgfsys.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgfsys.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfkeysfiltered.c\n",
      "ode.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgf.cfg)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgfsys-pdftex.d\n",
      "ef\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgfsys-common-p\n",
      "df.def)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgfsyssoftpath.\n",
      "code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/systemlayer/pgfsysprotocol.\n",
      "code.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcore.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmath.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathcalc.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathutil.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathparser.code.tex\n",
      ")\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.code.\n",
      "tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.basic\n",
      ".code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.trigo\n",
      "nometric.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.rando\n",
      "m.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.compa\n",
      "rison.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.base.\n",
      "code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.round\n",
      ".code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.misc.\n",
      "code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfunctions.integ\n",
      "erarithmetics.code.tex)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmathfloat.code.tex)\n",
      ")\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepoints.co\n",
      "de.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathconst\n",
      "ruct.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathusage\n",
      ".code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorescopes.co\n",
      "de.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoregraphicst\n",
      "ate.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoretransform\n",
      "ations.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorequick.cod\n",
      "e.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreobjects.c\n",
      "ode.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepathproce\n",
      "ssing.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorearrows.co\n",
      "de.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreshade.cod\n",
      "e.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreimage.cod\n",
      "e.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoreexternal.\n",
      "code.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorelayers.co\n",
      "de.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcoretranspare\n",
      "ncy.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/basiclayer/pgfcorepatterns.\n",
      "code.tex)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/modules/pgfmoduleshapes.cod\n",
      "e.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/modules/pgfmoduleplot.code.\n",
      "tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/compatibility/pgfcomp-version\n",
      "-0-65.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/compatibility/pgfcomp-version\n",
      "-1-18.sty)) (/usr/local/texlive/2018/texmf-dist/tex/latex/tools/verbatim.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/environ/environ.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/trimspaces/trimspaces.sty)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcbminted.code.tex\n",
      "Library (tcolorbox): 'tcbminted.code.tex' version '4.13'\n",
      "\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcblistingscore.code.te\n",
      "x\n",
      "Library (tcolorbox): 'tcblistingscore.code.tex' version '4.13'\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcbprocessing.code.tex\n",
      "Library (tcolorbox): 'tcbprocessing.code.tex' version '4.13'\n",
      ")))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcbbreakable.code.tex\n",
      "Library (tcolorbox): 'tcbbreakable.code.tex' version '4.13'\n",
      ") (/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcbskins.code.tex\n",
      "Library (tcolorbox): 'tcbskins.code.tex' version '4.13'\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/frontendlayer/tikz.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/utilities/pgffor.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/utilities/pgfkeys.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgfkeys.code.tex)\n",
      ") (/usr/local/texlive/2018/texmf-dist/tex/latex/pgf/math/pgfmath.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmath.code.tex))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/utilities/pgffor.code.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/math/pgfmath.code.tex)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/frontendlayer/tikz/tikz.cod\n",
      "e.tex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/libraries/pgflibraryplothan\n",
      "dlers.code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/modules/pgfmodulematrix.cod\n",
      "e.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/pgf/frontendlayer/tikz/librarie\n",
      "s/tikzlibrarytopaths.code.tex)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/tcolorbox/tcbskinsjigsaw.code.tex\n",
      "Library (tcolorbox): 'tcbskinsjigsaw.code.tex' version '4.13'\n",
      ")) (/usr/local/texlive/2018/texmf-dist/tex/latex/mdframed/mdframed.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/l3packages/xparse/xparse.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/l3kernel/expl3.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/l3kernel/expl3-code.tex)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/l3kernel/l3pdfmode.def)))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/zref-abspage.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/zref-base.sty))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/needspace/needspace.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/mdframed/md-frame-1.mdf))/Users/nandoide/miniconda3/bin/pygmentize\n",
      "\n",
      "(./environmental_population_analysis.aux)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/ts1cmr.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/t1ppl.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/context/base/mkii/supp-pdf.mkii\n",
      "[Loading MPS to PDF converter (version 2006.09.02).]\n",
      ") (/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/epstopdf-base.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/oberdiek/grfext.sty)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/caption/ltcaption.sty)\n",
      "*geometry* driver: auto-detecting\n",
      "*geometry* detected driver: pdftex\n",
      "\n",
      "Package geometry Warning: The marginal notes overrun the paper.\n",
      "     Add 25.63757pt and more to the right margin.\n",
      "\n",
      "*geometry* verbose mode - [ preamble ] result:\n",
      "* driver: pdftex\n",
      "* paper: <default>\n",
      "* layout: <same size as paper>\n",
      "* layoutoffset:(h,v)=(0.0pt,0.0pt)\n",
      "* modes: \n",
      "* h-part:(L,W,R)=(43.36243pt, 527.57014pt, 43.36243pt)\n",
      "* v-part:(T,H,B)=(72.26999pt, 650.43001pt, 72.26999pt)\n",
      "* \\paperwidth=614.295pt\n",
      "* \\paperheight=794.96999pt\n",
      "* \\textwidth=527.57014pt\n",
      "* \\textheight=650.43001pt\n",
      "* \\oddsidemargin=-28.90756pt\n",
      "* \\evensidemargin=-28.90756pt\n",
      "* \\topmargin=-37.0pt\n",
      "* \\headheight=12.0pt\n",
      "* \\headsep=25.0pt\n",
      "* \\topskip=11.0pt\n",
      "* \\footskip=30.0pt\n",
      "* \\marginparwidth=59.0pt\n",
      "* \\marginparsep=10.0pt\n",
      "* \\columnsep=10.0pt\n",
      "* \\skip\\footins=10.0pt plus 4.0pt minus 2.0pt\n",
      "* \\hoffset=0.0pt\n",
      "* \\voffset=0.0pt\n",
      "* \\mag=1000\n",
      "* \\@twocolumnfalse\n",
      "* \\@twosidefalse\n",
      "* \\@mparswitchfalse\n",
      "* \\@reversemarginfalse\n",
      "* (1in=72.27pt=25.4mm, 1cm=28.453pt)\n",
      "\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ucs/ucsencs.def)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/hyperref/nameref.sty\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/generic/oberdiek/gettitlestring.sty))\n",
      "(./environmental_population_analysis.out)\n",
      "(./environmental_population_analysis.out)\n",
      "ABD: EveryShipout initializing macros\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/ot1ppl.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/omlzplm.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/omszplm.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/omxzplm.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/psnfss/ot1zplm.fd)\n",
      "(./environmental_population_analysis.toc) [1{/usr/local/texlive/2018/texmf-var/\n",
      "fonts/map/pdftex/updmap/pdftex.map}] [2 <./img01.jpg>]\n",
      "(./_minted-environmental_population_analysis/default-pyg-prefix.pygstyle)\n",
      "(./_minted-environmental_population_analysis/default.pygstyle)\n",
      "(./_minted-environmental_population_analysis/8732FDB1896F32A2EB2BF846C3275CADAD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/t1cmtt.fd)\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/base/ts1cmtt.fd)) [3]\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 588--589\n",
      "\n",
      "\n",
      "(./_minted-environmental_population_analysis/31E49A7286ABA809EEA5BD1E8F06F8D3AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex) [4 <./environmental_population_analysis_\n",
      "files/environmental_population_analysis_3_0.png>]\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 606--607\n",
      "\n",
      "\n",
      "(./_minted-environmental_population_analysis/7D9B9DCE49F359D20B19DCEEBF628061AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex) [5 <./environmental_population_analysis_\n",
      "files/environmental_population_analysis_5_0.png>]\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 623--624\n",
      "\n",
      "\n",
      "(./_minted-environmental_population_analysis/8CF4790162334710A2FB600E8786CEA7AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex) [6 <./environmental_population_analysis_\n",
      "files/environmental_population_analysis_7_0.png>]\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 663--664\n",
      "\n",
      "\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 668--669\n",
      "\n",
      "[7 <./environmental_population_analysis_files/environmental_population_analysis\n",
      "_9_0.png> <./environmental_population_analysis_files/environmental_population_a\n",
      "nalysis_9_1.png>]\n",
      "(./_minted-environmental_population_analysis/7C0A7EE566D273A2FC3D675259263615AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex) [8]\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 709--710\n",
      "\n",
      "\n",
      "Underfull \\hbox (badness 10000) in paragraph at lines 714--715\n",
      "\n",
      "[9 <./environmental_population_analysis_files/environmental_population_analysis\n",
      "_11_0.png> <./environmental_population_analysis_files/environmental_population_\n",
      "analysis_11_1.png>]\n",
      "(./_minted-environmental_population_analysis/016FA97D008ADE451765E4B33D165901AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex)\n",
      "(./_minted-environmental_population_analysis/BEF14BFEB06535D39CCBD5EA77953DD2F5\n",
      "1B5F4C7AC0F35D40821A8A30C1D5E4.pygtex\n",
      "(/usr/local/texlive/2018/texmf-dist/tex/latex/ucs/data/uni-32.def)) [10]\n",
      "[11] [12]pdfTeX warning (ext4): destination with the same identifier (name{prep\n",
      "rocessing-and-quality-check}) has been already used, duplicate ignored\n",
      "<to be read again> \n",
      "                   \\relax \n",
      "l.910 ...}\\label{preprocessing-and-quality-check}}\n",
      "                                                  \n",
      "(./_minted-environmental_population_analysis/FB7F56FDA6C734FF4E84B65429CA97B9AD\n",
      "816AA5B8E26EAA98CD0343164964A6.pygtex)\n",
      "(./_minted-environmental_population_analysis/14329B0D71D1957C68F1ABE8EBC44A32F5\n",
      "1B5F4C7AC0F35D40821A8A30C1D5E4.pygtex\n",
      "! Argument of \\FV@Break@Next has an extra }.\n",
      "<inserted text> \n",
      "                \\par \n",
      "l.11 \\PYG{c+c1}{\\PYGZsh{}mkdir ${FILE_ID}_Quality}\n",
      "                                                  \n",
      "? Type <return> to proceed, S to scroll future error messages,\n",
      "R to run without stopping, Q to run quietly,\n",
      "I to insert something, E to edit your file,\n",
      "1 or ... or 9 to ignore the next 1 to 9 tokens of input,\n",
      "H for help, X to quit.\n",
      "? \n",
      "! Emergency stop.\n",
      "<inserted text> \n",
      "                \\par \n",
      "l.11 \\PYG{c+c1}{\\PYGZsh{}mkdir ${FILE_ID}_Quality}\n",
      "                                                  \n",
      "!  ==> Fatal error occurred, no output PDF file produced!\n",
      "Transcript written on environmental_population_analysis.log.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[NbConvertApp] Converting notebook environmental_population_analysis.ipynb to latex\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "/Users/nandoide/miniconda3/lib/python3.7/site-packages/nbconvert/utils/pandoc.py:52: RuntimeWarning: You are using an unsupported version of pandoc (2.2.3.2).\n",
      "Your version must be at least (1.12.1) but less than (2.0.0).\n",
      "Refer to http://pandoc.org/installing.html.\n",
      "Continuing with doubts...\n",
      "  check_pandoc_version()\n",
      "[NbConvertApp] Support files will be in environmental_population_analysis_files/\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Making directory environmental_population_analysis_files\n",
      "[NbConvertApp] Writing 110179 bytes to environmental_population_analysis.tex\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'jupyter nbconvert --to=latex --template=~/report.tplx environmental_population_analysis.ipynb\\n/Library/TeX/texbin/pdflatex -shell-escape environmental_population_analysis\\njupyter nbconvert --to html_toc environmental_population_analysis.ipynb 1>/dev/null 2>/dev/null\\n'' returned non-zero exit status 1.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-10-89affa512c8a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'jupyter nbconvert --to=latex --template=~/report.tplx environmental_population_analysis.ipynb\\n/Library/TeX/texbin/pdflatex -shell-escape environmental_population_analysis\\njupyter nbconvert --to html_toc environmental_population_analysis.ipynb 1>/dev/null 2>/dev/null\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2350\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2351\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2352\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2353\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2354\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m</Users/nandoide/miniconda3/lib/python3.7/site-packages/decorator.py:decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'jupyter nbconvert --to=latex --template=~/report.tplx environmental_population_analysis.ipynb\\n/Library/TeX/texbin/pdflatex -shell-escape environmental_population_analysis\\njupyter nbconvert --to html_toc environmental_population_analysis.ipynb 1>/dev/null 2>/dev/null\\n'' returned non-zero exit status 1."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "jupyter nbconvert --to=latex --template=~/report.tplx environmental_population_analysis.ipynb\n",
    "/Library/TeX/texbin/pdflatex -shell-escape environmental_population_analysis\n",
    "jupyter nbconvert --to html_toc environmental_population_analysis.ipynb 1>/dev/null 2>/dev/null"
   ]
  }
 ],
 "metadata": {
  "celltoolbar": "Edit Metadata",
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "latex_metadata": {
   "author": "Fernando Freire",
   "title": "Environmental Population Analysis"
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
