{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Whole Genome Shotgun metagenomics: de novo Assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have two fastqc files to process:\n",
    "\n",
    "*Vir1_100k* that contains $100,000$ paired end reads from the same saliva sample but after purification of viral particles. So this is a virome.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import pandas as pd\n",
    "FILE_ID = \"ECTV\"\n",
    "FASTQ_STR = \"@HWUSI-EAS1752R\"\n",
    "MIN_LEN = \"70\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preprocessing and quality check\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=ECTV, FASTQ_STR=@HWUSI-EAS1752R\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "NGGATCTCCGATTTCTTTACGATATGGATCTATACCGGACGAATTAATAAACAAACATCCAAAAAAATATGGAATT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "#*(.(,+,,+@@@@@@222@@@@@@@@:@@@@@@@:::<<71757<<:<:22@222:@@@8518500000:::8:5\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_1:N:0:GCCAAT\n",
      "NGAGTAATACGGTTCAAAATCATAAATGTGATAGTTTCCAGACTGGTATCCGAGTTTTTCTTGGATGATGGATACT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_1:N:0:GCCAAT\n",
      "#,*))33322@C@@@@C@@C@C@C@@@@@@@@CC@@@C@C@@@@@C@C@@@@@@@C@@CC222@@@@@@C@@@@@@\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010_1:N:0:GCCAAT\n",
      "NTGGAGTCGTAAAAAAGTTTTATCTCTTTCTCTCTTTGATGGTCTCATAAAAAAGTTTTACAAAAATATTTTTATT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "NGGATCTCCGATTTCTTTACGATATGGATCTATACCGGACGAATTAATAAACAAACATCCAAAAAAATATGGAATT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "#*(.(,+,,+@@@@@@222@@@@@@@@:@@@@@@@:::<<71757<<:<:22@222:@@@8518500000:::8:5\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 1:N:0:GCCAAT\n",
      "NGAGTAATACGGTTCAAAATCATAAATGTGATAGTTTCCAGACTGGTATCCGAGTTTTTCTTGGATGATGGATACT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 1:N:0:GCCAAT\n",
      "#,*))33322@C@@@@C@@C@C@C@@@@@@@@CC@@@C@C@@@@@C@C@@@@@@@C@@CC222@@@@@@C@@@@@@\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010 1:N:0:GCCAAT\n",
      "NTGGAGTCGTAAAAAAGTTTTATCTCTTTCTCTCTTTGATGGTCTCATAAAAAAGTTTTACAAAAATATTTTTATT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "ATCCTCTAACGTGGCAGTCATTAAAAACATAGAATCTATTTTCGTATGATGCTTTCTCGCTACTGCTATAATGATA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "HHIHII@GGIIDIGDIGDGGIDIGGGIIHGG?GGDIGEGIEGDEGHIHIIGGIIIIHHH>@DDGGIEHHFFIGBDD\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_2:N:0:GCCAAT\n",
      "CGATGATGTAAGTCTTACAATGTTCGAACCAGTACCAGATCTTAATTTGGAGGCCTCCGTAGAACTAGGGGAGGGA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_2:N:0:GCCAAT\n",
      "IBIIIFIHIGHGIGDIIIB3EGGDEEGGGGBIFGBDGD@CDDDDAHIGHEEHHEGDGDDEEEDBD@9AAB?#####\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010_2:N:0:GCCAAT\n",
      "CAGAAAGAGAATAAAAATATTTTTGTAAAACTTTTTTATGAGACCATCAAAGAGAGAAAGAGATAAAACTTTTTTA\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "ATCCTCTAACGTGGCAGTCATTAAAAACATAGAATCTATTTTCGTATGATGCTTTCTCGCTACTGCTATAATGATA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "HHIHII@GGIIDIGDIGDGGIDIGGGIIHGG?GGDIGEGIEGDEGHIHIIGGIIIIHHH>@DDGGIEHHFFIGBDD\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 2:N:0:GCCAAT\n",
      "CGATGATGTAAGTCTTACAATGTTCGAACCAGTACCAGATCTTAATTTGGAGGCCTCCGTAGAACTAGGGGAGGGA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 2:N:0:GCCAAT\n",
      "IBIIIFIHIGHGIGDIIIB3EGGDEEGGGGBIFGBDGD@CDDDDAHIGHEEHHEGDGDDEEEDBD@9AAB?#####\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010 2:N:0:GCCAAT\n",
      "CAGAAAGAGAATAAAAATATTTTTGTAAAACTTTTTTATGAGACCATCAAAGAGAGAAAGAGATAAAACTTTTTTA\n",
      "ECTV_R1_.fastq:50000\n",
      "ECTV_R1.fastq:50000\n",
      "ECTV_R2_.fastq:50000\n",
      "ECTV_R2.fastq:50000\n",
      "#### Compute quality\n",
      "Analysis complete for ECTV_R1.fastq\n",
      "Analysis complete for ECTV_R2.fastq\n",
      "#### Replace ' ' by '_' in header\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "cd Documentos/Tema_3\n",
    "head ${FILE_ID}*fastq\n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq\n",
    "\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_Quality\n",
    "fastqc ${FILE_ID}_R1.fastq -o ${FILE_ID}_Quality/\n",
    "fastqc ${FILE_ID}_R2.fastq -o ${FILE_ID}_Quality/\n",
    "\n",
    "echo \"#### Replace ' ' by '_' in header\"\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "cat ${FILE_ID}_R1.fastq | sed 's/ /_/g' > ${FILE_ID}_R1_.fastq\n",
    "cat ${FILE_ID}_R2.fastq | sed 's/ /_/g' > ${FILE_ID}_R2_.fastq\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming and decontaminating\n",
    "Trimming poor quality ends and short sequences (**Trimmomatic**) and removal of reads aligning to the human and phiX174 genomes (***bowtie2**). The later one is a contaminant used as spike by Illumina kits to control quality of the sequencing process.\n",
    "\n",
    "We are only filtering only R1 files because forward reads have usually better quality than reverse reads. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Trimming and decontaminating FILE_ID=ECTV MIN_LEN=70\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq ): 50000\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq ): 50000\n",
      "Running Trimmomatic ... \n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq ): 41996\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq ): 41996\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq ): 4107\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq ): 1791\n",
      "Decontaminating ...\n",
      "Running bowtie2 ... \n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 41839\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 41839\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq ): 41839\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq ): 41839\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 12\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq ): 12\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 5851\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq ): 5851\n",
      "\n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "echo \"#### Trimming and decontaminating FILE_ID=${FILE_ID} MIN_LEN=${MIN_LEN}\"\n",
    "kneaddata -i ${FILE_ID}_R1_.fastq -i ${FILE_ID}_R2_.fastq \\\n",
    "-o kneaddata_out -db /home/shared/bowtiedb/GRCh38_PhiX \\\n",
    "--trimmomatic /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/ \\\n",
    "-t 2 --trimmomatic-options \"SLIDINGWINDOW:4:20 MINLEN:${MIN_LEN}\" \\\n",
    "--bowtie2-options \"--very-sensitive --dovetail\" --remove-intermediate-output\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - INFO: Running kneaddata v0.6.1\n",
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - INFO: Output files will be written to: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - DEBUG: Running with the following arguments: \n",
      "verbose = False\n",
      "bmtagger_path = None\n",
      "minscore = 50\n",
      "bowtie2_path = /home/microbioinf/miniconda3/bin/bowtie2\n",
      "maxperiod = 500\n",
      "no_discordant = False\n",
      "serial = False\n",
      "fastqc_start = False\n",
      "bmtagger = False\n",
      "cat_final_output = False\n",
      "log_level = DEBUG\n",
      "log = /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.log\n",
      "max_memory = 500m\n",
      "remove_intermediate_output = True\n",
      "fastqc_path = None\n",
      "output_dir = /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "trf_path = None\n",
      "remove_temp_output = True\n",
      "reference_db = /home/shared/bowtiedb/GRCh38_PhiX\n",
      "input = /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq\n",
      "pi = 10\n",
      "reorder = False\n",
      "pm = 80\n",
      "trimmomatic_path = /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar\n",
      "store_temp_output = False\n",
      "cat_pairs = False\n",
      "mismatch = 7\n",
      "threads = 2\n",
      "delta = 7\n",
      "bowtie2_options = --very-sensitive --dovetail --phred33\n",
      "bypass_trim = False\n",
      "processes = 1\n",
      "trimmomatic_quality_scores = -phred33\n",
      "fastqc_end = False\n",
      "trf = False\n",
      "trimmomatic_options = SLIDINGWINDOW:4:20 MINLEN:70\n",
      "output_prefix = ECTV_R1__kneaddata\n",
      "match = 2\n",
      "\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: raw pair1 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq ): 50000\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: raw pair2 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq ): 50000\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Running Trimmomatic ... \n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Execute command: java -Xmx500m -d64 -jar /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar PE -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:70\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: TrimmomaticPE: Started with arguments:\n",
      " -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:70\n",
      "Input Read Pairs: 50000 Both Surviving: 41996 (83,99%) Forward Only Surviving: 4107 (8,21%) Reverse Only Surviving: 1791 (3,58%) Dropped: 2106 (4,21%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed pair1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq ): 41996\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed pair2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq ): 41996\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq ): 4107\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq ): 1791\n",
      "04/26/2019 08:44:27 AM - kneaddata.run - INFO: Decontaminating ...\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Running bowtie2 ... \n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Execute command: kneaddata_bowtie2_discordant_pairs --bowtie2 /home/microbioinf/miniconda3/bin/bowtie2 --threads 2 -x /home/shared/bowtiedb/GRCh38_PhiX --bowtie2-options \"--very-sensitive --dovetail --phred33\" -1 /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq -2 /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq --un-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_%.fastq --al-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_%.fastq -U /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq,/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq --un-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_clean.fastq --al-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_contam.fastq -S /dev/null\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: 47894 reads; of these:\n",
      "  41996 (87.69%) were paired; of these:\n",
      "    41881 (99.73%) aligned concordantly 0 times\n",
      "    106 (0.25%) aligned concordantly exactly 1 time\n",
      "    9 (0.02%) aligned concordantly >1 times\n",
      "    ----\n",
      "    41881 pairs aligned concordantly 0 times; of these:\n",
      "      10 (0.02%) aligned discordantly 1 time\n",
      "    ----\n",
      "    41871 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      83742 mates make up the pairs; of these:\n",
      "        83709 (99.96%) aligned 0 times\n",
      "        24 (0.03%) aligned exactly 1 time\n",
      "        9 (0.01%) aligned >1 times\n",
      "  5898 (12.31%) were unpaired; of these:\n",
      "    5832 (98.88%) aligned 0 times\n",
      "    59 (1.00%) aligned exactly 1 time\n",
      "    7 (0.12%) aligned >1 times\n",
      "0.39% overall alignment rate\n",
      "pair1_aligned : 115\n",
      "pair2_aligned : 115\n",
      "orphan1_unaligned : 12\n",
      "orphan2_unaligned : 5851\n",
      "orphan2_aligned : 89\n",
      "pair2_unaligned : 41839\n",
      "pair1_unaligned : 41839\n",
      "orphan1_aligned : 30\n",
      "\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq ) : 115\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq ) : 115\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq ) : 30\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq ) : 89\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final pair1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final pair2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 12\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final orphan1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq ): 12\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 5851\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final orphan2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq ): 5851\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.knead_data - INFO: \n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq\n",
      "\n",
      "Reading log: ./ECTV_R1__kneaddata.log\n",
      "Read count table written: kneaddata_read_countsECTV.txt\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:30\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:89\n",
      "ECTV_R1__kneaddata_paired_1.fastq:41839\n",
      "ECTV_R1__kneaddata_paired_2.fastq:41839\n",
      "ECTV_R1__kneaddata_unmatched_1.fastq:12\n",
      "ECTV_R1__kneaddata_unmatched_2.fastq:5851\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "l\n",
    "less *log\n",
    "kneaddata_read_count_table --input ./ --output kneaddata_read_counts${FILE_ID}.txt \n",
    "grep -c $FASTQ_STR *fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <td>ECTV_R1__kneaddata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair1</th>\n",
       "      <td>50000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair2</th>\n",
       "      <td>50000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair1</th>\n",
       "      <td>41996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair2</th>\n",
       "      <td>41996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan1</th>\n",
       "      <td>4107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan2</th>\n",
       "      <td>1791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair1</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair2</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan1</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan2</th>\n",
       "      <td>5851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair1</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair2</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan1</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan2</th>\n",
       "      <td>5851</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     0\n",
       "Sample                              ECTV_R1__kneaddata\n",
       "raw pair1                                        50000\n",
       "raw pair2                                        50000\n",
       "trimmed pair1                                    41996\n",
       "trimmed pair2                                    41996\n",
       "trimmed orphan1                                   4107\n",
       "trimmed orphan2                                   1791\n",
       "decontaminated GRCh38_PhiX pair1                 41839\n",
       "decontaminated GRCh38_PhiX pair2                 41839\n",
       "decontaminated GRCh38_PhiX orphan1                  12\n",
       "decontaminated GRCh38_PhiX orphan2                5851\n",
       "final pair1                                      41839\n",
       "final pair2                                      41839\n",
       "final orphan1                                       12\n",
       "final orphan2                                     5851"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/kneaddata_read_counts%s.txt\n",
    "EOT\n",
    "\"\"\" % FILE_ID\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_knead = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_knead.style.hide_index().set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"11pt\"}).\\\n",
    "set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"12pt\"}, subset=['Sample'])\n",
    "df_knead.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check number of reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With grep we can identify the non-contaminated high-quality files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:30\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:89\n",
      "ECTV_R1__kneaddata_paired_1.fastq:41839\n",
      "ECTV_R1__kneaddata_paired_2.fastq:41839\n",
      "ECTV_R1__kneaddata_unmatched_1.fastq:12\n",
      "ECTV_R1__kneaddata_unmatched_2.fastq:5851\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "grep -c $FASTQ_STR *fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute quality\n",
      "Analysis complete for ECTV_R1__kneaddata_paired_1.fastq\n",
      "Analysis complete for ECTV_R1__kneaddata_paired_2.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_HighQuality\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_1.fastq -o ${FILE_ID}_HighQuality/\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_2.fastq -o ${FILE_ID}_HighQuality/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assembly\n",
    "\n",
    " We are going to use a Refseq database of viral proteins (around 100Mb) from ncbi (ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/), and you have to download it in two separated files that can be joined into one with cat.\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process (spades)\n",
    "\n",
    "In this step we run command **spades** with the paired high-quality and free of known contaminants reads."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### K_MER = 35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "K_MER = 35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.000     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.368     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.368     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.729     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.729     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.734    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.779    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.779    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.786    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.912    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.921    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.921    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.973    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.207    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.391    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.392    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.399    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.448    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.548    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.551    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.551    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.703    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.705    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.705    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.394    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17286 new k-mers.\n",
      "  0:00:06.048    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.701    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.701    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.701    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.701    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.821    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.248    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.302    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.332    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.332    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.333    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "  0:00:07.334    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.174    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.285    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.429   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.454     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.579   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.585     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.706   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.713     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.713     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.713     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.714     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.714     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.714     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.714     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.962     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.962     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.966    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.966    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.979    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:00.980    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.984    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.985    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.985    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.985    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.986    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.986    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.084     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 380324 kmers\n",
      "  0:00:01.084     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 380324 kmers.\n",
      "  0:00:01.092    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.126    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380925 kmers in total.\n",
      "  0:00:01.126    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.133    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.233    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.247    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185224 bytes occupied (3.88998 bits per kmer).\n",
      "  0:00:01.248    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.310    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.310    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.310    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.390    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 31022 sequences extracted\n",
      "  0:00:01.433    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.448    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.475    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.475    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.475    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.511    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177904 bytes occupied (3.74216 bits per kmer).\n",
      "  0:00:01.511    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.683    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.727    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 62044 edges\n",
      "  0:00:01.729    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.766    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.766    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 9.00852\n",
      "  0:00:01.766    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 6\n",
      "  0:00:01.767    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 6\n",
      "  0:00:01.767    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.768    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.768    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.768    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.768    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.769    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.769    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.769    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.812     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 62044 edges\n",
      "  0:00:01.812     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 62044 sequences.\n",
      "  0:00:01.817    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.852    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:01.852    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.860    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.958    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.967    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184936 bytes occupied (3.89007 bits per kmer).\n",
      "  0:00:01.970    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.016    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.016    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.040    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.147    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.147    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.154    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.186    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 107 candidates\n",
      "  0:00:02.189    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.190    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.190    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.190    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.195    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.195    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.195    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.195    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.195    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.195    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13586 times\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.306    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 17 times\n",
      "  0:00:02.306    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 774 times\n",
      "  0:00:02.323    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.323    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 110 times\n",
      "  0:00:02.326    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 94 times\n",
      "  0:00:02.330    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 42 times\n",
      "  0:00:02.332    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:02.333    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.334    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.334    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.335    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.337    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.337    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.337    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.337    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.376     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3470 edges\n",
      "  0:00:02.376     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3470 sequences.\n",
      "  0:00:02.380    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.411    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 266582 kmers in total.\n",
      "  0:00:02.411    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.417    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.507    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.523    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132168 bytes occupied (3.9663 bits per kmer).\n",
      "  0:00:02.527    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.544    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.544    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.546    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.616    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.616    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.618    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.620    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 96 candidates\n",
      "  0:00:02.664    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.664    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.716    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.718    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.720    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.727    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.727    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.728    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.728    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.733    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.733    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.733    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.734    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.736    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.738    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.740    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.742    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.742    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.746    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.746    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.747    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.747    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.748    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.748    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.748    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.748    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.749    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.749    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.749    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.750    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.791    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1673 times\n",
      "  0:00:02.791    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.791    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 16.3895\n",
      "  0:00:02.791    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.791    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:02.792    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.014    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:03.019    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:03.019    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:03.019    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.019    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.022    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:03.024    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:03.031    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:03.034    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:03.036    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23095\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.064    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.226    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.447    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.447    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   16485 paired reads (39.4039% of all) aligned to long edges\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.556, deviation = 88.9493, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.458    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.562    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.567    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.568    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.568    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.568    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.570    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.570    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 3694\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.572    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.572    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.573    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 43; contradictional = 64\n",
      "  0:00:04.575    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 10; contradictional = 20\n",
      "  0:00:04.575    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.575    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 169\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.586    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.586    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 26 (0%)\n",
      "  0:00:04.587    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3 paths from 26 (11%)\n",
      "  0:00:04.589    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 26 (23%)\n",
      "  0:00:04.590    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 9 paths from 26 (34%)\n",
      "  0:00:04.590    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 26 (46%)\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 26 (57%)\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 26 (69%)\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 21 paths from 26 (80%)\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 26 (92%)\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.592    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.592    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.593    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.593    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.593    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.593    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.593    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.593    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.593    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.595    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:04.607    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/final_contigs.paths\n",
      "  0:00:04.611    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/scaffolds.paths\n",
      "  0:00:04.614    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.616    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"35\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -k ${K_MER} -o ${FILE_ID}-Assembly${K_MER}\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check output K_MER=35\n",
      ">NODE_1_length_61627_cov_16.136024\n",
      ">NODE_2_length_37406_cov_14.491584\n",
      ">NODE_3_length_35119_cov_16.619741\n",
      ">NODE_4_length_23130_cov_14.144966\n",
      ">NODE_5_length_14681_cov_16.895125\n",
      ">NODE_6_length_12066_cov_15.782063\n",
      ">NODE_7_length_9054_cov_26.310456\n",
      ">NODE_8_length_1060_cov_59.155122\n",
      ">NODE_1_length_61627_cov_16.136024\n",
      ">NODE_2_length_60827_cov_14.296881\n",
      ">NODE_3_length_47271_cov_16.363811\n",
      ">NODE_4_length_14681_cov_16.895125\n",
      ">NODE_5_length_9054_cov_26.310456\n",
      ">NODE_6_length_1060_cov_59.155122\n",
      ">NODE_7_length_945_cov_8.277551\n",
      ">NODE_8_length_294_cov_3.567568\n",
      "scaffolds.fasta:ACCTCAAAAAATTTGTTAATAATGGTATCTTNNNNNNNNNNTATGAATACAAAAGAATTG\n",
      "scaffolds.fasta:TTTACTATCTGNNNNNNNNNNAAGGTTGTAACATTTTATTACCGTGTGGGATATTAATTT\n",
      "scaffolds.fasta:AGATAANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTGGTATAGTACAGG\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MER\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Check output K_MER=${K_MER}\"\n",
    "cd ${FILE_ID}-Assembly${K_MER}\n",
    "rep -c \">\" *fasta\n",
    "grep \">\" -m 8 contigs.fasta \n",
    "grep \">\" -m 8 scaffolds.fasta \n",
    "grep \"NN\" *fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### List of different K_MER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly with no specified M_MER\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [21, 33, 55]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.001     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.395     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.395     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.743     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.743     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.747    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.794    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.794    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.801    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.935    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.944    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.944    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:01.003    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.214    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.387    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.388    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.394    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.621    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.834    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.837    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.837    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.992    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.996    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.996    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.749    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17287 new k-mers.\n",
      "  0:00:06.388    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:07.194    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.649    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.707    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.738    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.738    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.740    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/corrected.yaml\n",
      "  0:00:07.740    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K21/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.175    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.290    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.436   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.460     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.593   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.600     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.723   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:01.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:01.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.030    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:01.030    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.041    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 373295 kmers in total.\n",
      "  0:00:01.041    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.044    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:01.129     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 373295 kmers\n",
      "  0:00:01.129     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 373295 kmers.\n",
      "  0:00:01.132    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 371539 kmers in total.\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.163    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.264    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.269    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 180880 bytes occupied (3.89472 bits per kmer).\n",
      "  0:00:01.269    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.317    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.317    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:01.317    12M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:01.317    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:01.413    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   34023 22-mers were removed by early tip clipper\n",
      "  0:00:01.413    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.413    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.485    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 11711 sequences extracted\n",
      "  0:00:01.520    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.533    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:00:01.542    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.542    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.542    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.577    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 174632 bytes occupied (3.7425 bits per kmer).\n",
      "  0:00:01.578    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.790    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.821    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 23423 edges\n",
      "  0:00:01.823    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.832    12M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 9\n",
      "  0:00:01.832    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 13.4339\n",
      "  0:00:01.832    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 9\n",
      "  0:00:01.833    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 9\n",
      "  0:00:01.833    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.833    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.834    12M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.834    12M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.834    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.836    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.836    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.838    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.840    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 109 times\n",
      "  0:00:01.840    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.857    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 77 times\n",
      "  0:00:01.858    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.906    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2844 times\n",
      "  0:00:01.906    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.907    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.907    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.908    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.909    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.910    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 269 times\n",
      "  0:00:01.916    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.920    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 69 times\n",
      "  0:00:01.920    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.920    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.923    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 29 times\n",
      "  0:00:01.923    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.923    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.924    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.924    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.925    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.925    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.927    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:01.928    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.929    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.929    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
      "  0:00:01.930    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.932    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
      "  0:00:01.932    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.934    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.934    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.934    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.936    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.938    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:01.938    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.947    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.947    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.951    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.953    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.953    12M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.953    12M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.953    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.962    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1587 times\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 22.1978\n",
      "  0:00:01.962    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.964    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/before_rr.fasta\n",
      "  0:00:01.967    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:01.975    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:01.977    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/final_contigs.fasta\n",
      "  0:00:01.980    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.980    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.982    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/before_rr.fasta\n",
      "  0:00:01.984    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:01.992    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:01.994    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/final_contigs.fasta\n",
      "  0:00:01.997    12M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:02.000    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 2 seconds\n",
      "Max read length detected as 76\n",
      "\n",
      "== Running assembler: K33\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K33/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K33/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.256     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.256     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.297    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.297    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.311    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 381522 kmers in total.\n",
      "  0:00:00.312    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.316    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.317    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.317    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.317    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.318    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.318    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:00.392     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 381522 kmers\n",
      "  0:00:00.393     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 381522 kmers.\n",
      "  0:00:00.397    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.434    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 381759 kmers in total.\n",
      "  0:00:00.434    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.441    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.537    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.550    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185632 bytes occupied (3.89004 bits per kmer).\n",
      "  0:00:00.550    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.604    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.604    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:00.604    12M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:00.604    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.703    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   72893 34-mers were removed by early tip clipper\n",
      "  0:00:00.703    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:00.704    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.759    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 6128 sequences extracted\n",
      "  0:00:00.790    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.804    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:00:00.811    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:00.811    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.811    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.846    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 178448 bytes occupied (3.74181 bits per kmer).\n",
      "  0:00:00.847    12M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.023    12M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.054    12M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 12259 edges\n",
      "  0:00:01.057    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.060    12M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 8\n",
      "  0:00:01.060    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 11.3934\n",
      "  0:00:01.060    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 8\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 8\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.061    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.061    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.063    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.063    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.063    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.065    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 138 times\n",
      "  0:00:01.065    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.072    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 22 times\n",
      "  0:00:01.072    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.073    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1131 times\n",
      "  0:00:01.094    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.096    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.096    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 150 times\n",
      "  0:00:01.100    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 50 times\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:01.104    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:01.104    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.105    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.111    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.111    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.112    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.112    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.116    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.116    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.123    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.124    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.131    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.131    12M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.131    12M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.131    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.137    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1638 times\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 17.0928\n",
      "  0:00:01.137    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.139    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/before_rr.fasta\n",
      "  0:00:01.142    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:01.149    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:01.152    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/final_contigs.fasta\n",
      "  0:00:01.154    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.154    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.156    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/before_rr.fasta\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:01.167    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:01.169    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/final_contigs.fasta\n",
      "  0:00:01.172    12M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:01.173    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "== Running assembler: K55\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K55/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K55/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=55\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.144     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.144     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.184    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.184    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.200    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 336055 kmers in total.\n",
      "  0:00:00.200    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.204    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:00.269     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 336055 kmers\n",
      "  0:00:00.269     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 336055 kmers.\n",
      "  0:00:00.275    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.302    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 339904 kmers in total.\n",
      "  0:00:00.302    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.309    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.401    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.409    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 166192 bytes occupied (3.9115 bits per kmer).\n",
      "  0:00:00.410    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.458    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.458    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:00.459    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.525    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 28030 sequences extracted\n",
      "  0:00:00.564    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.576    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.597    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:00.597    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.597    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.628    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 157384 bytes occupied (3.74662 bits per kmer).\n",
      "  0:00:00.629    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.734    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.774    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 56060 edges\n",
      "  0:00:00.776    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:00.808    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 6\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 5.21539\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 3\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 3\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:00.808    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:00.808    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:00.847     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 56060 edges\n",
      "  0:00:00.847     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 56060 sequences.\n",
      "  0:00:00.852    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.881    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 336055 kmers in total.\n",
      "  0:00:00.882    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.890    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.981    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.990    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 164400 bytes occupied (3.91365 bits per kmer).\n",
      "  0:00:00.997    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.035    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.036    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.057    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:01.127    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:01.128    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:01.135    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:01.163    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 47 candidates\n",
      "  0:00:01.171    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.172    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.173    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.179    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.179    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.180    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.180    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.180    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 12451 times\n",
      "  0:00:01.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.282    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.284    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.284    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.286    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.293    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:01.293    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.300    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.301    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.338     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 7778 edges\n",
      "  0:00:01.338     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 7778 sequences.\n",
      "  0:00:01.341    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.370    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 279578 kmers in total.\n",
      "  0:00:01.370    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.376    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.463    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.470    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 138304 bytes occupied (3.95751 bits per kmer).\n",
      "  0:00:01.473    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.504    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.505    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.512    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:01.568    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:01.568    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:01.572    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:01.578    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 1 gaps after checking 145 candidates\n",
      "  0:00:01.589    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:01.724    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:01.729    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:01.738    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:01.742    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:01.745    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:01.746    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.749    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.749    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.755    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.755    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.761    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.761    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.762    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.762    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:01.762    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:01.767    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:01.772    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:01.778    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:01.782    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:01.783    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:01.783    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.784    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.784    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.790    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.790    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.791    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.791    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.792    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.792    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.793    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.793    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.795    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.795    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.795    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.797    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.854    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 3857 times\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 8.08614\n",
      "  0:00:01.854    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:02.002    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.005    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/before_rr.fasta\n",
      "  0:00:02.007    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:02.015    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/simplified_contigs.fasta\n",
      "  0:00:02.017    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/intermediate_contigs.fasta\n",
      "  0:00:02.019    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 30396\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:02.045    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:02.189    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:03.404    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:03.404    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   17950 paired reads (42.9056% of all) aligned to long edges\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 396.253, deviation = 87.446, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:03.413    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.493    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:03.498    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:03.498    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:03.498    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:03.498    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 182\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:03.504    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 8; contradictional = 8\n",
      "  0:00:03.505    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 4; contradictional = 8\n",
      "  0:00:03.505    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:03.505    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 79\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:03.508    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 15 (0%)\n",
      "  0:00:03.511    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2 paths from 15 (13%)\n",
      "  0:00:03.511    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4 paths from 15 (26%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 15 (40%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 8 paths from 15 (53%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 10 paths from 15 (66%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 15 (80%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 14 paths from 15 (93%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:03.513    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.515    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/before_rr.fasta\n",
      "  0:00:03.517    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:03.526    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/final_contigs.paths\n",
      "  0:00:03.530    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/scaffolds.paths\n",
      "  0:00:03.533    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:03.534    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 3 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=25\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t25\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [25]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.000     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.378     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.378     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.733     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.734     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.741    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.790    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.791    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.798    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.925    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.942    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.942    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.988    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.353    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.530    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.530    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.538    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.655    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.775    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.778    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.778    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.932    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.932    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.553    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17290 new k-mers.\n",
      "  0:00:06.156    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.888    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.295    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.349    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.378    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.378    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.380    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/corrected.yaml\n",
      "  0:00:07.380    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K25\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/K25/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/K25/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/dataset.info) with K=25\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.174    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.291    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.435   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.460     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.583   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.589     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.711   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:00.993     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.993     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.996    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.996    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 378642 kmers in total.\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.011    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:01.077     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 378642 kmers\n",
      "  0:00:01.077     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 378642 kmers.\n",
      "  0:00:01.082    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 377541 kmers in total.\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.120    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.217    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.222    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 183648 bytes occupied (3.89146 bits per kmer).\n",
      "  0:00:01.223    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.276    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.276    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.276    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.344    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 32682 sequences extracted\n",
      "  0:00:01.383    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.397    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.420    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.420    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.420    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.457    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177120 bytes occupied (3.74222 bits per kmer).\n",
      "  0:00:01.458    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.664    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.705    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 65363 edges\n",
      "  0:00:01.707    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.744    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 8\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 11.2584\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 7\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 7\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.744    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.744    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 927213\n",
      "  0:00:01.792     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 65363 edges\n",
      "  0:00:01.793     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 65363 sequences.\n",
      "  0:00:01.796    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.826    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 378642 kmers in total.\n",
      "  0:00:01.827    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.832    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.933    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.939    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184088 bytes occupied (3.88944 bits per kmer).\n",
      "  0:00:01.942    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.979    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.979    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.005    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.122    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.122    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.127    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.161    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 2 gaps after checking 112 candidates\n",
      "  0:00:02.163    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.164    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.164    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.164    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.254    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 12086 times\n",
      "  0:00:02.254    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.273    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 59 times\n",
      "  0:00:02.273    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2209 times\n",
      "  0:00:02.318    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.320    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.320    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 276 times\n",
      "  0:00:02.326    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.327    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.327    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 54 times\n",
      "  0:00:02.329    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 49 times\n",
      "  0:00:02.331    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:02.331    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 18 times\n",
      "  0:00:02.332    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.333    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.333    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.334    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.336    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 927213\n",
      "  0:00:02.375     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3384 edges\n",
      "  0:00:02.375     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3384 sequences.\n",
      "  0:00:02.380    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.409    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 281787 kmers in total.\n",
      "  0:00:02.409    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.413    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.507    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.511    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 139344 bytes occupied (3.95601 bits per kmer).\n",
      "  0:00:02.522    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.545    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.621    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.621    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.622    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.625    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 56 candidates\n",
      "  0:00:02.631    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.700    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.702    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.705    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.709    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.710    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.710    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.713    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.713    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.716    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:02.716    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.718    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.718    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.718    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.719    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.721    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.723    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.731    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.732    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.733    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.781    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1607 times\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 20.4915\n",
      "  0:00:02.781    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 2630 kmers to process\n",
      "  0:00:02.782    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.021    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:03.024    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:03.024    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.024    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.026    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/before_rr.fasta\n",
      "  0:00:03.028    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/simplified_contigs.fasta\n",
      "  0:00:03.038    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/intermediate_contigs.fasta\n",
      "  0:00:03.041    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 2630 kmers to process\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23104\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.068    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.244    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.494    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.495    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   14296 paired reads (34.1715% of all) aligned to long edges\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 396.296, deviation = 87.6388, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.505    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.610    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.615    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.615    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.615    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.615    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.620    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.620    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 6870\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.631    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 115; contradictional = 104\n",
      "  0:00:04.632    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 12; contradictional = 12\n",
      "  0:00:04.633    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.633    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 292\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.652    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 43 (0%)\n",
      "  0:00:04.653    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5 paths from 43 (11%)\n",
      "  0:00:04.654    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 10 paths from 43 (23%)\n",
      "  0:00:04.658    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 43 (34%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 20 paths from 43 (46%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 25 paths from 43 (58%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 30 paths from 43 (69%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 35 paths from 43 (81%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 40 paths from 43 (93%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.660    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.662    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.662    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.664    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/before_rr.fasta\n",
      "  0:00:04.666    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:04.675    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/final_contigs.paths\n",
      "  0:00:04.679    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/scaffolds.paths\n",
      "  0:00:04.683    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.685    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 25 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.000     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.359     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.359     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.712    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.769    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.769    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.776    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.906    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.914    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.914    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.971    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.196    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.365    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.365    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.372    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.452    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.526    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.528    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.528    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.675    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.675    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.302    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17286 new k-mers.\n",
      "  0:00:05.924    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.583    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.583    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.584    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.584    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.704    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.106    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.161    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.191    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.192    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.194    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "  0:00:07.194    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.168    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.278    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.422   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.446     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.573   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.581     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.717   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.971     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.971     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.975    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.975    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.988    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:00.989    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.993    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.994    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.995    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.065     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 380324 kmers\n",
      "  0:00:01.066     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 380324 kmers.\n",
      "  0:00:01.070    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380925 kmers in total.\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.109    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.208    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.219    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185224 bytes occupied (3.88998 bits per kmer).\n",
      "  0:00:01.219    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.266    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.267    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.267    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.358    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 31022 sequences extracted\n",
      "  0:00:01.403    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.419    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.442    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.442    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.443    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.479    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177904 bytes occupied (3.74216 bits per kmer).\n",
      "  0:00:01.479    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.660    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.707    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 62044 edges\n",
      "  0:00:01.710    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.740    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 9.00852\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 6\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 6\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.740    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.740    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.784     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 62044 edges\n",
      "  0:00:01.784     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 62044 sequences.\n",
      "  0:00:01.789    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.824    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:01.824    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.831    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.932    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.942    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184936 bytes occupied (3.89007 bits per kmer).\n",
      "  0:00:01.950    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.991    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.991    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.014    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.124    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.125    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.131    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.165    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 107 candidates\n",
      "  0:00:02.173    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.179    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13586 times\n",
      "  0:00:02.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 17 times\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 774 times\n",
      "  0:00:02.299    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.301    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.301    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 110 times\n",
      "  0:00:02.304    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 94 times\n",
      "  0:00:02.308    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 42 times\n",
      "  0:00:02.311    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:02.312    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.314    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.314    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.315    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.316    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.324    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.324    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.360     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3470 edges\n",
      "  0:00:02.360     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3470 sequences.\n",
      "  0:00:02.364    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.392    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 266582 kmers in total.\n",
      "  0:00:02.392    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.398    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.490    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.497    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132168 bytes occupied (3.9663 bits per kmer).\n",
      "  0:00:02.501    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.525    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.525    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.527    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.601    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.601    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.604    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.606    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 96 candidates\n",
      "  0:00:02.663    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.717    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.719    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.721    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.736    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.736    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.737    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.737    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.744    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.744    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.744    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.746    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.747    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.750    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.753    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.753    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.758    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.759    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.760    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.801    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1673 times\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 16.3895\n",
      "  0:00:02.801    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:02.802    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.005    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:03.006    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:03.006    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:03.006    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.007    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.009    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:03.011    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:03.018    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:03.020    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:03.023    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23095\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.049    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.210    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.441    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.442    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   16485 paired reads (39.4039% of all) aligned to long edges\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.556, deviation = 88.9493, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.447    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.454    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.566    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.571    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.571    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.575    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.575    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 3694\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.578    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 43; contradictional = 64\n",
      "  0:00:04.579    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 10; contradictional = 20\n",
      "  0:00:04.579    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.579    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.590    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 169\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.591    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 26 (0%)\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3 paths from 26 (11%)\n",
      "  0:00:04.594    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 26 (23%)\n",
      "  0:00:04.594    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 9 paths from 26 (34%)\n",
      "  0:00:04.595    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 26 (46%)\n",
      "  0:00:04.595    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 26 (57%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 26 (69%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 21 paths from 26 (80%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 26 (92%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.597    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.599    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:04.602    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:04.609    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/final_contigs.paths\n",
      "  0:00:04.613    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/scaffolds.paths\n",
      "  0:00:04.616    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.618    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=45\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t45\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [45]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.003     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.357     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.357     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.712    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.759    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.759    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.766    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.890    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.899    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.900    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.950    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.140    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.324    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.325    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.333    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.401    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.457    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.461    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.462    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.602    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.602    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.604    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.604    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.605    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.605    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.268    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17286 new k-mers.\n",
      "  0:00:05.887    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.521    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.521    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.522    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.522    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.640    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.064    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.119    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.150    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.150    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.152    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/corrected.yaml\n",
      "  0:00:07.152    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K45\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/K45/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/K45/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/dataset.info) with K=45\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.175    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.289    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.435   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.461     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.586   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.593     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.726   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.741     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.941     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.942     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.945    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.945    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.957    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 365508 kmers in total.\n",
      "  0:00:00.958    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.963    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.964    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.964    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.033     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 365508 kmers\n",
      "  0:00:01.034     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 365508 kmers.\n",
      "  0:00:01.039    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.074    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 367735 kmers in total.\n",
      "  0:00:01.075    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.081    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.181    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.192    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 179184 bytes occupied (3.89811 bits per kmer).\n",
      "  0:00:01.192    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.244    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.244    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.244    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.312    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 29619 sequences extracted\n",
      "  0:00:01.356    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.368    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:01.392    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.392    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.392    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.425    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 171064 bytes occupied (3.74414 bits per kmer).\n",
      "  0:00:01.426    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.564    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.611    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 59240 edges\n",
      "  0:00:01.613    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.641    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 7.08441\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 4\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 4\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.641    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.641    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.683     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 59240 edges\n",
      "  0:00:01.683     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 59240 sequences.\n",
      "  0:00:01.691    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.739    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 365508 kmers in total.\n",
      "  0:00:01.739    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.746    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.843    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.852    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 178104 bytes occupied (3.89822 bits per kmer).\n",
      "  0:00:01.863    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.899    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.899    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.921    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.008    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.008    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.015    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.047    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 144 candidates\n",
      "  0:00:02.052    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.053    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.053    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.053    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.058    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.058    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.058    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.160    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13887 times\n",
      "  0:00:02.160    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.162    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:02.162    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:02.163    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 34 times\n",
      "  0:00:02.165    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 32 times\n",
      "  0:00:02.166    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:02.167    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 22 times\n",
      "  0:00:02.168    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:02.170    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:02.170    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.174    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.219     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 4828 edges\n",
      "  0:00:02.219     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 4828 sequences.\n",
      "  0:00:02.224    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.249    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 267502 kmers in total.\n",
      "  0:00:02.250    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.256    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.346    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.353    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132568 bytes occupied (3.96462 bits per kmer).\n",
      "  0:00:02.360    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.379    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.379    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.383    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.449    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.449    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.453    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.459    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 43 gaps after checking 338 candidates\n",
      "  0:00:02.543    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.611    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.614    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.618    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.622    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:02.622    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.626    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.626    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.627    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.627    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.628    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:02.628    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.629    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.629    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.632    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.634    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.638    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.641    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.642    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.642    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.643    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.643    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.648    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.648    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.649    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.649    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.650    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.650    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.655    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.656    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.701    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 2284 times\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 12.2732\n",
      "  0:00:02.701    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 76 kmers to process\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:02.870    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.873    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/before_rr.fasta\n",
      "  0:00:02.875    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:02.882    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/simplified_contigs.fasta\n",
      "  0:00:02.885    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/intermediate_contigs.fasta\n",
      "  0:00:02.888    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 76 kmers to process\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 8231\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:02.890    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:02.916    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.077    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.302    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.302    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   15225 paired reads (36.3921% of all) aligned to long edges\n",
      "  0:00:04.305    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.263, deviation = 89.6726, left quantile = 305, right quantile = 505, read length = 76\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 305, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.312    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.408    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.414    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.414    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.414    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.414    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.417    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.418    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 992\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.421    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 28; contradictional = 8\n",
      "  0:00:04.423    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 3; contradictional = 10\n",
      "  0:00:04.423    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.423    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 361\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.442    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 57 (0%)\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 57 (10%)\n",
      "  0:00:04.447    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 57 (21%)\n",
      "  0:00:04.448    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 57 (31%)\n",
      "  0:00:04.449    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 57 (42%)\n",
      "  0:00:04.450    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 30 paths from 57 (52%)\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 36 paths from 57 (63%)\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 42 paths from 57 (73%)\n",
      "  0:00:04.453    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 48 paths from 57 (84%)\n",
      "  0:00:04.453    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 54 paths from 57 (94%)\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.457    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.457    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.459    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/before_rr.fasta\n",
      "  0:00:04.461    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:04.469    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/final_contigs.paths\n",
      "  0:00:04.473    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/scaffolds.paths\n",
      "  0:00:04.476    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.478    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 45 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly with no specified K_MER\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -o ${FILE_ID}-Assembly${K_MER}\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -k ${K_MER} -o ${FILE_ID}-Assembly${K_MER}\n",
    "done\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ",25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"\",\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "#### Check output K_MER=\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=25\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=35\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=45\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "    echo\n",
    "    echo \"#### Check output K_MER=${K_MER}\"\n",
    "    cd ${FILE_ID}-Assembly${K_MER}\n",
    "    rep -c \">\" *fasta\n",
    "    grep \">\" -m 8 contigs.fasta \n",
    "    grep \">\" -m 8 scaffolds.fasta \n",
    "    grep \"NN\" *fasta\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Comparison of assemblies (*quast*)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compare assemblies FILE_ID=ECTV\n",
      "Processing ECTV-Assembly file...\n",
      "Processing ECTV-Assembly25 file...\n",
      "Processing ECTV-Assembly35 file...\n",
      "Processing ECTV-Assembly45 file...\n",
      "/home/microbioinf/miniconda3/lib/python2.7/site-packages/quast-5.0.2-py2.7.egg-info/scripts/quast.py contigs-ECTV-Assembly25.fasta contigs-ECTV-Assembly35.fasta contigs-ECTV-Assembly45.fasta contigs-ECTV-Assembly.fasta scaffolds-ECTV-Assembly25.fasta scaffolds-ECTV-Assembly35.fasta scaffolds-ECTV-Assembly45.fasta scaffolds-ECTV-Assembly.fasta -R ../ECTV-MoscowGenome.fasta\n",
      "\n",
      "Version: 5.0.2\n",
      "\n",
      "System information:\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-debian-buster-sid (linux_64)\n",
      "  Python version: 2.7.11\n",
      "  CPUs number: 3\n",
      "\n",
      "Started: 2019-04-26 13:12:24\n",
      "\n",
      "Logging to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/quast.log\n",
      "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "Main parameters: \n",
      "  MODE: default, threads: 1, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Reference:\n",
      "  /home/microbioinf/Documentos/Tema_3/ECTV-MoscowGenome.fasta ==> ECTV_MoscowGenome\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  contigs-ECTV-Assembly25.fasta ==> contigs_ECTV_Assembly25\n",
      "  2  contigs-ECTV-Assembly35.fasta ==> contigs_ECTV_Assembly35\n",
      "  3  contigs-ECTV-Assembly45.fasta ==> contigs_ECTV_Assembly45\n",
      "  4  contigs-ECTV-Assembly.fasta ==> contigs_ECTV_Assembly\n",
      "  5  scaffolds-ECTV-Assembly25.fasta ==> scaffolds_ECTV_Assembly25\n",
      "  6  scaffolds-ECTV-Assembly35.fasta ==> scaffolds_ECTV_Assembly35\n",
      "  7  scaffolds-ECTV-Assembly45.fasta ==> scaffolds_ECTV_Assembly45\n",
      "  8  scaffolds-ECTV-Assembly.fasta ==> scaffolds_ECTV_Assembly\n",
      "\n",
      "2019-04-26 13:12:24\n",
      "Running Basic statistics processor...\n",
      "  Reference genome:\n",
      "    ECTV_MoscowGenome.fasta, length = 209771, num fragments = 1, GC % = 33.18\n",
      "  Contig files: \n",
      "    1  contigs_ECTV_Assembly25\n",
      "    2  contigs_ECTV_Assembly35\n",
      "    3  contigs_ECTV_Assembly45\n",
      "    4  contigs_ECTV_Assembly\n",
      "    5  scaffolds_ECTV_Assembly25\n",
      "    6  scaffolds_ECTV_Assembly35\n",
      "    7  scaffolds_ECTV_Assembly45\n",
      "    8  scaffolds_ECTV_Assembly\n",
      "  Calculating N50 and L50...\n",
      "    1  contigs_ECTV_Assembly25, N50 = 37092, L50 = 2, Total length = 195922, GC % = 33.23, # N's per 100 kbp =  0.00\n",
      "    2  contigs_ECTV_Assembly35, N50 = 37406, L50 = 2, Total length = 195088, GC % = 33.21, # N's per 100 kbp =  0.00\n",
      "    3  contigs_ECTV_Assembly45, N50 = 35007, L50 = 3, Total length = 193001, GC % = 33.22, # N's per 100 kbp =  0.00\n",
      "    4  contigs_ECTV_Assembly, N50 = 61624, L50 = 2, Total length = 197062, GC % = 33.20, # N's per 100 kbp =  0.00\n",
      "    5  scaffolds_ECTV_Assembly25, N50 = 43137, L50 = 2, Total length = 196526, GC % = 33.22, # N's per 100 kbp =  157.74\n",
      "    6  scaffolds_ECTV_Assembly35, N50 = 60827, L50 = 2, Total length = 195465, GC % = 33.20, # N's per 100 kbp =  61.39\n",
      "    7  scaffolds_ECTV_Assembly45, N50 = 35007, L50 = 3, Total length = 194789, GC % = 33.23, # N's per 100 kbp =  186.87\n",
      "    8  scaffolds_ECTV_Assembly, N50 = 75358, L50 = 2, Total length = 197822, GC % = 33.21, # N's per 100 kbp =  56.11\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/Nx_plot.pdf\n",
      "  Drawing NGx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/NGx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly25_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly35_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly45_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly25_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly35_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly45_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly25_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly35_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly45_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly25_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly35_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly45_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:29\n",
      "Running Contig analyzer...\n",
      "  1  contigs_ECTV_Assembly25\n",
      "  1  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly25.stdout and contigs_report_contigs_ECTV_Assembly25.stderr...\n",
      "  1  Aligning contigs to the reference\n",
      "  1  Analysis is finished.\n",
      "  2  contigs_ECTV_Assembly35\n",
      "  2  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly35.stdout and contigs_report_contigs_ECTV_Assembly35.stderr...\n",
      "  2  Aligning contigs to the reference\n",
      "  2  Analysis is finished.\n",
      "  3  contigs_ECTV_Assembly45\n",
      "  3  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly45.stdout and contigs_report_contigs_ECTV_Assembly45.stderr...\n",
      "  3  Aligning contigs to the reference\n",
      "  3  Analysis is finished.\n",
      "  4  contigs_ECTV_Assembly\n",
      "  4  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly.stdout and contigs_report_contigs_ECTV_Assembly.stderr...\n",
      "  4  Aligning contigs to the reference\n",
      "  4  Analysis is finished.\n",
      "  5  scaffolds_ECTV_Assembly25\n",
      "  5  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly25.stdout and contigs_report_scaffolds_ECTV_Assembly25.stderr...\n",
      "  5  Aligning contigs to the reference\n",
      "  5  Analysis is finished.\n",
      "  6  scaffolds_ECTV_Assembly35\n",
      "  6  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly35.stdout and contigs_report_scaffolds_ECTV_Assembly35.stderr...\n",
      "  6  Aligning contigs to the reference\n",
      "  6  Analysis is finished.\n",
      "  7  scaffolds_ECTV_Assembly45\n",
      "  7  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly45.stdout and contigs_report_scaffolds_ECTV_Assembly45.stderr...\n",
      "  7  Aligning contigs to the reference\n",
      "  7  Analysis is finished.\n",
      "  8  scaffolds_ECTV_Assembly\n",
      "  8  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly.stdout and contigs_report_scaffolds_ECTV_Assembly.stderr...\n",
      "  8  Aligning contigs to the reference\n",
      "  8  Analysis is finished.\n",
      "  Creating total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_report.txt, misassemblies_report.tsv, and misassemblies_report.tex\n",
      "  Transposed version of total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/transposed_report_misassemblies.txt, transposed_report_misassemblies.tsv, and transposed_report_misassemblies.tex\n",
      "  Creating total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/unaligned_report.txt, unaligned_report.tsv, and unaligned_report.tex\n",
      "  Drawing misassemblies by types plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_plot.pdf\n",
      "  Drawing misassemblies FRCurve plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_frcurve_plot.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:31\n",
      "Running NA-NGA calculation...\n",
      "  1  contigs_ECTV_Assembly25, Largest alignment = 60903, NA50 = 37092, NGA50 = 23116, LA50 = 2, LGA50 = 3\n",
      "  2  contigs_ECTV_Assembly35, Largest alignment = 61098, NA50 = 37373, NGA50 = 29501, LA50 = 2, LGA50 = 3\n",
      "  3  contigs_ECTV_Assembly45, Largest alignment = 47868, NA50 = 29379, NGA50 = 29379, LA50 = 3, LGA50 = 3\n",
      "  4  contigs_ECTV_Assembly, Largest alignment = 92154, NA50 = 61624, NGA50 = 61624, LA50 = 2, LGA50 = 2\n",
      "  5  scaffolds_ECTV_Assembly25, Largest alignment = 60903, NA50 = 42859, NGA50 = 37092, LA50 = 2, LGA50 = 3\n",
      "  6  scaffolds_ECTV_Assembly35, Largest alignment = 61098, NA50 = 60827, NGA50 = 60827, LA50 = 2, LGA50 = 2\n",
      "  7  scaffolds_ECTV_Assembly45, Largest alignment = 59747, NA50 = 29379, NGA50 = 29379, LA50 = 3, LGA50 = 3\n",
      "  8  scaffolds_ECTV_Assembly, Largest alignment = 92542, NA50 = 75267, NGA50 = 75267, LA50 = 2, LGA50 = 2\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/cumulative_plot.pdf\n",
      "  Drawing NAx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/NAx_plot.pdf\n",
      "  Drawing NGAx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/NGAx_plot.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:31\n",
      "Running Genome analyzer...\n",
      "  NOTICE: No file with genomic features were provided. Use the --features option if you want to specify it.\n",
      "\n",
      "  NOTICE: No file with operons were provided. Use the -O option if you want to specify it.\n",
      "  1  contigs_ECTV_Assembly25\n",
      "  1  Analysis is finished.\n",
      "  2  contigs_ECTV_Assembly35\n",
      "  2  Analysis is finished.\n",
      "  3  contigs_ECTV_Assembly45\n",
      "  3  Analysis is finished.\n",
      "  4  contigs_ECTV_Assembly\n",
      "  4  Analysis is finished.\n",
      "  5  scaffolds_ECTV_Assembly25\n",
      "  5  Analysis is finished.\n",
      "  6  scaffolds_ECTV_Assembly35\n",
      "  6  Analysis is finished.\n",
      "  7  scaffolds_ECTV_Assembly45\n",
      "  7  Analysis is finished.\n",
      "  8  scaffolds_ECTV_Assembly\n",
      "  8  Analysis is finished.\n",
      "  Drawing Genome fraction, % histogram...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/genome_stats/genome_fraction_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2019-04-26 13:12:32\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2019-04-26 13:12:37\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.html\n",
      "  PDF version (tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/icarus.html\n",
      "  Log is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/quast.log\n",
      "\n",
      "Finished: 2019-04-26 13:12:37\n",
      "Elapsed time: 0:00:13.291234\n",
      "NOTICEs: 4; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MER\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compare assemblies FILE_ID=${FILE_ID}\"\n",
    "for assembly in ${FILE_ID}-Assembly*; \n",
    "    do echo \"Processing $assembly file...\"; \n",
    "    cp ${assembly}/contigs.fasta contigs-${assembly}.fasta\n",
    "    cp ${assembly}/scaffolds.fasta scaffolds-${assembly}.fasta\n",
    "done\n",
    "quast.py contigs* scaffolds* -R ../ECTV-MoscowGenome.fasta\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style  type=\"text/css\" >\n",
       "    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col0 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col1 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col2 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col3 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col4 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col5 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col6 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col7 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col8 {\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }</style><table id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30\" ><thead>    <tr>        <th class=\"col_heading level0 col0\" >Assembly</th>        <th class=\"col_heading level0 col1\" >contigs_ECTV_Assembly25</th>        <th class=\"col_heading level0 col2\" >contigs_ECTV_Assembly35</th>        <th class=\"col_heading level0 col3\" >contigs_ECTV_Assembly45</th>        <th class=\"col_heading level0 col4\" >contigs_ECTV_Assembly</th>        <th class=\"col_heading level0 col5\" >scaffolds_ECTV_Assembly25</th>        <th class=\"col_heading level0 col6\" >scaffolds_ECTV_Assembly35</th>        <th class=\"col_heading level0 col7\" >scaffolds_ECTV_Assembly45</th>        <th class=\"col_heading level0 col8\" >scaffolds_ECTV_Assembly</th>    </tr></thead><tbody>\n",
       "                <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col0\" class=\"data row0 col0\" ># contigs (>= 0 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col1\" class=\"data row0 col1\" >38</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col2\" class=\"data row0 col2\" >22</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col3\" class=\"data row0 col3\" >34</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col4\" class=\"data row0 col4\" >10</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col5\" class=\"data row0 col5\" >33</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col6\" class=\"data row0 col6\" >19</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col7\" class=\"data row0 col7\" >27</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row0_col8\" class=\"data row0 col8\" >7</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col0\" class=\"data row1 col0\" ># contigs (>= 1000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col1\" class=\"data row1 col1\" >11</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col2\" class=\"data row1 col2\" >8</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col3\" class=\"data row1 col3\" >13</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col4\" class=\"data row1 col4\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col5\" class=\"data row1 col5\" >9</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col6\" class=\"data row1 col6\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col7\" class=\"data row1 col7\" >10</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row1_col8\" class=\"data row1 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col0\" class=\"data row2 col0\" ># contigs (>= 5000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col1\" class=\"data row2 col1\" >8</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col2\" class=\"data row2 col2\" >7</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col3\" class=\"data row2 col3\" >9</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col4\" class=\"data row2 col4\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col5\" class=\"data row2 col5\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col6\" class=\"data row2 col6\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col7\" class=\"data row2 col7\" >8</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row2_col8\" class=\"data row2 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col0\" class=\"data row3 col0\" ># contigs (>= 10000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col1\" class=\"data row3 col1\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col2\" class=\"data row3 col2\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col3\" class=\"data row3 col3\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col4\" class=\"data row3 col4\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col5\" class=\"data row3 col5\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col6\" class=\"data row3 col6\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col7\" class=\"data row3 col7\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row3_col8\" class=\"data row3 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col0\" class=\"data row4 col0\" ># contigs (>= 25000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col1\" class=\"data row4 col1\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col2\" class=\"data row4 col2\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col3\" class=\"data row4 col3\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col4\" class=\"data row4 col4\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col5\" class=\"data row4 col5\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col6\" class=\"data row4 col6\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col7\" class=\"data row4 col7\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row4_col8\" class=\"data row4 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col0\" class=\"data row5 col0\" ># contigs (>= 50000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col1\" class=\"data row5 col1\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col2\" class=\"data row5 col2\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col3\" class=\"data row5 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col4\" class=\"data row5 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col5\" class=\"data row5 col5\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col6\" class=\"data row5 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col7\" class=\"data row5 col7\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row5_col8\" class=\"data row5 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col0\" class=\"data row6 col0\" >Total length (>= 0 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col1\" class=\"data row6 col1\" >198533</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col2\" class=\"data row6 col2\" >196511</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col3\" class=\"data row6 col3\" >196309</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col4\" class=\"data row6 col4\" >198368</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col5\" class=\"data row6 col5\" >198845</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col6\" class=\"data row6 col6\" >196617</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col7\" class=\"data row6 col7\" >196673</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row6_col8\" class=\"data row6 col8\" >198479</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col0\" class=\"data row7 col0\" >Total length (>= 1000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col1\" class=\"data row7 col1\" >193592</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col2\" class=\"data row7 col2\" >194143</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col3\" class=\"data row7 col3\" >190456</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col4\" class=\"data row7 col4\" >197062</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col5\" class=\"data row7 col5\" >194196</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col6\" class=\"data row7 col6\" >194520</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col7\" class=\"data row7 col7\" >192259</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row7_col8\" class=\"data row7 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col0\" class=\"data row8 col0\" >Total length (>= 5000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col1\" class=\"data row8 col1\" >186718</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col2\" class=\"data row8 col2\" >193083</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col3\" class=\"data row8 col3\" >182355</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col4\" class=\"data row8 col4\" >197062</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col5\" class=\"data row8 col5\" >187322</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col6\" class=\"data row8 col6\" >193460</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col7\" class=\"data row8 col7\" >187864</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row8_col8\" class=\"data row8 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col0\" class=\"data row9 col0\" >Total length (>= 10000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col1\" class=\"data row9 col1\" >171220</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col2\" class=\"data row9 col2\" >184029</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col3\" class=\"data row9 col3\" >158872</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col4\" class=\"data row9 col4\" >197062</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col5\" class=\"data row9 col5\" >179237</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col6\" class=\"data row9 col6\" >184406</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col7\" class=\"data row9 col7\" >164381</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row9_col8\" class=\"data row9 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col0\" class=\"data row10 col0\" >Total length (>= 25000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col1\" class=\"data row10 col1\" >97996</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col2\" class=\"data row10 col2\" >134152</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col3\" class=\"data row10 col3\" >118339</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col4\" class=\"data row10 col4\" >183657</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col5\" class=\"data row10 col5\" >141133</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col6\" class=\"data row10 col6\" >169725</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col7\" class=\"data row10 col7\" >131153</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row10_col8\" class=\"data row10 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col0\" class=\"data row11 col0\" >Total length (>= 50000 bp)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col1\" class=\"data row11 col1\" >60904</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col2\" class=\"data row11 col2\" >61627</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col3\" class=\"data row11 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col4\" class=\"data row11 col4\" >154070</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col5\" class=\"data row11 col5\" >60904</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col6\" class=\"data row11 col6\" >122454</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col7\" class=\"data row11 col7\" >59859</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row11_col8\" class=\"data row11 col8\" >168235</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col0\" class=\"data row12 col0\" ># contigs</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col1\" class=\"data row12 col1\" >14</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col2\" class=\"data row12 col2\" >9</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col3\" class=\"data row12 col3\" >16</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col4\" class=\"data row12 col4\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col5\" class=\"data row12 col5\" >12</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col6\" class=\"data row12 col6\" >7</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col7\" class=\"data row12 col7\" >13</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row12_col8\" class=\"data row12 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col0\" class=\"data row13 col0\" >Largest contig</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col1\" class=\"data row13 col1\" >60904</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col2\" class=\"data row13 col2\" >61627</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col3\" class=\"data row13 col3\" >47869</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col4\" class=\"data row13 col4\" >92446</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col5\" class=\"data row13 col5\" >60904</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col6\" class=\"data row13 col6\" >61627</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col7\" class=\"data row13 col7\" >59859</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row13_col8\" class=\"data row13 col8\" >92877</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col0\" class=\"data row14 col0\" >Total length</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col1\" class=\"data row14 col1\" >195922</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col2\" class=\"data row14 col2\" >195088</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col3\" class=\"data row14 col3\" >193001</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col4\" class=\"data row14 col4\" >197062</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col5\" class=\"data row14 col5\" >196526</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col6\" class=\"data row14 col6\" >195465</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col7\" class=\"data row14 col7\" >194789</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row14_col8\" class=\"data row14 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col0\" class=\"data row15 col0\" >Reference length</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col1\" class=\"data row15 col1\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col2\" class=\"data row15 col2\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col3\" class=\"data row15 col3\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col4\" class=\"data row15 col4\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col5\" class=\"data row15 col5\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col6\" class=\"data row15 col6\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col7\" class=\"data row15 col7\" >209771</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row15_col8\" class=\"data row15 col8\" >209771</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col0\" class=\"data row16 col0\" >GC (%)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col1\" class=\"data row16 col1\" >33.23</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col2\" class=\"data row16 col2\" >33.21</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col3\" class=\"data row16 col3\" >33.22</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col4\" class=\"data row16 col4\" >33.20</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col5\" class=\"data row16 col5\" >33.22</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col6\" class=\"data row16 col6\" >33.20</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col7\" class=\"data row16 col7\" >33.23</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row16_col8\" class=\"data row16 col8\" >33.21</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col0\" class=\"data row17 col0\" >Reference GC (%)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col1\" class=\"data row17 col1\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col2\" class=\"data row17 col2\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col3\" class=\"data row17 col3\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col4\" class=\"data row17 col4\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col5\" class=\"data row17 col5\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col6\" class=\"data row17 col6\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col7\" class=\"data row17 col7\" >33.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row17_col8\" class=\"data row17 col8\" >33.18</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col0\" class=\"data row18 col0\" >N50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col1\" class=\"data row18 col1\" >37092</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col2\" class=\"data row18 col2\" >37406</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col3\" class=\"data row18 col3\" >35007</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col4\" class=\"data row18 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col5\" class=\"data row18 col5\" >43137</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col6\" class=\"data row18 col6\" >60827</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col7\" class=\"data row18 col7\" >35007</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row18_col8\" class=\"data row18 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col0\" class=\"data row19 col0\" >NG50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col1\" class=\"data row19 col1\" >23129</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col2\" class=\"data row19 col2\" >35119</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col3\" class=\"data row19 col3\" >35007</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col4\" class=\"data row19 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col5\" class=\"data row19 col5\" >37092</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col6\" class=\"data row19 col6\" >60827</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col7\" class=\"data row19 col7\" >35007</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row19_col8\" class=\"data row19 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col0\" class=\"data row20 col0\" >N75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col1\" class=\"data row20 col1\" >14681</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col2\" class=\"data row20 col2\" >23130</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col3\" class=\"data row20 col3\" >11899</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col4\" class=\"data row20 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col5\" class=\"data row20 col5\" >23423</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col6\" class=\"data row20 col6\" >47271</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col7\" class=\"data row20 col7\" >20230</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row20_col8\" class=\"data row20 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col0\" class=\"data row21 col0\" >NG75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col1\" class=\"data row21 col1\" >14681</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col2\" class=\"data row21 col2\" >14681</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col3\" class=\"data row21 col3\" >11016</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col4\" class=\"data row21 col4\" >29587</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col5\" class=\"data row21 col5\" >23423</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col6\" class=\"data row21 col6\" >47271</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col7\" class=\"data row21 col7\" >12998</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row21_col8\" class=\"data row21 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col0\" class=\"data row22 col0\" >L50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col1\" class=\"data row22 col1\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col2\" class=\"data row22 col2\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col3\" class=\"data row22 col3\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col4\" class=\"data row22 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col5\" class=\"data row22 col5\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col6\" class=\"data row22 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col7\" class=\"data row22 col7\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row22_col8\" class=\"data row22 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col0\" class=\"data row23 col0\" >LG50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col1\" class=\"data row23 col1\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col2\" class=\"data row23 col2\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col3\" class=\"data row23 col3\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col4\" class=\"data row23 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col5\" class=\"data row23 col5\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col6\" class=\"data row23 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col7\" class=\"data row23 col7\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row23_col8\" class=\"data row23 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col0\" class=\"data row24 col0\" >L75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col1\" class=\"data row24 col1\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col2\" class=\"data row24 col2\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col3\" class=\"data row24 col3\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col4\" class=\"data row24 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col5\" class=\"data row24 col5\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col6\" class=\"data row24 col6\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col7\" class=\"data row24 col7\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row24_col8\" class=\"data row24 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col0\" class=\"data row25 col0\" >LG75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col1\" class=\"data row25 col1\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col2\" class=\"data row25 col2\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col3\" class=\"data row25 col3\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col4\" class=\"data row25 col4\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col5\" class=\"data row25 col5\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col6\" class=\"data row25 col6\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col7\" class=\"data row25 col7\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row25_col8\" class=\"data row25 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col0\" class=\"data row26 col0\" ># misassemblies</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col1\" class=\"data row26 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col2\" class=\"data row26 col2\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col3\" class=\"data row26 col3\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col4\" class=\"data row26 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col5\" class=\"data row26 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col6\" class=\"data row26 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col7\" class=\"data row26 col7\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row26_col8\" class=\"data row26 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col0\" class=\"data row27 col0\" ># misassembled contigs</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col1\" class=\"data row27 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col2\" class=\"data row27 col2\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col3\" class=\"data row27 col3\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col4\" class=\"data row27 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col5\" class=\"data row27 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col6\" class=\"data row27 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col7\" class=\"data row27 col7\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row27_col8\" class=\"data row27 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col0\" class=\"data row28 col0\" >Misassembled contigs length</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col1\" class=\"data row28 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col2\" class=\"data row28 col2\" >49800</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col3\" class=\"data row28 col3\" >35007</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col4\" class=\"data row28 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col5\" class=\"data row28 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col6\" class=\"data row28 col6\" >61952</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col7\" class=\"data row28 col7\" >48005</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row28_col8\" class=\"data row28 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col0\" class=\"data row29 col0\" ># local misassemblies</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col1\" class=\"data row29 col1\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col2\" class=\"data row29 col2\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col3\" class=\"data row29 col3\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col4\" class=\"data row29 col4\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col5\" class=\"data row29 col5\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col6\" class=\"data row29 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col7\" class=\"data row29 col7\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row29_col8\" class=\"data row29 col8\" >1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col0\" class=\"data row30 col0\" ># scaffold gap ext. mis.</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col1\" class=\"data row30 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col2\" class=\"data row30 col2\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col3\" class=\"data row30 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col4\" class=\"data row30 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col5\" class=\"data row30 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col6\" class=\"data row30 col6\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col7\" class=\"data row30 col7\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row30_col8\" class=\"data row30 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col0\" class=\"data row31 col0\" ># scaffold gap loc. mis.</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col1\" class=\"data row31 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col2\" class=\"data row31 col2\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col3\" class=\"data row31 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col4\" class=\"data row31 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col5\" class=\"data row31 col5\" >1</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col6\" class=\"data row31 col6\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col7\" class=\"data row31 col7\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row31_col8\" class=\"data row31 col8\" >1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col0\" class=\"data row32 col0\" ># unaligned mis. contigs</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col1\" class=\"data row32 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col2\" class=\"data row32 col2\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col3\" class=\"data row32 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col4\" class=\"data row32 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col5\" class=\"data row32 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col6\" class=\"data row32 col6\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col7\" class=\"data row32 col7\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row32_col8\" class=\"data row32 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col0\" class=\"data row33 col0\" ># unaligned contigs</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col1\" class=\"data row33 col1\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col2\" class=\"data row33 col2\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col3\" class=\"data row33 col3\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col4\" class=\"data row33 col4\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col5\" class=\"data row33 col5\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col6\" class=\"data row33 col6\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col7\" class=\"data row33 col7\" >0 + 0 part</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row33_col8\" class=\"data row33 col8\" >0 + 0 part</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col0\" class=\"data row34 col0\" >Unaligned length</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col1\" class=\"data row34 col1\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col2\" class=\"data row34 col2\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col3\" class=\"data row34 col3\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col4\" class=\"data row34 col4\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col5\" class=\"data row34 col5\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col6\" class=\"data row34 col6\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col7\" class=\"data row34 col7\" >0</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row34_col8\" class=\"data row34 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col0\" class=\"data row35 col0\" >Genome fraction (%)</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col1\" class=\"data row35 col1\" >93.206</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col2\" class=\"data row35 col2\" >92.602</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col3\" class=\"data row35 col3\" >91.811</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col4\" class=\"data row35 col4\" >93.799</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col5\" class=\"data row35 col5\" >93.393</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col6\" class=\"data row35 col6\" >92.789</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col7\" class=\"data row35 col7\" >92.526</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row35_col8\" class=\"data row35 col8\" >94.122</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col0\" class=\"data row36 col0\" >Duplication ratio</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col1\" class=\"data row36 col1\" >1.002</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col2\" class=\"data row36 col2\" >1.004</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col3\" class=\"data row36 col3\" >1.002</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col4\" class=\"data row36 col4\" >1.002</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col5\" class=\"data row36 col5\" >1.003</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col6\" class=\"data row36 col6\" >1.004</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col7\" class=\"data row36 col7\" >1.004</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row36_col8\" class=\"data row36 col8\" >1.002</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col0\" class=\"data row37 col0\" ># N's per 100 kbp</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col1\" class=\"data row37 col1\" >0.00</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col2\" class=\"data row37 col2\" >0.00</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col3\" class=\"data row37 col3\" >0.00</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col4\" class=\"data row37 col4\" >0.00</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col5\" class=\"data row37 col5\" >157.74</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col6\" class=\"data row37 col6\" >61.39</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col7\" class=\"data row37 col7\" >186.87</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row37_col8\" class=\"data row37 col8\" >56.11</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col0\" class=\"data row38 col0\" ># mismatches per 100 kbp</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col1\" class=\"data row38 col1\" >90.53</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col2\" class=\"data row38 col2\" >93.18</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col3\" class=\"data row38 col3\" >90.35</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col4\" class=\"data row38 col4\" >98.60</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col5\" class=\"data row38 col5\" >90.86</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col6\" class=\"data row38 col6\" >95.56</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col7\" class=\"data row38 col7\" >93.25</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row38_col8\" class=\"data row38 col8\" >102.31</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col0\" class=\"data row39 col0\" ># indels per 100 kbp</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col1\" class=\"data row39 col1\" >29.66</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col2\" class=\"data row39 col2\" >32.43</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col3\" class=\"data row39 col3\" >30.63</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col4\" class=\"data row39 col4\" >34.56</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col5\" class=\"data row39 col5\" >30.63</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col6\" class=\"data row39 col6\" >33.91</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col7\" class=\"data row39 col7\" >34.00</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row39_col8\" class=\"data row39 col8\" >37.99</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col0\" class=\"data row40 col0\" >Largest alignment</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col1\" class=\"data row40 col1\" >60903</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col2\" class=\"data row40 col2\" >61098</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col3\" class=\"data row40 col3\" >47868</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col4\" class=\"data row40 col4\" >92154</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col5\" class=\"data row40 col5\" >60903</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col6\" class=\"data row40 col6\" >61098</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col7\" class=\"data row40 col7\" >59747</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row40_col8\" class=\"data row40 col8\" >92542</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col0\" class=\"data row41 col0\" >Total aligned length</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col1\" class=\"data row41 col1\" >195525</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col2\" class=\"data row41 col2\" >194248</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col3\" class=\"data row41 col3\" >192697</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col4\" class=\"data row41 col4\" >196686</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col5\" class=\"data row41 col5\" >195901</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col6\" class=\"data row41 col6\" >194671</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col7\" class=\"data row41 col7\" >194199</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row41_col8\" class=\"data row41 col8\" >197344</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col0\" class=\"data row42 col0\" >NA50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col1\" class=\"data row42 col1\" >37092</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col2\" class=\"data row42 col2\" >37373</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col3\" class=\"data row42 col3\" >29379</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col4\" class=\"data row42 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col5\" class=\"data row42 col5\" >42859</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col6\" class=\"data row42 col6\" >60827</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col7\" class=\"data row42 col7\" >29379</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row42_col8\" class=\"data row42 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col0\" class=\"data row43 col0\" >NGA50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col1\" class=\"data row43 col1\" >23116</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col2\" class=\"data row43 col2\" >29501</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col3\" class=\"data row43 col3\" >29379</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col4\" class=\"data row43 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col5\" class=\"data row43 col5\" >37092</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col6\" class=\"data row43 col6\" >60827</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col7\" class=\"data row43 col7\" >29379</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row43_col8\" class=\"data row43 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col0\" class=\"data row44 col0\" >NA75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col1\" class=\"data row44 col1\" >14348</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col2\" class=\"data row44 col2\" >23117</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col3\" class=\"data row44 col3\" >11016</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col4\" class=\"data row44 col4\" >61624</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col5\" class=\"data row44 col5\" >23410</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col6\" class=\"data row44 col6\" >41653</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col7\" class=\"data row44 col7\" >12657</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row44_col8\" class=\"data row44 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col0\" class=\"data row45 col0\" >NGA75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col1\" class=\"data row45 col1\" >14348</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col2\" class=\"data row45 col2\" >12424</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col3\" class=\"data row45 col3\" >9074</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col4\" class=\"data row45 col4\" >29535</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col5\" class=\"data row45 col5\" >23410</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col6\" class=\"data row45 col6\" >41653</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col7\" class=\"data row45 col7\" >12657</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row45_col8\" class=\"data row45 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col0\" class=\"data row46 col0\" >LA50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col1\" class=\"data row46 col1\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col2\" class=\"data row46 col2\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col3\" class=\"data row46 col3\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col4\" class=\"data row46 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col5\" class=\"data row46 col5\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col6\" class=\"data row46 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col7\" class=\"data row46 col7\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row46_col8\" class=\"data row46 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col0\" class=\"data row47 col0\" >LGA50</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col1\" class=\"data row47 col1\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col2\" class=\"data row47 col2\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col3\" class=\"data row47 col3\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col4\" class=\"data row47 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col5\" class=\"data row47 col5\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col6\" class=\"data row47 col6\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col7\" class=\"data row47 col7\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row47_col8\" class=\"data row47 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col0\" class=\"data row48 col0\" >LA75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col1\" class=\"data row48 col1\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col2\" class=\"data row48 col2\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col3\" class=\"data row48 col3\" >6</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col4\" class=\"data row48 col4\" >2</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col5\" class=\"data row48 col5\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col6\" class=\"data row48 col6\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col7\" class=\"data row48 col7\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row48_col8\" class=\"data row48 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col0\" class=\"data row49 col0\" >LGA75</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col1\" class=\"data row49 col1\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col2\" class=\"data row49 col2\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col3\" class=\"data row49 col3\" >7</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col4\" class=\"data row49 col4\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col5\" class=\"data row49 col5\" >4</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col6\" class=\"data row49 col6\" >3</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col7\" class=\"data row49 col7\" >5</td>\n",
       "                        <td id=\"T_37eb6ede_6814_11e9_88cb_b8e8562afb30row49_col8\" class=\"data row49 col8\" >2</td>\n",
       "            </tr>\n",
       "    </tbody></table>"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x110a25c88>"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/quast*/latest/report.tsv\n",
    "EOT\n",
    "\"\"\"\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_quast = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_quast.style.hide_index().set_properties(**{'text-align': 'left', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"10pt\"}).\\\n",
    "set_properties(**{'text-align': 'left', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"10pt\"}, \\\n",
    "               subset=['Assembly'])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
