{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Whole Genome Shotgun metagenomics: de novo Assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have two fastqc files to process:\n",
    "\n",
    "*Vir1_100k* that contains $100,000$ paired end reads from the same saliva sample but after purification of viral particles. So this is a virome.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import pandas as pd\n",
    "FILE_ID = \"ECTV\"\n",
    "FASTQ_STR = \"@HWUSI-EAS1752R\"\n",
    "MIN_LEN = \"70\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preprocessing and quality check\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=ECTV, FASTQ_STR=@HWUSI-EAS1752R\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "NGGATCTCCGATTTCTTTACGATATGGATCTATACCGGACGAATTAATAAACAAACATCCAAAAAAATATGGAATT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "#*(.(,+,,+@@@@@@222@@@@@@@@:@@@@@@@:::<<71757<<:<:22@222:@@@8518500000:::8:5\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_1:N:0:GCCAAT\n",
      "NGAGTAATACGGTTCAAAATCATAAATGTGATAGTTTCCAGACTGGTATCCGAGTTTTTCTTGGATGATGGATACT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_1:N:0:GCCAAT\n",
      "#,*))33322@C@@@@C@@C@C@C@@@@@@@@CC@@@C@C@@@@@C@C@@@@@@@C@@CC222@@@@@@C@@@@@@\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010_1:N:0:GCCAAT\n",
      "NTGGAGTCGTAAAAAAGTTTTATCTCTTTCTCTCTTTGATGGTCTCATAAAAAAGTTTTACAAAAATATTTTTATT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "NGGATCTCCGATTTCTTTACGATATGGATCTATACCGGACGAATTAATAAACAAACATCCAAAAAAATATGGAATT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "#*(.(,+,,+@@@@@@222@@@@@@@@:@@@@@@@:::<<71757<<:<:22@222:@@@8518500000:::8:5\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 1:N:0:GCCAAT\n",
      "NGAGTAATACGGTTCAAAATCATAAATGTGATAGTTTCCAGACTGGTATCCGAGTTTTTCTTGGATGATGGATACT\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 1:N:0:GCCAAT\n",
      "#,*))33322@C@@@@C@@C@C@C@@@@@@@@CC@@@C@C@@@@@C@C@@@@@@@C@@CC222@@@@@@C@@@@@@\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010 1:N:0:GCCAAT\n",
      "NTGGAGTCGTAAAAAAGTTTTATCTCTTTCTCTCTTTGATGGTCTCATAAAAAAGTTTTACAAAAATATTTTTATT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "ATCCTCTAACGTGGCAGTCATTAAAAACATAGAATCTATTTTCGTATGATGCTTTCTCGCTACTGCTATAATGATA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "HHIHII@GGIIDIGDIGDGGIDIGGGIIHGG?GGDIGEGIEGDEGHIHIIGGIIIIHHH>@DDGGIEHHFFIGBDD\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_2:N:0:GCCAAT\n",
      "CGATGATGTAAGTCTTACAATGTTCGAACCAGTACCAGATCTTAATTTGGAGGCCTCCGTAGAACTAGGGGAGGGA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011_2:N:0:GCCAAT\n",
      "IBIIIFIHIGHGIGDIIIB3EGGDEEGGGGBIFGBDGD@CDDDDAHIGHEEHHEGDGDDEEEDBD@9AAB?#####\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010_2:N:0:GCCAAT\n",
      "CAGAAAGAGAATAAAAATATTTTTGTAAAACTTTTTTATGAGACCATCAAAGAGAGAAAGAGATAAAACTTTTTTA\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "ATCCTCTAACGTGGCAGTCATTAAAAACATAGAATCTATTTTCGTATGATGCTTTCTCGCTACTGCTATAATGATA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "HHIHII@GGIIDIGDIGDGGIDIGGGIIHGG?GGDIGEGIEGDEGHIHIIGGIIIIHHH>@DDGGIEHHFFIGBDD\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 2:N:0:GCCAAT\n",
      "CGATGATGTAAGTCTTACAATGTTCGAACCAGTACCAGATCTTAATTTGGAGGCCTCCGTAGAACTAGGGGAGGGA\n",
      "+HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3893:1011 2:N:0:GCCAAT\n",
      "IBIIIFIHIGHGIGDIIIB3EGGDEEGGGGBIFGBDGD@CDDDDAHIGHEEHHEGDGDDEEEDBD@9AAB?#####\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:5526:1010 2:N:0:GCCAAT\n",
      "CAGAAAGAGAATAAAAATATTTTTGTAAAACTTTTTTATGAGACCATCAAAGAGAGAAAGAGATAAAACTTTTTTA\n",
      "ECTV_R1_.fastq:50000\n",
      "ECTV_R1.fastq:50000\n",
      "ECTV_R2_.fastq:50000\n",
      "ECTV_R2.fastq:50000\n",
      "#### Compute quality\n",
      "Analysis complete for ECTV_R1.fastq\n",
      "Analysis complete for ECTV_R2.fastq\n",
      "#### Replace ' ' by '_' in header\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n",
      "==> ECTV_R1_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R1.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 1:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2_.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008_2:N:0:GCCAAT\n",
      "\n",
      "==> ECTV_R2.fastq <==\n",
      "@HWUSI-EAS1752R:23:FC62KHPAAXX:6:3:3542:1008 2:N:0:GCCAAT\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "cd Documentos/Tema_3\n",
    "head ${FILE_ID}*fastq\n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq\n",
    "\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_Quality\n",
    "fastqc ${FILE_ID}_R1.fastq -o ${FILE_ID}_Quality/\n",
    "fastqc ${FILE_ID}_R2.fastq -o ${FILE_ID}_Quality/\n",
    "\n",
    "echo \"#### Replace ' ' by '_' in header\"\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "cat ${FILE_ID}_R1.fastq | sed 's/ /_/g' > ${FILE_ID}_R1_.fastq\n",
    "cat ${FILE_ID}_R2.fastq | sed 's/ /_/g' > ${FILE_ID}_R2_.fastq\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming and decontaminating\n",
    "Trimming poor quality ends and short sequences (**Trimmomatic**) and removal of reads aligning to the human and phiX174 genomes (***bowtie2**). The later one is a contaminant used as spike by Illumina kits to control quality of the sequencing process.\n",
    "\n",
    "We are only filtering only R1 files because forward reads have usually better quality than reverse reads. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Trimming and decontaminating FILE_ID=ECTV MIN_LEN=70\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq ): 50000\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq ): 50000\n",
      "Running Trimmomatic ... \n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq ): 41996\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq ): 41996\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq ): 4107\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq ): 1791\n",
      "Decontaminating ...\n",
      "Running bowtie2 ... \n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 41839\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 41839\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq ): 41839\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq ): 41839\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 12\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq ): 12\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 5851\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq ): 5851\n",
      "\n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "echo \"#### Trimming and decontaminating FILE_ID=${FILE_ID} MIN_LEN=${MIN_LEN}\"\n",
    "kneaddata -i ${FILE_ID}_R1_.fastq -i ${FILE_ID}_R2_.fastq \\\n",
    "-o kneaddata_out -db /home/shared/bowtiedb/GRCh38_PhiX \\\n",
    "--trimmomatic /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/ \\\n",
    "-t 2 --trimmomatic-options \"SLIDINGWINDOW:4:20 MINLEN:${MIN_LEN}\" \\\n",
    "--bowtie2-options \"--very-sensitive --dovetail\" --remove-intermediate-output\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - INFO: Running kneaddata v0.6.1\n",
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - INFO: Output files will be written to: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "04/26/2019 08:44:27 AM - kneaddata.knead_data - DEBUG: Running with the following arguments: \n",
      "verbose = False\n",
      "bmtagger_path = None\n",
      "minscore = 50\n",
      "bowtie2_path = /home/microbioinf/miniconda3/bin/bowtie2\n",
      "maxperiod = 500\n",
      "no_discordant = False\n",
      "serial = False\n",
      "fastqc_start = False\n",
      "bmtagger = False\n",
      "cat_final_output = False\n",
      "log_level = DEBUG\n",
      "log = /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.log\n",
      "max_memory = 500m\n",
      "remove_intermediate_output = True\n",
      "fastqc_path = None\n",
      "output_dir = /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "trf_path = None\n",
      "remove_temp_output = True\n",
      "reference_db = /home/shared/bowtiedb/GRCh38_PhiX\n",
      "input = /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq\n",
      "pi = 10\n",
      "reorder = False\n",
      "pm = 80\n",
      "trimmomatic_path = /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar\n",
      "store_temp_output = False\n",
      "cat_pairs = False\n",
      "mismatch = 7\n",
      "threads = 2\n",
      "delta = 7\n",
      "bowtie2_options = --very-sensitive --dovetail --phred33\n",
      "bypass_trim = False\n",
      "processes = 1\n",
      "trimmomatic_quality_scores = -phred33\n",
      "fastqc_end = False\n",
      "trf = False\n",
      "trimmomatic_options = SLIDINGWINDOW:4:20 MINLEN:70\n",
      "output_prefix = ECTV_R1__kneaddata\n",
      "match = 2\n",
      "\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: raw pair1 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq ): 50000\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: raw pair2 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq ): 50000\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Running Trimmomatic ... \n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Execute command: java -Xmx500m -d64 -jar /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar PE -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:70\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: TrimmomaticPE: Started with arguments:\n",
      " -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/ECTV_R1_.fastq /home/microbioinf/Documentos/Tema_3/ECTV_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:70\n",
      "Input Read Pairs: 50000 Both Surviving: 41996 (83,99%) Forward Only Surviving: 4107 (8,21%) Reverse Only Surviving: 1791 (3,58%) Dropped: 2106 (4,21%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed pair1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq ): 41996\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed pair2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq ): 41996\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq ): 4107\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq ): 1791\n",
      "04/26/2019 08:44:27 AM - kneaddata.run - INFO: Decontaminating ...\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Running bowtie2 ... \n",
      "04/26/2019 08:44:27 AM - kneaddata.utilities - INFO: Execute command: kneaddata_bowtie2_discordant_pairs --bowtie2 /home/microbioinf/miniconda3/bin/bowtie2 --threads 2 -x /home/shared/bowtiedb/GRCh38_PhiX --bowtie2-options \"--very-sensitive --dovetail --phred33\" -1 /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.1.fastq -2 /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.2.fastq --un-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_%.fastq --al-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_%.fastq -U /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.1.fastq,/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata.trimmed.single.2.fastq --un-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_clean.fastq --al-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_contam.fastq -S /dev/null\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: 47894 reads; of these:\n",
      "  41996 (87.69%) were paired; of these:\n",
      "    41881 (99.73%) aligned concordantly 0 times\n",
      "    106 (0.25%) aligned concordantly exactly 1 time\n",
      "    9 (0.02%) aligned concordantly >1 times\n",
      "    ----\n",
      "    41881 pairs aligned concordantly 0 times; of these:\n",
      "      10 (0.02%) aligned discordantly 1 time\n",
      "    ----\n",
      "    41871 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      83742 mates make up the pairs; of these:\n",
      "        83709 (99.96%) aligned 0 times\n",
      "        24 (0.03%) aligned exactly 1 time\n",
      "        9 (0.01%) aligned >1 times\n",
      "  5898 (12.31%) were unpaired; of these:\n",
      "    5832 (98.88%) aligned 0 times\n",
      "    59 (1.00%) aligned exactly 1 time\n",
      "    7 (0.12%) aligned >1 times\n",
      "0.39% overall alignment rate\n",
      "pair1_aligned : 115\n",
      "pair2_aligned : 115\n",
      "orphan1_unaligned : 12\n",
      "orphan2_unaligned : 5851\n",
      "orphan2_aligned : 89\n",
      "pair2_unaligned : 41839\n",
      "pair1_unaligned : 41839\n",
      "orphan1_aligned : 30\n",
      "\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq ) : 115\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq ) : 115\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq ) : 30\n",
      "04/26/2019 08:44:38 AM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq ) : 89\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final pair1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final pair2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq ): 41839\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 12\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final orphan1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq ): 12\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 5851\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - INFO: READ COUNT: final orphan2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq ): 5851\n",
      "04/26/2019 08:44:38 AM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq\n",
      "04/26/2019 08:44:38 AM - kneaddata.knead_data - INFO: \n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_unmatched_2.fastq\n",
      "\n",
      "Reading log: ./ECTV_R1__kneaddata.log\n",
      "Read count table written: kneaddata_read_countsECTV.txt\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:30\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:89\n",
      "ECTV_R1__kneaddata_paired_1.fastq:41839\n",
      "ECTV_R1__kneaddata_paired_2.fastq:41839\n",
      "ECTV_R1__kneaddata_unmatched_1.fastq:12\n",
      "ECTV_R1__kneaddata_unmatched_2.fastq:5851\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "l\n",
    "less *log\n",
    "kneaddata_read_count_table --input ./ --output kneaddata_read_counts${FILE_ID}.txt \n",
    "grep -c $FASTQ_STR *fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <td>ECTV_R1__kneaddata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair1</th>\n",
       "      <td>50000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair2</th>\n",
       "      <td>50000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair1</th>\n",
       "      <td>41996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair2</th>\n",
       "      <td>41996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan1</th>\n",
       "      <td>4107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan2</th>\n",
       "      <td>1791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair1</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair2</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan1</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan2</th>\n",
       "      <td>5851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair1</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair2</th>\n",
       "      <td>41839</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan1</th>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan2</th>\n",
       "      <td>5851</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     0\n",
       "Sample                              ECTV_R1__kneaddata\n",
       "raw pair1                                        50000\n",
       "raw pair2                                        50000\n",
       "trimmed pair1                                    41996\n",
       "trimmed pair2                                    41996\n",
       "trimmed orphan1                                   4107\n",
       "trimmed orphan2                                   1791\n",
       "decontaminated GRCh38_PhiX pair1                 41839\n",
       "decontaminated GRCh38_PhiX pair2                 41839\n",
       "decontaminated GRCh38_PhiX orphan1                  12\n",
       "decontaminated GRCh38_PhiX orphan2                5851\n",
       "final pair1                                      41839\n",
       "final pair2                                      41839\n",
       "final orphan1                                       12\n",
       "final orphan2                                     5851"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/kneaddata_read_counts%s.txt\n",
    "EOT\n",
    "\"\"\" % FILE_ID\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_knead = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_knead.style.set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"11pt\"}).\\\n",
    "set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"12pt\"}, subset=['Sample'])\n",
    "\n",
    "df_knead.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check number of reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With grep we can identify the non-contaminated high-quality files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:115\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:30\n",
      "ECTV_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:89\n",
      "ECTV_R1__kneaddata_paired_1.fastq:41839\n",
      "ECTV_R1__kneaddata_paired_2.fastq:41839\n",
      "ECTV_R1__kneaddata_unmatched_1.fastq:12\n",
      "ECTV_R1__kneaddata_unmatched_2.fastq:5851\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "grep -c $FASTQ_STR *fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute quality\n",
      "Analysis complete for ECTV_R1__kneaddata_paired_1.fastq\n",
      "Analysis complete for ECTV_R1__kneaddata_paired_2.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_HighQuality\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_1.fastq -o ${FILE_ID}_HighQuality/\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_2.fastq -o ${FILE_ID}_HighQuality/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assembly\n",
    "\n",
    " We are going to use a Refseq database of viral proteins (around 100Mb) from ncbi (ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/), and you have to download it in two separated files that can be joined into one with cat.\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Set up the reference database for diamond\n",
    "\n",
    "This will create a binary DIAMOND database file with the name: *viralprotein.dmnd*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "K_MER = 35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.001     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.385     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.385     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.750     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.751     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.756    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.809    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.811    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.818    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.945    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.954    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.954    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:01.011    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.230    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.416    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.417    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.423    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.496    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.611    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.614    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.614    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.759    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.761    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.761    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.366    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17288 new k-mers.\n",
      "  0:00:05.978    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.601    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.601    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.601    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.601    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.720    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.131    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.184    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.214    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.214    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.215    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "  0:00:07.215    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.175    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.288    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.439   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.464     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.592   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.598     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.726   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.734     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.734     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.735     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.735     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.736     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.736     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.736     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.982     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.982     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.986    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.986    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.002    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:01.002    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.008    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.009    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.009    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.009    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.082     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 380324 kmers\n",
      "  0:00:01.082     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 380324 kmers.\n",
      "  0:00:01.086    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.120    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380925 kmers in total.\n",
      "  0:00:01.120    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.127    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.222    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.230    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185224 bytes occupied (3.88998 bits per kmer).\n",
      "  0:00:01.230    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.278    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.278    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.278    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.350    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 31022 sequences extracted\n",
      "  0:00:01.392    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.406    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.428    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.428    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.428    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.461    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177904 bytes occupied (3.74216 bits per kmer).\n",
      "  0:00:01.462    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.633    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.681    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 62044 edges\n",
      "  0:00:01.684    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.722    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 9.00852\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 6\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 6\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.722    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.722    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.722    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.771     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 62044 edges\n",
      "  0:00:01.771     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 62044 sequences.\n",
      "  0:00:01.776    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.809    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:01.809    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.816    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.909    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.917    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184936 bytes occupied (3.89007 bits per kmer).\n",
      "  0:00:01.921    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.964    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.964    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.985    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.096    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.097    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.102    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.137    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 107 candidates\n",
      "  0:00:02.141    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.142    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.142    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.142    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.147    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.147    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.148    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.148    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.148    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.148    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.252    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13586 times\n",
      "  0:00:02.252    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.259    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 17 times\n",
      "  0:00:02.260    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.262    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.262    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.264    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.277    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 774 times\n",
      "  0:00:02.278    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.279    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.280    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 110 times\n",
      "  0:00:02.283    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.284    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.284    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.284    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 94 times\n",
      "  0:00:02.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 42 times\n",
      "  0:00:02.292    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:02.294    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.294    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.296    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.297    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.298    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.298    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.298    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.298    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.300    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.306    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.306    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.307    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.307    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.307    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.307    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.346     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3470 edges\n",
      "  0:00:02.347     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3470 sequences.\n",
      "  0:00:02.351    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.381    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 266582 kmers in total.\n",
      "  0:00:02.381    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.387    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.475    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.481    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132168 bytes occupied (3.9663 bits per kmer).\n",
      "  0:00:02.491    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.510    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.510    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.513    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.583    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.584    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.586    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.588    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 96 candidates\n",
      "  0:00:02.633    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.633    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.684    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.687    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.690    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.695    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.696    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.696    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.697    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.697    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.699    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.699    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.700    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.700    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.700    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.700    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.701    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.701    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.702    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.702    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.702    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.703    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.705    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.709    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.710    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.720    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.720    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.728    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.728    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.731    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.772    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1673 times\n",
      "  0:00:02.772    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.772    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 16.3895\n",
      "  0:00:02.772    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.772    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:02.773    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.993    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:02.995    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:02.995    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:02.995    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:02.995    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.998    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:03.000    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:03.007    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:03.009    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:03.012    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.012    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:03.012    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.012    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23095\n",
      "  0:00:03.012    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.012    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.037    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.201    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.415    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.415    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   16485 paired reads (39.4039% of all) aligned to long edges\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.556, deviation = 88.9493, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.426    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.525    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.529    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.530    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.530    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.530    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.532    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.532    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 3694\n",
      "  0:00:04.533    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.533    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.533    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.533    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.534    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 43; contradictional = 64\n",
      "  0:00:04.535    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 10; contradictional = 20\n",
      "  0:00:04.535    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.535    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 169\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.547    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.547    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 26 (0%)\n",
      "  0:00:04.549    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3 paths from 26 (11%)\n",
      "  0:00:04.550    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 26 (23%)\n",
      "  0:00:04.551    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 9 paths from 26 (34%)\n",
      "  0:00:04.551    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 26 (46%)\n",
      "  0:00:04.552    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 26 (57%)\n",
      "  0:00:04.553    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 26 (69%)\n",
      "  0:00:04.553    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 21 paths from 26 (80%)\n",
      "  0:00:04.553    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 26 (92%)\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.554    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.554    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.554    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.559    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:04.562    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:04.570    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/final_contigs.paths\n",
      "  0:00:04.573    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/scaffolds.paths\n",
      "  0:00:04.577    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.579    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Check output K_MER=35\n",
      ">NODE_1_length_61627_cov_16.136024\n",
      ">NODE_2_length_37406_cov_14.491584\n",
      ">NODE_3_length_35119_cov_16.619741\n",
      ">NODE_4_length_23130_cov_14.144966\n",
      ">NODE_5_length_14681_cov_16.895125\n",
      ">NODE_6_length_12066_cov_15.782063\n",
      ">NODE_7_length_9054_cov_26.310456\n",
      ">NODE_8_length_1060_cov_59.155122\n",
      ">NODE_1_length_61627_cov_16.136024\n",
      ">NODE_2_length_60827_cov_14.296881\n",
      ">NODE_3_length_47271_cov_16.363811\n",
      ">NODE_4_length_14681_cov_16.895125\n",
      ">NODE_5_length_9054_cov_26.310456\n",
      ">NODE_6_length_1060_cov_59.155122\n",
      ">NODE_7_length_945_cov_8.277551\n",
      ">NODE_8_length_294_cov_3.567568\n",
      "scaffolds.fasta:ACCTCAAAAAATTTGTTAATAATGGTATCTTNNNNNNNNNNTATGAATACAAAAGAATTG\n",
      "scaffolds.fasta:TTTACTATCTGNNNNNNNNNNAAGGTTGTAACATTTTATTACCGTGTGGGATATTAATTT\n",
      "scaffolds.fasta:AGATAANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTGGTATAGTACAGG\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MER\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -k ${K_MER} -o ${FILE_ID}-Assembly${K_MER}\n",
    "\n",
    "echo \"#### Check output K_MER=${K_MER}\"\n",
    "cd ${FILE_ID}-Assembly${K_MER}\n",
    "rep -c \">\" *fasta\n",
    "grep \">\" -m 8 contigs.fasta \n",
    "grep \">\" -m 8 scaffolds.fasta \n",
    "grep \"NN\" *fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Blastx alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "diamond v0.8.36.98 | by Benjamin Buchfink <buchfink@gmail.com>\n",
      "Check http://github.com/bbuchfink/diamond for updates.\n",
      "\n",
      "#CPU threads: 3\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "#Target sequences to report alignments for: 25\n",
      "Temporary directory: \n",
      "Opening the database...  [1.5e-05s]\n",
      "Opening the input file...  [2.8e-05s]\n",
      "Opening the output file...  [4.1e-05s]\n",
      "Loading query sequences...  [0.235952s]\n",
      "Running complexity filter...  [3.78898s]\n",
      "Building query histograms...  [0.318931s]\n",
      "Allocating buffers...  [6.2e-05s]\n",
      "Loading reference sequences...  [0.107758s]\n",
      "Building reference histograms...  [1.18488s]\n",
      "Allocating buffers...  [6.1e-05s]\n",
      "Initializing temporary storage...  [0.000619s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 0.\n",
      "Building reference index...  [0.969958s]\n",
      "Building query index...  [0.217957s]\n",
      "Building seed filter...  [0.091635s]\n",
      "Searching alignments...  [0.137262s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 1.\n",
      "Building reference index...  [1.11468s]\n",
      "Building query index...  [0.246178s]\n",
      "Building seed filter...  [0.090143s]\n",
      "Searching alignments...  [0.138724s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 2.\n",
      "Building reference index...  [1.127s]\n",
      "Building query index...  [0.249563s]\n",
      "Building seed filter...  [0.109712s]\n",
      "Searching alignments...  [0.183329s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 3.\n",
      "Building reference index...  [1.26957s]\n",
      "Building query index...  [0.370472s]\n",
      "Building seed filter...  [0.150778s]\n",
      "Searching alignments...  [0.22253s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 0.\n",
      "Building reference index...  [1.34065s]\n",
      "Building query index...  [0.265761s]\n",
      "Building seed filter...  [0.113665s]\n",
      "Searching alignments...  [0.429611s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 1.\n",
      "Building reference index...  [1.4586s]\n",
      "Building query index...  [0.295433s]\n",
      "Building seed filter...  [0.120444s]\n",
      "Searching alignments...  [0.16581s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 2.\n",
      "Building reference index...  [1.45692s]\n",
      "Building query index...  [0.302227s]\n",
      "Building seed filter...  [0.110777s]\n",
      "Searching alignments...  [0.159347s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 3.\n",
      "Building reference index...  [1.3176s]\n",
      "Building query index...  [0.278914s]\n",
      "Building seed filter...  [0.115424s]\n",
      "Searching alignments...  [0.165368s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 0.\n",
      "Building reference index...  [1.33535s]\n",
      "Building query index...  [0.246887s]\n",
      "Building seed filter...  [0.117666s]\n",
      "Searching alignments...  [0.147365s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 1.\n",
      "Building reference index...  [1.31552s]\n",
      "Building query index...  [0.204659s]\n",
      "Building seed filter...  [0.087173s]\n",
      "Searching alignments...  [0.110421s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 2.\n",
      "Building reference index...  [1.10052s]\n",
      "Building query index...  [0.21389s]\n",
      "Building seed filter...  [0.086331s]\n",
      "Searching alignments...  [0.109901s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 3.\n",
      "Building reference index...  [0.938978s]\n",
      "Building query index...  [0.179356s]\n",
      "Building seed filter...  [0.085301s]\n",
      "Searching alignments...  [0.109947s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 0.\n",
      "Building reference index...  [0.920597s]\n",
      "Building query index...  [0.171162s]\n",
      "Building seed filter...  [0.085914s]\n",
      "Searching alignments...  [0.101702s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 1.\n",
      "Building reference index...  [1.04422s]\n",
      "Building query index...  [0.199129s]\n",
      "Building seed filter...  [0.084341s]\n",
      "Searching alignments...  [0.106007s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 2.\n",
      "Building reference index...  [1.0729s]\n",
      "Building query index...  [0.204229s]\n",
      "Building seed filter...  [0.085288s]\n",
      "Searching alignments...  [0.127639s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 3.\n",
      "Building reference index...  [0.903671s]\n",
      "Building query index...  [0.167927s]\n",
      "Building seed filter...  [0.089396s]\n",
      "Searching alignments...  [0.105627s]\n",
      "Deallocating buffers...  [0.009547s]\n",
      "Computing alignments...  [1.84423s]\n",
      "Deallocating reference...  [0.00306s]\n",
      "Loading reference sequences...  [2.2e-05s]\n",
      "Deallocating buffers...  [0.002337s]\n",
      "Deallocating queries...  [0.002664s]\n",
      "Loading query sequences...  [2.2e-05s]\n",
      "Closing the input file...  [7e-06s]\n",
      "Closing the output file...  [0.005769s]\n",
      "Closing the database file...  [1.7e-05s]\n",
      "Total time = 34.1638s\n",
      "Reported 75459 pairwise alignments, 75459 HSSPs.\n",
      "12725 queries aligned.\n",
      "diamond v0.8.36.98 | by Benjamin Buchfink <buchfink@gmail.com>\n",
      "Check http://github.com/bbuchfink/diamond for updates.\n",
      "\n",
      "#CPU threads: 3\n",
      "Scoring parameters: (Matrix=BLOSUM62 Lambda=0.267 K=0.041 Penalties=11/1)\n",
      "#Target sequences to report alignments for: 25\n",
      "Temporary directory: \n",
      "Opening the database...  [1.5e-05s]\n",
      "Opening the input file...  [2.8e-05s]\n",
      "Opening the output file...  [7.2e-05s]\n",
      "Loading query sequences...  [0.128431s]\n",
      "Running complexity filter...  [1.89243s]\n",
      "Building query histograms...  [0.160397s]\n",
      "Allocating buffers...  [7.4e-05s]\n",
      "Loading reference sequences...  [0.112703s]\n",
      "Building reference histograms...  [1.18714s]\n",
      "Allocating buffers...  [0.000131s]\n",
      "Initializing temporary storage...  [0.000738s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 0.\n",
      "Building reference index...  [0.973812s]\n",
      "Building query index...  [0.118915s]\n",
      "Building seed filter...  [0.085079s]\n",
      "Searching alignments...  [0.088929s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 1.\n",
      "Building reference index...  [1.2453s]\n",
      "Building query index...  [0.13635s]\n",
      "Building seed filter...  [0.087573s]\n",
      "Searching alignments...  [0.096991s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 2.\n",
      "Building reference index...  [1.17932s]\n",
      "Building query index...  [0.136485s]\n",
      "Building seed filter...  [0.08577s]\n",
      "Searching alignments...  [0.087322s]\n",
      "Processing query chunk 0, reference chunk 0, shape 0, index chunk 3.\n",
      "Building reference index...  [0.960882s]\n",
      "Building query index...  [0.116979s]\n",
      "Building seed filter...  [0.08987s]\n",
      "Searching alignments...  [0.086879s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 0.\n",
      "Building reference index...  [0.947816s]\n",
      "Building query index...  [0.125525s]\n",
      "Building seed filter...  [0.089527s]\n",
      "Searching alignments...  [0.081632s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 1.\n",
      "Building reference index...  [1.09524s]\n",
      "Building query index...  [0.129516s]\n",
      "Building seed filter...  [0.088017s]\n",
      "Searching alignments...  [0.080841s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 2.\n",
      "Building reference index...  [1.11805s]\n",
      "Building query index...  [0.12736s]\n",
      "Building seed filter...  [0.086305s]\n",
      "Searching alignments...  [0.083204s]\n",
      "Processing query chunk 0, reference chunk 0, shape 1, index chunk 3.\n",
      "Building reference index...  [0.932964s]\n",
      "Building query index...  [0.109149s]\n",
      "Building seed filter...  [0.082084s]\n",
      "Searching alignments...  [0.080282s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 0.\n",
      "Building reference index...  [0.93305s]\n",
      "Building query index...  [0.101445s]\n",
      "Building seed filter...  [0.081855s]\n",
      "Searching alignments...  [0.07451s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 1.\n",
      "Building reference index...  [1.04542s]\n",
      "Building query index...  [0.114735s]\n",
      "Building seed filter...  [0.081284s]\n",
      "Searching alignments...  [0.074126s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 2.\n",
      "Building reference index...  [1.08343s]\n",
      "Building query index...  [0.124619s]\n",
      "Building seed filter...  [0.082408s]\n",
      "Searching alignments...  [0.069044s]\n",
      "Processing query chunk 0, reference chunk 0, shape 2, index chunk 3.\n",
      "Building reference index...  [0.901484s]\n",
      "Building query index...  [0.104237s]\n",
      "Building seed filter...  [0.081542s]\n",
      "Searching alignments...  [0.070414s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 0.\n",
      "Building reference index...  [0.907908s]\n",
      "Building query index...  [0.094342s]\n",
      "Building seed filter...  [0.079934s]\n",
      "Searching alignments...  [0.070951s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 1.\n",
      "Building reference index...  [1.05012s]\n",
      "Building query index...  [0.109936s]\n",
      "Building seed filter...  [0.082596s]\n",
      "Searching alignments...  [0.07083s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 2.\n",
      "Building reference index...  [1.09294s]\n",
      "Building query index...  [0.117459s]\n",
      "Building seed filter...  [0.083417s]\n",
      "Searching alignments...  [0.073389s]\n",
      "Processing query chunk 0, reference chunk 0, shape 3, index chunk 3.\n",
      "Building reference index...  [0.929055s]\n",
      "Building query index...  [0.099267s]\n",
      "Building seed filter...  [0.082223s]\n",
      "Searching alignments...  [0.069398s]\n",
      "Deallocating buffers...  [0.010661s]\n",
      "Computing alignments...  [0.215504s]\n",
      "Deallocating reference...  [0.004664s]\n",
      "Loading reference sequences...  [5.2e-05s]\n",
      "Deallocating buffers...  [0.000984s]\n",
      "Deallocating queries...  [0.001405s]\n",
      "Loading query sequences...  [2.7e-05s]\n",
      "Closing the input file...  [1.9e-05s]\n",
      "Closing the output file...  [0.002279s]\n",
      "Closing the database file...  [1.7e-05s]\n",
      "Total time = 24.599s\n",
      "Reported 6449 pairwise alignments, 6449 HSSPs.\n",
      "1288 queries aligned.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ssh microbioinf@192.168.56.101 2>/dev/null /bin/bash <<'EOT'\n",
    "cd Documentos/Tema_4a\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "diamond blastx -d viralprotein.dmnd  -q kneaddata_out/Vir1_R1_100000_kneaddata.fastq -o Vir1HQ_Vs_viralprotein.m8\n",
    "diamond blastx -d viralprotein.dmnd  -q kneaddata_out/Bact1_R1_200000_kneaddata.fastq -o Bact1HQ_Vs_viralprotein.m8\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash --out output\n",
    "ssh microbioinf@192.168.56.101 2>/dev/null  /bin/bash <<'EOT'\n",
    "cd Documentos/Tema_4a\n",
    "head -n 1 *fq\n",
    "grep -c \"@M02255\" *fq\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> Bact1_R1_200000.fq <==\n",
      "@M02255:131:000000000-AJC6R:1:2102:25217:13392_1:N:0:ACAGTG\n",
      "\n",
      "==> Bact1_R2_200000.fq <==\n",
      "@M02255:131:000000000-AJC6R:1:2102:25217:13392_2:N:0:ACAGTG\n",
      "\n",
      "==> Vir1_R1_100000.fq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA\n",
      "\n",
      "==> Vir1_R2_100000.fq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA\n",
      "Bact1_R1_200000.fq:200000\n",
      "Bact1_R2_200000.fq:200000\n",
      "Vir1_R1_100000.fq:100000\n",
      "Vir1_R2_100000.fq:100000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['==> Bact1_R1_200000.fq <==',\n",
       " '@M02255:131:000000000-AJC6R:1:2102:25217:13392_1:N:0:ACAGTG',\n",
       " '',\n",
       " '==> Bact1_R2_200000.fq <==',\n",
       " '@M02255:131:000000000-AJC6R:1:2102:25217:13392_2:N:0:ACAGTG',\n",
       " '',\n",
       " '==> Vir1_R1_100000.fq <==',\n",
       " '@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA',\n",
       " '',\n",
       " '==> Vir1_R2_100000.fq <==',\n",
       " '@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA',\n",
       " 'Bact1_R1_200000.fq:200000',\n",
       " 'Bact1_R2_200000.fq:200000',\n",
       " 'Vir1_R1_100000.fq:100000',\n",
       " 'Vir1_R2_100000.fq:100000',\n",
       " '']"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alist = output.split('\\n')\n",
    "alist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Bacteria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "root: 1288.0\n",
    "  Viruses: 922.0\n",
    "    dsDNA viruses, no RNA stage: 898.0\n",
    "      Ascoviridae: 6.0\n",
    "      Baculoviridae: 1.0\n",
    "      Caudovirales: 619.0\n",
    "        Ackermannviridae: 1.0\n",
    "        Myoviridae: 343.0\n",
    "        Podoviridae: 37.0\n",
    "        Siphoviridae: 225.0\n",
    "        unclassified Caudovirales: 3.0\n",
    "      Herpesvirales: 3.0\n",
    "        Herpesviridae: 3.0\n",
    "      Iridoviridae: 8.0\n",
    "      Marseilleviridae: 2.0\n",
    "      Mimiviridae: 62.0\n",
    "      Nimaviridae: 1.0\n",
    "      Nudiviridae: 2.0\n",
    "      Phycodnaviridae: 123.0\n",
    "      Poxviridae: 18.0\n",
    "      unclassified archaeal dsDNA viruses: 4.0\n",
    "      unclassified dsDNA phages: 9.0\n",
    "      unclassified dsDNA viruses: 28.0\n",
    "        Pandoravirus: 15.0\n",
    "        Pithoviridae: 13.0\n",
    "    Ortervirales: 4.0\n",
    "      Retroviridae: 4.0\n",
    "    ssDNA viruses: 2.0\n",
    "      Inoviridae: 1.0\n",
    "      unclassified ssDNA bacterial viruses: 1.0\n",
    "    unclassified bacterial viruses: 12.0\n",
    "    unclassified viruses: 4.0\n",
    "      Kaumoebavirus: 2.0\n",
    "      Mollivirus sibericum: 2.0\n",
    "  Not assigned: 332.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "recomputed\n",
    "root: 1288.0\n",
    "  Viruses: 589.0\n",
    "    dsDNA viruses, no RNA stage: 579.0\n",
    "      Ascoviridae: 4.0\n",
    "      Caudovirales: 441.0\n",
    "        Myoviridae: 250.0\n",
    "        Podoviridae: 20.0\n",
    "        Siphoviridae: 163.0\n",
    "        unclassified Caudovirales: 2.0\n",
    "      Iridoviridae: 6.0\n",
    "      Mimiviridae: 35.0\n",
    "      Nudiviridae: 1.0\n",
    "      Phycodnaviridae: 69.0\n",
    "      Poxviridae: 1.0\n",
    "      unclassified archaeal dsDNA viruses: 1.0\n",
    "      unclassified dsDNA phages: 4.0\n",
    "      unclassified dsDNA viruses: 8.0\n",
    "        Pandoravirus: 8.0\n",
    "    ssDNA viruses: 1.0\n",
    "      Inoviridae: 1.0\n",
    "    unclassified bacterial viruses: 5.0\n",
    "    unclassified viruses: 3.0\n",
    "      Kaumoebavirus: 2.0\n",
    "      Mollivirus sibericum: 1.0\n",
    "  Not assigned: 674.0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Viruses**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "       Myoviridae: 564.0\n",
    "        Podoviridae: 1055.0\n",
    "        Siphoviridae: 7232.0\n",
    "      Mimiviridae: 11.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "New params:\n",
    "root: 12725.0\n",
    "  Viruses: 8224.0\n",
    "    dsDNA viruses, no RNA stage: 7338.0\n",
    "      Caudovirales: 7190.0\n",
    "        Myoviridae: 338.0\n",
    "        Podoviridae: 823.0\n",
    "        Siphoviridae: 5677.0\n",
    "        unclassified Caudovirales: 11.0\n",
    "      Phycodnaviridae: 9.0\n",
    "      unclassified archaeal dsDNA viruses: 9.0\n",
    "      unclassified dsDNA phages: 21.0\n",
    "    unclassified bacterial viruses: 862.0\n",
    "  Not assigned: 4375.0"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
