{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Whole Genome Shotgun metagenomics: de novo Assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have two fastqc files to process:\n",
    "\n",
    "*Vir1_100k* that contains $100,000$ paired end reads from the same saliva sample but after purification of viral particles. So this is a virome.\n",
    "\n",
    "Different pairs of index primers are added to individual samples in a second thermocycling step, after initial amplification of the target region. This allows you to mix many samples together (up to 96) and sequence them at the same time. Following sequencing, for example on an Illumina MiSeq, the software is able to identify these indexes on each sequence read and, because you have already told the machine which pair of index primers was added to each sample, the machine then knows which sample to associate that read to, allowing you to separate the reads for each different sample.\n",
    "\n",
    "Use kneaddata to run pre-processing tools. First Trimmomatic is run to remove low quality sequences. Then Bowtie2 is run to screen out contaminant sequences. Below we are screening out reads that map to the human or PhiX genomes. Note KneadData is being run below on all unstitched FASTQ pairs with parallel, you can see our quick tutorial on this tool here. For a detailed breakdown of the options in the below command see this page. The forward and reverse reads will be specified by \"_1\" and \"_2\" in the output files, ignore the \"R1\" in each filename. Note that the \\ characters at the end of each line are just to split the command over multiple lines to make it easier to read.\n",
    "\n",
    "parallel -j 1 --link 'kneaddata -i {1} -i {2} -o kneaddata_out/ \\\n",
    "-db /home/shared/bowtiedb/GRCh38_PhiX --trimmomatic /usr/local/prg/Trimmomatic-0.36/ \\\n",
    "-t 4 --trimmomatic-options \"SLIDINGWINDOW:4:20 MINLEN:50\" \\\n",
    "--bowtie2-options \"--very-sensitive --dovetail\" --remove-intermediate-output' \\\n",
    " ::: cat_lanes/*_R1.fastq ::: cat_lanes/*_R2.fastq\n",
    "Clean up the output directory (helps downstream commands) by moving the discarded sequences to a subfolder:\n",
    "\n",
    "mkdir kneaddata_out/contam_seq\n",
    "\n",
    "mv kneaddata_out/*_contam*.fastq kneaddata_out/contam_seq\n",
    "You can produce a logfile summarizing the kneadData output with this command:\n",
    "\n",
    "The current version of SPAdes works with Illumina or IonTorrent reads and is capable of providing hybrid assemblies using PacBio, Oxford Nanopore and Sanger reads. You can also provide additional contigs that will be used as long reads.\n",
    "\n",
    "Version 3.10.1 of SPAdes supports paired-end reads, mate-pairs and unpaired reads. SPAdes can take as input several paired-end and mate-pair libraries simultaneously. Note, that SPAdes was initially designed for small genomes. It was tested on bacterial (both single-cell MDA and standard isolates), fungal and other small genomes. SPAdes is not intended for larger genomes (e.g. mammalian size genomes). For such purposes you can use it at your own risk.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import pandas as pd\n",
    "#FILE_ID = \"ECTV\"\n",
    "#FASTQ_STR = \"@HWUSI-EAS1752R\"\n",
    "#MIN_LEN = \"70\"\n",
    "\n",
    "FILE_ID = \"VIR\"\n",
    "FASTQ_STR = \"@M02255\"\n",
    "MIN_LEN = \"200\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preprocessing and quality check\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check files FILE_ID=VIR, FASTQ_STR=@M02255\n",
      "==> VIR_R1.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA\n",
      "AACTGGCGTTACATGAAGGGCTCTGAGTTGATTGATGCTTTGGAGGAGTACCTGTGAAATGGCCGTCTGAGAAGGTTGTTAATGCGACCGTAAAGTATGGTGGTGTCGTGTTGAGACGTGGACCGTACGCATATTTCGATAAGGGGGGCATTCGATTGTGTGCTACAAGGCTTGGTCTCTCTTCATATATTGTGGAGAGTGATGATTGTGGTCCTGAGATTTATAGTGAGGATGGTATGATTGAGTTGGTGACGTCTTTATGATTCCTGTTACCGAGACTATCCTGAAAACTGCTTACCAT\n",
      "+\n",
      "CCCCCGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGFAGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGEFGGGGGGGGGGGGGFGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGCF@EGGGG9CEGGGFFGGGC?FGECFFGGGG9CFGGFEFCGGEGGCF?FG7EGCFGC:FFGF6<FGFGGFGGGFFFFFEGFF@7@)8@FFFBAFA=F<FDFF<157526))4?<39>B9?><?BAA?2>F\n",
      "@M02255:128:000000000-AG7E5:1:1112:17059:22756_1:N:0:AGTCAA\n",
      "TGTTGGTCTATATTGCGCTTTAGCAGCCTTGTTTTTAAAAGCTGATACAAACGGGAATATAAAAATAATTGATAAAGAAAACATAAAAACGGATCAACTTGCCGGGGCAAGTGAAGCCATGAAACACATAGAGAAGTCTTCTATTCCATTAATGACAACTTCAGATCTTGACATTTATTCAGATTCATTATAACACACCTTAGACAATCTTTCTAAACTAGCGCCATTGTATGAAGCTTTTCCGTAATATAGTGAAGTTAGGTCTTTTGCTGCATCCCTTCAAGGCATCTCTAAAGTCCT\n",
      "+\n",
      "CCCCC<FFGFFGCFGGGGGGGGGGCA8FGGGEAFGFGAFCCGGGGFG<F99FGGFF@EFCFCCFCFF6EFF9E9E,9,,9E,:EF9,CFG77@FEG@A9?EC,>C=CC7DE<CFFGGGF,<F,?EE,A,CE9<<EFG,,EAEBFCFBCFGFA?C@FGDEFGGGAE94,,@C8=,7EE,@3,8,,@,3@ED,7=E=@DC=,6>A,2,3+,@,@,7D<;CCF<C+*4;@=D@*,+?F>>C>+*+*/9:3*+3/+41*4198+=)0++=7+0*/:*)B6()0))27:)*6;(+)+)+)7.:)<\n",
      "@M02255:131:000000000-AJC6R:1:1101:8786:11380_1:N:0:AGTCAA\n",
      "AATACTGATGCCGCCGAGATAGGTGCGCCACCAAGCGATTCACCGTCACGAACCTTCTGCACCGGTCCAAGACTGCAACCGACACCCTGTGACTGATTGTCGTAGTAGTAGAAGTTCACAAGGACATTAGCGTACATACCGGAGTAGACTTGCGTATGATCCGTAATCGGATTGCCCGACATGTCGACGACCTCAACCTTATACTTCGCATCTTGCCAAGCGGTTAAGCTCCAGTGCCCCTTTGCTTCCGGGCCGACCTCTTTACCGGTCTGCGTGTTTCCTTCTCCGTCCCATACAGGCC\n",
      "\n",
      "==> VIR_R2.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA\n",
      "GATGAAATTCTGAAGCAACGGACTAATGATCGGCAGCGGCATGCTTCCTCCTCAATTTCTCCTTCAGGAATATGATTGTCCCGATTTCTGTCAATTGAATATCGACCTGTTCAAAAGTGCACTGCCAGAGATCCTCCTTAATTCTAATAATATCCACGAAGCGGTTTCCTGAATTAATGCATGCAGTAGCATTATCAGGGAAAAAGATGTGCCACCTGAAATGGTAAGCAGTTTTCAGGATAGTCTCGGTAACAGGAATCATAAAGACGTCACCAACTCAATCATATCAANCTCACTATA\n",
      "+\n",
      "CCCCCGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGGGGGGGGGGGDGGEGGGGGGGGGADFGCDGGC?F8@FGFGGDGFGGGFGFFFFFGGGFFGFFGF*96FFAFFCFFFF@FF;A?FA<?EA=4@F478A2>F@@CFBDD@B9EFFC).//4/8?EF0:@AEE?=;?.).5)#/(/6(6265\n",
      "@M02255:128:000000000-AG7E5:1:1112:17059:22756_2:N:0:AGTCAA\n",
      "GAACTTAGTTCTTCAAATGTTCGGTTTGGAATTTGGGAAGTCCTAGCACTTGGAAGGCCGTTTTCTTCTATGTCTAAAAGTAGGTTTCTTTCAGCTCGTTTAGCTTTCTTTTCTCTCCTGAATCTCCGGCGCGTGTTTCTTCTTTCCTTCCCGGTCAATGGCTAAACCCATAACTAAGTTCGGCATAAATACCGGGTTCCATCGTCTGTAGTATTCTATTCCTTTCGTATACTCTTTTCCTTTCTTCAACCTTGTCTGCATTGTCGTAGAGTCTAATTTTGATTGTTACACCTCACTTCTT\n",
      "+\n",
      "8---,;,C<CCEC,<C6<,C,,8@8,C,C<6C,E,C,C,,6C<,<CFG,C;C,C,CB<6+76,8,,E6,CEA,C9AFD,EF9B,CFEFFDC,BCCE,B@<7,A9,E,49B,C,,,4,,4>AFFGA<>886BEC4++,@;DG?,=E,=,@,+@==8@EC8B+,0,3,666+6,0,,,+,272=++03*,*79<+<3)*1*35++)**3*=7+=++2*:+:*+3:AF+:*-)),*.*+4++*41/4=***10(27)//173<++/1+,(()))))1)))))))(19).).))),(((.6))))\n",
      "@M02255:131:000000000-AJC6R:1:1101:8786:11380_2:N:0:AGTCAA\n",
      "ATATGAACGCCGAGTGCTCACCTTGGCACAGATGGAAAAGGCCGTCGGCAAGAAGGCCTTCGGCGAATTGGTCGGCGACCTTGTTGTGAAGAACCCCGGTAAGCCGACACTTGTTGAAGAATCGGATAAGCGTCCGAGGATAACGAATCAGCCGACAGCAGCTGACGTATTTAATTCTTAATTCATCGCGGTATTAACTATGTTGACACCGAAAGCTACAGAAGTATTAATGACGAATGTTAGTCTGTATTATTGATACGTACAGGACCTGTTTACGGGTTGTCAACGGGNCACGCCGAA\n",
      "VIR_R1.fastq:100000\n",
      "VIR_R2.fastq:100000\n",
      "#### Compute quality\n",
      "Analysis complete for VIR_R1.fastq\n",
      "Analysis complete for VIR_R2.fastq\n",
      "#### Replace ' ' by '_' in header\n",
      "==> VIR_R1.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA\n",
      "\n",
      "==> VIR_R2.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA\n",
      "==> VIR_R1_.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA\n",
      "\n",
      "==> VIR_R1.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_1:N:0:AGTCAA\n",
      "\n",
      "==> VIR_R2_.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA\n",
      "\n",
      "==> VIR_R2.fastq <==\n",
      "@M02255:131:000000000-AJC6R:1:1105:23249:10170_2:N:0:AGTCAA\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "echo \"#### Check files FILE_ID=${FILE_ID}, FASTQ_STR=$FASTQ_STR\"\n",
    "cd Documentos/Tema_3\n",
    "head ${FILE_ID}*fastq\n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq\n",
    "\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_Quality\n",
    "fastqc ${FILE_ID}_R1.fastq -o ${FILE_ID}_Quality/\n",
    "fastqc ${FILE_ID}_R2.fastq -o ${FILE_ID}_Quality/\n",
    "\n",
    "echo \"#### Replace ' ' by '_' in header\"\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "cat ${FILE_ID}_R1.fastq | sed 's/ /_/g' > ${FILE_ID}_R1_.fastq\n",
    "cat ${FILE_ID}_R2.fastq | sed 's/ /_/g' > ${FILE_ID}_R2_.fastq\n",
    "head -n 1 ${FILE_ID}*fastq\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming and decontaminating\n",
    "Trimming poor quality ends and short sequences (**Trimmomatic**) and removal of reads aligning to the human and phiX174 genomes (***bowtie2**). The later one is a contaminant used as spike by Illumina kits to control quality of the sequencing process.\n",
    "\n",
    "We are only filtering only R1 files because forward reads have usually better quality than reverse reads. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Trimming and decontaminating FILE_ID=VIR MIN_LEN=200\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq ): 100000\n",
      "Initial number of reads ( /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq ): 100000\n",
      "Running Trimmomatic ... \n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq ): 29521\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq ): 29521\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq ): 34081\n",
      "Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq ): 1926\n",
      "Decontaminating ...\n",
      "Running bowtie2 ... \n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 29310\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 29310\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq ): 29310\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq ): 29310\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 8\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_1.fastq ): 8\n",
      "Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 35922\n",
      "Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_2.fastq ): 35922\n",
      "\n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_2.fastq\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "echo \"#### Trimming and decontaminating FILE_ID=${FILE_ID} MIN_LEN=${MIN_LEN}\"\n",
    "kneaddata -i ${FILE_ID}_R1_.fastq -i ${FILE_ID}_R2_.fastq \\\n",
    "-o kneaddata_out -db /home/shared/bowtiedb/GRCh38_PhiX \\\n",
    "--trimmomatic /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/ \\\n",
    "-t 2 --trimmomatic-options \"SLIDINGWINDOW:4:20 MINLEN:${MIN_LEN}\" \\\n",
    "--bowtie2-options \"--very-sensitive --dovetail\" --remove-intermediate-output\n",
    "EOT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "04/26/2019 01:17:31 PM - kneaddata.knead_data - INFO: Running kneaddata v0.6.1\n",
      "04/26/2019 01:17:31 PM - kneaddata.knead_data - INFO: Output files will be written to: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "04/26/2019 01:17:31 PM - kneaddata.knead_data - DEBUG: Running with the following arguments: \n",
      "verbose = False\n",
      "bmtagger_path = None\n",
      "minscore = 50\n",
      "bowtie2_path = /home/microbioinf/miniconda3/bin/bowtie2\n",
      "maxperiod = 500\n",
      "no_discordant = False\n",
      "serial = False\n",
      "fastqc_start = False\n",
      "bmtagger = False\n",
      "cat_final_output = False\n",
      "log_level = DEBUG\n",
      "log = /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.log\n",
      "max_memory = 500m\n",
      "remove_intermediate_output = True\n",
      "fastqc_path = None\n",
      "output_dir = /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "trf_path = None\n",
      "remove_temp_output = True\n",
      "reference_db = /home/shared/bowtiedb/GRCh38_PhiX\n",
      "input = /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq\n",
      "pi = 10\n",
      "reorder = False\n",
      "pm = 80\n",
      "trimmomatic_path = /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar\n",
      "store_temp_output = False\n",
      "cat_pairs = False\n",
      "mismatch = 7\n",
      "threads = 2\n",
      "delta = 7\n",
      "bowtie2_options = --very-sensitive --dovetail --phred33\n",
      "bypass_trim = False\n",
      "processes = 1\n",
      "trimmomatic_quality_scores = -phred33\n",
      "fastqc_end = False\n",
      "trf = False\n",
      "trimmomatic_options = SLIDINGWINDOW:4:20 MINLEN:200\n",
      "output_prefix = VIR_R1__kneaddata\n",
      "match = 2\n",
      "\n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - INFO: READ COUNT: raw pair1 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq ): 100000\n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - INFO: READ COUNT: raw pair2 : Initial number of reads ( /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq ): 100000\n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq\n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - DEBUG: Checking input file to Trimmomatic : /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq\n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - INFO: Running Trimmomatic ... \n",
      "04/26/2019 01:17:31 PM - kneaddata.utilities - INFO: Execute command: java -Xmx500m -d64 -jar /home/microbioinf/miniconda3/pkgs/trimmomatic-0.38-1/share/trimmomatic-0.38-1/trimmomatic.jar PE -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:200\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: TrimmomaticPE: Started with arguments:\n",
      " -threads 2 -phred33 /home/microbioinf/Documentos/Tema_3/VIR_R1_.fastq /home/microbioinf/Documentos/Tema_3/VIR_R2_.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq SLIDINGWINDOW:4:20 MINLEN:200\n",
      "Input Read Pairs: 100000 Both Surviving: 29521 (29,52%) Forward Only Surviving: 34081 (34,08%) Reverse Only Surviving: 1926 (1,93%) Dropped: 34472 (34,47%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking output file from Trimmomatic : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: READ COUNT: trimmed pair1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq ): 29521\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: READ COUNT: trimmed pair2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq ): 29521\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan1 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq ): 34081\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: READ COUNT: trimmed orphan2 : Total reads after trimming ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq ): 1926\n",
      "04/26/2019 01:17:32 PM - kneaddata.run - INFO: Decontaminating ...\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - DEBUG: Checking input file to bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq\n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: Running bowtie2 ... \n",
      "04/26/2019 01:17:32 PM - kneaddata.utilities - INFO: Execute command: kneaddata_bowtie2_discordant_pairs --bowtie2 /home/microbioinf/miniconda3/bin/bowtie2 --threads 2 -x /home/shared/bowtiedb/GRCh38_PhiX --bowtie2-options \"--very-sensitive --dovetail --phred33\" -1 /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.1.fastq -2 /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.2.fastq --un-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_%.fastq --al-pair /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_%.fastq -U /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.1.fastq,/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata.trimmed.single.2.fastq --un-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_clean.fastq --al-single /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_%_contam.fastq -S /dev/null\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - DEBUG: 65528 reads; of these:\n",
      "  29521 (45.05%) were paired; of these:\n",
      "    29338 (99.38%) aligned concordantly 0 times\n",
      "    157 (0.53%) aligned concordantly exactly 1 time\n",
      "    26 (0.09%) aligned concordantly >1 times\n",
      "    ----\n",
      "    29338 pairs aligned concordantly 0 times; of these:\n",
      "      10 (0.03%) aligned discordantly 1 time\n",
      "    ----\n",
      "    29328 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      58656 mates make up the pairs; of these:\n",
      "        58636 (99.97%) aligned 0 times\n",
      "        15 (0.03%) aligned exactly 1 time\n",
      "        5 (0.01%) aligned >1 times\n",
      "  36007 (54.95%) were unpaired; of these:\n",
      "    35914 (99.74%) aligned 0 times\n",
      "    81 (0.22%) aligned exactly 1 time\n",
      "    12 (0.03%) aligned >1 times\n",
      "0.52% overall alignment rate\n",
      "pair1_aligned : 183\n",
      "pair2_aligned : 183\n",
      "orphan1_unaligned : 8\n",
      "orphan2_unaligned : 35922\n",
      "orphan2_aligned : 113\n",
      "pair2_unaligned : 29310\n",
      "pair1_unaligned : 29310\n",
      "orphan1_aligned : 20\n",
      "\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - DEBUG: Checking output file from bowtie2 : /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq ) : 183\n",
      "04/26/2019 01:17:48 PM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq ) : 183\n",
      "04/26/2019 01:17:48 PM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq ) : 20\n",
      "04/26/2019 01:17:48 PM - kneaddata.run - INFO: Total contaminate sequences in file ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq ) : 113\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq ): 29310\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX pair2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq ): 29310\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: final pair1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq ): 29310\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: final pair2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq ): 29310\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_1.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_clean_2.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan1 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq ): 8\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: final orphan1 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_1.fastq ): 8\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_clean.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: decontaminated GRCh38_PhiX orphan2 : Total reads after removing those found in reference database ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq ): 35922\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - INFO: READ COUNT: final orphan2 : Total reads after merging results from multiple databases ( /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_2.fastq ): 35922\n",
      "04/26/2019 01:17:48 PM - kneaddata.utilities - WARNING: Unable to remove file: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_clean.fastq\n",
      "04/26/2019 01:17:48 PM - kneaddata.knead_data - INFO: \n",
      "Final output files created: \n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_1.fastq\n",
      "/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_unmatched_2.fastq\n",
      "\n",
      "Reading log: ./VIR_R1__kneaddata.log\n",
      "Reading log: ./ECTV_R1__kneaddata.log\n",
      "Read count table written: kneaddata_read_countsVIR.txt\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:183\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:183\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:20\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:113\n",
      "VIR_R1__kneaddata_paired_1.fastq:29310\n",
      "VIR_R1__kneaddata_paired_2.fastq:29310\n",
      "VIR_R1__kneaddata_unmatched_1.fastq:8\n",
      "VIR_R1__kneaddata_unmatched_2.fastq:35922\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "cat ${FILE_ID}_R1*log\n",
    "kneaddata_read_count_table --input ./ --output kneaddata_read_counts${FILE_ID}.txt \n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Sample</th>\n",
       "      <td>ECTV_R1__kneaddata</td>\n",
       "      <td>VIR_R1__kneaddata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair1</th>\n",
       "      <td>50000</td>\n",
       "      <td>100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>raw pair2</th>\n",
       "      <td>50000</td>\n",
       "      <td>100000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair1</th>\n",
       "      <td>41996</td>\n",
       "      <td>29521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed pair2</th>\n",
       "      <td>41996</td>\n",
       "      <td>29521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan1</th>\n",
       "      <td>4107</td>\n",
       "      <td>34081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>trimmed orphan2</th>\n",
       "      <td>1791</td>\n",
       "      <td>1926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair1</th>\n",
       "      <td>41839</td>\n",
       "      <td>29310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX pair2</th>\n",
       "      <td>41839</td>\n",
       "      <td>29310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan1</th>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>decontaminated GRCh38_PhiX orphan2</th>\n",
       "      <td>5851</td>\n",
       "      <td>35922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair1</th>\n",
       "      <td>41839</td>\n",
       "      <td>29310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final pair2</th>\n",
       "      <td>41839</td>\n",
       "      <td>29310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan1</th>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>final orphan2</th>\n",
       "      <td>5851</td>\n",
       "      <td>35922</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     0                  1\n",
       "Sample                              ECTV_R1__kneaddata  VIR_R1__kneaddata\n",
       "raw pair1                                        50000             100000\n",
       "raw pair2                                        50000             100000\n",
       "trimmed pair1                                    41996              29521\n",
       "trimmed pair2                                    41996              29521\n",
       "trimmed orphan1                                   4107              34081\n",
       "trimmed orphan2                                   1791               1926\n",
       "decontaminated GRCh38_PhiX pair1                 41839              29310\n",
       "decontaminated GRCh38_PhiX pair2                 41839              29310\n",
       "decontaminated GRCh38_PhiX orphan1                  12                  8\n",
       "decontaminated GRCh38_PhiX orphan2                5851              35922\n",
       "final pair1                                      41839              29310\n",
       "final pair2                                      41839              29310\n",
       "final orphan1                                       12                  8\n",
       "final orphan2                                     5851              35922"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/kneaddata_read_counts%s.txt\n",
    "EOT\n",
    "\"\"\" % FILE_ID\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_knead = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_knead.style.hide_index().set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"11pt\"}).\\\n",
    "set_properties(**{'text-align': 'right', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"12pt\"}, subset=['Sample'])\n",
    "df_knead.transpose()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check number of reads"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With grep we can identify the non-contaminated high-quality files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_1.fastq:183\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_paired_contam_2.fastq:183\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_1_contam.fastq:20\n",
      "VIR_R1__kneaddata_GRCh38_PhiX_bowtie2_unmatched_2_contam.fastq:113\n",
      "VIR_R1__kneaddata_paired_1.fastq:29310\n",
      "VIR_R1__kneaddata_paired_2.fastq:29310\n",
      "VIR_R1__kneaddata_unmatched_1.fastq:8\n",
      "VIR_R1__kneaddata_unmatched_2.fastq:35922\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "grep -c $FASTQ_STR ${FILE_ID}*fastq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check quality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute quality\n",
      "Analysis complete for VIR_R1__kneaddata_paired_1.fastq\n",
      "Analysis complete for VIR_R1__kneaddata_paired_2.fastq\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute quality\"\n",
    "mkdir ${FILE_ID}_HighQuality\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_1.fastq -o ${FILE_ID}_HighQuality/\n",
    "fastqc ${FILE_ID}_R1__kneaddata_paired_2.fastq -o ${FILE_ID}_HighQuality/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assembly\n",
    "\n",
    " We are going to use a Refseq database of viral proteins (around 100Mb) from ncbi (ftp://ftp.ncbi.nlm.nih.gov/refseq/release/viral/), and you have to download it in two separated files that can be joined into one with cat.\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process (*spades*)\n",
    "\n",
    "In this step we run command **spades** with the paired high-quality and free of known contaminants reads."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### K_MER = 35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "K_MER = 35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\t--sc\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.006     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.006     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.006     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:01.225     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:02.258     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 58620 reads\n",
      "  0:00:02.258     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58620 reads processed\n",
      "  0:00:02.266    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.568    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 11626452 kmers in total.\n",
      "  0:00:02.568    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.641    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.525    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.663    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 5399536 bytes occupied (3.71535 bits per kmer).\n",
      "  0:00:03.663    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:04.358   196M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:40.437   196M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:42.923   196M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 10832792\n",
      "  0:00:42.927   104M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:43.023   372M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:48.008   372M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:52.428   372M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:52.464   372M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 11626452 kmers in total. Among them 7209736 (62.0115%) are singletons.\n",
      "  0:00:52.464   372M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 6 non-read kmers were generated.\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 10149734. Among them 7959482 (78.4206%) are good\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 13363. Among them 13352 (99.9177%) are good\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 681455. Among them 468074 (68.6874%) are good\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.16702 kmers\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.01722\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 8440908\n",
      "  0:00:53.625   372M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.94211,0.022035,0.025852,0.0100032),(0.0125571,0.946744,0.00388487,0.036814),(0.0232144,0.00237136,0.952249,0.0221657),(0.00627067,0.0159369,0.0152383,0.962554))\n",
      "  0:00:53.641   372M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:53.641   372M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:56.715   372M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 71250 new k-mers.\n",
      "  0:00:59.807   372M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 356 new k-mers.\n",
      "  0:01:02.905   372M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:01:02.906   372M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:01:02.906   372M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:01:02.906   372M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:01:03.118   436M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 29310 reads.\n",
      "  0:01:05.538   444M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:01:05.610   444M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:01:05.641   372M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 15566 bases in 11498 reads.\n",
      "  0:01:05.641   372M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 14550035.\n",
      "  0:01:05.658    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/corrected/corrected.yaml\n",
      "  0:01:05.658    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.002     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.003     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.004     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.005     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.342    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.718   116M / 116M  INFO    General                 (binary_converter.hpp      : 117)   29310 reads written\n",
      "  0:00:00.742     4M / 116M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.873   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.879     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:01.005   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:01.011     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:01.011     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 248.209\n",
      "  0:00:01.012     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:01.013     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:01.015     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.015     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:01.015     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:02.013     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 117240 reads\n",
      "  0:00:02.013     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:02.025    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 117240 reads\n",
      "  0:00:02.025    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.136    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5453684 kmers in total.\n",
      "  0:00:02.137    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.200    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:02.201    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.201    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.201    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.201    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:02.202    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:03.133     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 5453684 kmers\n",
      "  0:00:03.134     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 5453684 kmers.\n",
      "  0:00:03.151    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:03.281    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5465187 kmers in total.\n",
      "  0:00:03.281    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:03.348    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.804    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.908    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2542568 bytes occupied (3.72184 bits per kmer).\n",
      "  0:00:03.912    20M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:04.886    20M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:04.886    20M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:04.886    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:06.287    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 33406 sequences extracted\n",
      "  0:00:06.910    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:07.157    20M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:07.190    32M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:07.190    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:07.190    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:07.596    36M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2530384 bytes occupied (3.71182 bits per kmer).\n",
      "  0:00:07.605    60M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:08.619    60M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:09.554    60M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 66809 edges\n",
      "  0:00:09.581    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:09.621    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 5\n",
      "  0:00:09.622    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 129)   Proper threshold was not found. Threshold set to 0.1 of average coverage\n",
      "  0:00:09.634    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 2.29172\n",
      "  0:00:09.634    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 0.229172\n",
      "  0:00:09.635    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 0.229172\n",
      "  0:00:09.635    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:09.635    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:09.635    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:09.635    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:09.636    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:09.636    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:09.636    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:09.636    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:10.074     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 66809 edges\n",
      "  0:00:10.075     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 66809 sequences.\n",
      "  0:00:10.082    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:10.221    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5453684 kmers in total.\n",
      "  0:00:10.221    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:10.289    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:10.804    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:10.949    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2537304 bytes occupied (3.72197 bits per kmer).\n",
      "  0:00:11.057   152M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:11.626   152M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:11.626   152M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:11.662   160M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:11.817   160M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 29310 paired reads\n",
      "  0:00:11.817   160M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:11.844   156M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:11.902   156M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 102 gaps after checking 2826 candidates\n",
      "  0:00:11.983   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:11.993    28M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:11.995    28M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:11.995    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:12.002    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:12.003    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:12.003    28M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:12.003    28M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:12.003    28M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:12.003    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.071    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 5401 times\n",
      "  0:00:12.072    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.350    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2195 times\n",
      "  0:00:12.350    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.355    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.355    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:12.355    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.357    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 69 times\n",
      "  0:00:12.357    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.358    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.359    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:12.359    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.369    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.369    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.381    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 8 times\n",
      "  0:00:12.381    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.385    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:12.385    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:12.385    32M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:12.385    32M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:12.386    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:12.386    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:12.386    32M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:12.386    32M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.329861 Gb\n",
      "  0:00:12.386    32M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 461179\n",
      "  0:00:12.823     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 32103 edges\n",
      "  0:00:12.823     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 32103 sequences.\n",
      "  0:00:12.830    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:12.973    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 5057548 kmers in total.\n",
      "  0:00:12.974    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:13.049    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:13.534    36M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:13.610    36M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2353336 bytes occupied (3.72249 bits per kmer).\n",
      "  0:00:13.665   152M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:14.288   152M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:14.288   152M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:14.321   156M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:14.652   156M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 29310 paired reads\n",
      "  0:00:14.652   156M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:14.678   156M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:14.709   156M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 18 gaps after checking 1096 candidates\n",
      "  0:00:14.738   152M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:14.738   152M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:15.138   152M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:15.227   152M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:15.271   152M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:15.300   152M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:15.307   152M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 1 times\n",
      "  0:00:15.307   152M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:15.313   152M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 2 times\n",
      "  0:00:15.313   152M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:15.669   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 130 times\n",
      "  0:00:15.669   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.681   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 6 times\n",
      "  0:00:15.681   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:15.691   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 2 times\n",
      "  0:00:15.691   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.702   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
      "  0:00:15.702   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:15.753   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 57 times\n",
      "  0:00:15.753   156M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:15.753   156M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:15.776   156M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:15.800   156M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:15.842   156M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:15.863   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:15.868   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:15.868   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:15.874   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:15.874   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:15.905   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 6 times\n",
      "  0:00:15.905   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.920   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 10 times\n",
      "  0:00:15.920   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:15.925   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:15.925   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.931   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.931   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:15.935   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:15.935   156M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:15.935   156M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:15.952   156M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:15.971   156M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:16.010   156M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:16.028   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:16.033   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:16.033   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.036   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:16.036   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.060   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:16.060   156M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:16.064   156M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:16.075   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:17.241   156M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 9671 times\n",
      "  0:00:17.241   156M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:17.244   156M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.28207\n",
      "  0:00:17.244   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:17.244   156M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 276352 kmers to process\n",
      "  0:00:17.392   156M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:19.535   164M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:19.612   156M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:19.616   156M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 27 nucleotides\n",
      "  0:00:19.616   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:19.617   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:19.658   156M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/before_rr.fasta\n",
      "  0:00:19.709   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:19.872   156M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:19.932   156M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:19.986   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:20.007   156M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 277766 kmers to process\n",
      "  0:00:20.175   156M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:20.180   156M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 540\n",
      "  0:00:20.180   156M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:20.181   156M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:20.221   220M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.142   220M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 29310 reads processed\n",
      "  0:00:22.420   220M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:22.420   220M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   14675 paired reads (50.0682% of all) aligned to long edges\n",
      "  0:00:22.424   156M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 511.615, deviation = 167.518, left quantile = 312, right quantile = 746, read length = 301\n",
      "  0:00:22.424   156M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:22.424   156M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:22.425   156M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 312, right insert size quantile 746, filtering threshold 0, rounding threshold 0\n",
      "  0:00:22.431   168M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:23.249   168M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 29310 reads processed\n",
      "  0:00:23.254   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:23.254   156M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:23.254   156M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:23.256   156M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:23.283   156M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:23.283   156M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 19934\n",
      "  0:00:23.293   160M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:23.293   156M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:23.294   156M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:23.294   156M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:23.377   156M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 326; contradictional = 8\n",
      "  0:00:23.429   156M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 4; contradictional = 0\n",
      "  0:00:23.429   156M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:23.429   156M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:23.451   156M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:23.451   156M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 11111\n",
      "  0:00:23.456   160M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:23.456   156M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:23.458   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:23.458   156M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:23.458   156M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:23.478   180M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:23.484   180M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:23.484   180M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:23.484   180M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:23.484   180M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:23.484   180M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 5435 (0%)\n",
      "  0:00:23.528   180M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 5435 (2%)\n",
      "  0:00:23.541   184M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 5435 (4%)\n",
      "  0:00:23.565   184M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 5435 (9%)\n",
      "  0:00:23.567   184M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 544 paths from 5435 (10%)\n",
      "  0:00:23.592   188M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 5435 (18%)\n",
      "  0:00:23.596   188M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1088 paths from 5435 (20%)\n",
      "  0:00:23.621   192M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1632 paths from 5435 (30%)\n",
      "  0:00:23.637   196M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 5435 (37%)\n",
      "  0:00:23.643   200M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2176 paths from 5435 (40%)\n",
      "  0:00:23.662   204M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2720 paths from 5435 (50%)\n",
      "  0:00:23.682   208M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3264 paths from 5435 (60%)\n",
      "  0:00:23.701   212M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3808 paths from 5435 (70%)\n",
      "  0:00:23.714   212M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 5435 (75%)\n",
      "  0:00:23.723   216M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4352 paths from 5435 (80%)\n",
      "  0:00:23.747   220M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4896 paths from 5435 (90%)\n",
      "  0:00:23.799   224M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:23.799   224M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:23.811   224M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:23.811   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:23.811   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:23.814   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:23.814   224M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:23.820   224M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:23.824   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:23.826   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:23.828   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:23.836   224M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:23.836   224M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:23.853   224M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:23.860   224M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:23.884   224M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 10 loops\n",
      "  0:00:23.884   224M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:23.884   224M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:23.887   224M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:23.887   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:23.887   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:23.890   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:23.890   224M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:23.896   224M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:23.903   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:23.903   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:23.907   224M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:23.913   224M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:23.927   224M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:23.952   196M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:23.953   196M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:23.992   196M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/before_rr.fasta\n",
      "  0:00:24.046   196M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:24.216   200M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/final_contigs.paths\n",
      "  0:00:24.296   196M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35//K35/scaffolds.paths\n",
      "  0:00:24.372   196M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:24.483    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 24 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"35\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -k ${K_MER} -o ${FILE_ID}-Assembly${K_MER}\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Check output K_MER=35\n",
      ">NODE_1_length_87312_cov_13.958695\n",
      ">NODE_2_length_43671_cov_9.036942\n",
      ">NODE_3_length_24905_cov_6.568195\n",
      ">NODE_4_length_15602_cov_12.480182\n",
      ">NODE_5_length_14953_cov_8.458238\n",
      ">NODE_6_length_13800_cov_8.779295\n",
      ">NODE_7_length_12853_cov_6.110704\n",
      ">NODE_8_length_12581_cov_26.042962\n",
      ">NODE_1_length_87312_cov_13.958695\n",
      ">NODE_2_length_43671_cov_9.036942\n",
      ">NODE_3_length_28485_cov_10.565624\n",
      ">NODE_4_length_24905_cov_6.568195\n",
      ">NODE_5_length_16556_cov_8.484898\n",
      ">NODE_6_length_15602_cov_12.480182\n",
      ">NODE_7_length_12853_cov_6.110704\n",
      ">NODE_8_length_12581_cov_26.042962\n",
      "scaffolds.fasta:CCCACAAGGGCCGTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNACTTTA\n",
      "scaffolds.fasta:TCCTCTTCTTTCGCGCGTTCNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:TTTTANNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGTGTATTCTGTGATC\n",
      "scaffolds.fasta:AACNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTCTGCTTCTGGTCCTAG\n",
      "scaffolds.fasta:GTTGAACATGATGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGTTCCTT\n",
      "scaffolds.fasta:CCCAGCACACACAGCTGGATAGGATCAGGATATATTGATTNNNNNNNNNNCGCTCGATGG\n",
      "scaffolds.fasta:TGTGCCCCGCGGATTAACTTAAATAAANNNNNNNNNNGAATTCATATTAGTTGTTTTTTT\n",
      "scaffolds.fasta:AGGGAGCGCAGCTTGATCCACGTTCTTCCCCACTNNNNNNNNNNTTACTTTAGCGCGGAG\n",
      "scaffolds.fasta:NNNNNNNNNNGTTGATATGTATAAGTATTAGATAATAAATACTAATGATATTATAAAGAC\n",
      "scaffolds.fasta:GAATGTACGCAATTCTTGGTTCATGTACCATACAGGNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNAATGTACGCAATTCTTGGTTCATGTACCATACAGGACGGCAGCC\n",
      "scaffolds.fasta:GCCCGATGTAGCGGGGCAGGAGGATAGCCATGTACATGTCCGACAAGGAACGGATCTNNN\n",
      "scaffolds.fasta:NNNNNNNTAAATGCGAACGGCTTAAAATACTGCCGCACGTACTCCAGCTGCTCGACGGCC\n",
      "scaffolds.fasta:TNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAATGACTGGAGACACTCAT\n",
      "scaffolds.fasta:GGGTGATGACGCATGACGTTTGAATATTTGCGCNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNGCGGGTGATGACGCATGACGTTTGAATATTTGCGCAATCTAAACCGT\n",
      "scaffolds.fasta:CAACAAAGAAGTTGACATTTAATTTATATTAAGAACAATNNNNNNNNNNAAAGGAAATGA\n",
      "scaffolds.fasta:TTTCCTTAGATTCTTCNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTAG\n",
      "scaffolds.fasta:AGCTATGCTGGAATTGTTGNNNNNNNNNNGTAAAGGATAATGATGGAATATAGTTGTATC\n",
      "scaffolds.fasta:ATGACTGGTGAAGAGGTCCCTGAGAAGTACAGCGANNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNATGACTGGTGAAGAGGTCCCTGAGAAGTACAGCGAGAAGTGGCTT\n",
      "scaffolds.fasta:GGTCGCCAATGATGCAGCNNNNNNNNNNGCCCGCCGTACAGCTTGGTGCGCCCGTTGATT\n",
      "scaffolds.fasta:CTNNNNNNNNNNCTCGAAGCGGCGCTGTTCATCCATCGCGTTGGCAACGTCGAGGCCTTG\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNTTTGAACCCCATGAAGAGCGT\n",
      "scaffolds.fasta:GGTAGCTTAGTTGCATACATAATGGGCATTACNNNNNNNNNNGTATTTGAACGATTTACC\n",
      "scaffolds.fasta:CGGTTTGGNNNNNNNNNNATGTGGTGCGGTCAGGAGAAGTATATTATCTAGGAGAAGCAG\n",
      "scaffolds.fasta:NNNNNNNNNGCTTCACATATGTTGAAGAGAGTTAAGTTAAATCCTACAGCTCTAATAACA\n",
      "scaffolds.fasta:TTAATATTCTACCAGGCTCTTCTTGAATCNNNNNNNNNNAGTCTCAACTGGTATATAATT\n",
      "scaffolds.fasta:TTCGAGCCAGGTATGTTTCGAAACCGAACCTGTCAATGACGANNNNNNNNNNCTCTAACC\n",
      "scaffolds.fasta:ATCCTCTTGATCTCGAGGATGTCGCCCAGACACGATCGGTTACTACTCGCATGTTCNNNN\n",
      "scaffolds.fasta:NNNNNNCCTCGAGTTCGGAGAGGATGACCGGAAGTTCGTCGGACGCGTCGGGAGCTTCCT\n",
      "scaffolds.fasta:GCCGGTTNNNNNNNNNNCGCACCATGCTGCGTCTTCCGGGAGCCACTCGTCAGAAGTACG\n",
      "scaffolds.fasta:TTGGAAGAAATACATTTTCANNNNNNNNNNTTTTCAACTAACCTGGAACCAAAAAGTTTA\n",
      "scaffolds.fasta:TAGNNNNNNNNNNCCAGCAAAGTAATGCGTAACCGCCGTACGGAAAACCGCACTGTCGAA\n",
      "scaffolds.fasta:TGCCTTTAGTAATGATNNNNNNNNNNTATTAGACTTACTATCAAGATCTAATTGATCTAC\n",
      "scaffolds.fasta:CTTGCCATCGCAAGTCTTGNNNNNNNNNNTTCAATTCTTCAACTTCTGCTACTAATTGAT\n",
      "scaffolds.fasta:TACCAATCAGCGACAAGGTGAACGCGTAACAGCTTACGAAGTGGAAGTTNNNNNNNNNNG\n",
      "scaffolds.fasta:ATGGATCCGGAGTTTTTAACTATACAGTTAGACGATATAGAATATTGGGATNNNNNNNNN\n",
      "scaffolds.fasta:TTTACCAACATTATTTTGTTGTGAGATTGTCAAATGTGGTTGACCGTCTGCACNNNNNNN\n",
      "scaffolds.fasta:NNNCTTGTTGGCGTTTCATATTCAACATAGCCATTTGATTATTATTCATGTACCTATCAG\n",
      "scaffolds.fasta:CGATTTCAGCAATAGTTTCTACGACCTCATNNNNNNNNNNCTTATATCGAATTGAAAACC\n",
      "scaffolds.fasta:GGGCTTCGATATTATGTNNNNNNNNNNCTGAGTTTGTAGGGCTGTACTATACAGCTTACG\n",
      "scaffolds.fasta:TCGTGCATCCGTTTCATAATTGTTATAAGTCGTATACGNNNNNNNNNNCGAAATGGGTTG\n",
      "scaffolds.fasta:ACTGTATCTTTAGAGGGAGAAAACTCTTCTAAATATATGCTTTCATTAANNNNNNNNNNT\n",
      "scaffolds.fasta:ATGGAAATTGTAAATGAAATTTCTGAGCCAACTACAATTGAGATTCCTAANNNNNNNNNN\n",
      "scaffolds.fasta:TATAGCACCAGCAGCGATGCCCTGANNNNNNNNNNGCACTGAAACCATACCTGCCGATTG\n",
      "scaffolds.fasta:CAATCAGATCAAGACGGCCGTCGCCAAGCAGAACAAGCCCCNNNNNNNNNNGGTGAGAAG\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MER\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Check output K_MER=${K_MER}\"\n",
    "cd ${FILE_ID}-Assembly${K_MER}\n",
    "rep -c \">\" *fasta\n",
    "grep \">\" -m 8 contigs.fasta \n",
    "grep \">\" -m 8 scaffolds.fasta \n",
    "grep \"NN\" *fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### List of different K_MER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly with no specified M_MER\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [21, 33, 55]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.001     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.395     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.395     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.743     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.743     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.747    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.794    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.794    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.801    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.935    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.944    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.944    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:01.003    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.214    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.387    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.388    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.394    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.621    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.834    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.837    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.837    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.992    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.993    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.994    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.996    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.996    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.749    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17287 new k-mers.\n",
      "  0:00:06.388    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:07.062    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:07.194    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.649    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.707    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.738    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.738    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.740    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/corrected.yaml\n",
      "  0:00:07.740    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K21/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.175    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.290    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.436   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.460     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.593   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.600     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.723   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.730     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:01.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:01.027     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.030    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:01.030    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.041    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 373295 kmers in total.\n",
      "  0:00:01.041    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.044    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.044    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:01.129     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 373295 kmers\n",
      "  0:00:01.129     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 373295 kmers.\n",
      "  0:00:01.132    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 371539 kmers in total.\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.163    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.264    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.269    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 180880 bytes occupied (3.89472 bits per kmer).\n",
      "  0:00:01.269    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.317    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.317    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:01.317    12M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:01.317    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:01.413    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   34023 22-mers were removed by early tip clipper\n",
      "  0:00:01.413    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.413    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.485    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 11711 sequences extracted\n",
      "  0:00:01.520    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.533    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:00:01.542    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.542    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.542    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.577    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 174632 bytes occupied (3.7425 bits per kmer).\n",
      "  0:00:01.578    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.790    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.821    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 23423 edges\n",
      "  0:00:01.823    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.832    12M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 9\n",
      "  0:00:01.832    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 13.4339\n",
      "  0:00:01.832    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 9\n",
      "  0:00:01.833    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 9\n",
      "  0:00:01.833    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.833    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.834    12M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.834    12M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.834    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.836    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.836    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.837    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.838    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.840    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 109 times\n",
      "  0:00:01.840    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.857    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 77 times\n",
      "  0:00:01.858    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.862    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.906    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2844 times\n",
      "  0:00:01.906    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.907    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.907    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.908    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.909    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.910    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 269 times\n",
      "  0:00:01.916    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.916    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.917    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.920    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 69 times\n",
      "  0:00:01.920    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.920    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.921    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.923    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 29 times\n",
      "  0:00:01.923    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.923    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.924    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.924    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.925    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.925    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.927    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:01.928    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.928    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.929    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.929    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
      "  0:00:01.930    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.930    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.931    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.932    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 7 times\n",
      "  0:00:01.932    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.933    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.934    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.934    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.934    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.935    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.936    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.936    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.937    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.937    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.937    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.938    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:01.938    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.947    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.947    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.948    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.949    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.951    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.952    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.953    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.953    12M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.953    12M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.953    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.962    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1587 times\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 22.1978\n",
      "  0:00:01.962    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.962    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.964    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/before_rr.fasta\n",
      "  0:00:01.967    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:01.975    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:01.977    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/final_contigs.fasta\n",
      "  0:00:01.980    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.980    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.982    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/before_rr.fasta\n",
      "  0:00:01.984    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:01.992    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:01.994    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K21/final_contigs.fasta\n",
      "  0:00:01.997    12M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:02.000    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 2 seconds\n",
      "Max read length detected as 76\n",
      "\n",
      "== Running assembler: K33\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K33/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K33/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.256     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.256     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.297    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.297    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.311    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 381522 kmers in total.\n",
      "  0:00:00.312    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.316    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.317    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.317    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.317    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.318    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.318    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:00.392     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 381522 kmers\n",
      "  0:00:00.393     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 381522 kmers.\n",
      "  0:00:00.397    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.434    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 381759 kmers in total.\n",
      "  0:00:00.434    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.441    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.537    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.550    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185632 bytes occupied (3.89004 bits per kmer).\n",
      "  0:00:00.550    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.604    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.604    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:00.604    12M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:00.604    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:00.703    12M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   72893 34-mers were removed by early tip clipper\n",
      "  0:00:00.703    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:00.704    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.759    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 6128 sequences extracted\n",
      "  0:00:00.790    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.804    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:00:00.811    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:00.811    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.811    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.846    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 178448 bytes occupied (3.74181 bits per kmer).\n",
      "  0:00:00.847    12M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.023    12M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.054    12M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 12259 edges\n",
      "  0:00:01.057    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.060    12M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 8\n",
      "  0:00:01.060    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 11.3934\n",
      "  0:00:01.060    12M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 8\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 8\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.061    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.061    12M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.061    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.063    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.063    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.063    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.063    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.065    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 138 times\n",
      "  0:00:01.065    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.072    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 22 times\n",
      "  0:00:01.072    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.073    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.073    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1131 times\n",
      "  0:00:01.094    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:01.094    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.096    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.096    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 150 times\n",
      "  0:00:01.100    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.100    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 50 times\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.102    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:01.104    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:01.104    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.104    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.105    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.105    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.106    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.106    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.111    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.111    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.112    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.112    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.115    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.115    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.116    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.116    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.120    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.121    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.123    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.124    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.130    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.131    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.131    12M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.131    12M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.131    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.137    12M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1638 times\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 17.0928\n",
      "  0:00:01.137    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.137    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.139    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/before_rr.fasta\n",
      "  0:00:01.142    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:01.149    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:01.152    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/final_contigs.fasta\n",
      "  0:00:01.154    12M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:01.154    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:01.156    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/before_rr.fasta\n",
      "  0:00:01.159    12M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:01.167    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:01.169    12M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K33/final_contigs.fasta\n",
      "  0:00:01.172    12M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:01.173    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 1 seconds\n",
      "\n",
      "== Running assembler: K55\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K55/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/K55/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/dataset.info) with K=55\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.144     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.144     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.184    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.184    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.200    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 336055 kmers in total.\n",
      "  0:00:00.200    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.204    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.204    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:00.269     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 336055 kmers\n",
      "  0:00:00.269     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 336055 kmers.\n",
      "  0:00:00.275    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.302    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 339904 kmers in total.\n",
      "  0:00:00.302    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.309    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.401    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.409    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 166192 bytes occupied (3.9115 bits per kmer).\n",
      "  0:00:00.410    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:00.458    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:00.458    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:00.459    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:00.525    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 28030 sequences extracted\n",
      "  0:00:00.564    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:00.576    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:00.597    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:00.597    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.597    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.628    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 157384 bytes occupied (3.74662 bits per kmer).\n",
      "  0:00:00.629    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:00.734    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:00.774    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 56060 edges\n",
      "  0:00:00.776    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:00.808    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 6\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 5.21539\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 3\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 3\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:00.808    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:00.808    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:00.808    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:00.847     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 56060 edges\n",
      "  0:00:00.847     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 56060 sequences.\n",
      "  0:00:00.852    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.881    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 336055 kmers in total.\n",
      "  0:00:00.882    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.890    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.981    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.990    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 164400 bytes occupied (3.91365 bits per kmer).\n",
      "  0:00:00.997    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.035    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.036    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.057    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:01.127    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:01.128    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:01.135    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:01.163    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 47 candidates\n",
      "  0:00:01.171    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:01.172    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:01.173    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:01.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:01.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:01.179    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:01.179    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:01.180    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.180    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:01.180    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 12451 times\n",
      "  0:00:01.278    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.282    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.284    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.284    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.286    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:01.288    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2 times\n",
      "  0:00:01.289    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:01.289    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.293    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:01.293    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:01.300    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:01.300    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.300    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.301    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.301    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.338     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 7778 edges\n",
      "  0:00:01.338     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 7778 sequences.\n",
      "  0:00:01.341    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.370    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 279578 kmers in total.\n",
      "  0:00:01.370    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.376    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.463    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.470    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 138304 bytes occupied (3.95751 bits per kmer).\n",
      "  0:00:01.473    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.504    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.505    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.512    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:01.568    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:01.568    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:01.572    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:01.578    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 1 gaps after checking 145 candidates\n",
      "  0:00:01.589    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:01.590    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:01.724    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:01.729    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:01.738    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:01.742    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:01.745    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:01.746    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.749    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.749    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.755    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.755    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.761    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.761    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.762    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.762    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:01.762    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:01.767    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:01.772    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:01.778    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:01.782    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:01.783    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:01.783    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:01.784    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:01.784    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:01.790    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:01.790    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:01.791    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:01.791    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:01.792    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:01.792    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:01.793    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:01.793    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:01.795    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:01.795    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:01.795    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:01.797    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:01.854    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 3857 times\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 8.08614\n",
      "  0:00:01.854    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
      "  0:00:01.854    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:02.002    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:02.002    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.005    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/before_rr.fasta\n",
      "  0:00:02.007    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:02.015    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/simplified_contigs.fasta\n",
      "  0:00:02.017    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/intermediate_contigs.fasta\n",
      "  0:00:02.019    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 0 kmers to process\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 30396\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:02.019    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:02.045    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:02.189    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:03.404    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:03.404    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   17950 paired reads (42.9056% of all) aligned to long edges\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 396.253, deviation = 87.446, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:03.407    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:03.413    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.493    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:03.498    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:03.498    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:03.498    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:03.498    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 182\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:03.500    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:03.504    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 8; contradictional = 8\n",
      "  0:00:03.505    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 4; contradictional = 8\n",
      "  0:00:03.505    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:03.505    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 79\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:03.508    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:03.508    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:03.509    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 15 (0%)\n",
      "  0:00:03.511    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2 paths from 15 (13%)\n",
      "  0:00:03.511    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4 paths from 15 (26%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 15 (40%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 8 paths from 15 (53%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 10 paths from 15 (66%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 15 (80%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 14 paths from 15 (93%)\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:03.512    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:03.512    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:03.513    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:03.513    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.513    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.515    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/before_rr.fasta\n",
      "  0:00:03.517    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:03.526    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/final_contigs.paths\n",
      "  0:00:03.530    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly//K55/scaffolds.paths\n",
      "  0:00:03.533    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:03.534    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 3 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=25\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t25\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [25]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.000     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.378     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.378     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.733     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.734     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.741    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.790    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.791    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.798    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.925    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.942    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.942    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.988    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.353    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.530    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.530    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.538    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.655    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.775    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.778    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.778    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.931    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.932    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.932    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.553    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17290 new k-mers.\n",
      "  0:00:06.156    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.772    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.888    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.295    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.349    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.378    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.378    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.380    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/corrected.yaml\n",
      "  0:00:07.380    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K25\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/K25/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/K25/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/dataset.info) with K=25\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.174    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.291    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.435   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.460     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.583   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.589     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.711   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.719     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:00.993     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.993     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.996    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.996    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 378642 kmers in total.\n",
      "  0:00:01.008    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.011    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:01.077     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 378642 kmers\n",
      "  0:00:01.077     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 378642 kmers.\n",
      "  0:00:01.082    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 377541 kmers in total.\n",
      "  0:00:01.115    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.120    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.217    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.222    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 183648 bytes occupied (3.89146 bits per kmer).\n",
      "  0:00:01.223    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.276    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.276    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.276    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.344    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 32682 sequences extracted\n",
      "  0:00:01.383    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.397    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.420    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.420    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.420    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.457    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177120 bytes occupied (3.74222 bits per kmer).\n",
      "  0:00:01.458    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.664    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.705    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 65363 edges\n",
      "  0:00:01.707    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.744    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 8\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 11.2584\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 7\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 7\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.744    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.744    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.744    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 927213\n",
      "  0:00:01.792     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 65363 edges\n",
      "  0:00:01.793     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 65363 sequences.\n",
      "  0:00:01.796    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.826    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 378642 kmers in total.\n",
      "  0:00:01.827    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.832    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.933    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.939    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184088 bytes occupied (3.88944 bits per kmer).\n",
      "  0:00:01.942    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.979    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.979    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.005    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.122    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.122    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.127    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.161    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 2 gaps after checking 112 candidates\n",
      "  0:00:02.163    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.164    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.164    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.164    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.254    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 12086 times\n",
      "  0:00:02.254    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.273    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 59 times\n",
      "  0:00:02.273    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.276    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2209 times\n",
      "  0:00:02.318    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.320    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.320    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 276 times\n",
      "  0:00:02.326    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.327    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.327    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 54 times\n",
      "  0:00:02.329    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 49 times\n",
      "  0:00:02.331    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:02.331    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.331    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 18 times\n",
      "  0:00:02.332    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.333    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.333    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.333    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.334    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.334    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.336    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.336    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.336    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 927213\n",
      "  0:00:02.375     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3384 edges\n",
      "  0:00:02.375     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3384 sequences.\n",
      "  0:00:02.380    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.409    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 281787 kmers in total.\n",
      "  0:00:02.409    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.413    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.507    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.511    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 139344 bytes occupied (3.95601 bits per kmer).\n",
      "  0:00:02.522    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.545    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.621    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.621    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.622    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.625    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 56 candidates\n",
      "  0:00:02.631    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.631    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.700    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.702    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.705    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.709    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.710    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.710    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:02.711    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.713    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.713    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.714    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.716    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:02.716    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.717    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.718    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.718    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.718    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.719    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.721    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.723    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.725    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.729    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.731    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.731    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.732    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.733    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.781    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1607 times\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 20.4915\n",
      "  0:00:02.781    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.781    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 2630 kmers to process\n",
      "  0:00:02.782    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.021    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:03.024    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:03.024    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.024    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.026    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/before_rr.fasta\n",
      "  0:00:03.028    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:03.036    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/simplified_contigs.fasta\n",
      "  0:00:03.038    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/intermediate_contigs.fasta\n",
      "  0:00:03.041    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 2630 kmers to process\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23104\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.041    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.068    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.244    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.494    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.495    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   14296 paired reads (34.1715% of all) aligned to long edges\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 396.296, deviation = 87.6388, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.498    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.505    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.610    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.615    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.615    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.615    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.615    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.620    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.620    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 6870\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.622    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.631    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 115; contradictional = 104\n",
      "  0:00:04.632    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 12; contradictional = 12\n",
      "  0:00:04.633    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.633    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 292\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.652    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.652    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 43 (0%)\n",
      "  0:00:04.653    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5 paths from 43 (11%)\n",
      "  0:00:04.654    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 10 paths from 43 (23%)\n",
      "  0:00:04.658    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 43 (34%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 20 paths from 43 (46%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 25 paths from 43 (58%)\n",
      "  0:00:04.659    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 30 paths from 43 (69%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 35 paths from 43 (81%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 40 paths from 43 (93%)\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.660    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.660    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.660    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.661    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.661    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.662    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.662    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.664    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/before_rr.fasta\n",
      "  0:00:04.666    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:04.675    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/final_contigs.paths\n",
      "  0:00:04.679    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25//K25/scaffolds.paths\n",
      "  0:00:04.683    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.685    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 25 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly25/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.000     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.359     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.359     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.712    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.769    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.769    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.776    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.906    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.914    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.914    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.971    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.196    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.365    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.365    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.372    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.452    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.526    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.528    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.528    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.674    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.675    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.675    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.302    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17286 new k-mers.\n",
      "  0:00:05.924    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.583    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.583    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.584    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.584    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.704    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.106    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.161    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.191    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.192    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.194    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "  0:00:07.194    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.168    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.278    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.422   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.446     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.573   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.581     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.717   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.724     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.725     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.971     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.971     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.975    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.975    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.988    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:00.989    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.993    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.994    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.994    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.995    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.065     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 380324 kmers\n",
      "  0:00:01.066     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 380324 kmers.\n",
      "  0:00:01.070    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380925 kmers in total.\n",
      "  0:00:01.102    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.109    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.208    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.219    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 185224 bytes occupied (3.88998 bits per kmer).\n",
      "  0:00:01.219    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.266    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.267    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.267    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.358    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 31022 sequences extracted\n",
      "  0:00:01.403    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.419    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:01.442    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.442    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.443    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.479    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 177904 bytes occupied (3.74216 bits per kmer).\n",
      "  0:00:01.479    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.660    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.707    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 62044 edges\n",
      "  0:00:01.710    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.740    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 9.00852\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 6\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 6\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.740    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.740    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.740    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.784     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 62044 edges\n",
      "  0:00:01.784     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 62044 sequences.\n",
      "  0:00:01.789    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.824    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 380324 kmers in total.\n",
      "  0:00:01.824    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.831    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.932    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.942    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 184936 bytes occupied (3.89007 bits per kmer).\n",
      "  0:00:01.950    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.991    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.991    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.014    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.124    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.125    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.131    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.165    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 3 gaps after checking 107 candidates\n",
      "  0:00:02.173    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.179    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.179    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.179    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13586 times\n",
      "  0:00:02.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 17 times\n",
      "  0:00:02.283    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.285    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 774 times\n",
      "  0:00:02.299    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.301    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.301    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 110 times\n",
      "  0:00:02.304    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.304    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 94 times\n",
      "  0:00:02.308    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.308    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 42 times\n",
      "  0:00:02.311    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.311    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 17 times\n",
      "  0:00:02.312    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.314    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.314    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.314    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 4 times\n",
      "  0:00:02.315    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 5 times\n",
      "  0:00:02.316    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.319    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.324    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.324    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.324    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.360     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 3470 edges\n",
      "  0:00:02.360     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 3470 sequences.\n",
      "  0:00:02.364    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.392    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 266582 kmers in total.\n",
      "  0:00:02.392    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.398    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.490    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.497    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132168 bytes occupied (3.9663 bits per kmer).\n",
      "  0:00:02.501    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.525    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.525    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.527    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.601    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.601    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.604    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.606    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 96 candidates\n",
      "  0:00:02.663    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.663    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.717    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.719    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.721    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.726    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.736    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.736    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.737    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.737    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.743    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.744    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.744    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.744    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.746    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.747    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.750    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.752    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.753    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.753    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.756    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.757    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.758    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.758    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.759    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.760    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.801    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 1673 times\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 16.3895\n",
      "  0:00:02.801    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.801    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:02.802    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.005    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:03.006    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:03.006    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:03.006    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:03.007    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:03.009    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:03.011    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:03.018    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:03.020    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:03.023    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 1036 kmers to process\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 23095\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:03.023    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:03.049    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.210    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.441    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.442    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   16485 paired reads (39.4039% of all) aligned to long edges\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.556, deviation = 88.9493, left quantile = 307, right quantile = 505, read length = 76\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.447    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 307, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.454    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.566    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.571    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.571    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.571    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.575    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.575    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 3694\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.576    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.578    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 43; contradictional = 64\n",
      "  0:00:04.579    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 10; contradictional = 20\n",
      "  0:00:04.579    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.579    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.590    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 169\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.591    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.591    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 26 (0%)\n",
      "  0:00:04.592    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3 paths from 26 (11%)\n",
      "  0:00:04.594    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 26 (23%)\n",
      "  0:00:04.594    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 9 paths from 26 (34%)\n",
      "  0:00:04.595    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 26 (46%)\n",
      "  0:00:04.595    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 15 paths from 26 (57%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 26 (69%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 21 paths from 26 (80%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 26 (92%)\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.596    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.597    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.597    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.597    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.599    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/before_rr.fasta\n",
      "  0:00:04.602    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:04.609    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/final_contigs.paths\n",
      "  0:00:04.613    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35//K35/scaffolds.paths\n",
      "  0:00:04.616    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.618    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=45\n",
      "Command line: /home/microbioinf/miniconda3/bin/spades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\t--sc\t-k\t45\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Single-cell mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [45]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.001     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 930853\n",
      "  0:00:00.003     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.357     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 41839 reads\n",
      "  0:00:00.357     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 83678 reads\n",
      "  0:00:00.707     1G / 1G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 83678 reads processed\n",
      "  0:00:00.712    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.759    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 813404 kmers in total.\n",
      "  0:00:00.759    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.766    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:00.890    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:00.899    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 385536 bytes occupied (3.79183 bits per kmer).\n",
      "  0:00:00.900    12M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:00.950    28M / 1G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:02.140    28M / 1G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:02.324    28M / 1G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 680939\n",
      "  0:00:02.325    20M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:02.333    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq\n",
      "  0:00:03.401    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:04.457    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:04.461    40M / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 813404 kmers in total. Among them 395506 (48.6236%) are singletons.\n",
      "  0:00:04.462    40M / 1G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:04.602    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 2 non-read kmers were generated.\n",
      "  0:00:04.602    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 575510. Among them 506778 (88.0572%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 492. Among them 491 (99.7967%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 105402. Among them 95369 (90.4812%) are good\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.25703 kmers\n",
      "  0:00:04.603    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.00441\n",
      "  0:00:04.604    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 602638\n",
      "  0:00:04.604    40M / 1G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.9542,0.00963162,0.0188003,0.0173676),(0.0187514,0.945333,0.00912106,0.026795),(0.0256299,0.00902639,0.946196,0.0191482),(0.0171353,0.0182203,0.00943027,0.955214))\n",
      "  0:00:04.605    40M / 1G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:04.605    40M / 1G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:05.268    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 17286 new k-mers.\n",
      "  0:00:05.887    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 139 new k-mers.\n",
      "  0:00:06.521    40M / 1G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:06.521    40M / 1G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:06.522    40M / 1G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:06.522    40M / 1G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV_R1__kneaddata_paired_2.fastq\n",
      "  0:00:06.640    88M / 1G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 41839 reads.\n",
      "  0:00:07.064    96M / 1G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:07.119    96M / 1G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:07.150    40M / 1G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 2082 bases in 2045 reads.\n",
      "  0:00:07.150    40M / 1G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 6355000.\n",
      "  0:00:07.152    12M / 1G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/corrected.yaml\n",
      "  0:00:07.152    12M / 1G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K45\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/K45/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/K45/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/dataset.info) with K=45\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.175    84M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.289    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   32768 reads processed\n",
      "  0:00:00.435   100M / 100M  INFO    General                 (binary_converter.hpp      : 117)   41836 reads written\n",
      "  0:00:00.461     4M / 100M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.586   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   3 reads written\n",
      "  0:00:00.593     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.726   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 76\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 75.946\n",
      "  0:00:00.739     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.740     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.741     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.941     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 167350 reads\n",
      "  0:00:00.942     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:00.945    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 167350 reads\n",
      "  0:00:00.945    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:00.957    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 365508 kmers in total.\n",
      "  0:00:00.958    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:00.963    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:00.963    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.964    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:00.964    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:01.033     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 365508 kmers\n",
      "  0:00:01.034     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 365508 kmers.\n",
      "  0:00:01.039    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.074    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 367735 kmers in total.\n",
      "  0:00:01.075    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.081    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.181    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.192    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 179184 bytes occupied (3.89811 bits per kmer).\n",
      "  0:00:01.192    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:01.244    12M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:01.244    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:01.244    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:01.312    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 29619 sequences extracted\n",
      "  0:00:01.356    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:01.368    12M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:01.392    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:01.392    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.392    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.425    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 171064 bytes occupied (3.74414 bits per kmer).\n",
      "  0:00:01.426    16M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:01.564    16M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:01.611    16M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 59240 edges\n",
      "  0:00:01.613    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == EC Threshold Finding\n",
      "  0:00:01.641    16M / 1G    INFO   ThresholdFinder          (ec_threshold_finder.hpp   : 114)   Bucket size: 7\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  98)   Average edge coverage: 7.08441\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   :  99)   Graph threshold: 4\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 127)   EC coverage threshold value was calculated as 4\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (genomic_info_filler.cpp   : 128)   Trusted kmer low bound: 0\n",
      "  0:00:01.641    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:01.641    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:01.641    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:01.683     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 59240 edges\n",
      "  0:00:01.683     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 59240 sequences.\n",
      "  0:00:01.691    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.739    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 365508 kmers in total.\n",
      "  0:00:01.739    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.746    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:01.843    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:01.852    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 178104 bytes occupied (3.89822 bits per kmer).\n",
      "  0:00:01.863    28M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:01.899    28M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:01.899    28M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:01.921    28M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.008    28M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.008    28M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.015    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.047    28M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 144 candidates\n",
      "  0:00:02.052    28M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:02.053    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:02.053    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:02.053    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:02.058    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:02.058    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.058    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:02.058    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.160    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13887 times\n",
      "  0:00:02.160    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.162    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:02.162    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1 times\n",
      "  0:00:02.163    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.163    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 34 times\n",
      "  0:00:02.165    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.165    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 32 times\n",
      "  0:00:02.166    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.166    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 21 times\n",
      "  0:00:02.167    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.167    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 22 times\n",
      "  0:00:02.168    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.168    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 6 times\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 11 times\n",
      "  0:00:02.169    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.169    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:02.170    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 3 times\n",
      "  0:00:02.170    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:00:02.170    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.173    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:02.174    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:02.174    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:02.174    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:02.174    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:02.219     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 4828 edges\n",
      "  0:00:02.219     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 4828 sequences.\n",
      "  0:00:02.224    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.249    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 267502 kmers in total.\n",
      "  0:00:02.250    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.256    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.346    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.353    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 132568 bytes occupied (3.96462 bits per kmer).\n",
      "  0:00:02.360    24M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:02.379    24M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:02.379    24M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:02.383    24M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:02.449    24M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 41836 paired reads\n",
      "  0:00:02.449    24M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:02.453    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:02.459    24M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 43 gaps after checking 338 candidates\n",
      "  0:00:02.543    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 453)   Disconnection of relatively low covered edges disabled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 489)   Complex tip clipping disabled\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.543    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.611    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.614    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.618    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.621    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.622    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:02.622    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.626    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.626    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.627    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.627    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.628    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:02.628    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.629    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.629    24M / 1G    INFO    General                 (simplification.cpp        :  99)   Topology-based removal procedures\n",
      "  0:00:02.629    24M / 1G    INFO    General                 (single_cell_simplification:  47)   Removing connections based on topology\n",
      "  0:00:02.632    24M / 1G    INFO    General                 (single_cell_simplification:  96)   Removing connections based on topology and reliable coverage\n",
      "  0:00:02.634    24M / 1G    INFO    General                 (single_cell_simplification:  79)   Removing interstrand connections\n",
      "  0:00:02.638    24M / 1G    INFO    General                 (single_cell_simplification:  62)   Removing connections based on topological multiplicity counting\n",
      "  0:00:02.641    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Topology tip clipper\n",
      "  0:00:02.642    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Topology tip clipper triggered 0 times\n",
      "  0:00:02.642    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:02.643    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:02.643    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:02.647    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:02.648    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:02.648    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:02.649    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:02.649    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:02.650    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:02.650    24M / 1G    INFO    General                 (single_cell_simplification: 132)   Removing hidden erroneous connections\n",
      "  0:00:02.655    24M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:02.656    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:02.701    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 2284 times\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 12.2732\n",
      "  0:00:02.701    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 76 kmers to process\n",
      "  0:00:02.701    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 0 nucleotides\n",
      "  0:00:02.870    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:02.870    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:02.873    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/before_rr.fasta\n",
      "  0:00:02.875    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:02.882    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/simplified_contigs.fasta\n",
      "  0:00:02.885    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/intermediate_contigs.fasta\n",
      "  0:00:02.888    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 76 kmers to process\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 8231\n",
      "  0:00:02.889    20M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:02.890    20M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:02.916    84M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:03.077    84M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.302    84M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:04.302    84M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   15225 paired reads (36.3921% of all) aligned to long edges\n",
      "  0:00:04.305    20M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 395.263, deviation = 89.6726, left quantile = 305, right quantile = 505, read length = 76\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:04.306    20M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 305, right insert size quantile 505, filtering threshold 0, rounding threshold 0\n",
      "  0:00:04.312    32M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:04.408    32M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 41836 reads processed\n",
      "  0:00:04.414    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:04.414    20M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:04.414    20M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:04.414    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:04.417    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.418    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 992\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:04.419    20M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:04.421    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 28; contradictional = 8\n",
      "  0:00:04.423    20M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 3; contradictional = 10\n",
      "  0:00:04.423    20M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:04.423    20M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 361\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:04.442    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:04.442    20M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:04.443    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 57 (0%)\n",
      "  0:00:04.446    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6 paths from 57 (10%)\n",
      "  0:00:04.447    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 12 paths from 57 (21%)\n",
      "  0:00:04.448    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 18 paths from 57 (31%)\n",
      "  0:00:04.449    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 24 paths from 57 (42%)\n",
      "  0:00:04.450    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 30 paths from 57 (52%)\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 36 paths from 57 (63%)\n",
      "  0:00:04.451    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 42 paths from 57 (73%)\n",
      "  0:00:04.453    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 48 paths from 57 (84%)\n",
      "  0:00:04.453    20M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 54 paths from 57 (94%)\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.455    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.455    20M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 0 loops\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 298)   Sorting paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:04.456    20M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:04.456    20M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:04.457    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:04.457    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:04.459    20M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/before_rr.fasta\n",
      "  0:00:04.461    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:04.469    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/final_contigs.paths\n",
      "  0:00:04.473    20M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45//K45/scaffolds.paths\n",
      "  0:00:04.476    20M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:04.478    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 4 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 45 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/ECTV-Assembly45/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly with no specified K_MER\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -o ${FILE_ID}-Assembly${K_MER}\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "spades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--sc -k ${K_MER} -o ${FILE_ID}-Assembly${K_MER}\n",
    "done\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ",25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"\",\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "#### Check output K_MER=\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=25\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=35\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n",
      "\n",
      "#### Check output K_MER=45\n",
      ">NODE_1_length_92446_cov_8.851382\n",
      ">NODE_2_length_61624_cov_7.032646\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_13405_cov_7.334082\n",
      ">NODE_5_length_421_cov_1.073770\n",
      ">NODE_6_length_261_cov_1.072816\n",
      ">NODE_7_length_228_cov_1.942197\n",
      ">NODE_8_length_227_cov_0.610465\n",
      ">NODE_1_length_92877_cov_8.814516\n",
      ">NODE_2_length_75358_cov_7.054686\n",
      ">NODE_3_length_29587_cov_8.324665\n",
      ">NODE_4_length_261_cov_1.072816\n",
      ">NODE_5_length_227_cov_0.610465\n",
      ">NODE_6_length_90_cov_11.685714\n",
      ">NODE_7_length_79_cov_49.625000\n",
      "scaffolds.fasta:TNNNNNNNNNNTACCGCCATTATGGTGGCTAGTGATGTTTGTAAAAAAAATTTGGATTTA\n",
      "scaffolds.fasta:ATAGNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGTGGATCCGACGTTTCAACTATTG\n",
      "scaffolds.fasta:GTTACTGTTACTACTAAAGACTANNNNNNNNNNCTTTTAGAATGACGTCTTGTAATATCA\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "    echo\n",
    "    echo \"#### Check output K_MER=${K_MER}\"\n",
    "    cd ${FILE_ID}-Assembly${K_MER}\n",
    "    rep -c \">\" *fasta\n",
    "    grep \">\" -m 8 contigs.fasta \n",
    "    grep \">\" -m 8 scaffolds.fasta \n",
    "    grep \"NN\" *fasta\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Comparison of assemblies (*quast*)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compare assemblies FILE_ID=ECTV\n",
      "Processing ECTV-Assembly file...\n",
      "Processing ECTV-Assembly25 file...\n",
      "Processing ECTV-Assembly35 file...\n",
      "Processing ECTV-Assembly45 file...\n",
      "/home/microbioinf/miniconda3/lib/python2.7/site-packages/quast-5.0.2-py2.7.egg-info/scripts/quast.py contigs-ECTV-Assembly25.fasta contigs-ECTV-Assembly35.fasta contigs-ECTV-Assembly45.fasta contigs-ECTV-Assembly.fasta scaffolds-ECTV-Assembly25.fasta scaffolds-ECTV-Assembly35.fasta scaffolds-ECTV-Assembly45.fasta scaffolds-ECTV-Assembly.fasta -R ../ECTV-MoscowGenome.fasta\n",
      "\n",
      "Version: 5.0.2\n",
      "\n",
      "System information:\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-debian-buster-sid (linux_64)\n",
      "  Python version: 2.7.11\n",
      "  CPUs number: 3\n",
      "\n",
      "Started: 2019-04-26 13:12:24\n",
      "\n",
      "Logging to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/quast.log\n",
      "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "Main parameters: \n",
      "  MODE: default, threads: 1, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Reference:\n",
      "  /home/microbioinf/Documentos/Tema_3/ECTV-MoscowGenome.fasta ==> ECTV_MoscowGenome\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  contigs-ECTV-Assembly25.fasta ==> contigs_ECTV_Assembly25\n",
      "  2  contigs-ECTV-Assembly35.fasta ==> contigs_ECTV_Assembly35\n",
      "  3  contigs-ECTV-Assembly45.fasta ==> contigs_ECTV_Assembly45\n",
      "  4  contigs-ECTV-Assembly.fasta ==> contigs_ECTV_Assembly\n",
      "  5  scaffolds-ECTV-Assembly25.fasta ==> scaffolds_ECTV_Assembly25\n",
      "  6  scaffolds-ECTV-Assembly35.fasta ==> scaffolds_ECTV_Assembly35\n",
      "  7  scaffolds-ECTV-Assembly45.fasta ==> scaffolds_ECTV_Assembly45\n",
      "  8  scaffolds-ECTV-Assembly.fasta ==> scaffolds_ECTV_Assembly\n",
      "\n",
      "2019-04-26 13:12:24\n",
      "Running Basic statistics processor...\n",
      "  Reference genome:\n",
      "    ECTV_MoscowGenome.fasta, length = 209771, num fragments = 1, GC % = 33.18\n",
      "  Contig files: \n",
      "    1  contigs_ECTV_Assembly25\n",
      "    2  contigs_ECTV_Assembly35\n",
      "    3  contigs_ECTV_Assembly45\n",
      "    4  contigs_ECTV_Assembly\n",
      "    5  scaffolds_ECTV_Assembly25\n",
      "    6  scaffolds_ECTV_Assembly35\n",
      "    7  scaffolds_ECTV_Assembly45\n",
      "    8  scaffolds_ECTV_Assembly\n",
      "  Calculating N50 and L50...\n",
      "    1  contigs_ECTV_Assembly25, N50 = 37092, L50 = 2, Total length = 195922, GC % = 33.23, # N's per 100 kbp =  0.00\n",
      "    2  contigs_ECTV_Assembly35, N50 = 37406, L50 = 2, Total length = 195088, GC % = 33.21, # N's per 100 kbp =  0.00\n",
      "    3  contigs_ECTV_Assembly45, N50 = 35007, L50 = 3, Total length = 193001, GC % = 33.22, # N's per 100 kbp =  0.00\n",
      "    4  contigs_ECTV_Assembly, N50 = 61624, L50 = 2, Total length = 197062, GC % = 33.20, # N's per 100 kbp =  0.00\n",
      "    5  scaffolds_ECTV_Assembly25, N50 = 43137, L50 = 2, Total length = 196526, GC % = 33.22, # N's per 100 kbp =  157.74\n",
      "    6  scaffolds_ECTV_Assembly35, N50 = 60827, L50 = 2, Total length = 195465, GC % = 33.20, # N's per 100 kbp =  61.39\n",
      "    7  scaffolds_ECTV_Assembly45, N50 = 35007, L50 = 3, Total length = 194789, GC % = 33.23, # N's per 100 kbp =  186.87\n",
      "    8  scaffolds_ECTV_Assembly, N50 = 75358, L50 = 2, Total length = 197822, GC % = 33.21, # N's per 100 kbp =  56.11\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/Nx_plot.pdf\n",
      "  Drawing NGx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/NGx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly25_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly35_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly45_GC_content_plot.pdf\n",
      "  Drawing contigs_ECTV_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly25_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly35_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly45_GC_content_plot.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly25_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly35_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly45_coverage_histogram.pdf\n",
      "  Drawing contigs_ECTV_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/contigs_ECTV_Assembly_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly25_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly35_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly45_coverage_histogram.pdf\n",
      "  Drawing scaffolds_ECTV_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/basic_stats/scaffolds_ECTV_Assembly_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:29\n",
      "Running Contig analyzer...\n",
      "  1  contigs_ECTV_Assembly25\n",
      "  1  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly25.stdout and contigs_report_contigs_ECTV_Assembly25.stderr...\n",
      "  1  Aligning contigs to the reference\n",
      "  1  Analysis is finished.\n",
      "  2  contigs_ECTV_Assembly35\n",
      "  2  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly35.stdout and contigs_report_contigs_ECTV_Assembly35.stderr...\n",
      "  2  Aligning contigs to the reference\n",
      "  2  Analysis is finished.\n",
      "  3  contigs_ECTV_Assembly45\n",
      "  3  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly45.stdout and contigs_report_contigs_ECTV_Assembly45.stderr...\n",
      "  3  Aligning contigs to the reference\n",
      "  3  Analysis is finished.\n",
      "  4  contigs_ECTV_Assembly\n",
      "  4  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_contigs_ECTV_Assembly.stdout and contigs_report_contigs_ECTV_Assembly.stderr...\n",
      "  4  Aligning contigs to the reference\n",
      "  4  Analysis is finished.\n",
      "  5  scaffolds_ECTV_Assembly25\n",
      "  5  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly25.stdout and contigs_report_scaffolds_ECTV_Assembly25.stderr...\n",
      "  5  Aligning contigs to the reference\n",
      "  5  Analysis is finished.\n",
      "  6  scaffolds_ECTV_Assembly35\n",
      "  6  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly35.stdout and contigs_report_scaffolds_ECTV_Assembly35.stderr...\n",
      "  6  Aligning contigs to the reference\n",
      "  6  Analysis is finished.\n",
      "  7  scaffolds_ECTV_Assembly45\n",
      "  7  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly45.stdout and contigs_report_scaffolds_ECTV_Assembly45.stderr...\n",
      "  7  Aligning contigs to the reference\n",
      "  7  Analysis is finished.\n",
      "  8  scaffolds_ECTV_Assembly\n",
      "  8  Logging to files /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/contigs_report_scaffolds_ECTV_Assembly.stdout and contigs_report_scaffolds_ECTV_Assembly.stderr...\n",
      "  8  Aligning contigs to the reference\n",
      "  8  Analysis is finished.\n",
      "  Creating total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_report.txt, misassemblies_report.tsv, and misassemblies_report.tex\n",
      "  Transposed version of total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/transposed_report_misassemblies.txt, transposed_report_misassemblies.tsv, and transposed_report_misassemblies.tex\n",
      "  Creating total report...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/unaligned_report.txt, unaligned_report.tsv, and unaligned_report.tex\n",
      "  Drawing misassemblies by types plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_plot.pdf\n",
      "  Drawing misassemblies FRCurve plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/contigs_reports/misassemblies_frcurve_plot.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:31\n",
      "Running NA-NGA calculation...\n",
      "  1  contigs_ECTV_Assembly25, Largest alignment = 60903, NA50 = 37092, NGA50 = 23116, LA50 = 2, LGA50 = 3\n",
      "  2  contigs_ECTV_Assembly35, Largest alignment = 61098, NA50 = 37373, NGA50 = 29501, LA50 = 2, LGA50 = 3\n",
      "  3  contigs_ECTV_Assembly45, Largest alignment = 47868, NA50 = 29379, NGA50 = 29379, LA50 = 3, LGA50 = 3\n",
      "  4  contigs_ECTV_Assembly, Largest alignment = 92154, NA50 = 61624, NGA50 = 61624, LA50 = 2, LGA50 = 2\n",
      "  5  scaffolds_ECTV_Assembly25, Largest alignment = 60903, NA50 = 42859, NGA50 = 37092, LA50 = 2, LGA50 = 3\n",
      "  6  scaffolds_ECTV_Assembly35, Largest alignment = 61098, NA50 = 60827, NGA50 = 60827, LA50 = 2, LGA50 = 2\n",
      "  7  scaffolds_ECTV_Assembly45, Largest alignment = 59747, NA50 = 29379, NGA50 = 29379, LA50 = 3, LGA50 = 3\n",
      "  8  scaffolds_ECTV_Assembly, Largest alignment = 92542, NA50 = 75267, NGA50 = 75267, LA50 = 2, LGA50 = 2\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/cumulative_plot.pdf\n",
      "  Drawing NAx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/NAx_plot.pdf\n",
      "  Drawing NGAx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/aligned_stats/NGAx_plot.pdf\n",
      "Done.\n",
      "\n",
      "2019-04-26 13:12:31\n",
      "Running Genome analyzer...\n",
      "  NOTICE: No file with genomic features were provided. Use the --features option if you want to specify it.\n",
      "\n",
      "  NOTICE: No file with operons were provided. Use the -O option if you want to specify it.\n",
      "  1  contigs_ECTV_Assembly25\n",
      "  1  Analysis is finished.\n",
      "  2  contigs_ECTV_Assembly35\n",
      "  2  Analysis is finished.\n",
      "  3  contigs_ECTV_Assembly45\n",
      "  3  Analysis is finished.\n",
      "  4  contigs_ECTV_Assembly\n",
      "  4  Analysis is finished.\n",
      "  5  scaffolds_ECTV_Assembly25\n",
      "  5  Analysis is finished.\n",
      "  6  scaffolds_ECTV_Assembly35\n",
      "  6  Analysis is finished.\n",
      "  7  scaffolds_ECTV_Assembly45\n",
      "  7  Analysis is finished.\n",
      "  8  scaffolds_ECTV_Assembly\n",
      "  8  Analysis is finished.\n",
      "  Drawing Genome fraction, % histogram...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/genome_stats/genome_fraction_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2019-04-26 13:12:32\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2019-04-26 13:12:37\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.html\n",
      "  PDF version (tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/icarus.html\n",
      "  Log is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_13_12_24/quast.log\n",
      "\n",
      "Finished: 2019-04-26 13:12:37\n",
      "Elapsed time: 0:00:13.291234\n",
      "NOTICEs: 4; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MER\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MER=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compare assemblies FILE_ID=${FILE_ID}\"\n",
    "for assembly in ${FILE_ID}-Assembly*; \n",
    "    do echo \"Processing $assembly file...\"; \n",
    "    cp ${assembly}/contigs.fasta contigs-${assembly}.fasta\n",
    "    cp ${assembly}/scaffolds.fasta scaffolds-${assembly}.fasta\n",
    "done\n",
    "quast.py contigs* scaffolds* -R ../ECTV-MoscowGenome.fasta\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style  type=\"text/css\" >\n",
       "    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }</style><table id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30\" ><thead>    <tr>        <th class=\"col_heading level0 col0\" >Assembly</th>        <th class=\"col_heading level0 col1\" >contigs_ECTV_Assembly25</th>        <th class=\"col_heading level0 col2\" >contigs_ECTV_Assembly35</th>        <th class=\"col_heading level0 col3\" >contigs_ECTV_Assembly45</th>        <th class=\"col_heading level0 col4\" >contigs_ECTV_Assembly</th>        <th class=\"col_heading level0 col5\" >scaffolds_ECTV_Assembly25</th>        <th class=\"col_heading level0 col6\" >scaffolds_ECTV_Assembly35</th>        <th class=\"col_heading level0 col7\" >scaffolds_ECTV_Assembly45</th>        <th class=\"col_heading level0 col8\" >scaffolds_ECTV_Assembly</th>    </tr></thead><tbody>\n",
       "                <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col0\" class=\"data row0 col0\" ># contigs (>= 0 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col1\" class=\"data row0 col1\" >38</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col2\" class=\"data row0 col2\" >22</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col3\" class=\"data row0 col3\" >34</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col4\" class=\"data row0 col4\" >10</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col5\" class=\"data row0 col5\" >33</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col6\" class=\"data row0 col6\" >19</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col7\" class=\"data row0 col7\" >27</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row0_col8\" class=\"data row0 col8\" >7</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col0\" class=\"data row1 col0\" ># contigs (>= 1000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col1\" class=\"data row1 col1\" >11</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col2\" class=\"data row1 col2\" >8</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col3\" class=\"data row1 col3\" >13</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col4\" class=\"data row1 col4\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col5\" class=\"data row1 col5\" >9</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col6\" class=\"data row1 col6\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col7\" class=\"data row1 col7\" >10</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row1_col8\" class=\"data row1 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col0\" class=\"data row2 col0\" ># contigs (>= 5000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col1\" class=\"data row2 col1\" >8</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col2\" class=\"data row2 col2\" >7</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col3\" class=\"data row2 col3\" >9</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col4\" class=\"data row2 col4\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col5\" class=\"data row2 col5\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col6\" class=\"data row2 col6\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col7\" class=\"data row2 col7\" >8</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row2_col8\" class=\"data row2 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col0\" class=\"data row3 col0\" ># contigs (>= 10000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col1\" class=\"data row3 col1\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col2\" class=\"data row3 col2\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col3\" class=\"data row3 col3\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col4\" class=\"data row3 col4\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col5\" class=\"data row3 col5\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col6\" class=\"data row3 col6\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col7\" class=\"data row3 col7\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row3_col8\" class=\"data row3 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col0\" class=\"data row4 col0\" ># contigs (>= 25000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col1\" class=\"data row4 col1\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col2\" class=\"data row4 col2\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col3\" class=\"data row4 col3\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col4\" class=\"data row4 col4\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col5\" class=\"data row4 col5\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col6\" class=\"data row4 col6\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col7\" class=\"data row4 col7\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row4_col8\" class=\"data row4 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col0\" class=\"data row5 col0\" ># contigs (>= 50000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col1\" class=\"data row5 col1\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col2\" class=\"data row5 col2\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col3\" class=\"data row5 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col4\" class=\"data row5 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col5\" class=\"data row5 col5\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col6\" class=\"data row5 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col7\" class=\"data row5 col7\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row5_col8\" class=\"data row5 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col0\" class=\"data row6 col0\" >Total length (>= 0 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col1\" class=\"data row6 col1\" >198533</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col2\" class=\"data row6 col2\" >196511</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col3\" class=\"data row6 col3\" >196309</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col4\" class=\"data row6 col4\" >198368</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col5\" class=\"data row6 col5\" >198845</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col6\" class=\"data row6 col6\" >196617</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col7\" class=\"data row6 col7\" >196673</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row6_col8\" class=\"data row6 col8\" >198479</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col0\" class=\"data row7 col0\" >Total length (>= 1000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col1\" class=\"data row7 col1\" >193592</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col2\" class=\"data row7 col2\" >194143</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col3\" class=\"data row7 col3\" >190456</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col4\" class=\"data row7 col4\" >197062</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col5\" class=\"data row7 col5\" >194196</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col6\" class=\"data row7 col6\" >194520</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col7\" class=\"data row7 col7\" >192259</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row7_col8\" class=\"data row7 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col0\" class=\"data row8 col0\" >Total length (>= 5000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col1\" class=\"data row8 col1\" >186718</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col2\" class=\"data row8 col2\" >193083</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col3\" class=\"data row8 col3\" >182355</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col4\" class=\"data row8 col4\" >197062</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col5\" class=\"data row8 col5\" >187322</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col6\" class=\"data row8 col6\" >193460</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col7\" class=\"data row8 col7\" >187864</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row8_col8\" class=\"data row8 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col0\" class=\"data row9 col0\" >Total length (>= 10000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col1\" class=\"data row9 col1\" >171220</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col2\" class=\"data row9 col2\" >184029</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col3\" class=\"data row9 col3\" >158872</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col4\" class=\"data row9 col4\" >197062</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col5\" class=\"data row9 col5\" >179237</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col6\" class=\"data row9 col6\" >184406</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col7\" class=\"data row9 col7\" >164381</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row9_col8\" class=\"data row9 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col0\" class=\"data row10 col0\" >Total length (>= 25000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col1\" class=\"data row10 col1\" >97996</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col2\" class=\"data row10 col2\" >134152</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col3\" class=\"data row10 col3\" >118339</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col4\" class=\"data row10 col4\" >183657</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col5\" class=\"data row10 col5\" >141133</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col6\" class=\"data row10 col6\" >169725</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col7\" class=\"data row10 col7\" >131153</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row10_col8\" class=\"data row10 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col0\" class=\"data row11 col0\" >Total length (>= 50000 bp)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col1\" class=\"data row11 col1\" >60904</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col2\" class=\"data row11 col2\" >61627</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col3\" class=\"data row11 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col4\" class=\"data row11 col4\" >154070</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col5\" class=\"data row11 col5\" >60904</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col6\" class=\"data row11 col6\" >122454</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col7\" class=\"data row11 col7\" >59859</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row11_col8\" class=\"data row11 col8\" >168235</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col0\" class=\"data row12 col0\" ># contigs</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col1\" class=\"data row12 col1\" >14</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col2\" class=\"data row12 col2\" >9</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col3\" class=\"data row12 col3\" >16</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col4\" class=\"data row12 col4\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col5\" class=\"data row12 col5\" >12</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col6\" class=\"data row12 col6\" >7</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col7\" class=\"data row12 col7\" >13</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row12_col8\" class=\"data row12 col8\" >3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col0\" class=\"data row13 col0\" >Largest contig</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col1\" class=\"data row13 col1\" >60904</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col2\" class=\"data row13 col2\" >61627</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col3\" class=\"data row13 col3\" >47869</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col4\" class=\"data row13 col4\" >92446</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col5\" class=\"data row13 col5\" >60904</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col6\" class=\"data row13 col6\" >61627</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col7\" class=\"data row13 col7\" >59859</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row13_col8\" class=\"data row13 col8\" >92877</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col0\" class=\"data row14 col0\" >Total length</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col1\" class=\"data row14 col1\" >195922</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col2\" class=\"data row14 col2\" >195088</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col3\" class=\"data row14 col3\" >193001</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col4\" class=\"data row14 col4\" >197062</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col5\" class=\"data row14 col5\" >196526</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col6\" class=\"data row14 col6\" >195465</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col7\" class=\"data row14 col7\" >194789</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row14_col8\" class=\"data row14 col8\" >197822</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col0\" class=\"data row15 col0\" >Reference length</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col1\" class=\"data row15 col1\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col2\" class=\"data row15 col2\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col3\" class=\"data row15 col3\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col4\" class=\"data row15 col4\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col5\" class=\"data row15 col5\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col6\" class=\"data row15 col6\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col7\" class=\"data row15 col7\" >209771</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row15_col8\" class=\"data row15 col8\" >209771</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col0\" class=\"data row16 col0\" >GC (%)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col1\" class=\"data row16 col1\" >33.23</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col2\" class=\"data row16 col2\" >33.21</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col3\" class=\"data row16 col3\" >33.22</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col4\" class=\"data row16 col4\" >33.20</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col5\" class=\"data row16 col5\" >33.22</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col6\" class=\"data row16 col6\" >33.20</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col7\" class=\"data row16 col7\" >33.23</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row16_col8\" class=\"data row16 col8\" >33.21</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col0\" class=\"data row17 col0\" >Reference GC (%)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col1\" class=\"data row17 col1\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col2\" class=\"data row17 col2\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col3\" class=\"data row17 col3\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col4\" class=\"data row17 col4\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col5\" class=\"data row17 col5\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col6\" class=\"data row17 col6\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col7\" class=\"data row17 col7\" >33.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row17_col8\" class=\"data row17 col8\" >33.18</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col0\" class=\"data row18 col0\" >N50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col1\" class=\"data row18 col1\" >37092</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col2\" class=\"data row18 col2\" >37406</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col3\" class=\"data row18 col3\" >35007</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col4\" class=\"data row18 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col5\" class=\"data row18 col5\" >43137</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col6\" class=\"data row18 col6\" >60827</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col7\" class=\"data row18 col7\" >35007</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row18_col8\" class=\"data row18 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col0\" class=\"data row19 col0\" >NG50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col1\" class=\"data row19 col1\" >23129</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col2\" class=\"data row19 col2\" >35119</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col3\" class=\"data row19 col3\" >35007</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col4\" class=\"data row19 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col5\" class=\"data row19 col5\" >37092</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col6\" class=\"data row19 col6\" >60827</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col7\" class=\"data row19 col7\" >35007</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row19_col8\" class=\"data row19 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col0\" class=\"data row20 col0\" >N75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col1\" class=\"data row20 col1\" >14681</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col2\" class=\"data row20 col2\" >23130</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col3\" class=\"data row20 col3\" >11899</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col4\" class=\"data row20 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col5\" class=\"data row20 col5\" >23423</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col6\" class=\"data row20 col6\" >47271</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col7\" class=\"data row20 col7\" >20230</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row20_col8\" class=\"data row20 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col0\" class=\"data row21 col0\" >NG75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col1\" class=\"data row21 col1\" >14681</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col2\" class=\"data row21 col2\" >14681</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col3\" class=\"data row21 col3\" >11016</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col4\" class=\"data row21 col4\" >29587</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col5\" class=\"data row21 col5\" >23423</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col6\" class=\"data row21 col6\" >47271</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col7\" class=\"data row21 col7\" >12998</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row21_col8\" class=\"data row21 col8\" >75358</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col0\" class=\"data row22 col0\" >L50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col1\" class=\"data row22 col1\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col2\" class=\"data row22 col2\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col3\" class=\"data row22 col3\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col4\" class=\"data row22 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col5\" class=\"data row22 col5\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col6\" class=\"data row22 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col7\" class=\"data row22 col7\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row22_col8\" class=\"data row22 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col0\" class=\"data row23 col0\" >LG50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col1\" class=\"data row23 col1\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col2\" class=\"data row23 col2\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col3\" class=\"data row23 col3\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col4\" class=\"data row23 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col5\" class=\"data row23 col5\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col6\" class=\"data row23 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col7\" class=\"data row23 col7\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row23_col8\" class=\"data row23 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col0\" class=\"data row24 col0\" >L75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col1\" class=\"data row24 col1\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col2\" class=\"data row24 col2\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col3\" class=\"data row24 col3\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col4\" class=\"data row24 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col5\" class=\"data row24 col5\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col6\" class=\"data row24 col6\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col7\" class=\"data row24 col7\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row24_col8\" class=\"data row24 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col0\" class=\"data row25 col0\" >LG75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col1\" class=\"data row25 col1\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col2\" class=\"data row25 col2\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col3\" class=\"data row25 col3\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col4\" class=\"data row25 col4\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col5\" class=\"data row25 col5\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col6\" class=\"data row25 col6\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col7\" class=\"data row25 col7\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row25_col8\" class=\"data row25 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col0\" class=\"data row26 col0\" ># misassemblies</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col1\" class=\"data row26 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col2\" class=\"data row26 col2\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col3\" class=\"data row26 col3\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col4\" class=\"data row26 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col5\" class=\"data row26 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col6\" class=\"data row26 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col7\" class=\"data row26 col7\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row26_col8\" class=\"data row26 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col0\" class=\"data row27 col0\" ># misassembled contigs</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col1\" class=\"data row27 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col2\" class=\"data row27 col2\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col3\" class=\"data row27 col3\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col4\" class=\"data row27 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col5\" class=\"data row27 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col6\" class=\"data row27 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col7\" class=\"data row27 col7\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row27_col8\" class=\"data row27 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col0\" class=\"data row28 col0\" >Misassembled contigs length</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col1\" class=\"data row28 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col2\" class=\"data row28 col2\" >49800</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col3\" class=\"data row28 col3\" >35007</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col4\" class=\"data row28 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col5\" class=\"data row28 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col6\" class=\"data row28 col6\" >61952</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col7\" class=\"data row28 col7\" >48005</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row28_col8\" class=\"data row28 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col0\" class=\"data row29 col0\" ># local misassemblies</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col1\" class=\"data row29 col1\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col2\" class=\"data row29 col2\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col3\" class=\"data row29 col3\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col4\" class=\"data row29 col4\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col5\" class=\"data row29 col5\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col6\" class=\"data row29 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col7\" class=\"data row29 col7\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row29_col8\" class=\"data row29 col8\" >1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col0\" class=\"data row30 col0\" ># scaffold gap ext. mis.</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col1\" class=\"data row30 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col2\" class=\"data row30 col2\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col3\" class=\"data row30 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col4\" class=\"data row30 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col5\" class=\"data row30 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col6\" class=\"data row30 col6\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col7\" class=\"data row30 col7\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row30_col8\" class=\"data row30 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col0\" class=\"data row31 col0\" ># scaffold gap loc. mis.</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col1\" class=\"data row31 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col2\" class=\"data row31 col2\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col3\" class=\"data row31 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col4\" class=\"data row31 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col5\" class=\"data row31 col5\" >1</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col6\" class=\"data row31 col6\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col7\" class=\"data row31 col7\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row31_col8\" class=\"data row31 col8\" >1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col0\" class=\"data row32 col0\" ># unaligned mis. contigs</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col1\" class=\"data row32 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col2\" class=\"data row32 col2\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col3\" class=\"data row32 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col4\" class=\"data row32 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col5\" class=\"data row32 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col6\" class=\"data row32 col6\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col7\" class=\"data row32 col7\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row32_col8\" class=\"data row32 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col0\" class=\"data row33 col0\" ># unaligned contigs</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col1\" class=\"data row33 col1\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col2\" class=\"data row33 col2\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col3\" class=\"data row33 col3\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col4\" class=\"data row33 col4\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col5\" class=\"data row33 col5\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col6\" class=\"data row33 col6\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col7\" class=\"data row33 col7\" >0 + 0 part</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row33_col8\" class=\"data row33 col8\" >0 + 0 part</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col0\" class=\"data row34 col0\" >Unaligned length</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col1\" class=\"data row34 col1\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col2\" class=\"data row34 col2\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col3\" class=\"data row34 col3\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col4\" class=\"data row34 col4\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col5\" class=\"data row34 col5\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col6\" class=\"data row34 col6\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col7\" class=\"data row34 col7\" >0</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row34_col8\" class=\"data row34 col8\" >0</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col0\" class=\"data row35 col0\" >Genome fraction (%)</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col1\" class=\"data row35 col1\" >93.206</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col2\" class=\"data row35 col2\" >92.602</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col3\" class=\"data row35 col3\" >91.811</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col4\" class=\"data row35 col4\" >93.799</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col5\" class=\"data row35 col5\" >93.393</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col6\" class=\"data row35 col6\" >92.789</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col7\" class=\"data row35 col7\" >92.526</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row35_col8\" class=\"data row35 col8\" >94.122</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col0\" class=\"data row36 col0\" >Duplication ratio</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col1\" class=\"data row36 col1\" >1.002</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col2\" class=\"data row36 col2\" >1.004</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col3\" class=\"data row36 col3\" >1.002</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col4\" class=\"data row36 col4\" >1.002</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col5\" class=\"data row36 col5\" >1.003</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col6\" class=\"data row36 col6\" >1.004</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col7\" class=\"data row36 col7\" >1.004</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row36_col8\" class=\"data row36 col8\" >1.002</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col0\" class=\"data row37 col0\" ># N's per 100 kbp</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col1\" class=\"data row37 col1\" >0.00</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col2\" class=\"data row37 col2\" >0.00</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col3\" class=\"data row37 col3\" >0.00</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col4\" class=\"data row37 col4\" >0.00</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col5\" class=\"data row37 col5\" >157.74</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col6\" class=\"data row37 col6\" >61.39</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col7\" class=\"data row37 col7\" >186.87</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row37_col8\" class=\"data row37 col8\" >56.11</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col0\" class=\"data row38 col0\" ># mismatches per 100 kbp</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col1\" class=\"data row38 col1\" >90.53</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col2\" class=\"data row38 col2\" >93.18</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col3\" class=\"data row38 col3\" >90.35</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col4\" class=\"data row38 col4\" >98.60</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col5\" class=\"data row38 col5\" >90.86</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col6\" class=\"data row38 col6\" >95.56</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col7\" class=\"data row38 col7\" >93.25</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row38_col8\" class=\"data row38 col8\" >102.31</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col0\" class=\"data row39 col0\" ># indels per 100 kbp</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col1\" class=\"data row39 col1\" >29.66</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col2\" class=\"data row39 col2\" >32.43</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col3\" class=\"data row39 col3\" >30.63</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col4\" class=\"data row39 col4\" >34.56</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col5\" class=\"data row39 col5\" >30.63</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col6\" class=\"data row39 col6\" >33.91</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col7\" class=\"data row39 col7\" >34.00</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row39_col8\" class=\"data row39 col8\" >37.99</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col0\" class=\"data row40 col0\" >Largest alignment</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col1\" class=\"data row40 col1\" >60903</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col2\" class=\"data row40 col2\" >61098</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col3\" class=\"data row40 col3\" >47868</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col4\" class=\"data row40 col4\" >92154</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col5\" class=\"data row40 col5\" >60903</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col6\" class=\"data row40 col6\" >61098</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col7\" class=\"data row40 col7\" >59747</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row40_col8\" class=\"data row40 col8\" >92542</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col0\" class=\"data row41 col0\" >Total aligned length</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col1\" class=\"data row41 col1\" >195525</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col2\" class=\"data row41 col2\" >194248</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col3\" class=\"data row41 col3\" >192697</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col4\" class=\"data row41 col4\" >196686</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col5\" class=\"data row41 col5\" >195901</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col6\" class=\"data row41 col6\" >194671</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col7\" class=\"data row41 col7\" >194199</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row41_col8\" class=\"data row41 col8\" >197344</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col0\" class=\"data row42 col0\" >NA50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col1\" class=\"data row42 col1\" >37092</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col2\" class=\"data row42 col2\" >37373</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col3\" class=\"data row42 col3\" >29379</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col4\" class=\"data row42 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col5\" class=\"data row42 col5\" >42859</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col6\" class=\"data row42 col6\" >60827</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col7\" class=\"data row42 col7\" >29379</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row42_col8\" class=\"data row42 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col0\" class=\"data row43 col0\" >NGA50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col1\" class=\"data row43 col1\" >23116</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col2\" class=\"data row43 col2\" >29501</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col3\" class=\"data row43 col3\" >29379</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col4\" class=\"data row43 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col5\" class=\"data row43 col5\" >37092</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col6\" class=\"data row43 col6\" >60827</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col7\" class=\"data row43 col7\" >29379</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row43_col8\" class=\"data row43 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col0\" class=\"data row44 col0\" >NA75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col1\" class=\"data row44 col1\" >14348</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col2\" class=\"data row44 col2\" >23117</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col3\" class=\"data row44 col3\" >11016</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col4\" class=\"data row44 col4\" >61624</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col5\" class=\"data row44 col5\" >23410</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col6\" class=\"data row44 col6\" >41653</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col7\" class=\"data row44 col7\" >12657</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row44_col8\" class=\"data row44 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col0\" class=\"data row45 col0\" >NGA75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col1\" class=\"data row45 col1\" >14348</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col2\" class=\"data row45 col2\" >12424</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col3\" class=\"data row45 col3\" >9074</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col4\" class=\"data row45 col4\" >29535</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col5\" class=\"data row45 col5\" >23410</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col6\" class=\"data row45 col6\" >41653</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col7\" class=\"data row45 col7\" >12657</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row45_col8\" class=\"data row45 col8\" >75267</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col0\" class=\"data row46 col0\" >LA50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col1\" class=\"data row46 col1\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col2\" class=\"data row46 col2\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col3\" class=\"data row46 col3\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col4\" class=\"data row46 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col5\" class=\"data row46 col5\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col6\" class=\"data row46 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col7\" class=\"data row46 col7\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row46_col8\" class=\"data row46 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col0\" class=\"data row47 col0\" >LGA50</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col1\" class=\"data row47 col1\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col2\" class=\"data row47 col2\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col3\" class=\"data row47 col3\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col4\" class=\"data row47 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col5\" class=\"data row47 col5\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col6\" class=\"data row47 col6\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col7\" class=\"data row47 col7\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row47_col8\" class=\"data row47 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col0\" class=\"data row48 col0\" >LA75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col1\" class=\"data row48 col1\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col2\" class=\"data row48 col2\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col3\" class=\"data row48 col3\" >6</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col4\" class=\"data row48 col4\" >2</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col5\" class=\"data row48 col5\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col6\" class=\"data row48 col6\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col7\" class=\"data row48 col7\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row48_col8\" class=\"data row48 col8\" >2</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col0\" class=\"data row49 col0\" >LGA75</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col1\" class=\"data row49 col1\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col2\" class=\"data row49 col2\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col3\" class=\"data row49 col3\" >7</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col4\" class=\"data row49 col4\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col5\" class=\"data row49 col5\" >4</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col6\" class=\"data row49 col6\" >3</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col7\" class=\"data row49 col7\" >5</td>\n",
       "                        <td id=\"T_dafa847c_6826_11e9_88cb_b8e8562afb30row49_col8\" class=\"data row49 col8\" >2</td>\n",
       "            </tr>\n",
       "    </tbody></table>"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x11097ecc0>"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/quast*/latest/report.tsv\n",
    "EOT\n",
    "\"\"\"\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_quast = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_quast.style.hide_index().set_properties(**{'text-align': 'rigth', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"10pt\"}).\\\n",
    "set_properties(**{'text-align': 'left', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"10pt\"}, \\\n",
    "               subset=['Assembly'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Process (*metaspades*)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### List of different K_MER"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compute assembly with no specified K_MER\n",
      "Command line: /home/microbioinf/miniconda3/bin/metaspades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\t--meta\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Metagenomic mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [21, 33, 55]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Counting           (kmer_data.cpp             : 280)   Estimating k-mer count\n",
      "  0:00:00.024    52M / 52M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.780    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:01.407    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 294)   Processed 58620 reads\n",
      "  0:00:01.407    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 299)   Total 58620 reads processed\n",
      "  0:00:02.688    12M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 311)   Filtering singleton k-mers\n",
      "38 8 0\n",
      "nslots: 1073741824\n",
      "bits per slot: 8 range: 0000004000000000\n",
      "  0:00:02.688     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:12.932     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:15.554     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 326)   Processed 58620 reads\n",
      "  0:00:15.554     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 331)   Total 58620 reads processed\n",
      "  0:00:15.554     1G / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:15.554     1G / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:15.554     1G / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:15.554     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.190972 Gb\n",
      "  0:00:15.554     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 533997\n",
      "  0:00:15.554     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:17.380     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:19.093     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 58620 reads\n",
      "  0:00:19.094     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58620 reads processed\n",
      "  0:00:19.098     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:19.276     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4417004 kmers in total.\n",
      "  0:00:19.276     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:19.305     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:19.737     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:19.815     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2056288 bytes occupied (3.72431 bits per kmer).\n",
      "  0:00:19.927    12M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:20.165    84M / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:29.815    84M / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:30.586    84M / 2G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 4295701\n",
      "  0:00:30.588    48M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:30.623   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:34.081   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:37.143   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:37.156   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 4417004 kmers in total. Among them 288 (0.00652026%) are singletons.\n",
      "  0:00:37.156   152M / 2G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:37.401   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:37.401   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:37.401   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4182962. Among them 4144882 (99.0896%) are good\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 12322. Among them 12322 (100%) are good\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 107706. Among them 107322 (99.6435%) are good\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.17297 kmers\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.06465\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4264526\n",
      "  0:00:37.402   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.946036,0.00722882,0.0352043,0.0115306),(0.00629838,0.940463,0.00446536,0.0487731),(0.0325448,0.00267922,0.954793,0.00998316),(0.00782649,0.0229528,0.00466325,0.964557))\n",
      "  0:00:37.408   152M / 2G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:37.408   152M / 2G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:39.684   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 23343 new k-mers.\n",
      "  0:00:41.999   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 165 new k-mers.\n",
      "  0:00:44.347   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:44.347   152M / 2G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:44.348   152M / 2G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:44.348   152M / 2G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:44.558   216M / 2G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 29310 reads.\n",
      "  0:00:46.656   220M / 2G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:46.730   220M / 2G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:46.759   152M / 2G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 18214 bases in 10929 reads.\n",
      "  0:00:46.759   152M / 2G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 14550035.\n",
      "  0:00:46.768    12M / 2G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/corrected/corrected.yaml\n",
      "  0:00:46.768    12M / 2G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K21\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K21/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K21/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K21/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/dataset.info) with K=21\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.349    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.636   108M / 108M  INFO    General                 (binary_converter.hpp      : 117)   24974 reads written\n",
      "  0:00:00.658     4M / 108M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.813   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   1853 reads written\n",
      "  0:00:00.821     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.948   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.956     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:01.788     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:01.788     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.794    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.794    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.883    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4100728 kmers in total.\n",
      "  0:00:01.883    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.911    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.911    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.911    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.911    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.911    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.911    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:02.511     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 4100728 kmers\n",
      "  0:00:02.512     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 4100728 kmers.\n",
      "  0:00:02.518    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.623    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4099450 kmers in total.\n",
      "  0:00:02.623    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.650    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.063    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.143    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1909128 bytes occupied (3.72563 bits per kmer).\n",
      "  0:00:03.146    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:03.768    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:03.769    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:03.769    16M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:03.769    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:05.114    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   280649 22-mers were removed by early tip clipper\n",
      "  0:00:05.114    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:05.114    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:06.035    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 27824 sequences extracted\n",
      "  0:00:06.425    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:06.583    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 19 loops collected\n",
      "  0:00:06.605    28M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:06.605    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:06.605    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:06.896    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1903032 bytes occupied (3.71257 bits per kmer).\n",
      "  0:00:06.903    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:07.784    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:08.356    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 55644 edges\n",
      "  0:00:08.370    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:08.370    24M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:08.371    24M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:08.371    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:08.376    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:08.377    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:08.384    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 159 times\n",
      "  0:00:08.384    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:08.452    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 3555 times\n",
      "  0:00:08.452    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:08.461    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 1224 times\n",
      "  0:00:08.462    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:08.462    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:08.463    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:08.463    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:08.463    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.468    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 270 times\n",
      "  0:00:08.468    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.554    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2886 times\n",
      "  0:00:08.554    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.558    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.558    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:08.558    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.560    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 65 times\n",
      "  0:00:08.560    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 34 times\n",
      "  0:00:08.561    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 5 times\n",
      "  0:00:08.561    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.562    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
      "  0:00:08.562    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.563    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 34 times\n",
      "  0:00:08.563    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:08.563    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.565    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 4 times\n",
      "  0:00:08.565    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.566    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 5 times\n",
      "  0:00:08.566    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.568    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.568    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:08.568    16M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:08.568    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:08.568    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.570    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 28 times\n",
      "  0:00:08.570    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.576    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 17 times\n",
      "  0:00:08.576    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.581    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.581    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.589    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 1 times\n",
      "  0:00:08.589    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.590    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 9 times\n",
      "  0:00:08.590    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.592    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:08.592    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.594    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.595    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.596    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.596    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.598    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:08.598    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.600    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 2 times\n",
      "  0:00:08.600    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.603    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.603    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.611    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:08.611    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.613    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:08.613    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.614    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:08.614    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.617    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.617    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.618    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.618    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.620    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:08.620    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.620    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:08.620    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.623    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.623    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.632    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.632    16M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:08.636    16M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:08.640    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:08.641    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 82 times\n",
      "  0:00:08.641    16M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:08.644    16M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.38891\n",
      "  0:00:08.644    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:08.644    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:08.687    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/before_rr.fasta\n",
      "  0:00:08.739    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:08.884    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:08.935    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/final_contigs.fasta\n",
      "  0:00:08.986    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:08.986    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:09.035    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/before_rr.fasta\n",
      "  0:00:09.091    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/assembly_graph.fastg\n",
      "  0:00:09.241    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/simplified_contigs.fasta\n",
      "  0:00:09.330    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K21/final_contigs.fasta\n",
      "  0:00:09.386    16M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:09.403    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 9 seconds\n",
      "Max read length detected as 301\n",
      "\n",
      "== Running assembler: K33\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K33/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K33/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K33/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/dataset.info) with K=33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 0\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.865     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:00.865     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.398    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.398    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.541    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4088588 kmers in total.\n",
      "  0:00:01.542    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.590    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.591    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.591    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.592    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.592    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.593    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:02.333     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 4088588 kmers\n",
      "  0:00:02.334     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 4088588 kmers.\n",
      "  0:00:02.344    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.454    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4091084 kmers in total.\n",
      "  0:00:02.454    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.505    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.847    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.980    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1905296 bytes occupied (3.72575 bits per kmer).\n",
      "  0:00:02.982    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:03.662    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:03.662    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Early tip clipping\n",
      "  0:00:03.662    16M / 1G    INFO    General                 (construction.cpp          : 253)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:03.662    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 181)   Early tip clipping\n",
      "  0:00:04.884    16M / 1G    INFO   Early tip clipping       (early_simplification.hpp  : 184)   305770 34-mers were removed by early tip clipper\n",
      "  0:00:04.884    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:04.903    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:05.814    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 18444 sequences extracted\n",
      "  0:00:06.217    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:06.389    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 15 loops collected\n",
      "  0:00:06.404    20M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:06.404    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:06.404    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:06.692    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1897400 bytes occupied (3.71258 bits per kmer).\n",
      "  0:00:06.698    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:07.547    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:08.127    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 36910 edges\n",
      "  0:00:08.144    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:08.144    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:08.144    16M / 1G    INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:00:08.144    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:08.150    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:08.150    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:08.158    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 210 times\n",
      "  0:00:08.158    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:08.199    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 2059 times\n",
      "  0:00:08.199    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:08.205    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 1716 times\n",
      "  0:00:08.205    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:08.205    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:08.205    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:08.205    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:08.205    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.212    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 172 times\n",
      "  0:00:08.212    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.253    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1306 times\n",
      "  0:00:08.253    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.255    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.256    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:08.256    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.258    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 36 times\n",
      "  0:00:08.258    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.258    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.258    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.259    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 13 times\n",
      "  0:00:08.259    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:08.259    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.260    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:08.260    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.260    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.260    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.261    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 20 times\n",
      "  0:00:08.261    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:08.261    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:08.263    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.265    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 3 times\n",
      "  0:00:08.265    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.266    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.266    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:08.266    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.266    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.266    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.267    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.267    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.267    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.268    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:08.268    16M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:08.268    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:08.268    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.270    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 12 times\n",
      "  0:00:08.270    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.272    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 9 times\n",
      "  0:00:08.272    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.275    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 77 times\n",
      "  0:00:08.286    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 9 times\n",
      "  0:00:08.288    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.290    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:08.290    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.291    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.291    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.293    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:08.297    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 1 times\n",
      "  0:00:08.299    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.302    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.303    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.309    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:08.310    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.313    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:08.313    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.315    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.316    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:08.317    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:08.317    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:08.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:08.318    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:08.320    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:08.321    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:08.328    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:08.328    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.329    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:08.330    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:08.330    16M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:08.331    16M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:08.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:08.338    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 28 times\n",
      "  0:00:08.338    16M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:08.341    16M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.20316\n",
      "  0:00:08.341    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:08.341    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:08.379    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/before_rr.fasta\n",
      "  0:00:08.430    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:08.573    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:08.623    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/final_contigs.fasta\n",
      "  0:00:08.671    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:08.672    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:08.711    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/before_rr.fasta\n",
      "  0:00:08.764    16M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/assembly_graph.fastg\n",
      "  0:00:08.918    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/simplified_contigs.fasta\n",
      "  0:00:08.985    16M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K33/final_contigs.fasta\n",
      "  0:00:09.044    16M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:09.058    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 9 seconds\n",
      "\n",
      "== Running assembler: K55\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K55/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K55/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/K55/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/dataset.info) with K=55\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 1\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  59)   Binary reads detected\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:00.817     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:00.817     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.368    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.368    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.499    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3988975 kmers in total.\n",
      "  0:00:01.499    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.544    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.544    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.544    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.544    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.544    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.544    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:02.244     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 3988975 kmers\n",
      "  0:00:02.244     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 3988975 kmers.\n",
      "  0:00:02.253    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.367    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3994257 kmers in total.\n",
      "  0:00:02.367    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.414    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:02.760    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:02.834    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1860384 bytes occupied (3.72612 bits per kmer).\n",
      "  0:00:02.839    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:03.571    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:03.571    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:03.572    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:04.549    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 26460 sequences extracted\n",
      "  0:00:04.968    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:05.134    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:05.160    28M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:05.160    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:05.160    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:05.447    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1851224 bytes occupied (3.71268 bits per kmer).\n",
      "  0:00:05.454    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:06.214    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:06.833    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 52920 edges\n",
      "  0:00:06.849    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:06.849    20M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:06.849    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:06.849    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:06.849    20M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:06.849    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331163 Gb\n",
      "  0:00:06.849    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462999\n",
      "  0:00:07.169     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 52920 edges\n",
      "  0:00:07.169     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 52920 sequences.\n",
      "  0:00:07.174    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:07.272    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3988975 kmers in total.\n",
      "  0:00:07.272    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:07.320    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:07.658    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:07.755    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1857904 bytes occupied (3.72608 bits per kmer).\n",
      "  0:00:07.794   116M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:08.189   116M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:08.189   116M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:08.216   120M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:08.344   120M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:08.344   120M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:08.362   116M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:08.420   116M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 43 gaps after checking 3062 candidates\n",
      "  0:00:08.454   116M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:08.458    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:08.458    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:08.462    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 2 times\n",
      "  0:00:08.463    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:08.539    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 5292 times\n",
      "  0:00:08.539    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:08.548    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 307 times\n",
      "  0:00:08.548    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with low flanking coverage\n",
      "  0:00:08.560    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with low flanking coverage triggered 586 times\n",
      "  0:00:08.560    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:08.575    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 2668 times\n",
      "  0:00:08.575    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Simplification\n",
      "  0:00:08.575    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:08.576    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:08.576    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:08.576    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.596    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1710 times\n",
      "  0:00:08.597    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.640    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 254 times\n",
      "  0:00:08.640    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.643    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.643    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:08.643    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.644    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 19 times\n",
      "  0:00:08.644    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.644    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.645    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.658    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 573 times\n",
      "  0:00:08.658    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:08.658    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.659    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 42 times\n",
      "  0:00:08.659    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.660    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 3 times\n",
      "  0:00:08.660    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.663    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 63 times\n",
      "  0:00:08.663    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:08.663    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.667    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:08.667    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.674    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:08.674    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.677    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:08.677    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:08.677    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:08.677    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:08.679    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:08.679    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:08.679    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:08.679    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:08.679    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:08.919     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 10980 edges\n",
      "  0:00:08.920     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 10980 sequences.\n",
      "  0:00:08.924    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:09.010    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 2989682 kmers in total.\n",
      "  0:00:09.010    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:09.047    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:09.308    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:09.383    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1394752 bytes occupied (3.73217 bits per kmer).\n",
      "  0:00:09.465    92M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:09.718    92M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:09.719    92M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:09.726    96M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:10.082   100M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:10.082   100M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:10.093   100M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:10.106    96M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 9 gaps after checking 726 candidates\n",
      "  0:00:10.137    96M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Paired Information Counting\n",
      "  0:00:10.172    96M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 37208 kmers to process\n",
      "  0:00:10.187    96M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:10.187    96M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:10.187    96M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:10.188    96M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:10.215   160M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:10.665   160M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:11.909   160M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:11.910   160M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   13114 paired reads (52.5106% of all) aligned to long edges\n",
      "  0:00:11.914    96M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 521.419, deviation = 170.64, left quantile = 313, right quantile = 751, read length = 301\n",
      "  0:00:11.978   288M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:11.979   288M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:12.362   288M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:12.362   288M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:12.363   288M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:12.363   288M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 313, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:12.370   300M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:12.756   304M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:12.769   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Distance Estimation\n",
      "  0:00:12.770   100M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:12.770   100M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:12.770   100M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:12.781   100M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:12.781   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 13206\n",
      "  0:00:12.793   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:12.793   100M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:12.796   100M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:12.796   100M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:13.290   100M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 161; contradictional = 0\n",
      "  0:00:13.758   100M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 7; contradictional = 0\n",
      "  0:00:13.758   100M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:13.758   100M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:13.773   100M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:13.773   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 11128\n",
      "  0:00:13.779   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:13.779   100M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:13.781   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Repeat Resolving\n",
      "  0:00:13.781   100M / 1G    INFO    General                 (repeat_resolving.cpp      :  56)   Setting up preliminary path extend settings\n",
      "  0:00:13.781   100M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:13.781   100M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:13.801   124M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:13.806   124M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:13.806   124M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:13.807   124M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:13.807   124M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:13.808   124M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 5473 (0%)\n",
      "  0:00:13.830   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 5473 (2%)\n",
      "  0:00:13.843   128M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 5473 (4%)\n",
      "  0:00:13.862   128M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 5473 (9%)\n",
      "  0:00:13.865   128M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 548 paths from 5473 (10%)\n",
      "  0:00:13.887   132M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 5473 (18%)\n",
      "  0:00:13.890   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1096 paths from 5473 (20%)\n",
      "  0:00:13.914   136M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1644 paths from 5473 (30%)\n",
      "  0:00:13.931   140M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 5473 (37%)\n",
      "  0:00:13.937   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2192 paths from 5473 (40%)\n",
      "  0:00:13.960   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2740 paths from 5473 (50%)\n",
      "  0:00:13.980   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3288 paths from 5473 (60%)\n",
      "  0:00:13.999   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3836 paths from 5473 (70%)\n",
      "  0:00:14.008   156M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 5473 (74%)\n",
      "  0:00:14.018   160M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4384 paths from 5473 (80%)\n",
      "  0:00:14.050   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4932 paths from 5473 (90%)\n",
      "  0:00:14.106   168M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:14.106   168M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:14.119   168M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:14.120   168M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:14.127   168M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:14.127   168M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:14.146   168M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:14.152   172M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:14.177   172M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:00:14.177   172M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:14.177   172M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:14.181   172M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:14.181   172M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:14.189   172M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:14.203   172M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:14.227   144M / 1G    INFO    General                 (repeat_resolving.cpp      :  75)   Restoring initial path extend settings\n",
      "  0:00:14.227   144M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:14.227   144M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:14.265   140M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/before_rr.fasta\n",
      "  0:00:14.313   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:14.471   144M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/final_contigs.paths\n",
      "  0:00:14.554   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/scaffolds.paths\n",
      "  0:00:14.633   144M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Second Phase Setup\n",
      "  0:00:14.633   144M / 1G    INFO    General                 (second_phase_setup.cpp    :  20)   Preparing second phase\n",
      "  0:00:14.656   100M / 1G    INFO    General                 (second_phase_setup.cpp    :  28)   Moving preliminary contigs from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/final_contigs.fasta to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/first_pe_contigs.fasta\n",
      "  0:00:14.656   100M / 1G    INFO    General                 (second_phase_setup.cpp    :  40)   Ready to run second phase\n",
      "  0:00:14.656   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:14.660    20M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:14.660    20M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:14.660    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:14.660    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:14.665    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:14.665    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:14.705    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 152 times\n",
      "  0:00:14.705    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:14.707    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:14.707    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:14.708    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:14.708    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:14.713    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:14.713    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:14.715    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:14.715    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:14.715    20M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:14.715    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:14.715    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:14.715    20M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:14.715    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331163 Gb\n",
      "  0:00:14.715    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462999\n",
      "  0:00:14.962     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 10050 edges\n",
      "  0:00:14.962     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 10050 sequences.\n",
      "  0:00:14.967    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:15.055    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 2974929 kmers in total.\n",
      "  0:00:15.056    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:15.091    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:15.359    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:15.413    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1387816 bytes occupied (3.73203 bits per kmer).\n",
      "  0:00:15.461    96M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:15.735    96M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:15.735    96M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:15.744    96M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:16.146   100M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:16.146   100M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:16.159    96M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:16.173    96M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 0 gaps after checking 698 candidates\n",
      "  0:00:16.175    96M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:16.176    96M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:16.176    96M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:16.176    96M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:16.177    96M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.298    96M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 5 times\n",
      "  0:00:16.299    96M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.350    96M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 3 times\n",
      "  0:00:16.351    96M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.360    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.361    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.695    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 45 times\n",
      "  0:00:16.696    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.699    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.699    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.702    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.702    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.704    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.705    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.881    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 17 times\n",
      "  0:00:16.882    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:16.883    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:16.883    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.885    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:16.885    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.887    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:16.887    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.889    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.890    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.906    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 2 times\n",
      "  0:00:16.907    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.916    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 3 times\n",
      "  0:00:16.916    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.919    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.919    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.921    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.921    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.927    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 1 times\n",
      "  0:00:16.927    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:16.930    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:16.930    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.932    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:16.932    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.932    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:16.932    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.938    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.938    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:16.948    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:16.948    92M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:16.954    92M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:16.958    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:16.994    92M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 197 times\n",
      "  0:00:16.995    92M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:16.998    92M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 2.97508\n",
      "  0:00:16.999    92M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:16.999    92M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 100744 kmers to process\n",
      "  0:00:17.047    92M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:18.380    96M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:18.402    92M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:18.403    92M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 2 nucleotides\n",
      "  0:00:18.403    92M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:18.403    92M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:18.441    92M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/before_rr.fasta\n",
      "  0:00:18.493    92M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:18.633    92M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/simplified_contigs.fasta\n",
      "  0:00:18.688    92M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/intermediate_contigs.fasta\n",
      "  0:00:18.737    92M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:18.772    96M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 100962 kmers to process\n",
      "  0:00:18.820    96M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:18.820    96M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:18.820    96M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:18.820    96M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:18.846   160M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:19.374   160M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:20.598   160M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:20.598   160M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   14716 paired reads (58.9253% of all) aligned to long edges\n",
      "  0:00:20.601    96M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 523.152, deviation = 170.769, left quantile = 313, right quantile = 751, read length = 301\n",
      "  0:00:20.665   288M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:20.665   288M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.106   288M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:21.106   288M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:21.106   288M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:21.106   288M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 313, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:21.113   300M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.531   308M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:21.543   104M / 1G    INFO    General                 (pair_info_count.cpp       : 327)   Mapping contigs library #1\n",
      "  0:00:21.543   104M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.778   104M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5088 reads processed\n",
      "  0:00:21.778   104M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:21.778   104M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:21.778   104M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:21.778   104M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:21.786   104M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:21.786   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 9598\n",
      "  0:00:21.794   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:21.794   104M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:21.795   104M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:21.795   104M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:22.162   104M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 37; contradictional = 0\n",
      "  0:00:22.400   104M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 3; contradictional = 0\n",
      "  0:00:22.400   104M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:22.400   104M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:22.412   104M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:22.412   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 9336\n",
      "  0:00:22.416   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:22.416   104M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:22.416   104M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:22.416   104M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:22.417   104M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:22.432   120M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:22.438   120M / 1G    INFO    General                 (extenders_logic.cpp       :  34)   resolvable_repeat_length_bound set to 10000\n",
      "  0:00:22.443   120M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:22.444   120M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:22.444   120M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 1 single read library\n",
      "  0:00:22.448   120M / 1G    INFO    General                 (launcher.cpp              : 359)   Filling backbone edges for long reads scaffolding...\n",
      "  0:00:22.448   120M / 1G    INFO    General                 (launcher.cpp              : 361)    with long reads paths\n",
      "  0:00:22.449   120M / 1G    INFO    General                 (launcher.cpp              : 370)   Removing fake unique with paired-end libs\n",
      "  0:00:22.449   120M / 1G    INFO    General                 (launcher.cpp              : 381)   0 unique edges\n",
      "  0:00:22.449   120M / 1G    INFO    General                 (extenders_logic.cpp       : 386)   Using 0 long reads scaffolding libraries\n",
      "  0:00:22.449   120M / 1G    INFO    General                 (extenders_logic.cpp       : 396)   Using additional coordinated coverage extender\n",
      "  0:00:22.450   120M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 5\n",
      "  0:00:22.450   120M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 4594 (0%)\n",
      "  0:00:22.477   120M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 4594 (2%)\n",
      "  0:00:22.491   120M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 4594 (5%)\n",
      "  0:00:22.511   124M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 460 paths from 4594 (10%)\n",
      "  0:00:22.518   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 4594 (11%)\n",
      "  0:00:22.548   128M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 920 paths from 4594 (20%)\n",
      "  0:00:22.554   128M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 4594 (22%)\n",
      "  0:00:22.574   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1380 paths from 4594 (30%)\n",
      "  0:00:22.602   136M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1840 paths from 4594 (40%)\n",
      "  0:00:22.613   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 4594 (44%)\n",
      "  0:00:22.626   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2300 paths from 4594 (50%)\n",
      "  0:00:22.649   144M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2760 paths from 4594 (60%)\n",
      "  0:00:22.672   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3220 paths from 4594 (70%)\n",
      "  0:00:22.699   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3680 paths from 4594 (80%)\n",
      "  0:00:22.722   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 4594 (89%)\n",
      "  0:00:22.725   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4140 paths from 4594 (90%)\n",
      "  0:00:22.757   156M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:22.758   156M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:22.770   156M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:22.770   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:22.770   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:22.770   156M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:22.775   156M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:22.781   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:22.782   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:22.784   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:22.795   156M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:22.795   156M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:22.811   156M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:22.817   160M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:22.841   160M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 3 loops\n",
      "  0:00:22.842   160M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:22.842   160M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:22.844   160M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:22.844   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:22.844   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:22.844   160M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:22.851   160M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:22.858   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:22.858   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:22.861   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:22.867   160M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:22.879   160M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:22.900   136M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:22.900   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:22.937   136M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/before_rr.fasta\n",
      "  0:00:22.986   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/assembly_graph.fastg\n",
      "  0:00:23.153   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/final_contigs.paths\n",
      "  0:00:23.233   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly//K55/scaffolds.paths\n",
      "  0:00:23.312   136M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:23.369    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 23 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 21, 33, 55 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=25\n",
      "Command line: /home/microbioinf/miniconda3/bin/metaspades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\t--meta\t-k\t25\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Metagenomic mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [25]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Counting           (kmer_data.cpp             : 280)   Estimating k-mer count\n",
      "  0:00:00.022    52M / 52M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.764    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:01.467    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 294)   Processed 58620 reads\n",
      "  0:00:01.467    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 299)   Total 58620 reads processed\n",
      "  0:00:02.787    12M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 311)   Filtering singleton k-mers\n",
      "38 8 0\n",
      "nslots: 1073741824\n",
      "bits per slot: 8 range: 0000004000000000\n",
      "  0:00:02.787     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:12.288     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:14.896     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 326)   Processed 58620 reads\n",
      "  0:00:14.896     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 331)   Total 58620 reads processed\n",
      "  0:00:14.896     1G / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:14.896     1G / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:14.896     1G / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:14.896     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.190972 Gb\n",
      "  0:00:14.896     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 533997\n",
      "  0:00:14.898     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:16.814     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:18.579     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 58620 reads\n",
      "  0:00:18.580     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58620 reads processed\n",
      "  0:00:18.588     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:18.768     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4417004 kmers in total.\n",
      "  0:00:18.770     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:18.798     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:19.230     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:19.277     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2056288 bytes occupied (3.72431 bits per kmer).\n",
      "  0:00:19.409    12M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:19.643    84M / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:29.767    84M / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:30.578    84M / 2G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 4295701\n",
      "  0:00:30.579    48M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:30.614   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:34.375   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:37.553   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:37.565   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 4417004 kmers in total. Among them 288 (0.00652026%) are singletons.\n",
      "  0:00:37.566   152M / 2G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:37.814   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:37.815   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:37.815   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4182962. Among them 4144882 (99.0896%) are good\n",
      "  0:00:37.815   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 12322. Among them 12322 (100%) are good\n",
      "  0:00:37.816   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 107706. Among them 107322 (99.6435%) are good\n",
      "  0:00:37.816   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.17297 kmers\n",
      "  0:00:37.816   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.06465\n",
      "  0:00:37.816   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4264526\n",
      "  0:00:37.816   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.946036,0.00722882,0.0352043,0.0115306),(0.00629838,0.940463,0.00446536,0.0487731),(0.0325448,0.00267922,0.954793,0.00998316),(0.00782649,0.0229528,0.00466325,0.964557))\n",
      "  0:00:37.824   152M / 2G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:37.825   152M / 2G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:40.124   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 23343 new k-mers.\n",
      "  0:00:42.425   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 165 new k-mers.\n",
      "  0:00:44.756   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:44.756   152M / 2G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:44.756   152M / 2G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:44.756   152M / 2G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:44.976   216M / 2G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 29310 reads.\n",
      "  0:00:47.161   220M / 2G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:47.240   220M / 2G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:47.267   152M / 2G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 18214 bases in 10929 reads.\n",
      "  0:00:47.267   152M / 2G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 14550035.\n",
      "  0:00:47.275    12M / 2G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/corrected/corrected.yaml\n",
      "  0:00:47.275    12M / 2G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K25\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/K25/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/K25/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/K25/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/dataset.info) with K=25\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 1\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.344    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.638   108M / 108M  INFO    General                 (binary_converter.hpp      : 117)   24974 reads written\n",
      "  0:00:00.659     4M / 108M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.808   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   1853 reads written\n",
      "  0:00:00.814     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.943   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.949     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 4964554\n",
      "  0:00:01.793     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:01.794     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.799    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.799    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.903    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4100246 kmers in total.\n",
      "  0:00:01.903    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:01.929    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:01.929    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:01.929    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:01.929    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:01.929    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:01.929    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 928426\n",
      "  0:00:02.557     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 4100246 kmers\n",
      "  0:00:02.557     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 4100246 kmers.\n",
      "  0:00:02.566    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.688    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4101167 kmers in total.\n",
      "  0:00:02.688    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.716    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.101    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.146    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1909944 bytes occupied (3.72566 bits per kmer).\n",
      "  0:00:03.152    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:03.812    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:03.812    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:03.812    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:04.838    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 36126 sequences extracted\n",
      "  0:00:05.271    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:05.448    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:05.476    28M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:05.476    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:05.476    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:05.795    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1902816 bytes occupied (3.71259 bits per kmer).\n",
      "  0:00:05.802    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:06.712    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:07.386    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 72229 edges\n",
      "  0:00:07.399    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:07.399    24M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:07.399    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:07.399    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:07.399    24M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:07.399    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.330729 Gb\n",
      "  0:00:07.399    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 924785\n",
      "  0:00:07.752     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 72229 edges\n",
      "  0:00:07.752     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 72229 sequences.\n",
      "  0:00:07.757    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:07.861    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4100246 kmers in total.\n",
      "  0:00:07.861    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:07.888    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:08.232    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:08.282    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1909496 bytes occupied (3.72562 bits per kmer).\n",
      "  0:00:08.348   120M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:08.757   120M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:08.757   120M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:08.786   124M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:08.944   124M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:08.944   124M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:08.959   124M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:09.007   124M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 58 gaps after checking 3089 candidates\n",
      "  0:00:09.049   120M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:09.054    24M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:09.054    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:09.060    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:09.060    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:09.118    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 4879 times\n",
      "  0:00:09.118    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:09.124    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
      "  0:00:09.124    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with low flanking coverage\n",
      "  0:00:09.127    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with low flanking coverage triggered 2 times\n",
      "  0:00:09.127    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:09.133    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 1392 times\n",
      "  0:00:09.133    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Simplification\n",
      "  0:00:09.134    20M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:09.134    20M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:09.134    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:09.134    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.153    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1860 times\n",
      "  0:00:09.153    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.304    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1497 times\n",
      "  0:00:09.304    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.310    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.310    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:09.310    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.311    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 41 times\n",
      "  0:00:09.311    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.311    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:09.311    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.354    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 2344 times\n",
      "  0:00:09.354    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:09.354    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.356    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 94 times\n",
      "  0:00:09.356    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.381    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 46 times\n",
      "  0:00:09.381    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.388    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 310 times\n",
      "  0:00:09.388    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:09.388    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.391    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 8 times\n",
      "  0:00:09.391    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.400    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 10 times\n",
      "  0:00:09.400    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.404    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.404    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:09.404    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.330729 Gb\n",
      "  0:00:09.404    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 924785\n",
      "  0:00:09.635     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 17169 edges\n",
      "  0:00:09.635     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 17169 sequences.\n",
      "  0:00:09.640    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:09.725    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3261913 kmers in total.\n",
      "  0:00:09.725    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:09.745    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:10.042    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:10.077    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1521016 bytes occupied (3.73037 bits per kmer).\n",
      "  0:00:10.114   108M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:10.403   108M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:10.403   108M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:10.411   108M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:10.779   108M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:10.779   108M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:10.790   108M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:10.803   108M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 12 gaps after checking 1063 candidates\n",
      "  0:00:10.816   108M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Paired Information Counting\n",
      "  0:00:10.879   112M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 134420 kmers to process\n",
      "  0:00:10.929   112M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:10.929   112M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:10.929   112M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:10.929   112M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:10.956   176M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:11.408   176M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:12.668   176M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:12.668   176M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   11241 paired reads (45.0108% of all) aligned to long edges\n",
      "  0:00:12.672   112M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 519.533, deviation = 170.608, left quantile = 313, right quantile = 751, read length = 301\n",
      "  0:00:12.739   304M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:12.739   304M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:13.160   304M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.161   304M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:13.161   304M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:13.161   304M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 313, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:13.167   316M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:13.563   316M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.576   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Distance Estimation\n",
      "  0:00:13.576   112M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:13.576   112M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:13.576   112M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:13.624   112M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:13.624   112M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 44878\n",
      "  0:00:13.644   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:13.644   112M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:13.649   112M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:13.649   112M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:14.159   112M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 1410; contradictional = 0\n",
      "  0:00:14.579   112M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 67; contradictional = 0\n",
      "  0:00:14.579   112M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:14.580   112M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:14.637   112M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:14.637   112M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 17613\n",
      "  0:00:14.645   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:14.645   112M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:14.648   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Repeat Resolving\n",
      "  0:00:14.648   112M / 1G    INFO    General                 (repeat_resolving.cpp      :  56)   Setting up preliminary path extend settings\n",
      "  0:00:14.649   112M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:14.649   112M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:14.679   148M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:14.686   148M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:14.686   148M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:14.686   148M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:14.686   148M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:14.687   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 8534 (0%)\n",
      "  0:00:14.724   148M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 8534 (1%)\n",
      "  0:00:14.754   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 8534 (2%)\n",
      "  0:00:14.777   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 8534 (5%)\n",
      "  0:00:14.808   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 854 paths from 8534 (10%)\n",
      "  0:00:14.819   156M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 8534 (11%)\n",
      "  0:00:14.858   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1708 paths from 8534 (20%)\n",
      "  0:00:14.873   164M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 8534 (23%)\n",
      "  0:00:14.894   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2562 paths from 8534 (30%)\n",
      "  0:00:14.936   176M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3416 paths from 8534 (40%)\n",
      "  0:00:14.968   184M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 8534 (47%)\n",
      "  0:00:14.974   184M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4270 paths from 8534 (50%)\n",
      "  0:00:15.023   192M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5124 paths from 8534 (60%)\n",
      "  0:00:15.132   200M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5978 paths from 8534 (70%)\n",
      "  0:00:15.233   204M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6832 paths from 8534 (80%)\n",
      "  0:00:15.310   212M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 7686 paths from 8534 (90%)\n",
      "  0:00:15.387   216M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 8192 paths from 8534 (95%)\n",
      "  0:00:15.404   216M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:15.404   216M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:15.423   216M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:15.423   216M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:15.434   216M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:15.434   216M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:15.461   216M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:15.470   216M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:15.506   216M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 1 loops\n",
      "  0:00:15.506   216M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:15.506   216M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:15.512   216M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:15.512   216M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:15.521   216M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:15.538   216M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:15.582   168M / 1G    INFO    General                 (repeat_resolving.cpp      :  75)   Restoring initial path extend settings\n",
      "  0:00:15.582   168M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:15.582   168M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:15.631   168M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/before_rr.fasta\n",
      "  0:00:15.695   168M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:15.905   172M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/final_contigs.paths\n",
      "  0:00:16.008   168M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/scaffolds.paths\n",
      "  0:00:16.117   168M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Second Phase Setup\n",
      "  0:00:16.117   168M / 1G    INFO    General                 (second_phase_setup.cpp    :  20)   Preparing second phase\n",
      "  0:00:16.153   108M / 1G    INFO    General                 (second_phase_setup.cpp    :  28)   Moving preliminary contigs from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/final_contigs.fasta to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/first_pe_contigs.fasta\n",
      "  0:00:16.153   108M / 1G    INFO    General                 (second_phase_setup.cpp    :  40)   Ready to run second phase\n",
      "  0:00:16.153   108M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:16.156    32M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:16.156    32M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:16.156    32M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:16.156    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.161    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.161    32M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.281    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 751 times\n",
      "  0:00:16.281    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:16.283    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:16.284    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:16.284    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.286    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 16 times\n",
      "  0:00:16.286    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.287    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.287    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:16.287    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:16.288    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:16.288    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.288    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.289    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.289    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.290    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:16.290    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:16.290    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:16.290    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.293    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:16.293    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.296    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.296    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:16.298    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:16.298    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:16.299    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.299    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:16.299    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.299    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.300    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:16.300    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:16.300    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:16.301    24M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:16.301    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:16.302    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:16.302    24M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:16.302    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.330729 Gb\n",
      "  0:00:16.303    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 924785\n",
      "  0:00:16.582     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 12603 edges\n",
      "  0:00:16.582     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 12603 sequences.\n",
      "  0:00:16.588    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:16.674    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3204310 kmers in total.\n",
      "  0:00:16.674    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:16.698    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:17.000    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:17.039    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1494328 bytes occupied (3.7308 bits per kmer).\n",
      "  0:00:17.075   108M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:17.360   108M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:17.360   108M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:17.367   108M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:17.839   108M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:17.839   108M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:17.849   108M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:17.859   108M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 1 gaps after checking 905 candidates\n",
      "  0:00:17.863   108M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:17.863   108M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:17.863   108M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:17.863   108M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:17.863   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:17.887   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 15 times\n",
      "  0:00:17.887   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:17.953   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 15 times\n",
      "  0:00:17.953   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:17.957   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:17.957   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:17.994   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 3 times\n",
      "  0:00:17.994   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:18.018   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 7 times\n",
      "  0:00:18.018   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:18.022   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:18.022   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:18.026   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:18.026   108M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:18.324   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 93 times\n",
      "  0:00:18.324   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:18.326   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:18.326   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:18.328   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:18.328   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:18.341   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 2 times\n",
      "  0:00:18.342   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:18.345   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:18.345   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:18.354   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:18.354   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:18.362   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 6 times\n",
      "  0:00:18.362   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:18.367   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:18.367   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:18.371   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:18.371   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:18.375   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:18.375   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:18.377   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:18.377   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:18.378   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:18.378   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:18.378   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:18.378   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:18.381   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:18.381   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:18.389   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:18.389   112M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:18.392   112M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:18.401   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:18.439   112M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 298 times\n",
      "  0:00:18.439   112M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:18.443   112M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.34002\n",
      "  0:00:18.443   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:18.443   112M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 278146 kmers to process\n",
      "  0:00:18.560   112M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:20.118   120M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:20.221   112M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:20.225   112M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 25 nucleotides\n",
      "  0:00:20.225   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:20.226   112M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:20.268   112M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/before_rr.fasta\n",
      "  0:00:20.321   112M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:20.469   112M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/simplified_contigs.fasta\n",
      "  0:00:20.522   112M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/intermediate_contigs.fasta\n",
      "  0:00:20.575   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:20.610   116M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 279012 kmers to process\n",
      "  0:00:20.734   116M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:20.734   116M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:20.734   116M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:20.735   116M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:20.760   180M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.351   180M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:22.696   180M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:22.696   180M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   13214 paired reads (52.911% of all) aligned to long edges\n",
      "  0:00:22.699   116M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 521.121, deviation = 171.109, left quantile = 313, right quantile = 751, read length = 301\n",
      "  0:00:22.767   308M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:22.767   308M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:23.290   308M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:23.291   308M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:23.291   308M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:23.291   308M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 313, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:23.298   320M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:23.780   320M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:23.793   116M / 1G    INFO    General                 (pair_info_count.cpp       : 327)   Mapping contigs library #1\n",
      "  0:00:23.794   116M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:24.127   116M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 7014 reads processed\n",
      "  0:00:24.127   116M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:24.127   116M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:24.127   116M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:24.127   116M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:24.148   116M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:24.149   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 26501\n",
      "  0:00:24.160   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:24.161   116M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:24.163   116M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:24.163   116M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:24.241   116M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 364; contradictional = 0\n",
      "  0:00:24.299   116M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 14; contradictional = 0\n",
      "  0:00:24.299   116M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:24.300   116M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:24.318   116M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:24.318   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 11466\n",
      "  0:00:24.322   116M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:24.323   116M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:24.326   116M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:24.326   116M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:24.326   116M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:24.344   140M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:24.352   140M / 1G    INFO    General                 (extenders_logic.cpp       :  34)   resolvable_repeat_length_bound set to 10000\n",
      "  0:00:24.362   140M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:24.363   140M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:24.363   140M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 1 single read library\n",
      "  0:00:24.366   140M / 1G    INFO    General                 (launcher.cpp              : 359)   Filling backbone edges for long reads scaffolding...\n",
      "  0:00:24.367   140M / 1G    INFO    General                 (launcher.cpp              : 361)    with long reads paths\n",
      "  0:00:24.367   140M / 1G    INFO    General                 (launcher.cpp              : 370)   Removing fake unique with paired-end libs\n",
      "  0:00:24.367   140M / 1G    INFO    General                 (launcher.cpp              : 381)   0 unique edges\n",
      "  0:00:24.367   140M / 1G    INFO    General                 (extenders_logic.cpp       : 386)   Using 0 long reads scaffolding libraries\n",
      "  0:00:24.368   140M / 1G    INFO    General                 (extenders_logic.cpp       : 396)   Using additional coordinated coverage extender\n",
      "  0:00:24.369   140M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 5\n",
      "  0:00:24.369   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 5586 (0%)\n",
      "  0:00:24.411   144M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 5586 (2%)\n",
      "  0:00:24.458   144M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 5586 (4%)\n",
      "  0:00:24.494   148M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 5586 (9%)\n",
      "  0:00:24.497   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 559 paths from 5586 (10%)\n",
      "  0:00:24.536   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 5586 (18%)\n",
      "  0:00:24.542   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1118 paths from 5586 (20%)\n",
      "  0:00:24.582   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1677 paths from 5586 (30%)\n",
      "  0:00:24.603   160M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 5586 (36%)\n",
      "  0:00:24.612   160M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2236 paths from 5586 (40%)\n",
      "  0:00:24.645   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2795 paths from 5586 (50%)\n",
      "  0:00:24.676   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3354 paths from 5586 (60%)\n",
      "  0:00:24.710   176M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3913 paths from 5586 (70%)\n",
      "  0:00:24.722   176M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 5586 (73%)\n",
      "  0:00:24.742   180M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4472 paths from 5586 (80%)\n",
      "  0:00:24.790   184M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5031 paths from 5586 (90%)\n",
      "  0:00:24.872   188M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:24.872   188M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:24.886   188M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:24.886   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:24.886   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:24.887   188M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:24.893   188M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:24.898   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:24.900   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:24.904   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:24.911   188M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:24.911   188M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:24.928   188M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:24.934   188M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:24.959   188M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 11 loops\n",
      "  0:00:24.960   188M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:24.960   188M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:24.963   188M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:24.963   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:24.963   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:24.963   188M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:24.969   188M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:24.976   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:24.978   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:24.981   188M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:24.988   188M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:25.002   188M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:25.030   156M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:25.030   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:25.071   156M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/before_rr.fasta\n",
      "  0:00:25.123   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/assembly_graph.fastg\n",
      "  0:00:25.292   160M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/final_contigs.paths\n",
      "  0:00:25.376   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25//K25/scaffolds.paths\n",
      "  0:00:25.463   156M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:25.556    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 25 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 25 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly25/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=35\n",
      "Command line: /home/microbioinf/miniconda3/bin/metaspades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\t--meta\t-k\t35\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Metagenomic mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [35]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Counting           (kmer_data.cpp             : 280)   Estimating k-mer count\n",
      "  0:00:00.021    52M / 52M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.793    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:01.459    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 294)   Processed 58620 reads\n",
      "  0:00:01.459    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 299)   Total 58620 reads processed\n",
      "  0:00:02.783    12M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 311)   Filtering singleton k-mers\n",
      "38 8 0\n",
      "nslots: 1073741824\n",
      "bits per slot: 8 range: 0000004000000000\n",
      "  0:00:02.783     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:13.633     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:16.413     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 326)   Processed 58620 reads\n",
      "  0:00:16.414     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 331)   Total 58620 reads processed\n",
      "  0:00:16.414     1G / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:16.414     1G / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:16.414     1G / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:16.415     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.190972 Gb\n",
      "  0:00:16.415     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 533997\n",
      "  0:00:16.415     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:18.396     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:20.111     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 58620 reads\n",
      "  0:00:20.112     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58620 reads processed\n",
      "  0:00:20.118     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:20.302     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4417004 kmers in total.\n",
      "  0:00:20.303     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:20.332     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:20.778     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:20.824     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2056288 bytes occupied (3.72431 bits per kmer).\n",
      "  0:00:20.915    12M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:21.152    84M / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:31.297    84M / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:32.123    84M / 2G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 4295701\n",
      "  0:00:32.125    48M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:32.170   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:35.861   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:39.035   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:39.046   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 4417004 kmers in total. Among them 288 (0.00652026%) are singletons.\n",
      "  0:00:39.046   152M / 2G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:39.300   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:39.301   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:39.301   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4182962. Among them 4144882 (99.0896%) are good\n",
      "  0:00:39.301   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 12322. Among them 12322 (100%) are good\n",
      "  0:00:39.302   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 107706. Among them 107322 (99.6435%) are good\n",
      "  0:00:39.302   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.17297 kmers\n",
      "  0:00:39.302   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.06465\n",
      "  0:00:39.302   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4264526\n",
      "  0:00:39.303   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.946036,0.00722882,0.0352043,0.0115306),(0.00629838,0.940463,0.00446536,0.0487731),(0.0325448,0.00267922,0.954793,0.00998316),(0.00782649,0.0229528,0.00466325,0.964557))\n",
      "  0:00:39.308   152M / 2G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:39.309   152M / 2G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:41.612   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 23343 new k-mers.\n",
      "  0:00:44.062   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 165 new k-mers.\n",
      "  0:00:46.516   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:46.518   152M / 2G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:46.519   152M / 2G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:46.519   152M / 2G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:46.739   216M / 2G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 29310 reads.\n",
      "  0:00:48.790   220M / 2G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:48.863   220M / 2G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:48.889   152M / 2G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 18214 bases in 10929 reads.\n",
      "  0:00:48.889   152M / 2G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 14550035.\n",
      "  0:00:48.897    12M / 2G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/corrected/corrected.yaml\n",
      "  0:00:48.897    12M / 2G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K35\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/K35/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/K35/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/K35/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/dataset.info) with K=35\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 1\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.353    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.655   108M / 108M  INFO    General                 (binary_converter.hpp      : 117)   24974 reads written\n",
      "  0:00:00.677     4M / 108M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.843   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   1853 reads written\n",
      "  0:00:00.850     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.981   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.987     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.988     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:01.883     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:01.883     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.893    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.893    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.992    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4068854 kmers in total.\n",
      "  0:00:01.992    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.042    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:02.042    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.042    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.042    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.042    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:02.042    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:02.740     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 4068854 kmers\n",
      "  0:00:02.740     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 4068854 kmers.\n",
      "  0:00:02.752    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.857    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4073339 kmers in total.\n",
      "  0:00:02.857    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.907    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.312    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.464    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1896880 bytes occupied (3.72545 bits per kmer).\n",
      "  0:00:03.467    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:04.139    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:04.139    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:04.139    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:05.121    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 28925 sequences extracted\n",
      "  0:00:05.565    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:05.744    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:05.775    28M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:05.776    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:05.776    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:06.064    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1888288 bytes occupied (3.71267 bits per kmer).\n",
      "  0:00:06.071    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:06.880    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:07.502    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 57847 edges\n",
      "  0:00:07.520    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:07.521    24M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:07.521    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:07.521    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:07.521    24M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:07.522    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.330729 Gb\n",
      "  0:00:07.522    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462392\n",
      "  0:00:07.840     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 57847 edges\n",
      "  0:00:07.841     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 57847 sequences.\n",
      "  0:00:07.847    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:07.943    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4068854 kmers in total.\n",
      "  0:00:07.943    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:07.993    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:08.336    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:08.447    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1894832 bytes occupied (3.72553 bits per kmer).\n",
      "  0:00:08.513   120M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:08.936   120M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:08.936   120M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:08.970   124M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:09.100   124M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:09.100   124M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:09.118   120M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:09.166   120M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 102 gaps after checking 3189 candidates\n",
      "  0:00:09.242   120M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:09.246    20M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:09.246    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:09.252    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:09.252    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:09.317    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 4629 times\n",
      "  0:00:09.317    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:09.322    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
      "  0:00:09.322    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with low flanking coverage\n",
      "  0:00:09.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with low flanking coverage triggered 0 times\n",
      "  0:00:09.326    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:09.335    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 2469 times\n",
      "  0:00:09.335    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Simplification\n",
      "  0:00:09.335    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:09.335    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:09.335    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:09.336    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.358    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1930 times\n",
      "  0:00:09.358    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.463    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 697 times\n",
      "  0:00:09.463    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.469    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.469    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:09.469    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.470    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 23 times\n",
      "  0:00:09.470    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.471    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:09.471    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.497    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 1280 times\n",
      "  0:00:09.497    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:09.497    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.499    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 51 times\n",
      "  0:00:09.499    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.503    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 15 times\n",
      "  0:00:09.503    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.507    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 171 times\n",
      "  0:00:09.507    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:09.508    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.511    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 7 times\n",
      "  0:00:09.511    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.517    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
      "  0:00:09.517    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.519    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.519    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:09.519    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.519    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:09.520    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.520    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:09.520    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.520    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.520    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:09.521    20M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:09.521    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:09.521    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:09.522    20M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:09.522    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331163 Gb\n",
      "  0:00:09.522    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462999\n",
      "  0:00:09.772     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 13830 edges\n",
      "  0:00:09.773     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 13830 sequences.\n",
      "  0:00:09.778    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:09.867    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3054401 kmers in total.\n",
      "  0:00:09.868    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:09.902    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:10.177    28M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:10.239    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1424768 bytes occupied (3.73171 bits per kmer).\n",
      "  0:00:10.296   100M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:10.624   100M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:10.624   100M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:10.635   100M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:11.067   100M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:11.067   100M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:11.077   100M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:11.089   100M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 15 gaps after checking 922 candidates\n",
      "  0:00:11.114   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Paired Information Counting\n",
      "  0:00:11.163   104M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 76246 kmers to process\n",
      "  0:00:11.197   104M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:11.197   104M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:11.198   104M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:11.198   104M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:11.228   168M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:11.742   168M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.050   168M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:13.051   168M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   11820 paired reads (47.3292% of all) aligned to long edges\n",
      "  0:00:13.055   104M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 520.586, deviation = 170.517, left quantile = 314, right quantile = 751, read length = 301\n",
      "  0:00:13.130   296M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:13.130   296M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:13.579   296M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.579   296M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:13.579   296M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:13.579   296M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 314, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:13.586   308M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:14.065   308M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:14.079   104M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Distance Estimation\n",
      "  0:00:14.079   104M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:14.079   104M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:14.079   104M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:14.099   104M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:14.099   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 20767\n",
      "  0:00:14.112   108M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:14.112   104M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:14.115   104M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:14.115   104M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:14.328   104M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 579; contradictional = 0\n",
      "  0:00:14.397   104M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 12; contradictional = 0\n",
      "  0:00:14.398   104M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:14.398   104M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:14.417   104M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:14.417   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 14066\n",
      "  0:00:14.422   104M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:14.422   104M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:14.423   104M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Repeat Resolving\n",
      "  0:00:14.423   104M / 1G    INFO    General                 (repeat_resolving.cpp      :  56)   Setting up preliminary path extend settings\n",
      "  0:00:14.423   104M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:14.423   104M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:14.449   132M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:14.454   132M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:14.454   132M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:14.454   132M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:14.454   132M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:14.454   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 6875 (0%)\n",
      "  0:00:14.488   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 6875 (1%)\n",
      "  0:00:14.508   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 6875 (3%)\n",
      "  0:00:14.534   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 6875 (7%)\n",
      "  0:00:14.547   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 688 paths from 6875 (10%)\n",
      "  0:00:14.565   140M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 6875 (14%)\n",
      "  0:00:14.588   144M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1376 paths from 6875 (20%)\n",
      "  0:00:14.617   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 6875 (29%)\n",
      "  0:00:14.617   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2064 paths from 6875 (30%)\n",
      "  0:00:14.649   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2752 paths from 6875 (40%)\n",
      "  0:00:14.677   160M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3440 paths from 6875 (50%)\n",
      "  0:00:14.705   168M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 6875 (59%)\n",
      "  0:00:14.707   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4128 paths from 6875 (60%)\n",
      "  0:00:14.737   172M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4816 paths from 6875 (70%)\n",
      "  0:00:14.818   180M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5504 paths from 6875 (80%)\n",
      "  0:00:14.892   184M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 6192 paths from 6875 (90%)\n",
      "  0:00:14.966   188M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:14.966   188M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:14.982   188M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:14.982   188M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:14.992   188M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:14.992   188M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:15.016   188M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:15.025   192M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:15.056   192M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 2 loops\n",
      "  0:00:15.056   192M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:15.056   192M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:15.061   192M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:15.061   192M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:15.072   192M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:15.088   192M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:15.120   152M / 1G    INFO    General                 (repeat_resolving.cpp      :  75)   Restoring initial path extend settings\n",
      "  0:00:15.120   152M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:15.120   152M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:15.169   152M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/before_rr.fasta\n",
      "  0:00:15.224   152M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:15.411   156M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/final_contigs.paths\n",
      "  0:00:15.511   152M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/scaffolds.paths\n",
      "  0:00:15.600   152M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Second Phase Setup\n",
      "  0:00:15.600   152M / 1G    INFO    General                 (second_phase_setup.cpp    :  20)   Preparing second phase\n",
      "  0:00:15.631   100M / 1G    INFO    General                 (second_phase_setup.cpp    :  28)   Moving preliminary contigs from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/final_contigs.fasta to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/first_pe_contigs.fasta\n",
      "  0:00:15.631   100M / 1G    INFO    General                 (second_phase_setup.cpp    :  40)   Ready to run second phase\n",
      "  0:00:15.631   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:15.635    28M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:15.635    28M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:15.635    28M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:15.635    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.637    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.637    28M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 396 times\n",
      "  0:00:15.730    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 13 times\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.732    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:15.732    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.734    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.734    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.738    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.742    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.742    24M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:15.742    24M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:15.742    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:15.742    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:15.742    24M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:15.742    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.330729 Gb\n",
      "  0:00:15.742    24M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462392\n",
      "  0:00:16.020     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 11400 edges\n",
      "  0:00:16.020     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 11400 sequences.\n",
      "  0:00:16.026    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:16.119    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 3019027 kmers in total.\n",
      "  0:00:16.119    24M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:16.156    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:16.423    32M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:16.484    32M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1408304 bytes occupied (3.73181 bits per kmer).\n",
      "  0:00:16.535   104M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:16.827   104M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:16.827   104M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:16.836   104M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:17.304   104M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:17.304   104M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:17.314   104M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:17.324   104M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 0 gaps after checking 789 candidates\n",
      "  0:00:17.329   104M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:17.329   104M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:17.329   104M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:17.329   104M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:17.329   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:17.381   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 14 times\n",
      "  0:00:17.381   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:17.516   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 11 times\n",
      "  0:00:17.516   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:17.520   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:17.520   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:17.674   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 74 times\n",
      "  0:00:17.674   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:17.704   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 10 times\n",
      "  0:00:17.705   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:17.723   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 1 times\n",
      "  0:00:17.723   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:17.725   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:17.725   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:17.955   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 37 times\n",
      "  0:00:17.955   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:17.956   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:17.956   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:17.959   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:17.959   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:17.964   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:17.964   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:17.969   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:17.969   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:17.977   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 1 times\n",
      "  0:00:17.977   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:17.982   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:17.982   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:17.984   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:17.984   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:17.985   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:17.985   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:17.988   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:17.988   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:17.989   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:17.989   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:17.991   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:17.991   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:17.991   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:17.991   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:17.994   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:17.994   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:18.001   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:18.001   104M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:18.011   104M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:18.016   104M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:18.055   100M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 286 times\n",
      "  0:00:18.055   100M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:18.059   100M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.29652\n",
      "  0:00:18.059   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:18.059   100M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 181954 kmers to process\n",
      "  0:00:18.162   100M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:19.697   104M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:19.754   100M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:19.756   100M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 10 nucleotides\n",
      "  0:00:19.756   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:19.756   100M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:19.796   100M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/before_rr.fasta\n",
      "  0:00:19.848   100M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:19.997   100M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/simplified_contigs.fasta\n",
      "  0:00:20.050   100M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/intermediate_contigs.fasta\n",
      "  0:00:20.103   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:20.136   104M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 182538 kmers to process\n",
      "  0:00:20.231   104M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:20.232   104M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:20.232   104M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:20.232   104M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:20.258   168M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:20.870   172M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:22.133   172M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:22.133   172M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   13685 paired reads (54.797% of all) aligned to long edges\n",
      "  0:00:22.136   104M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 521.604, deviation = 170.557, left quantile = 314, right quantile = 751, read length = 301\n",
      "  0:00:22.203   296M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:22.203   296M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:22.710   296M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:22.710   296M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:22.710   296M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:22.710   296M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 314, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:22.717   308M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:23.249   312M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:23.264   108M / 1G    INFO    General                 (pair_info_count.cpp       : 327)   Mapping contigs library #1\n",
      "  0:00:23.264   108M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:23.567   108M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5975 reads processed\n",
      "  0:00:23.567   108M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:23.568   108M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:23.568   108M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:23.568   108M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:23.578   108M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:23.579   108M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 12229\n",
      "  0:00:23.585   108M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:23.585   108M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:23.587   108M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:23.587   108M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:23.968   108M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 154; contradictional = 0\n",
      "  0:00:24.184   108M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 3; contradictional = 0\n",
      "  0:00:24.184   108M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:24.184   108M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:24.200   108M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:24.200   108M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 10188\n",
      "  0:00:24.205   108M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:24.205   108M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:24.206   108M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:24.206   108M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:24.206   108M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:24.224   128M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:24.231   132M / 1G    INFO    General                 (extenders_logic.cpp       :  34)   resolvable_repeat_length_bound set to 10000\n",
      "  0:00:24.238   132M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:24.238   132M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:24.238   132M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 1 single read library\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (launcher.cpp              : 359)   Filling backbone edges for long reads scaffolding...\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (launcher.cpp              : 361)    with long reads paths\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (launcher.cpp              : 370)   Removing fake unique with paired-end libs\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (launcher.cpp              : 381)   0 unique edges\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (extenders_logic.cpp       : 386)   Using 0 long reads scaffolding libraries\n",
      "  0:00:24.242   132M / 1G    INFO    General                 (extenders_logic.cpp       : 396)   Using additional coordinated coverage extender\n",
      "  0:00:24.244   132M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 5\n",
      "  0:00:24.244   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 4977 (0%)\n",
      "  0:00:24.288   132M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 4977 (2%)\n",
      "  0:00:24.305   132M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 4977 (5%)\n",
      "  0:00:24.332   136M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 498 paths from 4977 (10%)\n",
      "  0:00:24.335   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 4977 (10%)\n",
      "  0:00:24.369   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 996 paths from 4977 (20%)\n",
      "  0:00:24.371   140M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 4977 (20%)\n",
      "  0:00:24.399   144M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1494 paths from 4977 (30%)\n",
      "  0:00:24.427   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1992 paths from 4977 (40%)\n",
      "  0:00:24.429   148M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 4977 (41%)\n",
      "  0:00:24.452   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2490 paths from 4977 (50%)\n",
      "  0:00:24.477   156M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2988 paths from 4977 (60%)\n",
      "  0:00:24.505   160M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3486 paths from 4977 (70%)\n",
      "  0:00:24.529   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3984 paths from 4977 (80%)\n",
      "  0:00:24.534   164M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 4977 (82%)\n",
      "  0:00:24.553   168M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4482 paths from 4977 (90%)\n",
      "  0:00:24.607   172M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:24.607   172M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:24.619   172M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:24.619   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:24.619   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:24.619   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:24.625   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:24.629   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:24.630   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:24.632   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:24.638   172M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:24.638   172M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:24.655   172M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:24.661   172M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:24.684   172M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 8 loops\n",
      "  0:00:24.684   172M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:24.684   172M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:24.688   172M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:24.688   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:24.688   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:24.688   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:24.694   172M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:24.700   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:24.701   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:24.704   172M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:24.711   172M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:24.723   172M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:24.748   144M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:24.748   144M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:24.786   144M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/before_rr.fasta\n",
      "  0:00:24.835   144M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/assembly_graph.fastg\n",
      "  0:00:25.008   148M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/final_contigs.paths\n",
      "  0:00:25.079   144M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35//K35/scaffolds.paths\n",
      "  0:00:25.158   148M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:25.238    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 25 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 35 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly35/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n",
      "#### Compute assembly K_MER=45\n",
      "Command line: /home/microbioinf/miniconda3/bin/metaspades.py\t-1\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\t-2\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\t--meta\t-k\t45\t-o\t/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.13.0\n",
      "  Python version: 2.7.15\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-Ubuntu-18.04-bionic\n",
      "\n",
      "Output dir: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Metagenomic mode\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq']\n",
      "      right reads: ['/home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: [45]\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 3\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/spades.log\n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running read error correction tool: /home/microbioinf/miniconda3/share/spades-3.13.0-0/bin/spades-hammer /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  76)   Loading config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/corrected/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (hammer_tools.cpp          :  36)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     4M / 4M    INFO   K-mer Counting           (kmer_data.cpp             : 280)   Estimating k-mer count\n",
      "  0:00:00.021    52M / 52M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:00.777    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 285)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:01.430    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 294)   Processed 58620 reads\n",
      "  0:00:01.430    60M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 299)   Total 58620 reads processed\n",
      "  0:00:02.751    12M / 60M   INFO   K-mer Counting           (kmer_data.cpp             : 311)   Filtering singleton k-mers\n",
      "38 8 0\n",
      "nslots: 1073741824\n",
      "bits per slot: 8 range: 0000004000000000\n",
      "  0:00:02.751     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:12.225     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 317)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:14.810     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 326)   Processed 58620 reads\n",
      "  0:00:14.810     1G / 1G    INFO   K-mer Counting           (kmer_data.cpp             : 331)   Total 58620 reads processed\n",
      "  0:00:14.811     1G / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:14.811     1G / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:14.812     1G / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:14.812     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.190972 Gb\n",
      "  0:00:14.812     1G / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 533997\n",
      "  0:00:14.813     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:16.792     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:18.592     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 58620 reads\n",
      "  0:00:18.593     2G / 2G    INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 58620 reads processed\n",
      "  0:00:18.599     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:18.782     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4417004 kmers in total.\n",
      "  0:00:18.782     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:18.810     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:19.217     1G / 2G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:19.259     1G / 2G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 2056288 bytes occupied (3.72431 bits per kmer).\n",
      "  0:00:19.352    12M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 356)   Arranging kmers in hash map order\n",
      "  0:00:19.599    84M / 2G    INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:00:29.797    84M / 2G    INFO    General                 (main.cpp                  : 155)   Extracting clusters\n",
      "  0:00:30.640    84M / 2G    INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 4295701\n",
      "  0:00:30.642    48M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 376)   Collecting K-mer information, this takes a while.\n",
      "  0:00:30.686   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq\n",
      "  0:00:34.331   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 382)   Processing /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:37.606   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 389)   Collection done, postprocessing.\n",
      "  0:00:37.618   152M / 2G    INFO   K-mer Counting           (kmer_data.cpp             : 403)   There are 4417004 kmers in total. Among them 288 (0.00652026%) are singletons.\n",
      "  0:00:37.619   152M / 2G    INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:00:37.858   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 649)   Subclustering done. Total 0 non-read kmers were generated.\n",
      "  0:00:37.859   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering statistics:\n",
      "  0:00:37.859   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)     Total singleton hamming clusters: 4182962. Among them 4144882 (99.0896%) are good\n",
      "  0:00:37.859   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton subclusters: 12322. Among them 12322 (100%) are good\n",
      "  0:00:37.860   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total non-singleton subcluster centers: 107706. Among them 107322 (99.6435%) are good\n",
      "  0:00:37.860   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Average size of non-trivial subcluster: 2.17297 kmers\n",
      "  0:00:37.860   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average number of sub-clusters per non-singleton cluster: 1.06465\n",
      "  0:00:37.861   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Total solid k-mers: 4264526\n",
      "  0:00:37.861   152M / 2G    INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Substitution probabilities: [4,4]((0.946036,0.00722882,0.0352043,0.0115306),(0.00629838,0.940463,0.00446536,0.0487731),(0.0325448,0.00267922,0.954793,0.00998316),(0.00782649,0.0229528,0.00466325,0.964557))\n",
      "  0:00:37.868   152M / 2G    INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:00:37.869   152M / 2G    INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 3 threads.\n",
      "  0:00:40.162   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 23343 new k-mers.\n",
      "  0:00:42.583   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 165 new k-mers.\n",
      "  0:00:44.909   152M / 2G    INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 0 new k-mers.\n",
      "  0:00:44.909   152M / 2G    INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:00:44.909   152M / 2G    INFO    General                 (hammer_tools.cpp          : 220)   Starting read correction in 3 threads.\n",
      "  0:00:44.909   152M / 2G    INFO    General                 (hammer_tools.cpp          : 233)   Correcting pair of reads: /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_1.fastq and /home/microbioinf/Documentos/Tema_3/kneaddata_out/VIR_R1__kneaddata_paired_2.fastq\n",
      "  0:00:45.133   216M / 2G    INFO    General                 (hammer_tools.cpp          : 168)   Prepared batch 0 of 29310 reads.\n",
      "  0:00:47.224   220M / 2G    INFO    General                 (hammer_tools.cpp          : 175)   Processed batch 0\n",
      "  0:00:47.304   220M / 2G    INFO    General                 (hammer_tools.cpp          : 185)   Written batch 0\n",
      "  0:00:47.331   152M / 2G    INFO    General                 (hammer_tools.cpp          : 274)   Correction done. Changed 18214 bases in 10929 reads.\n",
      "  0:00:47.331   152M / 2G    INFO    General                 (hammer_tools.cpp          : 275)   Failed to correct 0 bases out of 14550035.\n",
      "  0:00:47.339    12M / 2G    INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/corrected/corrected.yaml\n",
      "  0:00:47.339    12M / 2G    INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "\n",
      "== Dataset description file was created: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/corrected/corrected.yaml\n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started.\n",
      "\n",
      "\n",
      "== Running assembler: K45\n",
      "\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/K45/configs/config.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/K45/configs/mda_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  74)   Loaded config from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/K45/configs/meta_mode.info\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (memory_limit.cpp          :  49)   Memory limit set to 3 Gb\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  87)   Starting SPAdes, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  88)   Maximum k-mer length: 128\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  89)   Assembling dataset (/home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/dataset.info) with K=45\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  :  90)   Maximum # of threads to use (adjusted due to OMP capabilities): 3\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  51)   SPAdes started\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  58)   Starting from stage: construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  65)   Two-step RR enabled: 1\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (launch.hpp                :  76)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     4M / 4M    INFO   StageManager             (stage.cpp                 : 132)   STAGE == de Bruijn graph construction\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  77)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     4M / 4M    INFO    General                 (read_converter.hpp        :  78)   Converting paired reads\n",
      "  0:00:00.350    96M / 100M  INFO    General                 (binary_converter.hpp      :  93)   16384 reads processed\n",
      "  0:00:00.656   108M / 108M  INFO    General                 (binary_converter.hpp      : 117)   24974 reads written\n",
      "  0:00:00.676     4M / 108M  INFO    General                 (read_converter.hpp        :  87)   Converting single reads\n",
      "  0:00:00.837   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   1853 reads written\n",
      "  0:00:00.844     4M / 180M  INFO    General                 (read_converter.hpp        :  95)   Converting merged reads\n",
      "  0:00:00.979   136M / 180M  INFO    General                 (binary_converter.hpp      : 117)   0 reads written\n",
      "  0:00:00.986     4M / 180M  INFO    General                 (construction.cpp          : 111)   Max read length 301\n",
      "  0:00:00.986     4M / 180M  INFO    General                 (construction.cpp          : 117)   Average read length 249.444\n",
      "  0:00:00.986     4M / 180M  INFO    General                 (stage.cpp                 : 101)   PROCEDURE == k+1-mer counting\n",
      "  0:00:00.987     4M / 180M  INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 9 files using 3 threads. This might take a while.\n",
      "  0:00:00.987     4M / 180M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:00.987     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332899 Gb\n",
      "  0:00:00.987     4M / 180M  INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 2482277\n",
      "  0:00:01.857     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 289)   Processed 103602 reads\n",
      "  0:00:01.857     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 295)   Adding contigs from previous K\n",
      "  0:00:01.867    12M / 1G    INFO    General                 (kmer_splitters.hpp        : 308)   Used 103602 reads\n",
      "  0:00:01.867    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:01.964    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4011725 kmers in total.\n",
      "  0:00:01.964    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.011    12M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Extension index construction\n",
      "  0:00:02.011    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:02.011    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:02.011    12M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:02.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.332031 Gb\n",
      "  0:00:02.011    12M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 464213\n",
      "  0:00:02.730     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 380)   Processed 4011725 kmers\n",
      "  0:00:02.730     1G / 1G    INFO    General                 (kmer_splitters.hpp        : 385)   Used 4011725 kmers.\n",
      "  0:00:02.740    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:02.853    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4018283 kmers in total.\n",
      "  0:00:02.853    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:02.899    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:03.294    12M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:03.365    12M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1871560 bytes occupied (3.72609 bits per kmer).\n",
      "  0:00:03.368    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build:  99)   Building k-mer extensions from k+1-mers\n",
      "  0:00:04.116    16M / 1G    INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 103)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:04.116    16M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Condensing graph\n",
      "  0:00:04.116    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 355)   Extracting unbranching paths\n",
      "  0:00:05.046    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 374)   Extracting unbranching paths finished. 25679 sequences extracted\n",
      "  0:00:05.459    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 310)   Collecting perfect loops\n",
      "  0:00:05.629    16M / 1G    INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 343)   Collecting perfect loops finished. 1 loops collected\n",
      "  0:00:05.654    24M / 1G    INFO    General                 (stage.cpp                 : 101)   PROCEDURE == Filling coverage indices (PHM)\n",
      "  0:00:05.655    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:05.655    24M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:05.964    28M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1861800 bytes occupied (3.71272 bits per kmer).\n",
      "  0:00:05.971    44M / 1G    INFO    General                 (construction.cpp          : 388)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:00:06.757    44M / 1G    INFO    General                 (construction.cpp          : 508)   Filling coverage and flanking coverage from PHM\n",
      "  0:00:07.408    44M / 1G    INFO    General                 (construction.cpp          : 464)   Processed 51360 edges\n",
      "  0:00:07.428    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:07.428    20M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:07.428    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:07.428    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:07.428    20M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:07.428    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331163 Gb\n",
      "  0:00:07.428    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462999\n",
      "  0:00:07.765     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 51360 edges\n",
      "  0:00:07.765     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 51360 sequences.\n",
      "  0:00:07.770    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:07.884    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 4011725 kmers in total.\n",
      "  0:00:07.885    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:07.939    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:08.282    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:08.360    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1868400 bytes occupied (3.72588 bits per kmer).\n",
      "  0:00:08.406   112M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:08.899   112M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:08.900   112M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:08.926   116M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:09.065   120M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:09.065   120M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:09.086   112M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:09.136   112M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 125 gaps after checking 3280 candidates\n",
      "  0:00:09.220   112M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Raw Simplification\n",
      "  0:00:09.225    16M / 1G    INFO    General                 (simplification.cpp        : 128)   PROCEDURE == InitialCleaning\n",
      "  0:00:09.226    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Self conjugate edge remover\n",
      "  0:00:09.230    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Self conjugate edge remover triggered 0 times\n",
      "  0:00:09.230    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial tip clipper\n",
      "  0:00:09.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial tip clipper triggered 4318 times\n",
      "  0:00:09.287    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial ec remover\n",
      "  0:00:09.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial ec remover triggered 0 times\n",
      "  0:00:09.292    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with low flanking coverage\n",
      "  0:00:09.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with low flanking coverage triggered 1 times\n",
      "  0:00:09.296    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Initial isolated edge remover\n",
      "  0:00:09.310    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Initial isolated edge remover triggered 3538 times\n",
      "  0:00:09.311    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Simplification\n",
      "  0:00:09.311    16M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:09.311    16M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:09.312    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:09.312    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1820 times\n",
      "  0:00:09.332    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.386    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 390 times\n",
      "  0:00:09.387    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.391    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.391    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:09.391    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.392    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 24 times\n",
      "  0:00:09.392    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.392    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:09.392    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.409    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 814 times\n",
      "  0:00:09.409    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:09.410    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.411    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 49 times\n",
      "  0:00:09.411    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.414    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 12 times\n",
      "  0:00:09.414    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.417    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 109 times\n",
      "  0:00:09.418    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:09.418    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.421    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 10 times\n",
      "  0:00:09.422    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.426    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 2 times\n",
      "  0:00:09.426    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.428    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.428    16M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:00:09.429    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:09.429    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:09.429    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:09.429    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:09.430    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:09.430    16M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:09.430    16M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:09.430    16M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:09.430    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:09.431    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:09.431    16M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:09.431    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331597 Gb\n",
      "  0:00:09.431    16M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 463606\n",
      "  0:00:09.657     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 11368 edges\n",
      "  0:00:09.657     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 11368 sequences.\n",
      "  0:00:09.662    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:09.748    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 2825242 kmers in total.\n",
      "  0:00:09.749    16M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:09.780    16M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:10.075    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:10.133    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1318632 bytes occupied (3.73386 bits per kmer).\n",
      "  0:00:10.171    88M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:10.447    88M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:10.447    88M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:10.455    92M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:10.880    96M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:10.882    96M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:10.894    92M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:10.907    92M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 25 gaps after checking 866 candidates\n",
      "  0:00:10.948    92M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Paired Information Counting\n",
      "  0:00:10.984    92M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 49574 kmers to process\n",
      "  0:00:11.001    92M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:11.001    92M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:11.001    92M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:11.001    92M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:11.028   156M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:11.540   152M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:12.811   152M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:12.812   152M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   12194 paired reads (48.8268% of all) aligned to long edges\n",
      "  0:00:12.816    88M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 521.705, deviation = 170.739, left quantile = 314, right quantile = 751, read length = 301\n",
      "  0:00:12.883   280M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:12.883   280M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:13.363   280M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.363   280M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:13.363   280M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:13.363   280M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 314, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:13.369   292M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:13.817   300M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:13.829    96M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Distance Estimation\n",
      "  0:00:13.829    96M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:13.829    96M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:13.829    96M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:13.847    96M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:13.847    96M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 14506\n",
      "  0:00:13.855    96M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:13.856    96M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:13.858    96M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:13.858    96M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:13.998   100M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 256; contradictional = 0\n",
      "  0:00:14.059   100M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 3; contradictional = 0\n",
      "  0:00:14.060   100M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:14.060   100M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:14.073   100M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:14.074   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 11554\n",
      "  0:00:14.078   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:14.079   100M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:14.080   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Preliminary Repeat Resolving\n",
      "  0:00:14.081   100M / 1G    INFO    General                 (repeat_resolving.cpp      :  56)   Setting up preliminary path extend settings\n",
      "  0:00:14.082   100M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:14.082   100M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:14.103   120M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:14.110   120M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:14.110   120M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:14.110   120M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 0 single read libraries\n",
      "  0:00:14.110   120M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 3\n",
      "  0:00:14.110   120M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 5647 (0%)\n",
      "  0:00:14.136   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 5647 (2%)\n",
      "  0:00:14.153   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 5647 (4%)\n",
      "  0:00:14.178   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 5647 (9%)\n",
      "  0:00:14.182   124M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 565 paths from 5647 (10%)\n",
      "  0:00:14.209   128M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 5647 (18%)\n",
      "  0:00:14.215   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1130 paths from 5647 (20%)\n",
      "  0:00:14.243   136M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1695 paths from 5647 (30%)\n",
      "  0:00:14.257   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 5647 (36%)\n",
      "  0:00:14.266   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2260 paths from 5647 (40%)\n",
      "  0:00:14.293   144M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2825 paths from 5647 (50%)\n",
      "  0:00:14.316   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3390 paths from 5647 (60%)\n",
      "  0:00:14.336   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3955 paths from 5647 (70%)\n",
      "  0:00:14.342   156M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 5647 (72%)\n",
      "  0:00:14.359   160M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4520 paths from 5647 (80%)\n",
      "  0:00:14.416   164M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 5085 paths from 5647 (90%)\n",
      "  0:00:14.484   168M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:14.484   168M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:14.498   168M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:14.498   168M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:14.507   168M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:14.507   168M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:14.525   168M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:14.531   168M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:14.558   168M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 2 loops\n",
      "  0:00:14.558   168M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:14.558   168M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:14.561   168M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:14.562   168M / 1G    INFO    General                 (launcher.cpp              : 247)   Overlaps will not be removed\n",
      "  0:00:14.569   168M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:14.583   168M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:14.609   140M / 1G    INFO    General                 (repeat_resolving.cpp      :  75)   Restoring initial path extend settings\n",
      "  0:00:14.609   140M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:14.610   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:14.650   140M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/before_rr.fasta\n",
      "  0:00:14.701   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:14.860   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/final_contigs.paths\n",
      "  0:00:14.946   140M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/scaffolds.paths\n",
      "  0:00:15.026   140M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Second Phase Setup\n",
      "  0:00:15.026   140M / 1G    INFO    General                 (second_phase_setup.cpp    :  20)   Preparing second phase\n",
      "  0:00:15.051    96M / 1G    INFO    General                 (second_phase_setup.cpp    :  28)   Moving preliminary contigs from /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/final_contigs.fasta to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/first_pe_contigs.fasta\n",
      "  0:00:15.051    96M / 1G    INFO    General                 (second_phase_setup.cpp    :  40)   Ready to run second phase\n",
      "  0:00:15.051    96M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification\n",
      "  0:00:15.055    24M / 1G    INFO    General                 (simplification.cpp        : 357)   Graph simplification started\n",
      "  0:00:15.055    24M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:15.055    24M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:00:15.055    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.057    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.057    24M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.094    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 192 times\n",
      "  0:00:15.094    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.096    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.096    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:00:15.096    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 4 times\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.097    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.097    20M / 1G    INFO    General                 (simplification.cpp        : 362)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:00:15.097    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:15.098    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:15.098    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:15.099    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:15.099    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Low coverage edge remover\n",
      "  0:00:15.102    20M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Low coverage edge remover triggered 0 times\n",
      "  0:00:15.102    20M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Gap Closer\n",
      "  0:00:15.102    20M / 1G    INFO    General                 (graph_pack.hpp            : 101)   Index refill\n",
      "  0:00:15.103    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 301)   Building kmer index\n",
      "  0:00:15.103    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 48 files using 3 threads. This might take a while.\n",
      "  0:00:15.103    20M / 1G    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024\n",
      "  0:00:15.103    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  89)   Memory available for splitting buffers: 0.331163 Gb\n",
      "  0:00:15.103    20M / 1G    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 462999\n",
      "  0:00:15.328     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  77)   Processed 10150 edges\n",
      "  0:00:15.328     1G / 1G    INFO    General                 (edge_index_builders.hpp   :  82)   Used 10150 sequences.\n",
      "  0:00:15.333    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.\n",
      "  0:00:15.421    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 2808259 kmers in total.\n",
      "  0:00:15.421    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.\n",
      "  0:00:15.456    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 314)   Building perfect hash indices\n",
      "  0:00:15.720    20M / 1G    INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.\n",
      "  0:00:15.788    20M / 1G    INFO   K-mer Index Building     (kmer_index_builder.hpp    : 336)   Index built. Total 1310616 bytes occupied (3.7336 bits per kmer).\n",
      "  0:00:15.836    88M / 1G    INFO    General                 (edge_index_builders.hpp   : 107)   Collecting edge information from graph, this takes a while.\n",
      "  0:00:16.120    88M / 1G    INFO    General                 (edge_index.hpp            :  92)   Index refilled\n",
      "  0:00:16.120    88M / 1G    INFO    General                 (gap_closer.cpp            : 159)   Preparing shift maps\n",
      "  0:00:16.129    88M / 1G    INFO    General                 (gap_closer.cpp            : 119)   Processing paired reads (takes a while)\n",
      "  0:00:16.568    96M / 1G    INFO    General                 (gap_closer.cpp            : 138)   Used 24974 paired reads\n",
      "  0:00:16.568    96M / 1G    INFO    General                 (gap_closer.cpp            : 140)   Merging paired indices\n",
      "  0:00:16.578    92M / 1G    INFO   GapCloser                (gap_closer.cpp            : 346)   Closing short gaps\n",
      "  0:00:16.591    92M / 1G    INFO   GapCloser                (gap_closer.cpp            : 380)   Closing short gaps complete: filled 0 gaps after checking 800 candidates\n",
      "  0:00:16.595    88M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Simplification Cleanup\n",
      "  0:00:16.595    88M / 1G    INFO    General                 (simplification.cpp        : 196)   PROCEDURE == Post simplification\n",
      "  0:00:16.596    88M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:16.596    88M / 1G    INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:00:16.596    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.654    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 9 times\n",
      "  0:00:16.654    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.713    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 5 times\n",
      "  0:00:16.714    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.719    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.719    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.897    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 55 times\n",
      "  0:00:16.897    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.903    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 2 times\n",
      "  0:00:16.903    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.906    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.906    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.908    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:16.909    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.957    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 14 times\n",
      "  0:00:16.957    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:16.961    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:16.961    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.967    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 1 times\n",
      "  0:00:16.967    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.974    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 1 times\n",
      "  0:00:16.974    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.976    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.976    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:16.983    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:16.983    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:16.986    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 1 times\n",
      "  0:00:16.986    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:16.987    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:16.987    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:16.990    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 1 times\n",
      "  0:00:16.990    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:16.992    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:16.992    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:16.993    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:16.993    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Relative coverage component remover\n",
      "  0:00:16.994    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Relative coverage component remover triggered 0 times\n",
      "  0:00:16.994    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Disconnecting edges with relatively low coverage\n",
      "  0:00:16.994    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Disconnecting edges with relatively low coverage triggered 0 times\n",
      "  0:00:16.994    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex tip clipper\n",
      "  0:00:16.997    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex tip clipper triggered 0 times\n",
      "  0:00:16.997    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Complex bulge remover\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Complex bulge remover triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Tip clipper\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Tip clipper triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final tip clipper\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final tip clipper triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Bulge remover\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Bulge remover triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Final bulge remover\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Final bulge remover triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Thorn remover (meta)\n",
      "  0:00:17.005    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Thorn remover (meta) triggered 0 times\n",
      "  0:00:17.005    88M / 1G    INFO    General                 (simplification.cpp        : 326)   Running Hidden EC remover (meta)\n",
      "  0:00:17.008    88M / 1G    INFO    General                 (simplification.cpp        : 330)   Disrupting self-conjugate edges\n",
      "  0:00:17.013    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 165)   Running Removing isolated edges\n",
      "  0:00:17.054    88M / 1G    INFO   Simplification           (parallel_processing.hpp   : 167)   Removing isolated edges triggered 271 times\n",
      "  0:00:17.054    88M / 1G    INFO    General                 (simplification.cpp        : 470)   Counting average coverage\n",
      "  0:00:17.057    88M / 1G    INFO    General                 (simplification.cpp        : 476)   Average coverage = 3.26928\n",
      "  0:00:17.057    88M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Mismatch Correction\n",
      "  0:00:17.057    88M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 109896 kmers to process\n",
      "  0:00:17.108    88M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:18.369    92M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 189)   Finished collecting potential mismatches positions\n",
      "  0:00:18.399    88M / 1G    INFO    General                 (mismatch_shall_not_pass.hp: 290)   All edges processed\n",
      "  0:00:18.400    88M / 1G    INFO    General                 (mismatch_correction.cpp   :  27)   Corrected 5 nucleotides\n",
      "  0:00:18.400    88M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:18.400    88M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:18.436    88M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/before_rr.fasta\n",
      "  0:00:18.483    88M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:18.626    88M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/simplified_contigs.fasta\n",
      "  0:00:18.675    88M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/intermediate_contigs.fasta\n",
      "  0:00:18.726    88M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Paired Information Counting\n",
      "  0:00:18.754    92M / 1G    INFO    General                 (graph_pack.hpp            : 109)   Normalizing k-mer map. Total 110340 kmers to process\n",
      "  0:00:18.805    92M / 1G    INFO    General                 (graph_pack.hpp            : 111)   Normalizing done\n",
      "  0:00:18.805    92M / 1G    INFO    General                 (pair_info_count.cpp       : 320)   Min edge length for estimation: 900\n",
      "  0:00:18.805    92M / 1G    INFO    General                 (pair_info_count.cpp       : 331)   Estimating insert size for library #0\n",
      "  0:00:18.805    92M / 1G    INFO    General                 (pair_info_count.cpp       : 190)   Estimating insert size (takes a while)\n",
      "  0:00:18.831   156M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:19.439   156M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:20.671   156M / 1G    INFO    General                 (pair_info_count.cpp       : 209)   Edge pairs: 67108864 (rough upper limit)\n",
      "  0:00:20.671   156M / 1G    INFO    General                 (pair_info_count.cpp       : 213)   13801 paired reads (55.2615% of all) aligned to long edges\n",
      "  0:00:20.675    92M / 1G    INFO    General                 (pair_info_count.cpp       : 354)     Insert size = 523.559, deviation = 170.802, left quantile = 314, right quantile = 751, read length = 301\n",
      "  0:00:20.744   284M / 1G    INFO    General                 (pair_info_count.cpp       : 371)   Filtering data for library #0\n",
      "  0:00:20.744   284M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.268   284M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:21.269   284M / 1G    INFO    General                 (pair_info_count.cpp       : 383)   Mapping library #0\n",
      "  0:00:21.269   284M / 1G    INFO    General                 (pair_info_count.cpp       : 385)   Mapping paired reads (takes a while)\n",
      "  0:00:21.269   284M / 1G    INFO    General                 (pair_info_count.cpp       : 289)   Left insert size quantile 314, right insert size quantile 751, filtering threshold 1, rounding threshold 0\n",
      "  0:00:21.277   296M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:21.763   300M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 24974 reads processed\n",
      "  0:00:21.776    96M / 1G    INFO    General                 (pair_info_count.cpp       : 327)   Mapping contigs library #1\n",
      "  0:00:21.776    96M / 1G    INFO    General                 (pair_info_count.cpp       :  39)   Selecting usual mapper\n",
      "  0:00:22.031    96M / 1G    INFO    General                 (sequence_mapper_notifier.h:  98)   Total 5123 reads processed\n",
      "  0:00:22.031    96M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Distance Estimation\n",
      "  0:00:22.031    96M / 1G    INFO    General                 (distance_estimation.cpp   : 173)   Processing library #0\n",
      "  0:00:22.032    96M / 1G    INFO    General                 (distance_estimation.cpp   : 149)   Weight Filter Done\n",
      "  0:00:22.032    96M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:00:22.041    96M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:22.041    96M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 9894\n",
      "  0:00:22.047    96M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:22.047    96M / 1G    INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:00:22.048    96M / 1G    INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:00:22.048    96M / 1G    INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:00:22.430    96M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 58; contradictional = 0\n",
      "  0:00:22.729    96M / 1G    INFO   PairInfoImprover         (pair_info_improver.hpp    : 103)   Paired info stats: missing = 1; contradictional = 0\n",
      "  0:00:22.729    96M / 1G    INFO    General                 (distance_estimation.cpp   : 103)   Filling scaffolding index\n",
      "  0:00:22.729    96M / 1G    INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:00:22.740   100M / 1G    INFO    General                 (distance_estimation.cpp   :  34)   Filtering info\n",
      "  0:00:22.740   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; index size: 9282\n",
      "  0:00:22.744   100M / 1G    INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering\n",
      "  0:00:22.744   100M / 1G    INFO    General                 (distance_estimation.cpp   : 182)   Clearing raw paired index\n",
      "  0:00:22.744   100M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Repeat Resolving\n",
      "  0:00:22.744   100M / 1G    INFO    General                 (repeat_resolving.cpp      :  69)   Using Path-Extend repeat resolving\n",
      "  0:00:22.744   100M / 1G    INFO    General                 (launcher.cpp              : 477)   ExSPAnder repeat resolving tool started\n",
      "  0:00:22.758   120M / 1G    INFO    General                 (launcher.cpp              : 392)   Creating main extenders, unique edge length = 2000\n",
      "  0:00:22.764   120M / 1G    INFO    General                 (extenders_logic.cpp       :  34)   resolvable_repeat_length_bound set to 10000\n",
      "  0:00:22.769   120M / 1G    INFO    General                 (extenders_logic.cpp       : 472)   Using 1 paired-end library\n",
      "  0:00:22.769   120M / 1G    INFO    General                 (extenders_logic.cpp       : 473)   Using 1 paired-end scaffolding library\n",
      "  0:00:22.769   120M / 1G    INFO    General                 (extenders_logic.cpp       : 474)   Using 1 single read library\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (launcher.cpp              : 359)   Filling backbone edges for long reads scaffolding...\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (launcher.cpp              : 361)    with long reads paths\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (launcher.cpp              : 370)   Removing fake unique with paired-end libs\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (launcher.cpp              : 381)   0 unique edges\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (extenders_logic.cpp       : 386)   Using 0 long reads scaffolding libraries\n",
      "  0:00:22.771   120M / 1G    INFO    General                 (extenders_logic.cpp       : 396)   Using additional coordinated coverage extender\n",
      "  0:00:22.772   120M / 1G    INFO    General                 (launcher.cpp              : 420)   Total number of extenders is 5\n",
      "  0:00:22.772   120M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 0 paths from 4541 (0%)\n",
      "  0:00:22.802   120M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 128 paths from 4541 (2%)\n",
      "  0:00:22.818   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 256 paths from 4541 (5%)\n",
      "  0:00:22.836   124M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 455 paths from 4541 (10%)\n",
      "  0:00:22.840   124M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 512 paths from 4541 (11%)\n",
      "  0:00:22.868   128M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 910 paths from 4541 (20%)\n",
      "  0:00:22.874   128M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 1024 paths from 4541 (22%)\n",
      "  0:00:22.892   132M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1365 paths from 4541 (30%)\n",
      "  0:00:22.917   136M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 1820 paths from 4541 (40%)\n",
      "  0:00:22.928   136M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 2048 paths from 4541 (45%)\n",
      "  0:00:22.938   140M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2275 paths from 4541 (50%)\n",
      "  0:00:22.961   144M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 2730 paths from 4541 (60%)\n",
      "  0:00:22.984   148M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3185 paths from 4541 (70%)\n",
      "  0:00:23.007   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 3640 paths from 4541 (80%)\n",
      "  0:00:23.028   152M / 1G    INFO    General                 (path_extender.hpp         : 885)   Processed 4095 paths from 4541 (90%)\n",
      "  0:00:23.028   152M / 1G    INFO    General                 (path_extender.hpp         : 883)   Processed 4096 paths from 4541 (90%)\n",
      "  0:00:23.063   156M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:23.063   156M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:23.073   156M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:23.073   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:23.073   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:23.073   156M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:23.078   156M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:23.081   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:23.082   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:23.084   156M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:23.089   156M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:23.089   156M / 1G    INFO    General                 (launcher.cpp              : 427)   Closing gaps in paths\n",
      "  0:00:23.104   156M / 1G    INFO    General                 (launcher.cpp              : 455)   Gap closing completed\n",
      "  0:00:23.109   160M / 1G    INFO    General                 (launcher.cpp              : 286)   Traversing tandem repeats\n",
      "  0:00:23.131   160M / 1G    INFO    General                 (launcher.cpp              : 296)   Traversed 6 loops\n",
      "  0:00:23.131   160M / 1G    INFO    General                 (launcher.cpp              : 234)   Finalizing paths\n",
      "  0:00:23.131   160M / 1G    INFO    General                 (launcher.cpp              : 236)   Deduplicating paths\n",
      "  0:00:23.134   160M / 1G    INFO    General                 (launcher.cpp              : 240)   Paths deduplicated\n",
      "  0:00:23.134   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 295)   Removing overlaps\n",
      "  0:00:23.134   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 305)   Marking overlaps\n",
      "  0:00:23.134   160M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 130)   Marking start/end overlaps\n",
      "  0:00:23.139   160M / 1G    INFO   OverlapRemover           (pe_resolver.hpp           : 133)   Marking remaining overlaps\n",
      "  0:00:23.144   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 308)   Splitting paths\n",
      "  0:00:23.144   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 313)   Deduplicating paths\n",
      "  0:00:23.147   160M / 1G    INFO   PEResolver               (pe_resolver.hpp           : 315)   Overlaps removed\n",
      "  0:00:23.152   160M / 1G    INFO    General                 (launcher.cpp              : 257)   Paths finalized\n",
      "  0:00:23.164   160M / 1G    INFO    General                 (launcher.cpp              : 529)   ExSPAnder repeat resolving tool finished\n",
      "  0:00:23.185   136M / 1G    INFO   StageManager             (stage.cpp                 : 132)   STAGE == Contig Output\n",
      "  0:00:23.186   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  40)   Writing GFA to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph_with_scaffolds.gfa\n",
      "  0:00:23.220   136M / 1G    INFO    General                 (contig_output.hpp         :  22)   Outputting contigs to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/before_rr.fasta\n",
      "  0:00:23.264   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  51)   Outputting FastG graph to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/assembly_graph.fastg\n",
      "  0:00:23.437   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/final_contigs.paths\n",
      "  0:00:23.506   136M / 1G    INFO    General                 (contig_output_stage.cpp   :  20)   Outputting FastG paths to /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45//K45/scaffolds.paths\n",
      "  0:00:23.580   136M / 1G    INFO    General                 (launch.hpp                : 149)   SPAdes finished\n",
      "  0:00:23.640    12M / 1G    INFO    General                 (main.cpp                  : 109)   Assembling time: 0 hours 0 minutes 23 seconds\n",
      "\n",
      "===== Assembling finished. Used k-mer sizes: 45 \n",
      "\n",
      " * Corrected reads are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/corrected/\n",
      " * Assembled contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/contigs.fasta\n",
      " * Assembled scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/scaffolds.fasta\n",
      " * Assembly graph is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/assembly_graph_with_scaffolds.gfa\n",
      " * Paths in the assembly graph corresponding to the contigs are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/scaffolds.paths\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /home/microbioinf/Documentos/Tema_3/kneaddata_out/meta-VIR-Assembly45/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compute assembly with no specified K_MER\"\n",
    "metaspades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--meta -o meta-${FILE_ID}-Assembly${K_MER}\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "echo \"#### Compute assembly K_MER=${K_MER}\"\n",
    "metaspades.py -1 ${FILE_ID}_R1__kneaddata_paired_1.fastq -2 ${FILE_ID}_R1__kneaddata_paired_2.fastq \\\n",
    "--meta -k ${K_MER} -o meta-${FILE_ID}-Assembly${K_MER}\n",
    "done\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ",25,35,45\n"
     ]
    }
   ],
   "source": [
    "K_MERS_LIST = [\"\",\"25\", \"35\", \"45\"]\n",
    "K_MERS =  \",\".join(K_MERS_LIST)\n",
    "print(K_MERS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "#### Check output K_MER=\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_31161_cov_5.416351\n",
      ">NODE_5_length_21925_cov_9.228166\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_18141_cov_10.684452\n",
      ">NODE_8_length_17748_cov_24.048381\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_40166_cov_9.849144\n",
      ">NODE_5_length_31161_cov_5.416351\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_17748_cov_24.048381\n",
      ">NODE_8_length_12677_cov_11.504754\n",
      "scaffolds.fasta:GAAAAGAGCTGGAAGAAGATGGTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNGAAGTGATCGGAAATATTTATGAGAATGAACTAGATTTGATAGTGTACGAGGCTT\n",
      "scaffolds.fasta:GGTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGCACTCATAGATTTTGC\n",
      "scaffolds.fasta:GTGTCATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCCTTCCTGCAT\n",
      "scaffolds.fasta:GAGCGNNNNNNNNNNAATCAATATATCCTGATCCTATCCAGCTGTGTGTGCTGGGATGCC\n",
      "scaffolds.fasta:ATAGCACATGAAAAAATCTAAACTTTTTGGTTCCAGGTTAGTTGAAAANNNNNNNNNNTG\n",
      "scaffolds.fasta:CTGTAGATACGCCTGGTGTATAGACNNNNNNNNNNAAAATGATAATCCTGGAGTTCAAAC\n",
      "scaffolds.fasta:CCTGCGCTGGAATTGGAGGGAGCGCAGCTTGATCCACGTTCTTCCCCACTNNNNNNNNNN\n",
      "scaffolds.fasta:TGGAGTACGTGCGGCAGTATTTTAAGCCGTTCGCATTTANNNNNNNNNNAGATCCGTTCC\n",
      "scaffolds.fasta:TTCTTTGTNNNNNNNNNNGTGACAGTAGATCGACGTAGGAATGAGTGCGCCTATAAAGAA\n",
      "scaffolds.fasta:ACGTGCTCACGTTCGTACTCGGTTGTACGCTCGCGCTGCTGTGGANNNNNNNNNNGTAAT\n",
      "scaffolds.fasta:TTTACNNNNNNNNNNCAACAATTCCAGCATAGCTAGATAATTCTGGTATAACTGAACTTT\n",
      "scaffolds.fasta:TACGGCGGGCNNNNNNNNNNGCTGCATCATTGGCGACCAGATGGTGAAGATGTATGACGA\n",
      "scaffolds.fasta:CTNNNNNNNNNNCTCGAAGCGGCGCTGTTCATCCATCGCGTTGGCAACGTCGAGGCCTTG\n",
      "scaffolds.fasta:ATTGTGCTTTTGATNNNNNNNNNNAGCTGACAAGGTTAGCACCAAGCATTGATTTGAACA\n",
      "scaffolds.fasta:GGGTTACGAAATCCACTGCTTCTCCTAGATAATATACTTCTCCTGACCGCACCACATNNN\n",
      "scaffolds.fasta:NNNNNNNCCAAACCGGTCGTAATTCCGCTCTACAATATCGCGAACCGTAGTGCTAATGGT\n",
      "scaffolds.fasta:NNNNNNNNNGCTTCACATATGTTGAAGAGAGTTAAGTTAAATCCTACAGCTCTAATAACA\n",
      "scaffolds.fasta:GGTAGCTTAGTTGCATACATAATGGGCATTACNNNNNNNNNNGTATTTGAACGATTTACC\n",
      "scaffolds.fasta:GATTACACGGACCATTAACAACAACGATAGTTAATTATATACCAGTTGAGACTNNNNNNN\n",
      "scaffolds.fasta:NNNGATTCAAGAAGAGCCTGGTAGAATATTAAATCACCCAGCTCAAGATAAACCTCAATT\n",
      "scaffolds.fasta:TTCGAGCCAGGTATGTTTCGAAACCGAACCTGTCAATGACGANNNNNNNNNNCTCTAACC\n",
      "scaffolds.fasta:CAACAAAGAAGTTGACATTTAATTTATATTAAGAACAATNNNNNNNNNNAAAGGAAATGA\n",
      "scaffolds.fasta:TACTCGCATGTTCNNNNNNNNNNCCTCGAGTTCGGAGAGGATGACCGGAAGTTCGTCGGA\n",
      "scaffolds.fasta:TCAGCAATGTTTCTTCTCCTACTGGTTCTAACATATTATCTACAATTGATNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNTTTAAGAAAGTTGATATGTATAAGTATTAGATAATAAATACTAATGATAT\n",
      "scaffolds.fasta:ACAACGGAAACAGGCAGGCCGAAGAATTACACCATTAAAATCAAAAACGNNNNNNNNNNC\n",
      "scaffolds.fasta:TAGNNNNNNNNNNCCAGCAAAGTAATGCGTAACCGCCGTACGGAAAACCGCACTGTCGAA\n",
      "scaffolds.fasta:TGCCTTTAGTAATGATNNNNNNNNNNTATTAGACTTACTATCAAGATCTAATTGATCTAC\n",
      "scaffolds.fasta:GAGAACGTATTGTTTTAAGAAATGCTTCTGCTACAACATATTGTTCTTCAATTGNNNNNN\n",
      "scaffolds.fasta:NNNNTTCAATTGTTTTGTACCGTTTTACTAATTGTTTGATAACAAAAATATCAATATGTT\n",
      "scaffolds.fasta:ATGGATCCGGAGTTTTTAACTATACAGTTAGACGATATAGAATATTGGGATNNNNNNNNN\n",
      "scaffolds.fasta:CTATNNNNNNNNNNATGGCTTCTTTATCACCACTAAGAATAGCGCTTGTGAGCTTATTAG\n",
      "scaffolds.fasta:CGATTTCAGCAATAGTTTCTACGACCTCATNNNNNNNNNNCTTATATCGAATGGAAAACC\n",
      "scaffolds.fasta:GGGCTTCGATATTATGTNNNNNNNNNNCTGAGTTTGTAGGGCTGTACTATACAGCTTACG\n",
      "scaffolds.fasta:ACTGTATCTTTAGAGGGAGAAAACTCTTCTAAATATATGCTTTCATTAANNNNNNNNNNT\n",
      "scaffolds.fasta:ATTCAATTTAGAATCATAAAAGNNNNNNNNNNTTAGGAATCTCAATTGTAGTTGGCTCAG\n",
      "scaffolds.fasta:TTTAGAGCNNNNNNNNNNATATATTCCATGCGCCTAGCGTATTTTAATGATTGTCTAATA\n",
      "scaffolds.fasta:CGTCTCACCGCTTCTCCCATTGTATCAGATGCAGATGCGAGAGNNNNNNNNNNGAGAGAT\n",
      "scaffolds.fasta:TGCNNNNNNNNNNTCAGGGCATCGCTGCTGGTGCTATACACTGGTGTATCGTCAGCGAGA\n",
      "scaffolds.fasta:GCGAGCGTTAAATCCGAATNNNNNNNNNNATCGTAAATTATGGTACTGACGACTCTACTG\n",
      "\n",
      "#### Check output K_MER=25\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_31161_cov_5.416351\n",
      ">NODE_5_length_21925_cov_9.228166\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_18141_cov_10.684452\n",
      ">NODE_8_length_17748_cov_24.048381\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_40166_cov_9.849144\n",
      ">NODE_5_length_31161_cov_5.416351\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_17748_cov_24.048381\n",
      ">NODE_8_length_12677_cov_11.504754\n",
      "scaffolds.fasta:GAAAAGAGCTGGAAGAAGATGGTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNGAAGTGATCGGAAATATTTATGAGAATGAACTAGATTTGATAGTGTACGAGGCTT\n",
      "scaffolds.fasta:GGTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGCACTCATAGATTTTGC\n",
      "scaffolds.fasta:GTGTCATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCCTTCCTGCAT\n",
      "scaffolds.fasta:GAGCGNNNNNNNNNNAATCAATATATCCTGATCCTATCCAGCTGTGTGTGCTGGGATGCC\n",
      "scaffolds.fasta:ATAGCACATGAAAAAATCTAAACTTTTTGGTTCCAGGTTAGTTGAAAANNNNNNNNNNTG\n",
      "scaffolds.fasta:CTGTAGATACGCCTGGTGTATAGACNNNNNNNNNNAAAATGATAATCCTGGAGTTCAAAC\n",
      "scaffolds.fasta:CCTGCGCTGGAATTGGAGGGAGCGCAGCTTGATCCACGTTCTTCCCCACTNNNNNNNNNN\n",
      "scaffolds.fasta:TGGAGTACGTGCGGCAGTATTTTAAGCCGTTCGCATTTANNNNNNNNNNAGATCCGTTCC\n",
      "scaffolds.fasta:TTCTTTGTNNNNNNNNNNGTGACAGTAGATCGACGTAGGAATGAGTGCGCCTATAAAGAA\n",
      "scaffolds.fasta:ACGTGCTCACGTTCGTACTCGGTTGTACGCTCGCGCTGCTGTGGANNNNNNNNNNGTAAT\n",
      "scaffolds.fasta:TTTACNNNNNNNNNNCAACAATTCCAGCATAGCTAGATAATTCTGGTATAACTGAACTTT\n",
      "scaffolds.fasta:TACGGCGGGCNNNNNNNNNNGCTGCATCATTGGCGACCAGATGGTGAAGATGTATGACGA\n",
      "scaffolds.fasta:CTNNNNNNNNNNCTCGAAGCGGCGCTGTTCATCCATCGCGTTGGCAACGTCGAGGCCTTG\n",
      "scaffolds.fasta:ATTGTGCTTTTGATNNNNNNNNNNAGCTGACAAGGTTAGCACCAAGCATTGATTTGAACA\n",
      "scaffolds.fasta:GGGTTACGAAATCCACTGCTTCTCCTAGATAATATACTTCTCCTGACCGCACCACATNNN\n",
      "scaffolds.fasta:NNNNNNNCCAAACCGGTCGTAATTCCGCTCTACAATATCGCGAACCGTAGTGCTAATGGT\n",
      "scaffolds.fasta:NNNNNNNNNGCTTCACATATGTTGAAGAGAGTTAAGTTAAATCCTACAGCTCTAATAACA\n",
      "scaffolds.fasta:GGTAGCTTAGTTGCATACATAATGGGCATTACNNNNNNNNNNGTATTTGAACGATTTACC\n",
      "scaffolds.fasta:GATTACACGGACCATTAACAACAACGATAGTTAATTATATACCAGTTGAGACTNNNNNNN\n",
      "scaffolds.fasta:NNNGATTCAAGAAGAGCCTGGTAGAATATTAAATCACCCAGCTCAAGATAAACCTCAATT\n",
      "scaffolds.fasta:TTCGAGCCAGGTATGTTTCGAAACCGAACCTGTCAATGACGANNNNNNNNNNCTCTAACC\n",
      "scaffolds.fasta:CAACAAAGAAGTTGACATTTAATTTATATTAAGAACAATNNNNNNNNNNAAAGGAAATGA\n",
      "scaffolds.fasta:TACTCGCATGTTCNNNNNNNNNNCCTCGAGTTCGGAGAGGATGACCGGAAGTTCGTCGGA\n",
      "scaffolds.fasta:TCAGCAATGTTTCTTCTCCTACTGGTTCTAACATATTATCTACAATTGATNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNTTTAAGAAAGTTGATATGTATAAGTATTAGATAATAAATACTAATGATAT\n",
      "scaffolds.fasta:ACAACGGAAACAGGCAGGCCGAAGAATTACACCATTAAAATCAAAAACGNNNNNNNNNNC\n",
      "scaffolds.fasta:TAGNNNNNNNNNNCCAGCAAAGTAATGCGTAACCGCCGTACGGAAAACCGCACTGTCGAA\n",
      "scaffolds.fasta:TGCCTTTAGTAATGATNNNNNNNNNNTATTAGACTTACTATCAAGATCTAATTGATCTAC\n",
      "scaffolds.fasta:GAGAACGTATTGTTTTAAGAAATGCTTCTGCTACAACATATTGTTCTTCAATTGNNNNNN\n",
      "scaffolds.fasta:NNNNTTCAATTGTTTTGTACCGTTTTACTAATTGTTTGATAACAAAAATATCAATATGTT\n",
      "scaffolds.fasta:ATGGATCCGGAGTTTTTAACTATACAGTTAGACGATATAGAATATTGGGATNNNNNNNNN\n",
      "scaffolds.fasta:CTATNNNNNNNNNNATGGCTTCTTTATCACCACTAAGAATAGCGCTTGTGAGCTTATTAG\n",
      "scaffolds.fasta:CGATTTCAGCAATAGTTTCTACGACCTCATNNNNNNNNNNCTTATATCGAATGGAAAACC\n",
      "scaffolds.fasta:GGGCTTCGATATTATGTNNNNNNNNNNCTGAGTTTGTAGGGCTGTACTATACAGCTTACG\n",
      "scaffolds.fasta:ACTGTATCTTTAGAGGGAGAAAACTCTTCTAAATATATGCTTTCATTAANNNNNNNNNNT\n",
      "scaffolds.fasta:ATTCAATTTAGAATCATAAAAGNNNNNNNNNNTTAGGAATCTCAATTGTAGTTGGCTCAG\n",
      "scaffolds.fasta:TTTAGAGCNNNNNNNNNNATATATTCCATGCGCCTAGCGTATTTTAATGATTGTCTAATA\n",
      "scaffolds.fasta:CGTCTCACCGCTTCTCCCATTGTATCAGATGCAGATGCGAGAGNNNNNNNNNNGAGAGAT\n",
      "scaffolds.fasta:TGCNNNNNNNNNNTCAGGGCATCGCTGCTGGTGCTATACACTGGTGTATCGTCAGCGAGA\n",
      "scaffolds.fasta:GCGAGCGTTAAATCCGAATNNNNNNNNNNATCGTAAATTATGGTACTGACGACTCTACTG\n",
      "\n",
      "#### Check output K_MER=35\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_31161_cov_5.416351\n",
      ">NODE_5_length_21925_cov_9.228166\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_18141_cov_10.684452\n",
      ">NODE_8_length_17748_cov_24.048381\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_40166_cov_9.849144\n",
      ">NODE_5_length_31161_cov_5.416351\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_17748_cov_24.048381\n",
      ">NODE_8_length_12677_cov_11.504754\n",
      "scaffolds.fasta:GAAAAGAGCTGGAAGAAGATGGTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNGAAGTGATCGGAAATATTTATGAGAATGAACTAGATTTGATAGTGTACGAGGCTT\n",
      "scaffolds.fasta:GGTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGCACTCATAGATTTTGC\n",
      "scaffolds.fasta:GTGTCATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCCTTCCTGCAT\n",
      "scaffolds.fasta:GAGCGNNNNNNNNNNAATCAATATATCCTGATCCTATCCAGCTGTGTGTGCTGGGATGCC\n",
      "scaffolds.fasta:ATAGCACATGAAAAAATCTAAACTTTTTGGTTCCAGGTTAGTTGAAAANNNNNNNNNNTG\n",
      "scaffolds.fasta:CTGTAGATACGCCTGGTGTATAGACNNNNNNNNNNAAAATGATAATCCTGGAGTTCAAAC\n",
      "scaffolds.fasta:CCTGCGCTGGAATTGGAGGGAGCGCAGCTTGATCCACGTTCTTCCCCACTNNNNNNNNNN\n",
      "scaffolds.fasta:TGGAGTACGTGCGGCAGTATTTTAAGCCGTTCGCATTTANNNNNNNNNNAGATCCGTTCC\n",
      "scaffolds.fasta:TTCTTTGTNNNNNNNNNNGTGACAGTAGATCGACGTAGGAATGAGTGCGCCTATAAAGAA\n",
      "scaffolds.fasta:ACGTGCTCACGTTCGTACTCGGTTGTACGCTCGCGCTGCTGTGGANNNNNNNNNNGTAAT\n",
      "scaffolds.fasta:TTTACNNNNNNNNNNCAACAATTCCAGCATAGCTAGATAATTCTGGTATAACTGAACTTT\n",
      "scaffolds.fasta:TACGGCGGGCNNNNNNNNNNGCTGCATCATTGGCGACCAGATGGTGAAGATGTATGACGA\n",
      "scaffolds.fasta:CTNNNNNNNNNNCTCGAAGCGGCGCTGTTCATCCATCGCGTTGGCAACGTCGAGGCCTTG\n",
      "scaffolds.fasta:ATTGTGCTTTTGATNNNNNNNNNNAGCTGACAAGGTTAGCACCAAGCATTGATTTGAACA\n",
      "scaffolds.fasta:GGGTTACGAAATCCACTGCTTCTCCTAGATAATATACTTCTCCTGACCGCACCACATNNN\n",
      "scaffolds.fasta:NNNNNNNCCAAACCGGTCGTAATTCCGCTCTACAATATCGCGAACCGTAGTGCTAATGGT\n",
      "scaffolds.fasta:NNNNNNNNNGCTTCACATATGTTGAAGAGAGTTAAGTTAAATCCTACAGCTCTAATAACA\n",
      "scaffolds.fasta:GGTAGCTTAGTTGCATACATAATGGGCATTACNNNNNNNNNNGTATTTGAACGATTTACC\n",
      "scaffolds.fasta:GATTACACGGACCATTAACAACAACGATAGTTAATTATATACCAGTTGAGACTNNNNNNN\n",
      "scaffolds.fasta:NNNGATTCAAGAAGAGCCTGGTAGAATATTAAATCACCCAGCTCAAGATAAACCTCAATT\n",
      "scaffolds.fasta:TTCGAGCCAGGTATGTTTCGAAACCGAACCTGTCAATGACGANNNNNNNNNNCTCTAACC\n",
      "scaffolds.fasta:CAACAAAGAAGTTGACATTTAATTTATATTAAGAACAATNNNNNNNNNNAAAGGAAATGA\n",
      "scaffolds.fasta:TACTCGCATGTTCNNNNNNNNNNCCTCGAGTTCGGAGAGGATGACCGGAAGTTCGTCGGA\n",
      "scaffolds.fasta:TCAGCAATGTTTCTTCTCCTACTGGTTCTAACATATTATCTACAATTGATNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNTTTAAGAAAGTTGATATGTATAAGTATTAGATAATAAATACTAATGATAT\n",
      "scaffolds.fasta:ACAACGGAAACAGGCAGGCCGAAGAATTACACCATTAAAATCAAAAACGNNNNNNNNNNC\n",
      "scaffolds.fasta:TAGNNNNNNNNNNCCAGCAAAGTAATGCGTAACCGCCGTACGGAAAACCGCACTGTCGAA\n",
      "scaffolds.fasta:TGCCTTTAGTAATGATNNNNNNNNNNTATTAGACTTACTATCAAGATCTAATTGATCTAC\n",
      "scaffolds.fasta:GAGAACGTATTGTTTTAAGAAATGCTTCTGCTACAACATATTGTTCTTCAATTGNNNNNN\n",
      "scaffolds.fasta:NNNNTTCAATTGTTTTGTACCGTTTTACTAATTGTTTGATAACAAAAATATCAATATGTT\n",
      "scaffolds.fasta:ATGGATCCGGAGTTTTTAACTATACAGTTAGACGATATAGAATATTGGGATNNNNNNNNN\n",
      "scaffolds.fasta:CTATNNNNNNNNNNATGGCTTCTTTATCACCACTAAGAATAGCGCTTGTGAGCTTATTAG\n",
      "scaffolds.fasta:CGATTTCAGCAATAGTTTCTACGACCTCATNNNNNNNNNNCTTATATCGAATGGAAAACC\n",
      "scaffolds.fasta:GGGCTTCGATATTATGTNNNNNNNNNNCTGAGTTTGTAGGGCTGTACTATACAGCTTACG\n",
      "scaffolds.fasta:ACTGTATCTTTAGAGGGAGAAAACTCTTCTAAATATATGCTTTCATTAANNNNNNNNNNT\n",
      "scaffolds.fasta:ATTCAATTTAGAATCATAAAAGNNNNNNNNNNTTAGGAATCTCAATTGTAGTTGGCTCAG\n",
      "scaffolds.fasta:TTTAGAGCNNNNNNNNNNATATATTCCATGCGCCTAGCGTATTTTAATGATTGTCTAATA\n",
      "scaffolds.fasta:CGTCTCACCGCTTCTCCCATTGTATCAGATGCAGATGCGAGAGNNNNNNNNNNGAGAGAT\n",
      "scaffolds.fasta:TGCNNNNNNNNNNTCAGGGCATCGCTGCTGGTGCTATACACTGGTGTATCGTCAGCGAGA\n",
      "scaffolds.fasta:GCGAGCGTTAAATCCGAATNNNNNNNNNNATCGTAAATTATGGTACTGACGACTCTACTG\n",
      "\n",
      "#### Check output K_MER=45\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_31161_cov_5.416351\n",
      ">NODE_5_length_21925_cov_9.228166\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_18141_cov_10.684452\n",
      ">NODE_8_length_17748_cov_24.048381\n",
      ">NODE_1_length_87312_cov_12.680977\n",
      ">NODE_2_length_49827_cov_8.457647\n",
      ">NODE_3_length_41290_cov_6.676852\n",
      ">NODE_4_length_40166_cov_9.849144\n",
      ">NODE_5_length_31161_cov_5.416351\n",
      ">NODE_6_length_19420_cov_25.583527\n",
      ">NODE_7_length_17748_cov_24.048381\n",
      ">NODE_8_length_12677_cov_11.504754\n",
      "scaffolds.fasta:GAAAAGAGCTGGAAGAAGATGGTTTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNGAAGTGATCGGAAATATTTATGAGAATGAACTAGATTTGATAGTGTACGAGGCTT\n",
      "scaffolds.fasta:GGTNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNGCACTCATAGATTTTGC\n",
      "scaffolds.fasta:GTGTCATNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAGCCTTCCTGCAT\n",
      "scaffolds.fasta:GAGCGNNNNNNNNNNAATCAATATATCCTGATCCTATCCAGCTGTGTGTGCTGGGATGCC\n",
      "scaffolds.fasta:ATAGCACATGAAAAAATCTAAACTTTTTGGTTCCAGGTTAGTTGAAAANNNNNNNNNNTG\n",
      "scaffolds.fasta:CTGTAGATACGCCTGGTGTATAGACNNNNNNNNNNAAAATGATAATCCTGGAGTTCAAAC\n",
      "scaffolds.fasta:CCTGCGCTGGAATTGGAGGGAGCGCAGCTTGATCCACGTTCTTCCCCACTNNNNNNNNNN\n",
      "scaffolds.fasta:TGGAGTACGTGCGGCAGTATTTTAAGCCGTTCGCATTTANNNNNNNNNNAGATCCGTTCC\n",
      "scaffolds.fasta:TTCTTTGTNNNNNNNNNNGTGACAGTAGATCGACGTAGGAATGAGTGCGCCTATAAAGAA\n",
      "scaffolds.fasta:ACGTGCTCACGTTCGTACTCGGTTGTACGCTCGCGCTGCTGTGGANNNNNNNNNNGTAAT\n",
      "scaffolds.fasta:TTTACNNNNNNNNNNCAACAATTCCAGCATAGCTAGATAATTCTGGTATAACTGAACTTT\n",
      "scaffolds.fasta:TACGGCGGGCNNNNNNNNNNGCTGCATCATTGGCGACCAGATGGTGAAGATGTATGACGA\n",
      "scaffolds.fasta:CTNNNNNNNNNNCTCGAAGCGGCGCTGTTCATCCATCGCGTTGGCAACGTCGAGGCCTTG\n",
      "scaffolds.fasta:ATTGTGCTTTTGATNNNNNNNNNNAGCTGACAAGGTTAGCACCAAGCATTGATTTGAACA\n",
      "scaffolds.fasta:GGGTTACGAAATCCACTGCTTCTCCTAGATAATATACTTCTCCTGACCGCACCACATNNN\n",
      "scaffolds.fasta:NNNNNNNCCAAACCGGTCGTAATTCCGCTCTACAATATCGCGAACCGTAGTGCTAATGGT\n",
      "scaffolds.fasta:NNNNNNNNNGCTTCACATATGTTGAAGAGAGTTAAGTTAAATCCTACAGCTCTAATAACA\n",
      "scaffolds.fasta:GGTAGCTTAGTTGCATACATAATGGGCATTACNNNNNNNNNNGTATTTGAACGATTTACC\n",
      "scaffolds.fasta:GATTACACGGACCATTAACAACAACGATAGTTAATTATATACCAGTTGAGACTNNNNNNN\n",
      "scaffolds.fasta:NNNGATTCAAGAAGAGCCTGGTAGAATATTAAATCACCCAGCTCAAGATAAACCTCAATT\n",
      "scaffolds.fasta:TTCGAGCCAGGTATGTTTCGAAACCGAACCTGTCAATGACGANNNNNNNNNNCTCTAACC\n",
      "scaffolds.fasta:CAACAAAGAAGTTGACATTTAATTTATATTAAGAACAATNNNNNNNNNNAAAGGAAATGA\n",
      "scaffolds.fasta:TACTCGCATGTTCNNNNNNNNNNCCTCGAGTTCGGAGAGGATGACCGGAAGTTCGTCGGA\n",
      "scaffolds.fasta:TCAGCAATGTTTCTTCTCCTACTGGTTCTAACATATTATCTACAATTGATNNNNNNNNNN\n",
      "scaffolds.fasta:NNNNNNNNNNTTTAAGAAAGTTGATATGTATAAGTATTAGATAATAAATACTAATGATAT\n",
      "scaffolds.fasta:ACAACGGAAACAGGCAGGCCGAAGAATTACACCATTAAAATCAAAAACGNNNNNNNNNNC\n",
      "scaffolds.fasta:TAGNNNNNNNNNNCCAGCAAAGTAATGCGTAACCGCCGTACGGAAAACCGCACTGTCGAA\n",
      "scaffolds.fasta:TGCCTTTAGTAATGATNNNNNNNNNNTATTAGACTTACTATCAAGATCTAATTGATCTAC\n",
      "scaffolds.fasta:GAGAACGTATTGTTTTAAGAAATGCTTCTGCTACAACATATTGTTCTTCAATTGNNNNNN\n",
      "scaffolds.fasta:NNNNTTCAATTGTTTTGTACCGTTTTACTAATTGTTTGATAACAAAAATATCAATATGTT\n",
      "scaffolds.fasta:ATGGATCCGGAGTTTTTAACTATACAGTTAGACGATATAGAATATTGGGATNNNNNNNNN\n",
      "scaffolds.fasta:CTATNNNNNNNNNNATGGCTTCTTTATCACCACTAAGAATAGCGCTTGTGAGCTTATTAG\n",
      "scaffolds.fasta:CGATTTCAGCAATAGTTTCTACGACCTCATNNNNNNNNNNCTTATATCGAATGGAAAACC\n",
      "scaffolds.fasta:GGGCTTCGATATTATGTNNNNNNNNNNCTGAGTTTGTAGGGCTGTACTATACAGCTTACG\n",
      "scaffolds.fasta:ACTGTATCTTTAGAGGGAGAAAACTCTTCTAAATATATGCTTTCATTAANNNNNNNNNNT\n",
      "scaffolds.fasta:ATTCAATTTAGAATCATAAAAGNNNNNNNNNNTTAGGAATCTCAATTGTAGTTGGCTCAG\n",
      "scaffolds.fasta:TTTAGAGCNNNNNNNNNNATATATTCCATGCGCCTAGCGTATTTTAATGATTGTCTAATA\n",
      "scaffolds.fasta:CGTCTCACCGCTTCTCCCATTGTATCAGATGCAGATGCGAGAGNNNNNNNNNNGAGAGAT\n",
      "scaffolds.fasta:TGCNNNNNNNNNNTCAGGGCATCGCTGCTGGTGCTATACACTGGTGTATCGTCAGCGAGA\n",
      "scaffolds.fasta:GCGAGCGTTAAATCCGAATNNNNNNNNNNATCGTAAATTATGGTACTGACGACTCTACTG\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\" \"$FASTQ_STR\" \"$MIN_LEN\" \"$K_MERS\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 FASTQ_STR=$2 MIN_LEN=$3 K_MERS=$4 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "IFS=\",\"\n",
    "for K_MER in ${K_MERS}\n",
    "do\n",
    "    echo\n",
    "    echo \"#### Check output K_MER=${K_MER}\"\n",
    "    cd meta-${FILE_ID}-Assembly${K_MER}\n",
    "    rep -c \">\" *fasta\n",
    "    grep \">\" -m 8 contigs.fasta \n",
    "    grep \">\" -m 8 scaffolds.fasta \n",
    "    grep \"NN\" *fasta\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Comparison of assemblies (*quast*)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#### Compare assemblies FILE_ID=VIR\n",
      "Processing meta-VIR-Assembly file...\n",
      "Processing meta-VIR-Assembly25 file...\n",
      "Processing meta-VIR-Assembly35 file...\n",
      "Processing meta-VIR-Assembly45 file...\n",
      "/home/microbioinf/miniconda3/lib/python2.7/site-packages/quast-5.0.2-py2.7.egg-info/scripts/quast.py meta-contigs-meta-VIR-Assembly25.fasta meta-contigs-meta-VIR-Assembly35.fasta meta-contigs-meta-VIR-Assembly45.fasta meta-contigs-meta-VIR-Assembly.fasta meta-scaffolds-meta-VIR-Assembly25.fasta meta-scaffolds-meta-VIR-Assembly35.fasta meta-scaffolds-meta-VIR-Assembly45.fasta meta-scaffolds-meta-VIR-Assembly.fasta\n",
      "\n",
      "Version: 5.0.2\n",
      "\n",
      "System information:\n",
      "  OS: Linux-4.15.0-47-generic-x86_64-with-debian-buster-sid (linux_64)\n",
      "  Python version: 2.7.11\n",
      "  CPUs number: 3\n",
      "\n",
      "Started: 2019-04-26 16:12:55\n",
      "\n",
      "Logging to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/quast.log\n",
      "NOTICE: Maximum number of threads is set to 1 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /home/microbioinf/Documentos/Tema_3/kneaddata_out\n",
      "Main parameters: \n",
      "  MODE: default, threads: 1, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  meta-contigs-meta-VIR-Assembly25.fasta ==> meta_contigs_meta_VIR_Assembly25\n",
      "  2  meta-contigs-meta-VIR-Assembly35.fasta ==> meta_contigs_meta_VIR_Assembly35\n",
      "  3  meta-contigs-meta-VIR-Assembly45.fasta ==> meta_contigs_meta_VIR_Assembly45\n",
      "  4  meta-contigs-meta-VIR-Assembly.fasta ==> meta_contigs_meta_VIR_Assembly\n",
      "  5  meta-scaffolds-meta-VIR-Assembly25.fasta ==> meta_scaffolds_meta_VIR_Assembly25\n",
      "  6  meta-scaffolds-meta-VIR-Assembly35.fasta ==> meta_scaffolds_meta_VIR_Assembly35\n",
      "  7  meta-scaffolds-meta-VIR-Assembly45.fasta ==> meta_scaffolds_meta_VIR_Assembly45\n",
      "  8  meta-scaffolds-meta-VIR-Assembly.fasta ==> meta_scaffolds_meta_VIR_Assembly\n",
      "\n",
      "2019-04-26 16:12:56\n",
      "Running Basic statistics processor...\n",
      "  Contig files: \n",
      "    1  meta_contigs_meta_VIR_Assembly25\n",
      "    2  meta_contigs_meta_VIR_Assembly35\n",
      "    3  meta_contigs_meta_VIR_Assembly45\n",
      "    4  meta_contigs_meta_VIR_Assembly\n",
      "    5  meta_scaffolds_meta_VIR_Assembly25\n",
      "    6  meta_scaffolds_meta_VIR_Assembly35\n",
      "    7  meta_scaffolds_meta_VIR_Assembly45\n",
      "    8  meta_scaffolds_meta_VIR_Assembly\n",
      "  Calculating N50 and L50...\n",
      "    1  meta_contigs_meta_VIR_Assembly25, N50 = 1242, L50 = 281, Total length = 1875605, GC % = 41.83, # N's per 100 kbp =  0.00\n",
      "    2  meta_contigs_meta_VIR_Assembly35, N50 = 1295, L50 = 245, Total length = 1792179, GC % = 41.77, # N's per 100 kbp =  0.00\n",
      "    3  meta_contigs_meta_VIR_Assembly45, N50 = 1336, L50 = 225, Total length = 1704277, GC % = 41.78, # N's per 100 kbp =  0.00\n",
      "    4  meta_contigs_meta_VIR_Assembly, N50 = 1233, L50 = 275, Total length = 1982665, GC % = 41.80, # N's per 100 kbp =  0.00\n",
      "    5  meta_scaffolds_meta_VIR_Assembly25, N50 = 1279, L50 = 265, Total length = 1893693, GC % = 41.83, # N's per 100 kbp =  86.08\n",
      "    6  meta_scaffolds_meta_VIR_Assembly35, N50 = 1384, L50 = 231, Total length = 1804462, GC % = 41.76, # N's per 100 kbp =  71.49\n",
      "    7  meta_scaffolds_meta_VIR_Assembly45, N50 = 1387, L50 = 210, Total length = 1714995, GC % = 41.77, # N's per 100 kbp =  57.73\n",
      "    8  meta_scaffolds_meta_VIR_Assembly, N50 = 1255, L50 = 260, Total length = 1990311, GC % = 41.79, # N's per 100 kbp =  32.16\n",
      "  Drawing Nx plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/Nx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/GC_content_plot.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly25_GC_content_plot.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly35_GC_content_plot.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly45_GC_content_plot.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly_GC_content_plot.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly25 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly25_GC_content_plot.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly35 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly35_GC_content_plot.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly45 GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly45_GC_content_plot.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly GC content plot...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/coverage_histogram.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly25_coverage_histogram.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly35_coverage_histogram.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly45_coverage_histogram.pdf\n",
      "  Drawing meta_contigs_meta_VIR_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_contigs_meta_VIR_Assembly_coverage_histogram.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly25 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly25_coverage_histogram.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly35 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly35_coverage_histogram.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly45 coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly45_coverage_histogram.pdf\n",
      "  Drawing meta_scaffolds_meta_VIR_Assembly coverage histogram (bin size: 1x)...\n",
      "    saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/basic_stats/meta_scaffolds_meta_VIR_Assembly_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2019-04-26 16:13:02\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2019-04-26 16:13:04\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/report.html\n",
      "  PDF version (tables and plots) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/report.pdf\n",
      "  Icarus (contig browser) is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/icarus.html\n",
      "  Log is saved to /home/microbioinf/Documentos/Tema_3/kneaddata_out/quast_results/results_2019_04_26_16_12_55/quast.log\n",
      "\n",
      "Finished: 2019-04-26 16:13:04\n",
      "Elapsed time: 0:00:09.363492\n",
      "NOTICEs: 2; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$FILE_ID\"\n",
    "ssh microbioinf@192.168.56.101 env FILE_ID=$1 2>/dev/null /bin/bash <<'EOT'\n",
    "export PATH=$PATH:/home/microbioinf/miniconda3/bin\n",
    "cd Documentos/Tema_3\n",
    "cd kneaddata_out/\n",
    "echo \"#### Compare assemblies FILE_ID=${FILE_ID}\"\n",
    "for assembly in meta-${FILE_ID}-Assembly*; \n",
    "    do echo \"Processing $assembly file...\"; \n",
    "    cp ${assembly}/contigs.fasta meta-contigs-${assembly}.fasta\n",
    "    cp ${assembly}/scaffolds.fasta meta-scaffolds-${assembly}.fasta\n",
    "done\n",
    "quast.py meta-contigs* meta-scaffolds*\n",
    "EOT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style  type=\"text/css\" >\n",
       "    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col0 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "            text-align:  left;\n",
       "            font-family:  courier;\n",
       "            color:  darkblue;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col1 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col2 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col3 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col4 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col5 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col6 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col7 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }    #T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col8 {\n",
       "            text-align:  rigth;\n",
       "            font-family:  courier;\n",
       "            color:  darkgreen;\n",
       "            font-size:  10pt;\n",
       "        }</style><table id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30\" ><thead>    <tr>        <th class=\"col_heading level0 col0\" >Assembly</th>        <th class=\"col_heading level0 col1\" >meta_contigs_meta_VIR_Assembly25</th>        <th class=\"col_heading level0 col2\" >meta_contigs_meta_VIR_Assembly35</th>        <th class=\"col_heading level0 col3\" >meta_contigs_meta_VIR_Assembly45</th>        <th class=\"col_heading level0 col4\" >meta_contigs_meta_VIR_Assembly</th>        <th class=\"col_heading level0 col5\" >meta_scaffolds_meta_VIR_Assembly25</th>        <th class=\"col_heading level0 col6\" >meta_scaffolds_meta_VIR_Assembly35</th>        <th class=\"col_heading level0 col7\" >meta_scaffolds_meta_VIR_Assembly45</th>        <th class=\"col_heading level0 col8\" >meta_scaffolds_meta_VIR_Assembly</th>    </tr></thead><tbody>\n",
       "                <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col0\" class=\"data row0 col0\" ># contigs (>= 0 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col1\" class=\"data row0 col1\" >5184</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col2\" class=\"data row0 col2\" >4744</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col3\" class=\"data row0 col3\" >4405</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col4\" class=\"data row0 col4\" >4503</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col5\" class=\"data row0 col5\" >5125</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col6\" class=\"data row0 col6\" >4696</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col7\" class=\"data row0 col7\" >4360</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row0_col8\" class=\"data row0 col8\" >4466</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col0\" class=\"data row1 col0\" ># contigs (>= 1000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col1\" class=\"data row1 col1\" >419</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col2\" class=\"data row1 col2\" >388</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col3\" class=\"data row1 col3\" >360</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col4\" class=\"data row1 col4\" >395</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col5\" class=\"data row1 col5\" >418</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col6\" class=\"data row1 col6\" >390</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col7\" class=\"data row1 col7\" >357</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row1_col8\" class=\"data row1 col8\" >392</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col0\" class=\"data row2 col0\" ># contigs (>= 5000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col1\" class=\"data row2 col1\" >24</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col2\" class=\"data row2 col2\" >27</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col3\" class=\"data row2 col3\" >27</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col4\" class=\"data row2 col4\" >32</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col5\" class=\"data row2 col5\" >27</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col6\" class=\"data row2 col6\" >28</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col7\" class=\"data row2 col7\" >30</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row2_col8\" class=\"data row2 col8\" >33</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col0\" class=\"data row3 col0\" ># contigs (>= 10000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col1\" class=\"data row3 col1\" >10</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col2\" class=\"data row3 col2\" >9</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col3\" class=\"data row3 col3\" >12</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col4\" class=\"data row3 col4\" >11</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col5\" class=\"data row3 col5\" >11</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col6\" class=\"data row3 col6\" >8</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col7\" class=\"data row3 col7\" >11</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row3_col8\" class=\"data row3 col8\" >10</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col0\" class=\"data row4 col0\" ># contigs (>= 25000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col1\" class=\"data row4 col1\" >3</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col2\" class=\"data row4 col2\" >3</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col3\" class=\"data row4 col3\" >3</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col4\" class=\"data row4 col4\" >4</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col5\" class=\"data row4 col5\" >3</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col6\" class=\"data row4 col6\" >4</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col7\" class=\"data row4 col7\" >4</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row4_col8\" class=\"data row4 col8\" >5</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col0\" class=\"data row5 col0\" ># contigs (>= 50000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col1\" class=\"data row5 col1\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col2\" class=\"data row5 col2\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col3\" class=\"data row5 col3\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col4\" class=\"data row5 col4\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col5\" class=\"data row5 col5\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col6\" class=\"data row5 col6\" >2</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col7\" class=\"data row5 col7\" >1</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row5_col8\" class=\"data row5 col8\" >1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col0\" class=\"data row6 col0\" >Total length (>= 0 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col1\" class=\"data row6 col1\" >3231524</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col2\" class=\"data row6 col2\" >3080704</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col3\" class=\"data row6 col3\" >2919721</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col4\" class=\"data row6 col4\" >3153756</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col5\" class=\"data row6 col5\" >3233154</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col6\" class=\"data row6 col6\" >3081960</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col7\" class=\"data row6 col7\" >2920711</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row6_col8\" class=\"data row6 col8\" >3154396</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col0\" class=\"data row7 col0\" >Total length (>= 1000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col1\" class=\"data row7 col1\" >1091752</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col2\" class=\"data row7 col2\" >1059009</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col3\" class=\"data row7 col3\" >1006115</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col4\" class=\"data row7 col4\" >1124948</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col5\" class=\"data row7 col5\" >1119353</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col6\" class=\"data row7 col6\" >1086350</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col7\" class=\"data row7 col7\" >1027274</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row7_col8\" class=\"data row7 col8\" >1144212</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col0\" class=\"data row8 col0\" >Total length (>= 5000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col1\" class=\"data row8 col1\" >370024</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col2\" class=\"data row8 col2\" >402336</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col3\" class=\"data row8 col3\" >385353</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col4\" class=\"data row8 col4\" >460108</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col5\" class=\"data row8 col5\" >400553</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col6\" class=\"data row8 col6\" >416267</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col7\" class=\"data row8 col7\" >415499</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row8_col8\" class=\"data row8 col8\" >482429</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col0\" class=\"data row9 col0\" >Total length (>= 10000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col1\" class=\"data row9 col1\" >282350</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col2\" class=\"data row9 col2\" >284826</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col3\" class=\"data row9 col3\" >293309</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col4\" class=\"data row9 col4\" >324571</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col5\" class=\"data row9 col5\" >298340</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col6\" class=\"data row9 col6\" >286198</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col7\" class=\"data row9 col7\" >293409</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row9_col8\" class=\"data row9 col8\" >324671</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col0\" class=\"data row10 col0\" >Total length (>= 25000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col1\" class=\"data row10 col1\" >177666</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col2\" class=\"data row10 col2\" >178601</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col3\" class=\"data row10 col3\" >161994</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col4\" class=\"data row10 col4\" >209590</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col5\" class=\"data row10 col5\" >177666</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col6\" class=\"data row10 col6\" >220055</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col7\" class=\"data row10 col7\" >190746</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row10_col8\" class=\"data row10 col8\" >249756</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col0\" class=\"data row11 col0\" >Total length (>= 50000 bp)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col1\" class=\"data row11 col1\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col2\" class=\"data row11 col2\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col3\" class=\"data row11 col3\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col4\" class=\"data row11 col4\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col5\" class=\"data row11 col5\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col6\" class=\"data row11 col6\" >138582</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col7\" class=\"data row11 col7\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row11_col8\" class=\"data row11 col8\" >87312</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col0\" class=\"data row12 col0\" ># contigs</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col1\" class=\"data row12 col1\" >1602</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col2\" class=\"data row12 col2\" >1496</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col3\" class=\"data row12 col3\" >1420</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col4\" class=\"data row12 col4\" >1696</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col5\" class=\"data row12 col5\" >1584</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col6\" class=\"data row12 col6\" >1475</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col7\" class=\"data row12 col7\" >1398</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row12_col8\" class=\"data row12 col8\" >1675</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col0\" class=\"data row13 col0\" >Largest contig</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col1\" class=\"data row13 col1\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col2\" class=\"data row13 col2\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col3\" class=\"data row13 col3\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col4\" class=\"data row13 col4\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col5\" class=\"data row13 col5\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col6\" class=\"data row13 col6\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col7\" class=\"data row13 col7\" >87312</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row13_col8\" class=\"data row13 col8\" >87312</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col0\" class=\"data row14 col0\" >Total length</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col1\" class=\"data row14 col1\" >1875605</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col2\" class=\"data row14 col2\" >1792179</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col3\" class=\"data row14 col3\" >1704277</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col4\" class=\"data row14 col4\" >1982665</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col5\" class=\"data row14 col5\" >1893693</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col6\" class=\"data row14 col6\" >1804462</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col7\" class=\"data row14 col7\" >1714995</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row14_col8\" class=\"data row14 col8\" >1990311</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col0\" class=\"data row15 col0\" >GC (%)</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col1\" class=\"data row15 col1\" >41.83</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col2\" class=\"data row15 col2\" >41.77</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col3\" class=\"data row15 col3\" >41.78</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col4\" class=\"data row15 col4\" >41.80</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col5\" class=\"data row15 col5\" >41.83</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col6\" class=\"data row15 col6\" >41.76</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col7\" class=\"data row15 col7\" >41.77</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row15_col8\" class=\"data row15 col8\" >41.79</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col0\" class=\"data row16 col0\" >N50</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col1\" class=\"data row16 col1\" >1242</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col2\" class=\"data row16 col2\" >1295</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col3\" class=\"data row16 col3\" >1336</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col4\" class=\"data row16 col4\" >1233</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col5\" class=\"data row16 col5\" >1279</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col6\" class=\"data row16 col6\" >1384</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col7\" class=\"data row16 col7\" >1387</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row16_col8\" class=\"data row16 col8\" >1255</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col0\" class=\"data row17 col0\" >N75</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col1\" class=\"data row17 col1\" >707</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col2\" class=\"data row17 col2\" >709</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col3\" class=\"data row17 col3\" >704</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col4\" class=\"data row17 col4\" >694</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col5\" class=\"data row17 col5\" >718</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col6\" class=\"data row17 col6\" >717</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col7\" class=\"data row17 col7\" >712</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row17_col8\" class=\"data row17 col8\" >699</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col0\" class=\"data row18 col0\" >L50</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col1\" class=\"data row18 col1\" >281</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col2\" class=\"data row18 col2\" >245</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col3\" class=\"data row18 col3\" >225</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col4\" class=\"data row18 col4\" >275</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col5\" class=\"data row18 col5\" >265</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col6\" class=\"data row18 col6\" >231</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col7\" class=\"data row18 col7\" >210</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row18_col8\" class=\"data row18 col8\" >260</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col0\" class=\"data row19 col0\" >L75</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col1\" class=\"data row19 col1\" >801</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col2\" class=\"data row19 col2\" >732</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col3\" class=\"data row19 col3\" >691</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col4\" class=\"data row19 col4\" >841</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col5\" class=\"data row19 col5\" >779</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col6\" class=\"data row19 col6\" >710</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col7\" class=\"data row19 col7\" >668</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row19_col8\" class=\"data row19 col8\" >819</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                                <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col0\" class=\"data row20 col0\" ># N's per 100 kbp</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col1\" class=\"data row20 col1\" >0.00</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col2\" class=\"data row20 col2\" >0.00</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col3\" class=\"data row20 col3\" >0.00</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col4\" class=\"data row20 col4\" >0.00</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col5\" class=\"data row20 col5\" >86.08</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col6\" class=\"data row20 col6\" >71.49</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col7\" class=\"data row20 col7\" >57.73</td>\n",
       "                        <td id=\"T_6cde309a_682d_11e9_88cb_b8e8562afb30row20_col8\" class=\"data row20 col8\" >32.16</td>\n",
       "            </tr>\n",
       "    </tbody></table>"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x1117bb208>"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = \"\"\"\n",
    "cat Documentos/Tema_3/kneaddata_out/quast*/latest/report.tsv\n",
    "EOT\n",
    "\"\"\"\n",
    "output = !ssh microbioinf@192.168.56.101 /bin/bash <<'EOT' {data}\n",
    "data = []\n",
    "# To list of lists\n",
    "for row in output:\n",
    "    data.append(row.split('\\t'))\n",
    "# To dataframe\n",
    "df_quast_meta = pd.DataFrame(data[1:], columns=data[0])\n",
    "df_quast_meta.style.hide_index().set_properties(**{'text-align': 'rigth', 'font-family' : 'courier', 'color' : 'darkgreen', \"font-size\" : \"10pt\"}).\\\n",
    "set_properties(**{'text-align': 'left', 'font-family' : 'courier', 'color' : 'darkblue', \"font-size\" : \"10pt\"}, \\\n",
    "               subset=['Assembly'])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  },
  "latex_envs": {
   "LaTeX_envs_menu_present": true,
   "autoclose": false,
   "autocomplete": true,
   "bibliofile": "biblio.bib",
   "cite_by": "apalike",
   "current_citInitial": 1,
   "eqLabelWithNumbers": true,
   "eqNumInitial": 1,
   "hotkeys": {
    "equation": "Ctrl-E",
    "itemize": "Ctrl-I"
   },
   "labels_anchors": false,
   "latex_user_defs": false,
   "report_style_numbering": false,
   "user_envs_cfg": false
  },
  "nbTranslate": {
   "displayLangs": [
    "*"
   ],
   "hotkey": "alt-t",
   "langInMainMenu": true,
   "sourceLang": "en",
   "targetLang": "fr",
   "useGoogleTranslate": true
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
